// @ziqni-tech/admin-api-client/dist/api/ActionTypesApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActionTypeResponse = _interopRequireDefault(require("../model/ActionTypeResponse"));

var _ApiResponse = _interopRequireDefault(require("../model/ApiResponse"));

var _CreateActionTypeRequest = _interopRequireDefault(require("../model/CreateActionTypeRequest"));

var _QueryRequest = _interopRequireDefault(require("../model/QueryRequest"));

var _UpdateActionTypeRequest = _interopRequireDefault(require("../model/UpdateActionTypeRequest"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
* ActionTypes service.
* @module api/ActionTypesApi
* @version 0.0.26
*/
var ActionTypesApi = /*#__PURE__*/function () {
  /**
  * Constructs a new ActionTypesApi. 
  * @alias module:api/ActionTypesApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function ActionTypesApi(apiClient) {
    _classCallCheck(this, ActionTypesApi);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Callback function to receive the result of the createActionTypes operation.
   * @callback module:api/ActionTypesApi~createActionTypesCallback
   * @param {String} error Error message, if any.
   * @param {module:model/ApiResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Create action types
   * Create new Action types in the Ziqni database
   * @param {Array.<module:model/CreateActionTypeRequest>} body Create Action Types in the Ziqni database
   * @param {module:api/ActionTypesApi~createActionTypesCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/ApiResponse}
   */


  _createClass(ActionTypesApi, [{
    key: "createActionTypes",
    value: function createActionTypes(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling createActionTypes");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/action-types', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteActionTypes operation.
     * @callback module:api/ActionTypesApi~deleteActionTypesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete action types
     * Delete Action types for a given identifier specified
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.id The unique identifiers of the resources
     * @param {module:api/ActionTypesApi~deleteActionTypesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteActionTypes",
    value: function deleteActionTypes(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'id': this.apiClient.buildCollectionParam(opts['id'], 'multi')
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/action-types', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteActionTypesByQuery operation.
     * @callback module:api/ActionTypesApi~deleteActionTypesByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete action types by query
     * Delete Action types from Ziqni database by unique Action types ID's or any other POST body parameters using the POST method
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Delete Action types from Ziqni database by unique Action types ID's or any other POST body parameters using the POST method
     * @param {module:api/ActionTypesApi~deleteActionTypesByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteActionTypesByQuery",
    value: function deleteActionTypesByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/action-types/delete', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getActionTypes operation.
     * @callback module:api/ActionTypesApi~getActionTypesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ActionTypeResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Get action types
     * Returns a list of Action Types. This assumes that action types have first been uploaded via a POST request or web console
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.id The unique identifiers of the resources
     * @param {Number} opts.limit Limit the returned total records found
     * @param {Number} opts.skip Skip the returned records found and return the next batch of records
     * @param {module:api/ActionTypesApi~getActionTypesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ActionTypeResponse}
     */

  }, {
    key: "getActionTypes",
    value: function getActionTypes(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'id': this.apiClient.buildCollectionParam(opts['id'], 'multi'),
        '_limit': opts['limit'],
        '_skip': opts['skip']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ActionTypeResponse["default"];
      return this.apiClient.callApi('/action-types', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getActionTypesByQuery operation.
     * @callback module:api/ActionTypesApi~getActionTypesByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ActionTypeResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Get action types by query
     * Retrieve Action types from Ziqni database by unique Action type ID's or any other POST body parameters using the POST method
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Retrieve Action types from Ziqni database by unique Action type ID's or any other POST body parameters using the POST method
     * @param {module:api/ActionTypesApi~getActionTypesByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ActionTypeResponse}
     */

  }, {
    key: "getActionTypesByQuery",
    value: function getActionTypesByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ActionTypeResponse["default"];
      return this.apiClient.callApi('/action-types/query', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the updateActionTypes operation.
     * @callback module:api/ActionTypesApi~updateActionTypesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Update action types
     * Update existing Action types in the Ziqni database
     * @param {Array.<module:model/UpdateActionTypeRequest>} body Update Action types in the Ziqni database.
     * @param {module:api/ActionTypesApi~updateActionTypesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "updateActionTypes",
    value: function updateActionTypes(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling updateActionTypes");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/action-types', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);

  return ActionTypesApi;
}();

exports["default"] = ActionTypesApi;
