// @ziqni-tech/admin-api-client/dist/api/ComputeEngineApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ApiResponse = _interopRequireDefault(require("../model/ApiResponse"));

var _ComputeEngineResponse = _interopRequireDefault(require("../model/ComputeEngineResponse"));

var _CreateComputeEngineRequest = _interopRequireDefault(require("../model/CreateComputeEngineRequest"));

var _QueryRequest = _interopRequireDefault(require("../model/QueryRequest"));

var _UpdateComputeEngineRequest = _interopRequireDefault(require("../model/UpdateComputeEngineRequest"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
* ComputeEngine service.
* @module api/ComputeEngineApi
* @version 0.0.26
*/
var ComputeEngineApi = /*#__PURE__*/function () {
  /**
  * Constructs a new ComputeEngineApi. 
  * @alias module:api/ComputeEngineApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function ComputeEngineApi(apiClient) {
    _classCallCheck(this, ComputeEngineApi);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Callback function to receive the result of the createComputeEngines operation.
   * @callback module:api/ComputeEngineApi~createComputeEnginesCallback
   * @param {String} error Error message, if any.
   * @param {module:model/ApiResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Create a new Compute Engine in the Ziqni system
   * @param {module:model/CreateComputeEngineRequest} body Create a Compute Engine in the Ziqni system
   * @param {module:api/ComputeEngineApi~createComputeEnginesCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/ApiResponse}
   */


  _createClass(ComputeEngineApi, [{
    key: "createComputeEngines",
    value: function createComputeEngines(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling createComputeEngines");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/compute-engines', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteComputeEnginesById operation.
     * @callback module:api/ComputeEngineApi~deleteComputeEnginesByIdCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete the Compute Engine for a given identifier specified
     * @param {String} id Unique identifier of the resource
     * @param {module:api/ComputeEngineApi~deleteComputeEnginesByIdCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteComputeEnginesById",
    value: function deleteComputeEnginesById(id, callback) {
      var postBody = null; // verify the required parameter 'id' is set

      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling deleteComputeEnginesById");
      }

      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/compute-engines/{id}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteComputeEnginesByQuery operation.
     * @callback module:api/ComputeEngineApi~deleteComputeEnginesByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * NOT AVAILABLE IN CURRENT RELEASE
     * Delete an Compute Engine or a list of Compute Engines from Ziqni by query
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Delete an  Compute Engine or a list of Compute Engines from Ziqni by unique Compute Engine ID's or any other POST body parameters using the POST method
     * @param {module:api/ComputeEngineApi~deleteComputeEnginesByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteComputeEnginesByQuery",
    value: function deleteComputeEnginesByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/compute-engines/delete', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getComputeEngines operation.
     * @callback module:api/ComputeEngineApi~getComputeEnginesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ComputeEngineResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Returns a list of Compute Engines. This assumes that api keys have first been created via a POST request or web console
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.id The unique identifiers of the resources
     * @param {Number} opts.limit Limit the returned total records found
     * @param {Number} opts.skip Skip the returned records found and return the next batch of records
     * @param {module:api/ComputeEngineApi~getComputeEnginesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ComputeEngineResponse}
     */

  }, {
    key: "getComputeEngines",
    value: function getComputeEngines(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'id': this.apiClient.buildCollectionParam(opts['id'], 'multi'),
        '_limit': opts['limit'],
        '_skip': opts['skip']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ComputeEngineResponse["default"];
      return this.apiClient.callApi('/compute-engines', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getComputeEnginesById operation.
     * @callback module:api/ComputeEngineApi~getComputeEnginesByIdCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ComputeEngineResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Returns an Compute Engine by identifier provided. This assumes that compute engines have first been created via a POST request or web console
     * @param {String} id Unique identifier of the resource
     * @param {module:api/ComputeEngineApi~getComputeEnginesByIdCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ComputeEngineResponse}
     */

  }, {
    key: "getComputeEnginesById",
    value: function getComputeEnginesById(id, callback) {
      var postBody = null; // verify the required parameter 'id' is set

      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getComputeEnginesById");
      }

      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ComputeEngineResponse["default"];
      return this.apiClient.callApi('/compute-engines/{id}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getComputeEnginesByQuery operation.
     * @callback module:api/ComputeEngineApi~getComputeEnginesByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ComputeEngineResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Retrieve an Compute Engine or a list of Compute Engines by query
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Retrieve an  Compute Engine or a list of Compute Engines from Ziqni by unique Copute Engine ID's or any other POST body parameters using the POST method
     * @param {module:api/ComputeEngineApi~getComputeEnginesByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ComputeEngineResponse}
     */

  }, {
    key: "getComputeEnginesByQuery",
    value: function getComputeEnginesByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ComputeEngineResponse["default"];
      return this.apiClient.callApi('/compute-engines/query', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the updateComputeEngines operation.
     * @callback module:api/ComputeEngineApi~updateComputeEnginesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Update an existing Compute Engine in the Ziqni system
     * @param {module:model/UpdateComputeEngineRequest} body Update an Compute Engine in the Ziqni system. * Any Put body Parameters that are excluded in the Request body field will be considered as empty and updated with an empty field
     * @param {module:api/ComputeEngineApi~updateComputeEnginesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "updateComputeEngines",
    value: function updateComputeEngines(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling updateComputeEngines");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/compute-engines', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);

  return ComputeEngineApi;
}();

exports["default"] = ComputeEngineApi;
