// @ziqni-tech/admin-api-client/dist/api/RewardTypesApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ApiResponse = _interopRequireDefault(require("../model/ApiResponse"));

var _CreateRewardTypeRequest = _interopRequireDefault(require("../model/CreateRewardTypeRequest"));

var _QueryRequest = _interopRequireDefault(require("../model/QueryRequest"));

var _RewardTypeResponse = _interopRequireDefault(require("../model/RewardTypeResponse"));

var _UpdateRewardTypeRequest = _interopRequireDefault(require("../model/UpdateRewardTypeRequest"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
* RewardTypes service.
* @module api/RewardTypesApi
* @version 0.0.26
*/
var RewardTypesApi = /*#__PURE__*/function () {
  /**
  * Constructs a new RewardTypesApi. 
  * @alias module:api/RewardTypesApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function RewardTypesApi(apiClient) {
    _classCallCheck(this, RewardTypesApi);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Callback function to receive the result of the createRewardTypes operation.
   * @callback module:api/RewardTypesApi~createRewardTypesCallback
   * @param {String} error Error message, if any.
   * @param {module:model/ApiResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Create reward types
   * Create new Reward types in the Ziqni database
   * @param {Array.<module:model/CreateRewardTypeRequest>} body Create Reward Types in the Ziqni database
   * @param {module:api/RewardTypesApi~createRewardTypesCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/ApiResponse}
   */


  _createClass(RewardTypesApi, [{
    key: "createRewardTypes",
    value: function createRewardTypes(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling createRewardTypes");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/reward-types', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteRewardTypes operation.
     * @callback module:api/RewardTypesApi~deleteRewardTypesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete reward types
     * Delete Reward types for a given identifier specified
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.id The unique identifiers of the resources
     * @param {module:api/RewardTypesApi~deleteRewardTypesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteRewardTypes",
    value: function deleteRewardTypes(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'id': this.apiClient.buildCollectionParam(opts['id'], 'multi')
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/reward-types', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteRewardTypesByQuery operation.
     * @callback module:api/RewardTypesApi~deleteRewardTypesByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete reward types by query
     * Delete Reward types from Ziqni database by unique Reward types ID's or any other POST body parameters using the POST method
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Delete Reward types from Ziqni database by unique Reward types ID's or any other POST body parameters using the POST method
     * @param {module:api/RewardTypesApi~deleteRewardTypesByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteRewardTypesByQuery",
    value: function deleteRewardTypesByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/reward-types/delete', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getRewardTypes operation.
     * @callback module:api/RewardTypesApi~getRewardTypesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/RewardTypeResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Get reward types
     * Returns a list of Reward Types. This assumes that reward types have first been uploaded via a POST request or web console
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.id The unique identifiers of the resources
     * @param {Number} opts.limit Limit the returned total records found
     * @param {Number} opts.skip Skip the returned records found and return the next batch of records
     * @param {module:api/RewardTypesApi~getRewardTypesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/RewardTypeResponse}
     */

  }, {
    key: "getRewardTypes",
    value: function getRewardTypes(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'id': this.apiClient.buildCollectionParam(opts['id'], 'multi'),
        '_limit': opts['limit'],
        '_skip': opts['skip']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _RewardTypeResponse["default"];
      return this.apiClient.callApi('/reward-types', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getRewardTypesByQuery operation.
     * @callback module:api/RewardTypesApi~getRewardTypesByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/RewardTypeResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Get reward types by query
     * Retrieve Reward types from Ziqni database by unique Reward type ID's or any other POST body parameters using the POST method
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Retrieve Reward types from Ziqni database by unique Reward type ID's or any other POST body parameters using the POST method
     * @param {module:api/RewardTypesApi~getRewardTypesByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/RewardTypeResponse}
     */

  }, {
    key: "getRewardTypesByQuery",
    value: function getRewardTypesByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _RewardTypeResponse["default"];
      return this.apiClient.callApi('/reward-types/query', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the updateRewardTypes operation.
     * @callback module:api/RewardTypesApi~updateRewardTypesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Update Reward Types
     * Update existing Reward types in the Ziqni database
     * @param {Array.<module:model/UpdateRewardTypeRequest>} body Update Reward types in the Ziqni database.
     * @param {module:api/RewardTypesApi~updateRewardTypesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "updateRewardTypes",
    value: function updateRewardTypes(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling updateRewardTypes");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/reward-types', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);

  return RewardTypesApi;
}();

exports["default"] = RewardTypesApi;
