// @ziqni-tech/admin-api-client/dist/api/UnitsOfMeasureApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ApiResponse = _interopRequireDefault(require("../model/ApiResponse"));

var _CreateUnitOfMeasureRequest = _interopRequireDefault(require("../model/CreateUnitOfMeasureRequest"));

var _QueryRequest = _interopRequireDefault(require("../model/QueryRequest"));

var _UnitOfMeasureResponse = _interopRequireDefault(require("../model/UnitOfMeasureResponse"));

var _UpdateUnitOfMeasureRequest = _interopRequireDefault(require("../model/UpdateUnitOfMeasureRequest"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
* UnitsOfMeasure service.
* @module api/UnitsOfMeasureApi
* @version 0.0.26
*/
var UnitsOfMeasureApi = /*#__PURE__*/function () {
  /**
  * Constructs a new UnitsOfMeasureApi. 
  * @alias module:api/UnitsOfMeasureApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function UnitsOfMeasureApi(apiClient) {
    _classCallCheck(this, UnitsOfMeasureApi);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Callback function to receive the result of the createUnitsOfMeasure operation.
   * @callback module:api/UnitsOfMeasureApi~createUnitsOfMeasureCallback
   * @param {String} error Error message, if any.
   * @param {module:model/ApiResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Create a new Unit of measure in the Ziqni database
   * @param {Array.<module:model/CreateUnitOfMeasureRequest>} body Create a Unit of measure in the Ziqni database
   * @param {module:api/UnitsOfMeasureApi~createUnitsOfMeasureCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/ApiResponse}
   */


  _createClass(UnitsOfMeasureApi, [{
    key: "createUnitsOfMeasure",
    value: function createUnitsOfMeasure(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling createUnitsOfMeasure");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/units-of-measure', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteCollaboratorsByQuery operation.
     * @callback module:api/UnitsOfMeasureApi~deleteCollaboratorsByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * NOT AVAILABLE IN CURRENT RELEASE
     * Remove collaborators from account
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Remove collaborators from account
     * @param {module:api/UnitsOfMeasureApi~deleteCollaboratorsByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteCollaboratorsByQuery",
    value: function deleteCollaboratorsByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/collaborators/delete', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteUnitsOfMeasure operation.
     * @callback module:api/UnitsOfMeasureApi~deleteUnitsOfMeasureCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete the Unit of measured for a given identifier specified
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.id The unique identifiers of the resources
     * @param {module:api/UnitsOfMeasureApi~deleteUnitsOfMeasureCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteUnitsOfMeasure",
    value: function deleteUnitsOfMeasure(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'id': this.apiClient.buildCollectionParam(opts['id'], 'multi')
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/units-of-measure', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteUnitsOfMeasureByQuery operation.
     * @callback module:api/UnitsOfMeasureApi~deleteUnitsOfMeasureByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * NOT AVAILABLE IN CURRENT RELEASE
     * Delete Units of measure from Ziqni database by unique Unit of measure ID's or any other POST body parameters using the POST method
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Delete Units of measure from Ziqni database by unique Unit of measure ID's or any other POST body parameters using the POST method
     * @param {module:api/UnitsOfMeasureApi~deleteUnitsOfMeasureByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "deleteUnitsOfMeasureByQuery",
    value: function deleteUnitsOfMeasureByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/units-of-measure/delete', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getUnitsOfMeasure operation.
     * @callback module:api/UnitsOfMeasureApi~getUnitsOfMeasureCallback
     * @param {String} error Error message, if any.
     * @param {module:model/UnitOfMeasureResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Returns a list of Units of measure. This assumes that units of measure have first been uploaded via a POST request or web console
     * @param {Object} opts Optional parameters
     * @param {Array.<String>} opts.id The unique identifiers of the resources
     * @param {Number} opts.limit Limit the returned total records found
     * @param {Number} opts.skip Skip the returned records found and return the next batch of records
     * @param {module:api/UnitsOfMeasureApi~getUnitsOfMeasureCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/UnitOfMeasureResponse}
     */

  }, {
    key: "getUnitsOfMeasure",
    value: function getUnitsOfMeasure(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'id': this.apiClient.buildCollectionParam(opts['id'], 'multi'),
        '_limit': opts['limit'],
        '_skip': opts['skip']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _UnitOfMeasureResponse["default"];
      return this.apiClient.callApi('/units-of-measure', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getUnitsOfMeasureByQuery operation.
     * @callback module:api/UnitsOfMeasureApi~getUnitsOfMeasureByQueryCallback
     * @param {String} error Error message, if any.
     * @param {module:model/UnitOfMeasureResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Retrieve Units of measure from Ziqni database by unique Unit of measure ID's or any other POST body parameters using the POST method
     * @param {Object} opts Optional parameters
     * @param {module:model/QueryRequest} opts.body Retrieve Units of measure from Ziqni database by unique Unit of measure ID's or any other POST body parameters using the POST method
     * @param {module:api/UnitsOfMeasureApi~getUnitsOfMeasureByQueryCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/UnitOfMeasureResponse}
     */

  }, {
    key: "getUnitsOfMeasureByQuery",
    value: function getUnitsOfMeasureByQuery(opts, callback) {
      opts = opts || {};
      var postBody = opts['body'];
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _UnitOfMeasureResponse["default"];
      return this.apiClient.callApi('/units-of-measure/query', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the updateUnitsOfMeasure operation.
     * @callback module:api/UnitsOfMeasureApi~updateUnitsOfMeasureCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ApiResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Update an existing Unit of measure in the Ziqni database
     * @param {Array.<module:model/UpdateUnitOfMeasureRequest>} body Update a Unit of measure in the Ziqni database.
     * @param {module:api/UnitsOfMeasureApi~updateUnitsOfMeasureCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ApiResponse}
     */

  }, {
    key: "updateUnitsOfMeasure",
    value: function updateUnitsOfMeasure(body, callback) {
      var postBody = body; // verify the required parameter 'body' is set

      if (body === undefined || body === null) {
        throw new Error("Missing the required parameter 'body' when calling updateUnitsOfMeasure");
      }

      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _ApiResponse["default"];
      return this.apiClient.callApi('/units-of-measure', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);

  return UnitsOfMeasureApi;
}();

exports["default"] = UnitsOfMeasureApi;
