// @ziqni-tech/admin-api-client/dist/api/UserApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccountMessageResponse = _interopRequireDefault(require("../model/AccountMessageResponse"));

var _ProfileResponse = _interopRequireDefault(require("../model/ProfileResponse"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
* User service.
* @module api/UserApi
* @version 0.0.26
*/
var UserApi = /*#__PURE__*/function () {
  /**
  * Constructs a new UserApi. 
  * @alias module:api/UserApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function UserApi(apiClient) {
    _classCallCheck(this, UserApi);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Callback function to receive the result of the getUser operation.
   * @callback module:api/UserApi~getUserCallback
   * @param {String} error Error message, if any.
   * @param {module:model/ProfileResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Returns the current users user. This assumes that a user has first been uploaded via a POST request or web console
   * @param {Object} opts Optional parameters
   * @param {Boolean} opts.includeSpaces Include Spaces
   * @param {module:api/UserApi~getUserCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/ProfileResponse}
   */


  _createClass(UserApi, [{
    key: "getUser",
    value: function getUser(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'includeSpaces': opts['includeSpaces']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ProfileResponse["default"];
      return this.apiClient.callApi('/user', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getUserMessages operation.
     * @callback module:api/UserApi~getUserMessagesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/AccountMessageResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * NOT AVAILABLE IN CURRENT RELEASE
     * Returns a list of the Users messages
     * @param {Object} opts Optional parameters
     * @param {Number} opts.limit Limit the returned total records found
     * @param {Number} opts.skip Skip the returned records found and return the next batch of records
     * @param {module:api/UserApi~getUserMessagesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/AccountMessageResponse}
     */

  }, {
    key: "getUserMessages",
    value: function getUserMessages(opts, callback) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        '_limit': opts['limit'],
        '_skip': opts['skip']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AccountMessageResponse["default"];
      return this.apiClient.callApi('/user/inbox', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getUserMessagesById operation.
     * @callback module:api/UserApi~getUserMessagesByIdCallback
     * @param {String} error Error message, if any.
     * @param {module:model/AccountMessageResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * NOT AVAILABLE IN CURRENT RELEASE
     * Returns a Users message for the identifier provided
     * @param {String} id Unique identifier of the resource
     * @param {module:api/UserApi~getUserMessagesByIdCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/AccountMessageResponse}
     */

  }, {
    key: "getUserMessagesById",
    value: function getUserMessagesById(id, callback) {
      var postBody = null; // verify the required parameter 'id' is set

      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getUserMessagesById");
      }

      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['OAuth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AccountMessageResponse["default"];
      return this.apiClient.callApi('/user/inbox/{id}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);

  return UserApi;
}();

exports["default"] = UserApi;
