// @ziqni-tech/admin-api-client/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AccountMessage", {
  enumerable: true,
  get: function get() {
    return _AccountMessage["default"];
  }
});
Object.defineProperty(exports, "AccountMessageAllOf", {
  enumerable: true,
  get: function get() {
    return _AccountMessageAllOf["default"];
  }
});
Object.defineProperty(exports, "AccountMessageResponse", {
  enumerable: true,
  get: function get() {
    return _AccountMessageResponse["default"];
  }
});
Object.defineProperty(exports, "AccountMessagesApi", {
  enumerable: true,
  get: function get() {
    return _AccountMessagesApi["default"];
  }
});
Object.defineProperty(exports, "Achievement", {
  enumerable: true,
  get: function get() {
    return _Achievement["default"];
  }
});
Object.defineProperty(exports, "AchievementAllOf", {
  enumerable: true,
  get: function get() {
    return _AchievementAllOf["default"];
  }
});
Object.defineProperty(exports, "AchievementIssuedCount", {
  enumerable: true,
  get: function get() {
    return _AchievementIssuedCount["default"];
  }
});
Object.defineProperty(exports, "AchievementIssuedResponse", {
  enumerable: true,
  get: function get() {
    return _AchievementIssuedResponse["default"];
  }
});
Object.defineProperty(exports, "AchievementLiveStatus", {
  enumerable: true,
  get: function get() {
    return _AchievementLiveStatus["default"];
  }
});
Object.defineProperty(exports, "AchievementLiveStatusRequest", {
  enumerable: true,
  get: function get() {
    return _AchievementLiveStatusRequest["default"];
  }
});
Object.defineProperty(exports, "AchievementLiveStatusRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _AchievementLiveStatusRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "AchievementReduced", {
  enumerable: true,
  get: function get() {
    return _AchievementReduced["default"];
  }
});
Object.defineProperty(exports, "AchievementReducedAllOf", {
  enumerable: true,
  get: function get() {
    return _AchievementReducedAllOf["default"];
  }
});
Object.defineProperty(exports, "AchievementReducedResponse", {
  enumerable: true,
  get: function get() {
    return _AchievementReducedResponse["default"];
  }
});
Object.defineProperty(exports, "AchievementResponse", {
  enumerable: true,
  get: function get() {
    return _AchievementResponse["default"];
  }
});
Object.defineProperty(exports, "AchievementStatus", {
  enumerable: true,
  get: function get() {
    return _AchievementStatus["default"];
  }
});
Object.defineProperty(exports, "AchievementsApi", {
  enumerable: true,
  get: function get() {
    return _AchievementsApi["default"];
  }
});
Object.defineProperty(exports, "ActionType", {
  enumerable: true,
  get: function get() {
    return _ActionType["default"];
  }
});
Object.defineProperty(exports, "ActionTypeAdjustmentFactor", {
  enumerable: true,
  get: function get() {
    return _ActionTypeAdjustmentFactor["default"];
  }
});
Object.defineProperty(exports, "ActionTypeAllOf", {
  enumerable: true,
  get: function get() {
    return _ActionTypeAllOf["default"];
  }
});
Object.defineProperty(exports, "ActionTypeResponse", {
  enumerable: true,
  get: function get() {
    return _ActionTypeResponse["default"];
  }
});
Object.defineProperty(exports, "ActionTypesApi", {
  enumerable: true,
  get: function get() {
    return _ActionTypesApi["default"];
  }
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "ApiKey", {
  enumerable: true,
  get: function get() {
    return _ApiKey["default"];
  }
});
Object.defineProperty(exports, "ApiKeyAllOf", {
  enumerable: true,
  get: function get() {
    return _ApiKeyAllOf["default"];
  }
});
Object.defineProperty(exports, "ApiKeyResponse", {
  enumerable: true,
  get: function get() {
    return _ApiKeyResponse["default"];
  }
});
Object.defineProperty(exports, "ApiKeysApi", {
  enumerable: true,
  get: function get() {
    return _ApiKeysApi["default"];
  }
});
Object.defineProperty(exports, "ApiResponse", {
  enumerable: true,
  get: function get() {
    return _ApiResponse["default"];
  }
});
Object.defineProperty(exports, "AppliesTo", {
  enumerable: true,
  get: function get() {
    return _AppliesTo["default"];
  }
});
Object.defineProperty(exports, "AvailableLanguageResponse", {
  enumerable: true,
  get: function get() {
    return _AvailableLanguageResponse["default"];
  }
});
Object.defineProperty(exports, "Award", {
  enumerable: true,
  get: function get() {
    return _Award["default"];
  }
});
Object.defineProperty(exports, "AwardAllOf", {
  enumerable: true,
  get: function get() {
    return _AwardAllOf["default"];
  }
});
Object.defineProperty(exports, "AwardResponse", {
  enumerable: true,
  get: function get() {
    return _AwardResponse["default"];
  }
});
Object.defineProperty(exports, "AwardsApi", {
  enumerable: true,
  get: function get() {
    return _AwardsApi["default"];
  }
});
Object.defineProperty(exports, "Bracket", {
  enumerable: true,
  get: function get() {
    return _Bracket["default"];
  }
});
Object.defineProperty(exports, "BracketResponse", {
  enumerable: true,
  get: function get() {
    return _BracketResponse["default"];
  }
});
Object.defineProperty(exports, "ChangePasswordRequest", {
  enumerable: true,
  get: function get() {
    return _ChangePasswordRequest["default"];
  }
});
Object.defineProperty(exports, "ClaimAwardRequest", {
  enumerable: true,
  get: function get() {
    return _ClaimAwardRequest["default"];
  }
});
Object.defineProperty(exports, "CloneAchievementResponse", {
  enumerable: true,
  get: function get() {
    return _CloneAchievementResponse["default"];
  }
});
Object.defineProperty(exports, "CloneCompetitionResponse", {
  enumerable: true,
  get: function get() {
    return _CloneCompetitionResponse["default"];
  }
});
Object.defineProperty(exports, "CloneContestResponse", {
  enumerable: true,
  get: function get() {
    return _CloneContestResponse["default"];
  }
});
Object.defineProperty(exports, "Collaborator", {
  enumerable: true,
  get: function get() {
    return _Collaborator["default"];
  }
});
Object.defineProperty(exports, "CollaboratorAllOf", {
  enumerable: true,
  get: function get() {
    return _CollaboratorAllOf["default"];
  }
});
Object.defineProperty(exports, "CollaboratorResponse", {
  enumerable: true,
  get: function get() {
    return _CollaboratorResponse["default"];
  }
});
Object.defineProperty(exports, "CollaboratorRolesAvailable", {
  enumerable: true,
  get: function get() {
    return _CollaboratorRolesAvailable["default"];
  }
});
Object.defineProperty(exports, "CollaboratorRolesAvailableAllOf", {
  enumerable: true,
  get: function get() {
    return _CollaboratorRolesAvailableAllOf["default"];
  }
});
Object.defineProperty(exports, "CollaboratorRolesResponse", {
  enumerable: true,
  get: function get() {
    return _CollaboratorRolesResponse["default"];
  }
});
Object.defineProperty(exports, "CollaboratorsApi", {
  enumerable: true,
  get: function get() {
    return _CollaboratorsApi["default"];
  }
});
Object.defineProperty(exports, "Competition", {
  enumerable: true,
  get: function get() {
    return _Competition["default"];
  }
});
Object.defineProperty(exports, "CompetitionAllOf", {
  enumerable: true,
  get: function get() {
    return _CompetitionAllOf["default"];
  }
});
Object.defineProperty(exports, "CompetitionLeaderboardResponse", {
  enumerable: true,
  get: function get() {
    return _CompetitionLeaderboardResponse["default"];
  }
});
Object.defineProperty(exports, "CompetitionReduced", {
  enumerable: true,
  get: function get() {
    return _CompetitionReduced["default"];
  }
});
Object.defineProperty(exports, "CompetitionReducedAllOf", {
  enumerable: true,
  get: function get() {
    return _CompetitionReducedAllOf["default"];
  }
});
Object.defineProperty(exports, "CompetitionReducedResponse", {
  enumerable: true,
  get: function get() {
    return _CompetitionReducedResponse["default"];
  }
});
Object.defineProperty(exports, "CompetitionResponse", {
  enumerable: true,
  get: function get() {
    return _CompetitionResponse["default"];
  }
});
Object.defineProperty(exports, "CompetitionStatus", {
  enumerable: true,
  get: function get() {
    return _CompetitionStatus["default"];
  }
});
Object.defineProperty(exports, "CompetitionStatusActions", {
  enumerable: true,
  get: function get() {
    return _CompetitionStatusActions["default"];
  }
});
Object.defineProperty(exports, "CompetitionType", {
  enumerable: true,
  get: function get() {
    return _CompetitionType["default"];
  }
});
Object.defineProperty(exports, "CompetitionsApi", {
  enumerable: true,
  get: function get() {
    return _CompetitionsApi["default"];
  }
});
Object.defineProperty(exports, "ComputeEngine", {
  enumerable: true,
  get: function get() {
    return _ComputeEngine["default"];
  }
});
Object.defineProperty(exports, "ComputeEngineAllOf", {
  enumerable: true,
  get: function get() {
    return _ComputeEngineAllOf["default"];
  }
});
Object.defineProperty(exports, "ComputeEngineApi", {
  enumerable: true,
  get: function get() {
    return _ComputeEngineApi["default"];
  }
});
Object.defineProperty(exports, "ComputeEngineResponse", {
  enumerable: true,
  get: function get() {
    return _ComputeEngineResponse["default"];
  }
});
Object.defineProperty(exports, "Condition", {
  enumerable: true,
  get: function get() {
    return _Condition["default"];
  }
});
Object.defineProperty(exports, "Connection", {
  enumerable: true,
  get: function get() {
    return _Connection["default"];
  }
});
Object.defineProperty(exports, "ConnectionResponse", {
  enumerable: true,
  get: function get() {
    return _ConnectionResponse["default"];
  }
});
Object.defineProperty(exports, "ConnectionState", {
  enumerable: true,
  get: function get() {
    return _ConnectionState["default"];
  }
});
Object.defineProperty(exports, "ConsumersApi", {
  enumerable: true,
  get: function get() {
    return _ConsumersApi["default"];
  }
});
Object.defineProperty(exports, "Contest", {
  enumerable: true,
  get: function get() {
    return _Contest["default"];
  }
});
Object.defineProperty(exports, "ContestAllOf", {
  enumerable: true,
  get: function get() {
    return _ContestAllOf["default"];
  }
});
Object.defineProperty(exports, "ContestLeaderboardResponse", {
  enumerable: true,
  get: function get() {
    return _ContestLeaderboardResponse["default"];
  }
});
Object.defineProperty(exports, "ContestReduced", {
  enumerable: true,
  get: function get() {
    return _ContestReduced["default"];
  }
});
Object.defineProperty(exports, "ContestReducedAllOf", {
  enumerable: true,
  get: function get() {
    return _ContestReducedAllOf["default"];
  }
});
Object.defineProperty(exports, "ContestReducedResponse", {
  enumerable: true,
  get: function get() {
    return _ContestReducedResponse["default"];
  }
});
Object.defineProperty(exports, "ContestResponse", {
  enumerable: true,
  get: function get() {
    return _ContestResponse["default"];
  }
});
Object.defineProperty(exports, "ContestStateOperations", {
  enumerable: true,
  get: function get() {
    return _ContestStateOperations["default"];
  }
});
Object.defineProperty(exports, "ContestStatus", {
  enumerable: true,
  get: function get() {
    return _ContestStatus["default"];
  }
});
Object.defineProperty(exports, "ContestsApi", {
  enumerable: true,
  get: function get() {
    return _ContestsApi["default"];
  }
});
Object.defineProperty(exports, "CountResponse", {
  enumerable: true,
  get: function get() {
    return _CountResponse["default"];
  }
});
Object.defineProperty(exports, "CreateAchievementRequest", {
  enumerable: true,
  get: function get() {
    return _CreateAchievementRequest["default"];
  }
});
Object.defineProperty(exports, "CreateAchievementRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateAchievementRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateActionTypeRequest", {
  enumerable: true,
  get: function get() {
    return _CreateActionTypeRequest["default"];
  }
});
Object.defineProperty(exports, "CreateActionTypeRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateActionTypeRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateApiKeyRequest", {
  enumerable: true,
  get: function get() {
    return _CreateApiKeyRequest["default"];
  }
});
Object.defineProperty(exports, "CreateApiKeyRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateApiKeyRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateCollaboratorRequest", {
  enumerable: true,
  get: function get() {
    return _CreateCollaboratorRequest["default"];
  }
});
Object.defineProperty(exports, "CreateCompetitionRequest", {
  enumerable: true,
  get: function get() {
    return _CreateCompetitionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateCompetitionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateCompetitionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateComputeEngineRequest", {
  enumerable: true,
  get: function get() {
    return _CreateComputeEngineRequest["default"];
  }
});
Object.defineProperty(exports, "CreateComputeEngineRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateComputeEngineRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateContestForCompetitionRequest", {
  enumerable: true,
  get: function get() {
    return _CreateContestForCompetitionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateContestForCompetitionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateContestForCompetitionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateContestRequest", {
  enumerable: true,
  get: function get() {
    return _CreateContestRequest["default"];
  }
});
Object.defineProperty(exports, "CreateContestRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateContestRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateCustomFieldRequest", {
  enumerable: true,
  get: function get() {
    return _CreateCustomFieldRequest["default"];
  }
});
Object.defineProperty(exports, "CreateEntityRewardRequest", {
  enumerable: true,
  get: function get() {
    return _CreateEntityRewardRequest["default"];
  }
});
Object.defineProperty(exports, "CreateEntityRewardRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateEntityRewardRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateEventRequest", {
  enumerable: true,
  get: function get() {
    return _CreateEventRequest["default"];
  }
});
Object.defineProperty(exports, "CreateFileObjectRequest", {
  enumerable: true,
  get: function get() {
    return _CreateFileObjectRequest["default"];
  }
});
Object.defineProperty(exports, "CreateFileObjectRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateFileObjectRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateKafkaConnectionRequest", {
  enumerable: true,
  get: function get() {
    return _CreateKafkaConnectionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateKafkaConnectionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateKafkaConnectionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateLanguageRequest", {
  enumerable: true,
  get: function get() {
    return _CreateLanguageRequest["default"];
  }
});
Object.defineProperty(exports, "CreateLanguageRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateLanguageRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateMemberMessageRequest", {
  enumerable: true,
  get: function get() {
    return _CreateMemberMessageRequest["default"];
  }
});
Object.defineProperty(exports, "CreateMemberMessageRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateMemberMessageRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateMemberRequest", {
  enumerable: true,
  get: function get() {
    return _CreateMemberRequest["default"];
  }
});
Object.defineProperty(exports, "CreateMemberRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateMemberRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateNotificationRequest", {
  enumerable: true,
  get: function get() {
    return _CreateNotificationRequest["default"];
  }
});
Object.defineProperty(exports, "CreateNotificationRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateNotificationRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateOptParamModels", {
  enumerable: true,
  get: function get() {
    return _CreateOptParamModels["default"];
  }
});
Object.defineProperty(exports, "CreateProductRequest", {
  enumerable: true,
  get: function get() {
    return _CreateProductRequest["default"];
  }
});
Object.defineProperty(exports, "CreateProductRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateProductRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateRabbitMqConnectionRequest", {
  enumerable: true,
  get: function get() {
    return _CreateRabbitMqConnectionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateRabbitMqConnectionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateRabbitMqConnectionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateRepositoryRequest", {
  enumerable: true,
  get: function get() {
    return _CreateRepositoryRequest["default"];
  }
});
Object.defineProperty(exports, "CreateRepositoryRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateRepositoryRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateRewardRequest", {
  enumerable: true,
  get: function get() {
    return _CreateRewardRequest["default"];
  }
});
Object.defineProperty(exports, "CreateRewardRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateRewardRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateRewardTypeRequest", {
  enumerable: true,
  get: function get() {
    return _CreateRewardTypeRequest["default"];
  }
});
Object.defineProperty(exports, "CreateRewardTypeRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateRewardTypeRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateSimpleCompetitionRequest", {
  enumerable: true,
  get: function get() {
    return _CreateSimpleCompetitionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateSimpleCompetitionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateSimpleCompetitionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateSpaceRequest", {
  enumerable: true,
  get: function get() {
    return _CreateSpaceRequest["default"];
  }
});
Object.defineProperty(exports, "CreateSqsConnectionRequest", {
  enumerable: true,
  get: function get() {
    return _CreateSqsConnectionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateSqsConnectionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateSqsConnectionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateTagRequest", {
  enumerable: true,
  get: function get() {
    return _CreateTagRequest["default"];
  }
});
Object.defineProperty(exports, "CreateTransformerRequest", {
  enumerable: true,
  get: function get() {
    return _CreateTransformerRequest["default"];
  }
});
Object.defineProperty(exports, "CreateTranslationRequest", {
  enumerable: true,
  get: function get() {
    return _CreateTranslationRequest["default"];
  }
});
Object.defineProperty(exports, "CreateUnitOfMeasureRequest", {
  enumerable: true,
  get: function get() {
    return _CreateUnitOfMeasureRequest["default"];
  }
});
Object.defineProperty(exports, "CreateUnitOfMeasureRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateUnitOfMeasureRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CreateWebhookRequest", {
  enumerable: true,
  get: function get() {
    return _CreateWebhookRequest["default"];
  }
});
Object.defineProperty(exports, "CreateWebhookRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _CreateWebhookRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomField", {
  enumerable: true,
  get: function get() {
    return _CustomField["default"];
  }
});
Object.defineProperty(exports, "CustomFieldAllOf", {
  enumerable: true,
  get: function get() {
    return _CustomFieldAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntry", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntry["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryNumber", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryNumber["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryNumberAllOf", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryNumberAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryNumberArray", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryNumberArray["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryNumberArrayAllOf", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryNumberArrayAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryString", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryString["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryStringAllOf", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryStringAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryStringArray", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryStringArray["default"];
  }
});
Object.defineProperty(exports, "CustomFieldEntryStringArrayAllOf", {
  enumerable: true,
  get: function get() {
    return _CustomFieldEntryStringArrayAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomFieldReduced", {
  enumerable: true,
  get: function get() {
    return _CustomFieldReduced["default"];
  }
});
Object.defineProperty(exports, "CustomFieldReducedAllOf", {
  enumerable: true,
  get: function get() {
    return _CustomFieldReducedAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomFieldResponse", {
  enumerable: true,
  get: function get() {
    return _CustomFieldResponse["default"];
  }
});
Object.defineProperty(exports, "CustomFieldsApi", {
  enumerable: true,
  get: function get() {
    return _CustomFieldsApi["default"];
  }
});
Object.defineProperty(exports, "DefaultApi", {
  enumerable: true,
  get: function get() {
    return _DefaultApi["default"];
  }
});
Object.defineProperty(exports, "Dependancy", {
  enumerable: true,
  get: function get() {
    return _Dependancy["default"];
  }
});
Object.defineProperty(exports, "DependantOn", {
  enumerable: true,
  get: function get() {
    return _DependantOn["default"];
  }
});
Object.defineProperty(exports, "EntityType", {
  enumerable: true,
  get: function get() {
    return _EntityType["default"];
  }
});
Object.defineProperty(exports, "Entrant", {
  enumerable: true,
  get: function get() {
    return _Entrant["default"];
  }
});
Object.defineProperty(exports, "EntrantAction", {
  enumerable: true,
  get: function get() {
    return _EntrantAction["default"];
  }
});
Object.defineProperty(exports, "EntrantRequest", {
  enumerable: true,
  get: function get() {
    return _EntrantRequest["default"];
  }
});
Object.defineProperty(exports, "EntrantResponse", {
  enumerable: true,
  get: function get() {
    return _EntrantResponse["default"];
  }
});
Object.defineProperty(exports, "EntrantStatus", {
  enumerable: true,
  get: function get() {
    return _EntrantStatus["default"];
  }
});
Object.defineProperty(exports, "Error", {
  enumerable: true,
  get: function get() {
    return _Error["default"];
  }
});
Object.defineProperty(exports, "ErrorDetail", {
  enumerable: true,
  get: function get() {
    return _ErrorDetail["default"];
  }
});
Object.defineProperty(exports, "Event", {
  enumerable: true,
  get: function get() {
    return _Event["default"];
  }
});
Object.defineProperty(exports, "EventAllOf", {
  enumerable: true,
  get: function get() {
    return _EventAllOf["default"];
  }
});
Object.defineProperty(exports, "EventRefType", {
  enumerable: true,
  get: function get() {
    return _EventRefType["default"];
  }
});
Object.defineProperty(exports, "EventResponse", {
  enumerable: true,
  get: function get() {
    return _EventResponse["default"];
  }
});
Object.defineProperty(exports, "EventsApi", {
  enumerable: true,
  get: function get() {
    return _EventsApi["default"];
  }
});
Object.defineProperty(exports, "FactDefault", {
  enumerable: true,
  get: function get() {
    return _FactDefault["default"];
  }
});
Object.defineProperty(exports, "FieldType", {
  enumerable: true,
  get: function get() {
    return _FieldType["default"];
  }
});
Object.defineProperty(exports, "FileObject", {
  enumerable: true,
  get: function get() {
    return _FileObject["default"];
  }
});
Object.defineProperty(exports, "FileObjectAllOf", {
  enumerable: true,
  get: function get() {
    return _FileObjectAllOf["default"];
  }
});
Object.defineProperty(exports, "FileObjectsApi", {
  enumerable: true,
  get: function get() {
    return _FileObjectsApi["default"];
  }
});
Object.defineProperty(exports, "FileObjectsResponse", {
  enumerable: true,
  get: function get() {
    return _FileObjectsResponse["default"];
  }
});
Object.defineProperty(exports, "FileRepositoriesApi", {
  enumerable: true,
  get: function get() {
    return _FileRepositoriesApi["default"];
  }
});
Object.defineProperty(exports, "HostingOptions", {
  enumerable: true,
  get: function get() {
    return _HostingOptions["default"];
  }
});
Object.defineProperty(exports, "KafkaConnection", {
  enumerable: true,
  get: function get() {
    return _KafkaConnection["default"];
  }
});
Object.defineProperty(exports, "KafkaConnectionAllOf", {
  enumerable: true,
  get: function get() {
    return _KafkaConnectionAllOf["default"];
  }
});
Object.defineProperty(exports, "KafkaConnectionResponse", {
  enumerable: true,
  get: function get() {
    return _KafkaConnectionResponse["default"];
  }
});
Object.defineProperty(exports, "Language", {
  enumerable: true,
  get: function get() {
    return _Language["default"];
  }
});
Object.defineProperty(exports, "LanguageResponse", {
  enumerable: true,
  get: function get() {
    return _LanguageResponse["default"];
  }
});
Object.defineProperty(exports, "Languages", {
  enumerable: true,
  get: function get() {
    return _Languages["default"];
  }
});
Object.defineProperty(exports, "LanguagesApi", {
  enumerable: true,
  get: function get() {
    return _LanguagesApi["default"];
  }
});
Object.defineProperty(exports, "Leaderboard", {
  enumerable: true,
  get: function get() {
    return _Leaderboard["default"];
  }
});
Object.defineProperty(exports, "LeaderboardAllOf", {
  enumerable: true,
  get: function get() {
    return _LeaderboardAllOf["default"];
  }
});
Object.defineProperty(exports, "LeaderboardEntry", {
  enumerable: true,
  get: function get() {
    return _LeaderboardEntry["default"];
  }
});
Object.defineProperty(exports, "LeaderboardMember", {
  enumerable: true,
  get: function get() {
    return _LeaderboardMember["default"];
  }
});
Object.defineProperty(exports, "LeaderboardResponseByContest", {
  enumerable: true,
  get: function get() {
    return _LeaderboardResponseByContest["default"];
  }
});
Object.defineProperty(exports, "Location", {
  enumerable: true,
  get: function get() {
    return _Location["default"];
  }
});
Object.defineProperty(exports, "Macro", {
  enumerable: true,
  get: function get() {
    return _Macro["default"];
  }
});
Object.defineProperty(exports, "MatchCondition", {
  enumerable: true,
  get: function get() {
    return _MatchCondition["default"];
  }
});
Object.defineProperty(exports, "MaxMin", {
  enumerable: true,
  get: function get() {
    return _MaxMin["default"];
  }
});
Object.defineProperty(exports, "Member", {
  enumerable: true,
  get: function get() {
    return _Member["default"];
  }
});
Object.defineProperty(exports, "MemberAchievementCount", {
  enumerable: true,
  get: function get() {
    return _MemberAchievementCount["default"];
  }
});
Object.defineProperty(exports, "MemberAchievementCountAllOf", {
  enumerable: true,
  get: function get() {
    return _MemberAchievementCountAllOf["default"];
  }
});
Object.defineProperty(exports, "MemberAchievementIssuedResponse", {
  enumerable: true,
  get: function get() {
    return _MemberAchievementIssuedResponse["default"];
  }
});
Object.defineProperty(exports, "MemberAllOf", {
  enumerable: true,
  get: function get() {
    return _MemberAllOf["default"];
  }
});
Object.defineProperty(exports, "MemberAward", {
  enumerable: true,
  get: function get() {
    return _MemberAward["default"];
  }
});
Object.defineProperty(exports, "MemberAwardAllOf", {
  enumerable: true,
  get: function get() {
    return _MemberAwardAllOf["default"];
  }
});
Object.defineProperty(exports, "MemberAwardResponse", {
  enumerable: true,
  get: function get() {
    return _MemberAwardResponse["default"];
  }
});
Object.defineProperty(exports, "MemberMessage", {
  enumerable: true,
  get: function get() {
    return _MemberMessage["default"];
  }
});
Object.defineProperty(exports, "MemberMessageAllOf", {
  enumerable: true,
  get: function get() {
    return _MemberMessageAllOf["default"];
  }
});
Object.defineProperty(exports, "MemberMessageResponse", {
  enumerable: true,
  get: function get() {
    return _MemberMessageResponse["default"];
  }
});
Object.defineProperty(exports, "MemberMessagesApi", {
  enumerable: true,
  get: function get() {
    return _MemberMessagesApi["default"];
  }
});
Object.defineProperty(exports, "MemberResponse", {
  enumerable: true,
  get: function get() {
    return _MemberResponse["default"];
  }
});
Object.defineProperty(exports, "MemberType", {
  enumerable: true,
  get: function get() {
    return _MemberType["default"];
  }
});
Object.defineProperty(exports, "MembersApi", {
  enumerable: true,
  get: function get() {
    return _MembersApi["default"];
  }
});
Object.defineProperty(exports, "MessageStatus", {
  enumerable: true,
  get: function get() {
    return _MessageStatus["default"];
  }
});
Object.defineProperty(exports, "MessageType", {
  enumerable: true,
  get: function get() {
    return _MessageType["default"];
  }
});
Object.defineProperty(exports, "MetaBasic", {
  enumerable: true,
  get: function get() {
    return _MetaBasic["default"];
  }
});
Object.defineProperty(exports, "MetaExtended", {
  enumerable: true,
  get: function get() {
    return _MetaExtended["default"];
  }
});
Object.defineProperty(exports, "MetaExtendedAllOf", {
  enumerable: true,
  get: function get() {
    return _MetaExtendedAllOf["default"];
  }
});
Object.defineProperty(exports, "MetaInfo", {
  enumerable: true,
  get: function get() {
    return _MetaInfo["default"];
  }
});
Object.defineProperty(exports, "ModelDefault", {
  enumerable: true,
  get: function get() {
    return _ModelDefault["default"];
  }
});
Object.defineProperty(exports, "Notification", {
  enumerable: true,
  get: function get() {
    return _Notification["default"];
  }
});
Object.defineProperty(exports, "NotificationAllOf", {
  enumerable: true,
  get: function get() {
    return _NotificationAllOf["default"];
  }
});
Object.defineProperty(exports, "NotificationResponse", {
  enumerable: true,
  get: function get() {
    return _NotificationResponse["default"];
  }
});
Object.defineProperty(exports, "NotificationStatus", {
  enumerable: true,
  get: function get() {
    return _NotificationStatus["default"];
  }
});
Object.defineProperty(exports, "NotificationsApi", {
  enumerable: true,
  get: function get() {
    return _NotificationsApi["default"];
  }
});
Object.defineProperty(exports, "OptParamModels", {
  enumerable: true,
  get: function get() {
    return _OptParamModels["default"];
  }
});
Object.defineProperty(exports, "Product", {
  enumerable: true,
  get: function get() {
    return _Product["default"];
  }
});
Object.defineProperty(exports, "ProductAllOf", {
  enumerable: true,
  get: function get() {
    return _ProductAllOf["default"];
  }
});
Object.defineProperty(exports, "ProductReduced", {
  enumerable: true,
  get: function get() {
    return _ProductReduced["default"];
  }
});
Object.defineProperty(exports, "ProductReducedAllOf", {
  enumerable: true,
  get: function get() {
    return _ProductReducedAllOf["default"];
  }
});
Object.defineProperty(exports, "ProductResponse", {
  enumerable: true,
  get: function get() {
    return _ProductResponse["default"];
  }
});
Object.defineProperty(exports, "ProductsApi", {
  enumerable: true,
  get: function get() {
    return _ProductsApi["default"];
  }
});
Object.defineProperty(exports, "ProfileResponse", {
  enumerable: true,
  get: function get() {
    return _ProfileResponse["default"];
  }
});
Object.defineProperty(exports, "QueryMultiple", {
  enumerable: true,
  get: function get() {
    return _QueryMultiple["default"];
  }
});
Object.defineProperty(exports, "QueryMultipleFields", {
  enumerable: true,
  get: function get() {
    return _QueryMultipleFields["default"];
  }
});
Object.defineProperty(exports, "QueryRequest", {
  enumerable: true,
  get: function get() {
    return _QueryRequest["default"];
  }
});
Object.defineProperty(exports, "QuerySingle", {
  enumerable: true,
  get: function get() {
    return _QuerySingle["default"];
  }
});
Object.defineProperty(exports, "QuerySortBy", {
  enumerable: true,
  get: function get() {
    return _QuerySortBy["default"];
  }
});
Object.defineProperty(exports, "RabbitMqConnection", {
  enumerable: true,
  get: function get() {
    return _RabbitMqConnection["default"];
  }
});
Object.defineProperty(exports, "RabbitMqConnectionAllOf", {
  enumerable: true,
  get: function get() {
    return _RabbitMqConnectionAllOf["default"];
  }
});
Object.defineProperty(exports, "RabbitMqConnectionResponse", {
  enumerable: true,
  get: function get() {
    return _RabbitMqConnectionResponse["default"];
  }
});
Object.defineProperty(exports, "RangeQuery", {
  enumerable: true,
  get: function get() {
    return _RangeQuery["default"];
  }
});
Object.defineProperty(exports, "RankingStrategy", {
  enumerable: true,
  get: function get() {
    return _RankingStrategy["default"];
  }
});
Object.defineProperty(exports, "Relation", {
  enumerable: true,
  get: function get() {
    return _Relation["default"];
  }
});
Object.defineProperty(exports, "Repository", {
  enumerable: true,
  get: function get() {
    return _Repository["default"];
  }
});
Object.defineProperty(exports, "RepositoryAllOf", {
  enumerable: true,
  get: function get() {
    return _RepositoryAllOf["default"];
  }
});
Object.defineProperty(exports, "RepositoryResponse", {
  enumerable: true,
  get: function get() {
    return _RepositoryResponse["default"];
  }
});
Object.defineProperty(exports, "Response", {
  enumerable: true,
  get: function get() {
    return _Response["default"];
  }
});
Object.defineProperty(exports, "Result", {
  enumerable: true,
  get: function get() {
    return _Result["default"];
  }
});
Object.defineProperty(exports, "Reward", {
  enumerable: true,
  get: function get() {
    return _Reward["default"];
  }
});
Object.defineProperty(exports, "RewardAllOf", {
  enumerable: true,
  get: function get() {
    return _RewardAllOf["default"];
  }
});
Object.defineProperty(exports, "RewardEntityType", {
  enumerable: true,
  get: function get() {
    return _RewardEntityType["default"];
  }
});
Object.defineProperty(exports, "RewardReduced", {
  enumerable: true,
  get: function get() {
    return _RewardReduced["default"];
  }
});
Object.defineProperty(exports, "RewardReducedAllOf", {
  enumerable: true,
  get: function get() {
    return _RewardReducedAllOf["default"];
  }
});
Object.defineProperty(exports, "RewardResponse", {
  enumerable: true,
  get: function get() {
    return _RewardResponse["default"];
  }
});
Object.defineProperty(exports, "RewardType", {
  enumerable: true,
  get: function get() {
    return _RewardType["default"];
  }
});
Object.defineProperty(exports, "RewardTypeAllOf", {
  enumerable: true,
  get: function get() {
    return _RewardTypeAllOf["default"];
  }
});
Object.defineProperty(exports, "RewardTypeReduced", {
  enumerable: true,
  get: function get() {
    return _RewardTypeReduced["default"];
  }
});
Object.defineProperty(exports, "RewardTypeReducedAllOf", {
  enumerable: true,
  get: function get() {
    return _RewardTypeReducedAllOf["default"];
  }
});
Object.defineProperty(exports, "RewardTypeResponse", {
  enumerable: true,
  get: function get() {
    return _RewardTypeResponse["default"];
  }
});
Object.defineProperty(exports, "RewardTypesApi", {
  enumerable: true,
  get: function get() {
    return _RewardTypesApi["default"];
  }
});
Object.defineProperty(exports, "RewardsApi", {
  enumerable: true,
  get: function get() {
    return _RewardsApi["default"];
  }
});
Object.defineProperty(exports, "Role", {
  enumerable: true,
  get: function get() {
    return _Role["default"];
  }
});
Object.defineProperty(exports, "RoundType", {
  enumerable: true,
  get: function get() {
    return _RoundType["default"];
  }
});
Object.defineProperty(exports, "Rule", {
  enumerable: true,
  get: function get() {
    return _Rule["default"];
  }
});
Object.defineProperty(exports, "RuleArgument", {
  enumerable: true,
  get: function get() {
    return _RuleArgument["default"];
  }
});
Object.defineProperty(exports, "RuleExpression", {
  enumerable: true,
  get: function get() {
    return _RuleExpression["default"];
  }
});
Object.defineProperty(exports, "RuleOperandLookupRequest", {
  enumerable: true,
  get: function get() {
    return _RuleOperandLookupRequest["default"];
  }
});
Object.defineProperty(exports, "RuleOperandLookupResponse", {
  enumerable: true,
  get: function get() {
    return _RuleOperandLookupResponse["default"];
  }
});
Object.defineProperty(exports, "RuleResponse", {
  enumerable: true,
  get: function get() {
    return _RuleResponse["default"];
  }
});
Object.defineProperty(exports, "RuleRoutine", {
  enumerable: true,
  get: function get() {
    return _RuleRoutine["default"];
  }
});
Object.defineProperty(exports, "RulesApi", {
  enumerable: true,
  get: function get() {
    return _RulesApi["default"];
  }
});
Object.defineProperty(exports, "ScheduleType", {
  enumerable: true,
  get: function get() {
    return _ScheduleType["default"];
  }
});
Object.defineProperty(exports, "Scheduling", {
  enumerable: true,
  get: function get() {
    return _Scheduling["default"];
  }
});
Object.defineProperty(exports, "ScoringStrategy", {
  enumerable: true,
  get: function get() {
    return _ScoringStrategy["default"];
  }
});
Object.defineProperty(exports, "SortOrder", {
  enumerable: true,
  get: function get() {
    return _SortOrder["default"];
  }
});
Object.defineProperty(exports, "SourceLanguage", {
  enumerable: true,
  get: function get() {
    return _SourceLanguage["default"];
  }
});
Object.defineProperty(exports, "Space", {
  enumerable: true,
  get: function get() {
    return _Space["default"];
  }
});
Object.defineProperty(exports, "SpaceAllOf", {
  enumerable: true,
  get: function get() {
    return _SpaceAllOf["default"];
  }
});
Object.defineProperty(exports, "SpaceResponse", {
  enumerable: true,
  get: function get() {
    return _SpaceResponse["default"];
  }
});
Object.defineProperty(exports, "SpaceTokenRequest", {
  enumerable: true,
  get: function get() {
    return _SpaceTokenRequest["default"];
  }
});
Object.defineProperty(exports, "SpacesApi", {
  enumerable: true,
  get: function get() {
    return _SpacesApi["default"];
  }
});
Object.defineProperty(exports, "SqsConnection", {
  enumerable: true,
  get: function get() {
    return _SqsConnection["default"];
  }
});
Object.defineProperty(exports, "SqsConnectionAllOf", {
  enumerable: true,
  get: function get() {
    return _SqsConnectionAllOf["default"];
  }
});
Object.defineProperty(exports, "SqsConnectionResponse", {
  enumerable: true,
  get: function get() {
    return _SqsConnectionResponse["default"];
  }
});
Object.defineProperty(exports, "StartEndDate", {
  enumerable: true,
  get: function get() {
    return _StartEndDate["default"];
  }
});
Object.defineProperty(exports, "Strategy", {
  enumerable: true,
  get: function get() {
    return _Strategy["default"];
  }
});
Object.defineProperty(exports, "StrategyType", {
  enumerable: true,
  get: function get() {
    return _StrategyType["default"];
  }
});
Object.defineProperty(exports, "Tag", {
  enumerable: true,
  get: function get() {
    return _Tag["default"];
  }
});
Object.defineProperty(exports, "TagAllOf", {
  enumerable: true,
  get: function get() {
    return _TagAllOf["default"];
  }
});
Object.defineProperty(exports, "TagResponse", {
  enumerable: true,
  get: function get() {
    return _TagResponse["default"];
  }
});
Object.defineProperty(exports, "TagsApi", {
  enumerable: true,
  get: function get() {
    return _TagsApi["default"];
  }
});
Object.defineProperty(exports, "Transformer", {
  enumerable: true,
  get: function get() {
    return _Transformer["default"];
  }
});
Object.defineProperty(exports, "TransformerAllOf", {
  enumerable: true,
  get: function get() {
    return _TransformerAllOf["default"];
  }
});
Object.defineProperty(exports, "TransformerResponse", {
  enumerable: true,
  get: function get() {
    return _TransformerResponse["default"];
  }
});
Object.defineProperty(exports, "TransformersApi", {
  enumerable: true,
  get: function get() {
    return _TransformersApi["default"];
  }
});
Object.defineProperty(exports, "TranslatedField", {
  enumerable: true,
  get: function get() {
    return _TranslatedField["default"];
  }
});
Object.defineProperty(exports, "Translation", {
  enumerable: true,
  get: function get() {
    return _Translation["default"];
  }
});
Object.defineProperty(exports, "TranslationEntry", {
  enumerable: true,
  get: function get() {
    return _TranslationEntry["default"];
  }
});
Object.defineProperty(exports, "TranslationResponse", {
  enumerable: true,
  get: function get() {
    return _TranslationResponse["default"];
  }
});
Object.defineProperty(exports, "TranslationValue", {
  enumerable: true,
  get: function get() {
    return _TranslationValue["default"];
  }
});
Object.defineProperty(exports, "TranslationsApi", {
  enumerable: true,
  get: function get() {
    return _TranslationsApi["default"];
  }
});
Object.defineProperty(exports, "Triggers", {
  enumerable: true,
  get: function get() {
    return _Triggers["default"];
  }
});
Object.defineProperty(exports, "UnitOfMeasure", {
  enumerable: true,
  get: function get() {
    return _UnitOfMeasure["default"];
  }
});
Object.defineProperty(exports, "UnitOfMeasureAllOf", {
  enumerable: true,
  get: function get() {
    return _UnitOfMeasureAllOf["default"];
  }
});
Object.defineProperty(exports, "UnitOfMeasureResponse", {
  enumerable: true,
  get: function get() {
    return _UnitOfMeasureResponse["default"];
  }
});
Object.defineProperty(exports, "UnitOfMeasureType", {
  enumerable: true,
  get: function get() {
    return _UnitOfMeasureType["default"];
  }
});
Object.defineProperty(exports, "UnitsOfMeasureApi", {
  enumerable: true,
  get: function get() {
    return _UnitsOfMeasureApi["default"];
  }
});
Object.defineProperty(exports, "UpdateAchievementRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateAchievementRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateAchievementRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateAchievementRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateActionTypeRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateActionTypeRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateActionTypeRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateActionTypeRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateApiKeyRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateApiKeyRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateApiKeyRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateApiKeyRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateCollaboratorRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateCollaboratorRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateCollaboratorRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateCollaboratorRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateCompetitionRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateCompetitionRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateCompetitionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateCompetitionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateCompetitionStatusRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateCompetitionStatusRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateCompetitionStatusRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateCompetitionStatusRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateComputeEngineRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateComputeEngineRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateComputeEngineRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateComputeEngineRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateConnectionStateRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateConnectionStateRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateContestRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateContestRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateContestRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateContestRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateContestStateRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateContestStateRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateCustomFieldRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateCustomFieldRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateCustomFieldRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateCustomFieldRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateEntityRewardRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateEntityRewardRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateEntityRewardRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateEntityRewardRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateFileObjectRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateFileObjectRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateFileObjectRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateFileObjectRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateKafkaConnectionRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateKafkaConnectionRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateKafkaConnectionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateKafkaConnectionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateLanguageRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateLanguageRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateMemberRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateMemberRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateMemberRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateMemberRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateMessageStatusRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateMessageStatusRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateMessageStatusRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateMessageStatusRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateModelDefault", {
  enumerable: true,
  get: function get() {
    return _UpdateModelDefault["default"];
  }
});
Object.defineProperty(exports, "UpdateNotificationStatusRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateNotificationStatusRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateNotificationStatusRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateNotificationStatusRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateOptParamModels", {
  enumerable: true,
  get: function get() {
    return _UpdateOptParamModels["default"];
  }
});
Object.defineProperty(exports, "UpdateProductRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateProductRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateProductRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateProductRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateRabbitMqConnectionRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateRabbitMqConnectionRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateRabbitMqConnectionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateRabbitMqConnectionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateRepositoryRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateRepositoryRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateRepositoryRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateRepositoryRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateRewardRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateRewardRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateRewardRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateRewardRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateRewardTypeRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateRewardTypeRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateRewardTypeRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateRewardTypeRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateRuleRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateRuleRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateRuleRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateRuleRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateSpaceRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateSpaceRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateSqsConnectionRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateSqsConnectionRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateSqsConnectionRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateSqsConnectionRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateStateRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateStateRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateStateRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateStateRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateTagRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateTagRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateTagRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateTagRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateTransformerRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateTransformerRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateTransformerRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateTransformerRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateTranslationRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateTranslationRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateTranslationRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateTranslationRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateUnitOfMeasureRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateUnitOfMeasureRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateUnitOfMeasureRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateUnitOfMeasureRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "UpdateWebhookRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateWebhookRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateWebhookRequestAllOf", {
  enumerable: true,
  get: function get() {
    return _UpdateWebhookRequestAllOf["default"];
  }
});
Object.defineProperty(exports, "User", {
  enumerable: true,
  get: function get() {
    return _User["default"];
  }
});
Object.defineProperty(exports, "UserAllOf", {
  enumerable: true,
  get: function get() {
    return _UserAllOf["default"];
  }
});
Object.defineProperty(exports, "UserApi", {
  enumerable: true,
  get: function get() {
    return _UserApi["default"];
  }
});
Object.defineProperty(exports, "UserModelDefault", {
  enumerable: true,
  get: function get() {
    return _UserModelDefault["default"];
  }
});
Object.defineProperty(exports, "UserResponse", {
  enumerable: true,
  get: function get() {
    return _UserResponse["default"];
  }
});
Object.defineProperty(exports, "ValueLookupResult", {
  enumerable: true,
  get: function get() {
    return _ValueLookupResult["default"];
  }
});
Object.defineProperty(exports, "Webhook", {
  enumerable: true,
  get: function get() {
    return _Webhook["default"];
  }
});
Object.defineProperty(exports, "WebhookAllOf", {
  enumerable: true,
  get: function get() {
    return _WebhookAllOf["default"];
  }
});
Object.defineProperty(exports, "WebhookResponse", {
  enumerable: true,
  get: function get() {
    return _WebhookResponse["default"];
  }
});
Object.defineProperty(exports, "WebhookTriggersResponse", {
  enumerable: true,
  get: function get() {
    return _WebhookTriggersResponse["default"];
  }
});
Object.defineProperty(exports, "WebhooksApi", {
  enumerable: true,
  get: function get() {
    return _WebhooksApi["default"];
  }
});

var _ApiClient = _interopRequireDefault(require("./ApiClient"));

var _AccountMessage = _interopRequireDefault(require("./model/AccountMessage"));

var _AccountMessageAllOf = _interopRequireDefault(require("./model/AccountMessageAllOf"));

var _AccountMessageResponse = _interopRequireDefault(require("./model/AccountMessageResponse"));

var _Achievement = _interopRequireDefault(require("./model/Achievement"));

var _AchievementAllOf = _interopRequireDefault(require("./model/AchievementAllOf"));

var _AchievementIssuedCount = _interopRequireDefault(require("./model/AchievementIssuedCount"));

var _AchievementIssuedResponse = _interopRequireDefault(require("./model/AchievementIssuedResponse"));

var _AchievementLiveStatus = _interopRequireDefault(require("./model/AchievementLiveStatus"));

var _AchievementLiveStatusRequest = _interopRequireDefault(require("./model/AchievementLiveStatusRequest"));

var _AchievementLiveStatusRequestAllOf = _interopRequireDefault(require("./model/AchievementLiveStatusRequestAllOf"));

var _AchievementReduced = _interopRequireDefault(require("./model/AchievementReduced"));

var _AchievementReducedAllOf = _interopRequireDefault(require("./model/AchievementReducedAllOf"));

var _AchievementReducedResponse = _interopRequireDefault(require("./model/AchievementReducedResponse"));

var _AchievementResponse = _interopRequireDefault(require("./model/AchievementResponse"));

var _AchievementStatus = _interopRequireDefault(require("./model/AchievementStatus"));

var _ActionType = _interopRequireDefault(require("./model/ActionType"));

var _ActionTypeAdjustmentFactor = _interopRequireDefault(require("./model/ActionTypeAdjustmentFactor"));

var _ActionTypeAllOf = _interopRequireDefault(require("./model/ActionTypeAllOf"));

var _ActionTypeResponse = _interopRequireDefault(require("./model/ActionTypeResponse"));

var _ApiKey = _interopRequireDefault(require("./model/ApiKey"));

var _ApiKeyAllOf = _interopRequireDefault(require("./model/ApiKeyAllOf"));

var _ApiKeyResponse = _interopRequireDefault(require("./model/ApiKeyResponse"));

var _ApiResponse = _interopRequireDefault(require("./model/ApiResponse"));

var _AppliesTo = _interopRequireDefault(require("./model/AppliesTo"));

var _AvailableLanguageResponse = _interopRequireDefault(require("./model/AvailableLanguageResponse"));

var _Award = _interopRequireDefault(require("./model/Award"));

var _AwardAllOf = _interopRequireDefault(require("./model/AwardAllOf"));

var _AwardResponse = _interopRequireDefault(require("./model/AwardResponse"));

var _Bracket = _interopRequireDefault(require("./model/Bracket"));

var _BracketResponse = _interopRequireDefault(require("./model/BracketResponse"));

var _ChangePasswordRequest = _interopRequireDefault(require("./model/ChangePasswordRequest"));

var _ClaimAwardRequest = _interopRequireDefault(require("./model/ClaimAwardRequest"));

var _CloneAchievementResponse = _interopRequireDefault(require("./model/CloneAchievementResponse"));

var _CloneCompetitionResponse = _interopRequireDefault(require("./model/CloneCompetitionResponse"));

var _CloneContestResponse = _interopRequireDefault(require("./model/CloneContestResponse"));

var _Collaborator = _interopRequireDefault(require("./model/Collaborator"));

var _CollaboratorAllOf = _interopRequireDefault(require("./model/CollaboratorAllOf"));

var _CollaboratorResponse = _interopRequireDefault(require("./model/CollaboratorResponse"));

var _CollaboratorRolesAvailable = _interopRequireDefault(require("./model/CollaboratorRolesAvailable"));

var _CollaboratorRolesAvailableAllOf = _interopRequireDefault(require("./model/CollaboratorRolesAvailableAllOf"));

var _CollaboratorRolesResponse = _interopRequireDefault(require("./model/CollaboratorRolesResponse"));

var _Competition = _interopRequireDefault(require("./model/Competition"));

var _CompetitionAllOf = _interopRequireDefault(require("./model/CompetitionAllOf"));

var _CompetitionLeaderboardResponse = _interopRequireDefault(require("./model/CompetitionLeaderboardResponse"));

var _CompetitionReduced = _interopRequireDefault(require("./model/CompetitionReduced"));

var _CompetitionReducedAllOf = _interopRequireDefault(require("./model/CompetitionReducedAllOf"));

var _CompetitionReducedResponse = _interopRequireDefault(require("./model/CompetitionReducedResponse"));

var _CompetitionResponse = _interopRequireDefault(require("./model/CompetitionResponse"));

var _CompetitionStatus = _interopRequireDefault(require("./model/CompetitionStatus"));

var _CompetitionStatusActions = _interopRequireDefault(require("./model/CompetitionStatusActions"));

var _CompetitionType = _interopRequireDefault(require("./model/CompetitionType"));

var _ComputeEngine = _interopRequireDefault(require("./model/ComputeEngine"));

var _ComputeEngineAllOf = _interopRequireDefault(require("./model/ComputeEngineAllOf"));

var _ComputeEngineResponse = _interopRequireDefault(require("./model/ComputeEngineResponse"));

var _Condition = _interopRequireDefault(require("./model/Condition"));

var _Connection = _interopRequireDefault(require("./model/Connection"));

var _ConnectionResponse = _interopRequireDefault(require("./model/ConnectionResponse"));

var _ConnectionState = _interopRequireDefault(require("./model/ConnectionState"));

var _Contest = _interopRequireDefault(require("./model/Contest"));

var _ContestAllOf = _interopRequireDefault(require("./model/ContestAllOf"));

var _ContestLeaderboardResponse = _interopRequireDefault(require("./model/ContestLeaderboardResponse"));

var _ContestReduced = _interopRequireDefault(require("./model/ContestReduced"));

var _ContestReducedAllOf = _interopRequireDefault(require("./model/ContestReducedAllOf"));

var _ContestReducedResponse = _interopRequireDefault(require("./model/ContestReducedResponse"));

var _ContestResponse = _interopRequireDefault(require("./model/ContestResponse"));

var _ContestStateOperations = _interopRequireDefault(require("./model/ContestStateOperations"));

var _ContestStatus = _interopRequireDefault(require("./model/ContestStatus"));

var _CountResponse = _interopRequireDefault(require("./model/CountResponse"));

var _CreateAchievementRequest = _interopRequireDefault(require("./model/CreateAchievementRequest"));

var _CreateAchievementRequestAllOf = _interopRequireDefault(require("./model/CreateAchievementRequestAllOf"));

var _CreateActionTypeRequest = _interopRequireDefault(require("./model/CreateActionTypeRequest"));

var _CreateActionTypeRequestAllOf = _interopRequireDefault(require("./model/CreateActionTypeRequestAllOf"));

var _CreateApiKeyRequest = _interopRequireDefault(require("./model/CreateApiKeyRequest"));

var _CreateApiKeyRequestAllOf = _interopRequireDefault(require("./model/CreateApiKeyRequestAllOf"));

var _CreateCollaboratorRequest = _interopRequireDefault(require("./model/CreateCollaboratorRequest"));

var _CreateCompetitionRequest = _interopRequireDefault(require("./model/CreateCompetitionRequest"));

var _CreateCompetitionRequestAllOf = _interopRequireDefault(require("./model/CreateCompetitionRequestAllOf"));

var _CreateComputeEngineRequest = _interopRequireDefault(require("./model/CreateComputeEngineRequest"));

var _CreateComputeEngineRequestAllOf = _interopRequireDefault(require("./model/CreateComputeEngineRequestAllOf"));

var _CreateContestForCompetitionRequest = _interopRequireDefault(require("./model/CreateContestForCompetitionRequest"));

var _CreateContestForCompetitionRequestAllOf = _interopRequireDefault(require("./model/CreateContestForCompetitionRequestAllOf"));

var _CreateContestRequest = _interopRequireDefault(require("./model/CreateContestRequest"));

var _CreateContestRequestAllOf = _interopRequireDefault(require("./model/CreateContestRequestAllOf"));

var _CreateCustomFieldRequest = _interopRequireDefault(require("./model/CreateCustomFieldRequest"));

var _CreateEntityRewardRequest = _interopRequireDefault(require("./model/CreateEntityRewardRequest"));

var _CreateEntityRewardRequestAllOf = _interopRequireDefault(require("./model/CreateEntityRewardRequestAllOf"));

var _CreateEventRequest = _interopRequireDefault(require("./model/CreateEventRequest"));

var _CreateFileObjectRequest = _interopRequireDefault(require("./model/CreateFileObjectRequest"));

var _CreateFileObjectRequestAllOf = _interopRequireDefault(require("./model/CreateFileObjectRequestAllOf"));

var _CreateKafkaConnectionRequest = _interopRequireDefault(require("./model/CreateKafkaConnectionRequest"));

var _CreateKafkaConnectionRequestAllOf = _interopRequireDefault(require("./model/CreateKafkaConnectionRequestAllOf"));

var _CreateLanguageRequest = _interopRequireDefault(require("./model/CreateLanguageRequest"));

var _CreateLanguageRequestAllOf = _interopRequireDefault(require("./model/CreateLanguageRequestAllOf"));

var _CreateMemberMessageRequest = _interopRequireDefault(require("./model/CreateMemberMessageRequest"));

var _CreateMemberMessageRequestAllOf = _interopRequireDefault(require("./model/CreateMemberMessageRequestAllOf"));

var _CreateMemberRequest = _interopRequireDefault(require("./model/CreateMemberRequest"));

var _CreateMemberRequestAllOf = _interopRequireDefault(require("./model/CreateMemberRequestAllOf"));

var _CreateNotificationRequest = _interopRequireDefault(require("./model/CreateNotificationRequest"));

var _CreateNotificationRequestAllOf = _interopRequireDefault(require("./model/CreateNotificationRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./model/CreateOptParamModels"));

var _CreateProductRequest = _interopRequireDefault(require("./model/CreateProductRequest"));

var _CreateProductRequestAllOf = _interopRequireDefault(require("./model/CreateProductRequestAllOf"));

var _CreateRabbitMqConnectionRequest = _interopRequireDefault(require("./model/CreateRabbitMqConnectionRequest"));

var _CreateRabbitMqConnectionRequestAllOf = _interopRequireDefault(require("./model/CreateRabbitMqConnectionRequestAllOf"));

var _CreateRepositoryRequest = _interopRequireDefault(require("./model/CreateRepositoryRequest"));

var _CreateRepositoryRequestAllOf = _interopRequireDefault(require("./model/CreateRepositoryRequestAllOf"));

var _CreateRewardRequest = _interopRequireDefault(require("./model/CreateRewardRequest"));

var _CreateRewardRequestAllOf = _interopRequireDefault(require("./model/CreateRewardRequestAllOf"));

var _CreateRewardTypeRequest = _interopRequireDefault(require("./model/CreateRewardTypeRequest"));

var _CreateRewardTypeRequestAllOf = _interopRequireDefault(require("./model/CreateRewardTypeRequestAllOf"));

var _CreateSimpleCompetitionRequest = _interopRequireDefault(require("./model/CreateSimpleCompetitionRequest"));

var _CreateSimpleCompetitionRequestAllOf = _interopRequireDefault(require("./model/CreateSimpleCompetitionRequestAllOf"));

var _CreateSpaceRequest = _interopRequireDefault(require("./model/CreateSpaceRequest"));

var _CreateSqsConnectionRequest = _interopRequireDefault(require("./model/CreateSqsConnectionRequest"));

var _CreateSqsConnectionRequestAllOf = _interopRequireDefault(require("./model/CreateSqsConnectionRequestAllOf"));

var _CreateTagRequest = _interopRequireDefault(require("./model/CreateTagRequest"));

var _CreateTransformerRequest = _interopRequireDefault(require("./model/CreateTransformerRequest"));

var _CreateTranslationRequest = _interopRequireDefault(require("./model/CreateTranslationRequest"));

var _CreateUnitOfMeasureRequest = _interopRequireDefault(require("./model/CreateUnitOfMeasureRequest"));

var _CreateUnitOfMeasureRequestAllOf = _interopRequireDefault(require("./model/CreateUnitOfMeasureRequestAllOf"));

var _CreateWebhookRequest = _interopRequireDefault(require("./model/CreateWebhookRequest"));

var _CreateWebhookRequestAllOf = _interopRequireDefault(require("./model/CreateWebhookRequestAllOf"));

var _CustomField = _interopRequireDefault(require("./model/CustomField"));

var _CustomFieldAllOf = _interopRequireDefault(require("./model/CustomFieldAllOf"));

var _CustomFieldEntry = _interopRequireDefault(require("./model/CustomFieldEntry"));

var _CustomFieldEntryNumber = _interopRequireDefault(require("./model/CustomFieldEntryNumber"));

var _CustomFieldEntryNumberAllOf = _interopRequireDefault(require("./model/CustomFieldEntryNumberAllOf"));

var _CustomFieldEntryNumberArray = _interopRequireDefault(require("./model/CustomFieldEntryNumberArray"));

var _CustomFieldEntryNumberArrayAllOf = _interopRequireDefault(require("./model/CustomFieldEntryNumberArrayAllOf"));

var _CustomFieldEntryString = _interopRequireDefault(require("./model/CustomFieldEntryString"));

var _CustomFieldEntryStringAllOf = _interopRequireDefault(require("./model/CustomFieldEntryStringAllOf"));

var _CustomFieldEntryStringArray = _interopRequireDefault(require("./model/CustomFieldEntryStringArray"));

var _CustomFieldEntryStringArrayAllOf = _interopRequireDefault(require("./model/CustomFieldEntryStringArrayAllOf"));

var _CustomFieldReduced = _interopRequireDefault(require("./model/CustomFieldReduced"));

var _CustomFieldReducedAllOf = _interopRequireDefault(require("./model/CustomFieldReducedAllOf"));

var _CustomFieldResponse = _interopRequireDefault(require("./model/CustomFieldResponse"));

var _Dependancy = _interopRequireDefault(require("./model/Dependancy"));

var _DependantOn = _interopRequireDefault(require("./model/DependantOn"));

var _EntityType = _interopRequireDefault(require("./model/EntityType"));

var _Entrant = _interopRequireDefault(require("./model/Entrant"));

var _EntrantAction = _interopRequireDefault(require("./model/EntrantAction"));

var _EntrantRequest = _interopRequireDefault(require("./model/EntrantRequest"));

var _EntrantResponse = _interopRequireDefault(require("./model/EntrantResponse"));

var _EntrantStatus = _interopRequireDefault(require("./model/EntrantStatus"));

var _Error = _interopRequireDefault(require("./model/Error"));

var _ErrorDetail = _interopRequireDefault(require("./model/ErrorDetail"));

var _Event = _interopRequireDefault(require("./model/Event"));

var _EventAllOf = _interopRequireDefault(require("./model/EventAllOf"));

var _EventRefType = _interopRequireDefault(require("./model/EventRefType"));

var _EventResponse = _interopRequireDefault(require("./model/EventResponse"));

var _FactDefault = _interopRequireDefault(require("./model/FactDefault"));

var _FieldType = _interopRequireDefault(require("./model/FieldType"));

var _FileObject = _interopRequireDefault(require("./model/FileObject"));

var _FileObjectAllOf = _interopRequireDefault(require("./model/FileObjectAllOf"));

var _FileObjectsResponse = _interopRequireDefault(require("./model/FileObjectsResponse"));

var _HostingOptions = _interopRequireDefault(require("./model/HostingOptions"));

var _KafkaConnection = _interopRequireDefault(require("./model/KafkaConnection"));

var _KafkaConnectionAllOf = _interopRequireDefault(require("./model/KafkaConnectionAllOf"));

var _KafkaConnectionResponse = _interopRequireDefault(require("./model/KafkaConnectionResponse"));

var _Language = _interopRequireDefault(require("./model/Language"));

var _LanguageResponse = _interopRequireDefault(require("./model/LanguageResponse"));

var _Languages = _interopRequireDefault(require("./model/Languages"));

var _Leaderboard = _interopRequireDefault(require("./model/Leaderboard"));

var _LeaderboardAllOf = _interopRequireDefault(require("./model/LeaderboardAllOf"));

var _LeaderboardEntry = _interopRequireDefault(require("./model/LeaderboardEntry"));

var _LeaderboardMember = _interopRequireDefault(require("./model/LeaderboardMember"));

var _LeaderboardResponseByContest = _interopRequireDefault(require("./model/LeaderboardResponseByContest"));

var _Location = _interopRequireDefault(require("./model/Location"));

var _Macro = _interopRequireDefault(require("./model/Macro"));

var _MatchCondition = _interopRequireDefault(require("./model/MatchCondition"));

var _MaxMin = _interopRequireDefault(require("./model/MaxMin"));

var _Member = _interopRequireDefault(require("./model/Member"));

var _MemberAchievementCount = _interopRequireDefault(require("./model/MemberAchievementCount"));

var _MemberAchievementCountAllOf = _interopRequireDefault(require("./model/MemberAchievementCountAllOf"));

var _MemberAchievementIssuedResponse = _interopRequireDefault(require("./model/MemberAchievementIssuedResponse"));

var _MemberAllOf = _interopRequireDefault(require("./model/MemberAllOf"));

var _MemberAward = _interopRequireDefault(require("./model/MemberAward"));

var _MemberAwardAllOf = _interopRequireDefault(require("./model/MemberAwardAllOf"));

var _MemberAwardResponse = _interopRequireDefault(require("./model/MemberAwardResponse"));

var _MemberMessage = _interopRequireDefault(require("./model/MemberMessage"));

var _MemberMessageAllOf = _interopRequireDefault(require("./model/MemberMessageAllOf"));

var _MemberMessageResponse = _interopRequireDefault(require("./model/MemberMessageResponse"));

var _MemberResponse = _interopRequireDefault(require("./model/MemberResponse"));

var _MemberType = _interopRequireDefault(require("./model/MemberType"));

var _MessageStatus = _interopRequireDefault(require("./model/MessageStatus"));

var _MessageType = _interopRequireDefault(require("./model/MessageType"));

var _MetaBasic = _interopRequireDefault(require("./model/MetaBasic"));

var _MetaExtended = _interopRequireDefault(require("./model/MetaExtended"));

var _MetaExtendedAllOf = _interopRequireDefault(require("./model/MetaExtendedAllOf"));

var _MetaInfo = _interopRequireDefault(require("./model/MetaInfo"));

var _ModelDefault = _interopRequireDefault(require("./model/ModelDefault"));

var _Notification = _interopRequireDefault(require("./model/Notification"));

var _NotificationAllOf = _interopRequireDefault(require("./model/NotificationAllOf"));

var _NotificationResponse = _interopRequireDefault(require("./model/NotificationResponse"));

var _NotificationStatus = _interopRequireDefault(require("./model/NotificationStatus"));

var _OptParamModels = _interopRequireDefault(require("./model/OptParamModels"));

var _Product = _interopRequireDefault(require("./model/Product"));

var _ProductAllOf = _interopRequireDefault(require("./model/ProductAllOf"));

var _ProductReduced = _interopRequireDefault(require("./model/ProductReduced"));

var _ProductReducedAllOf = _interopRequireDefault(require("./model/ProductReducedAllOf"));

var _ProductResponse = _interopRequireDefault(require("./model/ProductResponse"));

var _ProfileResponse = _interopRequireDefault(require("./model/ProfileResponse"));

var _QueryMultiple = _interopRequireDefault(require("./model/QueryMultiple"));

var _QueryMultipleFields = _interopRequireDefault(require("./model/QueryMultipleFields"));

var _QueryRequest = _interopRequireDefault(require("./model/QueryRequest"));

var _QuerySingle = _interopRequireDefault(require("./model/QuerySingle"));

var _QuerySortBy = _interopRequireDefault(require("./model/QuerySortBy"));

var _RabbitMqConnection = _interopRequireDefault(require("./model/RabbitMqConnection"));

var _RabbitMqConnectionAllOf = _interopRequireDefault(require("./model/RabbitMqConnectionAllOf"));

var _RabbitMqConnectionResponse = _interopRequireDefault(require("./model/RabbitMqConnectionResponse"));

var _RangeQuery = _interopRequireDefault(require("./model/RangeQuery"));

var _RankingStrategy = _interopRequireDefault(require("./model/RankingStrategy"));

var _Relation = _interopRequireDefault(require("./model/Relation"));

var _Repository = _interopRequireDefault(require("./model/Repository"));

var _RepositoryAllOf = _interopRequireDefault(require("./model/RepositoryAllOf"));

var _RepositoryResponse = _interopRequireDefault(require("./model/RepositoryResponse"));

var _Response = _interopRequireDefault(require("./model/Response"));

var _Result = _interopRequireDefault(require("./model/Result"));

var _Reward = _interopRequireDefault(require("./model/Reward"));

var _RewardAllOf = _interopRequireDefault(require("./model/RewardAllOf"));

var _RewardEntityType = _interopRequireDefault(require("./model/RewardEntityType"));

var _RewardReduced = _interopRequireDefault(require("./model/RewardReduced"));

var _RewardReducedAllOf = _interopRequireDefault(require("./model/RewardReducedAllOf"));

var _RewardResponse = _interopRequireDefault(require("./model/RewardResponse"));

var _RewardType = _interopRequireDefault(require("./model/RewardType"));

var _RewardTypeAllOf = _interopRequireDefault(require("./model/RewardTypeAllOf"));

var _RewardTypeReduced = _interopRequireDefault(require("./model/RewardTypeReduced"));

var _RewardTypeReducedAllOf = _interopRequireDefault(require("./model/RewardTypeReducedAllOf"));

var _RewardTypeResponse = _interopRequireDefault(require("./model/RewardTypeResponse"));

var _Role = _interopRequireDefault(require("./model/Role"));

var _RoundType = _interopRequireDefault(require("./model/RoundType"));

var _Rule = _interopRequireDefault(require("./model/Rule"));

var _RuleArgument = _interopRequireDefault(require("./model/RuleArgument"));

var _RuleExpression = _interopRequireDefault(require("./model/RuleExpression"));

var _RuleOperandLookupRequest = _interopRequireDefault(require("./model/RuleOperandLookupRequest"));

var _RuleOperandLookupResponse = _interopRequireDefault(require("./model/RuleOperandLookupResponse"));

var _RuleResponse = _interopRequireDefault(require("./model/RuleResponse"));

var _RuleRoutine = _interopRequireDefault(require("./model/RuleRoutine"));

var _ScheduleType = _interopRequireDefault(require("./model/ScheduleType"));

var _Scheduling = _interopRequireDefault(require("./model/Scheduling"));

var _ScoringStrategy = _interopRequireDefault(require("./model/ScoringStrategy"));

var _SortOrder = _interopRequireDefault(require("./model/SortOrder"));

var _SourceLanguage = _interopRequireDefault(require("./model/SourceLanguage"));

var _Space = _interopRequireDefault(require("./model/Space"));

var _SpaceAllOf = _interopRequireDefault(require("./model/SpaceAllOf"));

var _SpaceResponse = _interopRequireDefault(require("./model/SpaceResponse"));

var _SpaceTokenRequest = _interopRequireDefault(require("./model/SpaceTokenRequest"));

var _SqsConnection = _interopRequireDefault(require("./model/SqsConnection"));

var _SqsConnectionAllOf = _interopRequireDefault(require("./model/SqsConnectionAllOf"));

var _SqsConnectionResponse = _interopRequireDefault(require("./model/SqsConnectionResponse"));

var _StartEndDate = _interopRequireDefault(require("./model/StartEndDate"));

var _Strategy = _interopRequireDefault(require("./model/Strategy"));

var _StrategyType = _interopRequireDefault(require("./model/StrategyType"));

var _Tag = _interopRequireDefault(require("./model/Tag"));

var _TagAllOf = _interopRequireDefault(require("./model/TagAllOf"));

var _TagResponse = _interopRequireDefault(require("./model/TagResponse"));

var _Transformer = _interopRequireDefault(require("./model/Transformer"));

var _TransformerAllOf = _interopRequireDefault(require("./model/TransformerAllOf"));

var _TransformerResponse = _interopRequireDefault(require("./model/TransformerResponse"));

var _TranslatedField = _interopRequireDefault(require("./model/TranslatedField"));

var _Translation = _interopRequireDefault(require("./model/Translation"));

var _TranslationEntry = _interopRequireDefault(require("./model/TranslationEntry"));

var _TranslationResponse = _interopRequireDefault(require("./model/TranslationResponse"));

var _TranslationValue = _interopRequireDefault(require("./model/TranslationValue"));

var _Triggers = _interopRequireDefault(require("./model/Triggers"));

var _UnitOfMeasure = _interopRequireDefault(require("./model/UnitOfMeasure"));

var _UnitOfMeasureAllOf = _interopRequireDefault(require("./model/UnitOfMeasureAllOf"));

var _UnitOfMeasureResponse = _interopRequireDefault(require("./model/UnitOfMeasureResponse"));

var _UnitOfMeasureType = _interopRequireDefault(require("./model/UnitOfMeasureType"));

var _UpdateAchievementRequest = _interopRequireDefault(require("./model/UpdateAchievementRequest"));

var _UpdateAchievementRequestAllOf = _interopRequireDefault(require("./model/UpdateAchievementRequestAllOf"));

var _UpdateActionTypeRequest = _interopRequireDefault(require("./model/UpdateActionTypeRequest"));

var _UpdateActionTypeRequestAllOf = _interopRequireDefault(require("./model/UpdateActionTypeRequestAllOf"));

var _UpdateApiKeyRequest = _interopRequireDefault(require("./model/UpdateApiKeyRequest"));

var _UpdateApiKeyRequestAllOf = _interopRequireDefault(require("./model/UpdateApiKeyRequestAllOf"));

var _UpdateCollaboratorRequest = _interopRequireDefault(require("./model/UpdateCollaboratorRequest"));

var _UpdateCollaboratorRequestAllOf = _interopRequireDefault(require("./model/UpdateCollaboratorRequestAllOf"));

var _UpdateCompetitionRequest = _interopRequireDefault(require("./model/UpdateCompetitionRequest"));

var _UpdateCompetitionRequestAllOf = _interopRequireDefault(require("./model/UpdateCompetitionRequestAllOf"));

var _UpdateCompetitionStatusRequest = _interopRequireDefault(require("./model/UpdateCompetitionStatusRequest"));

var _UpdateCompetitionStatusRequestAllOf = _interopRequireDefault(require("./model/UpdateCompetitionStatusRequestAllOf"));

var _UpdateComputeEngineRequest = _interopRequireDefault(require("./model/UpdateComputeEngineRequest"));

var _UpdateComputeEngineRequestAllOf = _interopRequireDefault(require("./model/UpdateComputeEngineRequestAllOf"));

var _UpdateConnectionStateRequest = _interopRequireDefault(require("./model/UpdateConnectionStateRequest"));

var _UpdateContestRequest = _interopRequireDefault(require("./model/UpdateContestRequest"));

var _UpdateContestRequestAllOf = _interopRequireDefault(require("./model/UpdateContestRequestAllOf"));

var _UpdateContestStateRequest = _interopRequireDefault(require("./model/UpdateContestStateRequest"));

var _UpdateCustomFieldRequest = _interopRequireDefault(require("./model/UpdateCustomFieldRequest"));

var _UpdateCustomFieldRequestAllOf = _interopRequireDefault(require("./model/UpdateCustomFieldRequestAllOf"));

var _UpdateEntityRewardRequest = _interopRequireDefault(require("./model/UpdateEntityRewardRequest"));

var _UpdateEntityRewardRequestAllOf = _interopRequireDefault(require("./model/UpdateEntityRewardRequestAllOf"));

var _UpdateFileObjectRequest = _interopRequireDefault(require("./model/UpdateFileObjectRequest"));

var _UpdateFileObjectRequestAllOf = _interopRequireDefault(require("./model/UpdateFileObjectRequestAllOf"));

var _UpdateKafkaConnectionRequest = _interopRequireDefault(require("./model/UpdateKafkaConnectionRequest"));

var _UpdateKafkaConnectionRequestAllOf = _interopRequireDefault(require("./model/UpdateKafkaConnectionRequestAllOf"));

var _UpdateLanguageRequest = _interopRequireDefault(require("./model/UpdateLanguageRequest"));

var _UpdateMemberRequest = _interopRequireDefault(require("./model/UpdateMemberRequest"));

var _UpdateMemberRequestAllOf = _interopRequireDefault(require("./model/UpdateMemberRequestAllOf"));

var _UpdateMessageStatusRequest = _interopRequireDefault(require("./model/UpdateMessageStatusRequest"));

var _UpdateMessageStatusRequestAllOf = _interopRequireDefault(require("./model/UpdateMessageStatusRequestAllOf"));

var _UpdateModelDefault = _interopRequireDefault(require("./model/UpdateModelDefault"));

var _UpdateNotificationStatusRequest = _interopRequireDefault(require("./model/UpdateNotificationStatusRequest"));

var _UpdateNotificationStatusRequestAllOf = _interopRequireDefault(require("./model/UpdateNotificationStatusRequestAllOf"));

var _UpdateOptParamModels = _interopRequireDefault(require("./model/UpdateOptParamModels"));

var _UpdateProductRequest = _interopRequireDefault(require("./model/UpdateProductRequest"));

var _UpdateProductRequestAllOf = _interopRequireDefault(require("./model/UpdateProductRequestAllOf"));

var _UpdateRabbitMqConnectionRequest = _interopRequireDefault(require("./model/UpdateRabbitMqConnectionRequest"));

var _UpdateRabbitMqConnectionRequestAllOf = _interopRequireDefault(require("./model/UpdateRabbitMqConnectionRequestAllOf"));

var _UpdateRepositoryRequest = _interopRequireDefault(require("./model/UpdateRepositoryRequest"));

var _UpdateRepositoryRequestAllOf = _interopRequireDefault(require("./model/UpdateRepositoryRequestAllOf"));

var _UpdateRewardRequest = _interopRequireDefault(require("./model/UpdateRewardRequest"));

var _UpdateRewardRequestAllOf = _interopRequireDefault(require("./model/UpdateRewardRequestAllOf"));

var _UpdateRewardTypeRequest = _interopRequireDefault(require("./model/UpdateRewardTypeRequest"));

var _UpdateRewardTypeRequestAllOf = _interopRequireDefault(require("./model/UpdateRewardTypeRequestAllOf"));

var _UpdateRuleRequest = _interopRequireDefault(require("./model/UpdateRuleRequest"));

var _UpdateRuleRequestAllOf = _interopRequireDefault(require("./model/UpdateRuleRequestAllOf"));

var _UpdateSpaceRequest = _interopRequireDefault(require("./model/UpdateSpaceRequest"));

var _UpdateSqsConnectionRequest = _interopRequireDefault(require("./model/UpdateSqsConnectionRequest"));

var _UpdateSqsConnectionRequestAllOf = _interopRequireDefault(require("./model/UpdateSqsConnectionRequestAllOf"));

var _UpdateStateRequest = _interopRequireDefault(require("./model/UpdateStateRequest"));

var _UpdateStateRequestAllOf = _interopRequireDefault(require("./model/UpdateStateRequestAllOf"));

var _UpdateTagRequest = _interopRequireDefault(require("./model/UpdateTagRequest"));

var _UpdateTagRequestAllOf = _interopRequireDefault(require("./model/UpdateTagRequestAllOf"));

var _UpdateTransformerRequest = _interopRequireDefault(require("./model/UpdateTransformerRequest"));

var _UpdateTransformerRequestAllOf = _interopRequireDefault(require("./model/UpdateTransformerRequestAllOf"));

var _UpdateTranslationRequest = _interopRequireDefault(require("./model/UpdateTranslationRequest"));

var _UpdateTranslationRequestAllOf = _interopRequireDefault(require("./model/UpdateTranslationRequestAllOf"));

var _UpdateUnitOfMeasureRequest = _interopRequireDefault(require("./model/UpdateUnitOfMeasureRequest"));

var _UpdateUnitOfMeasureRequestAllOf = _interopRequireDefault(require("./model/UpdateUnitOfMeasureRequestAllOf"));

var _UpdateWebhookRequest = _interopRequireDefault(require("./model/UpdateWebhookRequest"));

var _UpdateWebhookRequestAllOf = _interopRequireDefault(require("./model/UpdateWebhookRequestAllOf"));

var _User = _interopRequireDefault(require("./model/User"));

var _UserAllOf = _interopRequireDefault(require("./model/UserAllOf"));

var _UserModelDefault = _interopRequireDefault(require("./model/UserModelDefault"));

var _UserResponse = _interopRequireDefault(require("./model/UserResponse"));

var _ValueLookupResult = _interopRequireDefault(require("./model/ValueLookupResult"));

var _Webhook = _interopRequireDefault(require("./model/Webhook"));

var _WebhookAllOf = _interopRequireDefault(require("./model/WebhookAllOf"));

var _WebhookResponse = _interopRequireDefault(require("./model/WebhookResponse"));

var _WebhookTriggersResponse = _interopRequireDefault(require("./model/WebhookTriggersResponse"));

var _AccountMessagesApi = _interopRequireDefault(require("./api/AccountMessagesApi"));

var _AchievementsApi = _interopRequireDefault(require("./api/AchievementsApi"));

var _ActionTypesApi = _interopRequireDefault(require("./api/ActionTypesApi"));

var _ApiKeysApi = _interopRequireDefault(require("./api/ApiKeysApi"));

var _AwardsApi = _interopRequireDefault(require("./api/AwardsApi"));

var _CollaboratorsApi = _interopRequireDefault(require("./api/CollaboratorsApi"));

var _CompetitionsApi = _interopRequireDefault(require("./api/CompetitionsApi"));

var _ComputeEngineApi = _interopRequireDefault(require("./api/ComputeEngineApi"));

var _ConsumersApi = _interopRequireDefault(require("./api/ConsumersApi"));

var _ContestsApi = _interopRequireDefault(require("./api/ContestsApi"));

var _CustomFieldsApi = _interopRequireDefault(require("./api/CustomFieldsApi"));

var _DefaultApi = _interopRequireDefault(require("./api/DefaultApi"));

var _EventsApi = _interopRequireDefault(require("./api/EventsApi"));

var _FileObjectsApi = _interopRequireDefault(require("./api/FileObjectsApi"));

var _FileRepositoriesApi = _interopRequireDefault(require("./api/FileRepositoriesApi"));

var _LanguagesApi = _interopRequireDefault(require("./api/LanguagesApi"));

var _MemberMessagesApi = _interopRequireDefault(require("./api/MemberMessagesApi"));

var _MembersApi = _interopRequireDefault(require("./api/MembersApi"));

var _NotificationsApi = _interopRequireDefault(require("./api/NotificationsApi"));

var _ProductsApi = _interopRequireDefault(require("./api/ProductsApi"));

var _RewardTypesApi = _interopRequireDefault(require("./api/RewardTypesApi"));

var _RewardsApi = _interopRequireDefault(require("./api/RewardsApi"));

var _RulesApi = _interopRequireDefault(require("./api/RulesApi"));

var _SpacesApi = _interopRequireDefault(require("./api/SpacesApi"));

var _TagsApi = _interopRequireDefault(require("./api/TagsApi"));

var _TransformersApi = _interopRequireDefault(require("./api/TransformersApi"));

var _TranslationsApi = _interopRequireDefault(require("./api/TranslationsApi"));

var _UnitsOfMeasureApi = _interopRequireDefault(require("./api/UnitsOfMeasureApi"));

var _UserApi = _interopRequireDefault(require("./api/UserApi"));

var _WebhooksApi = _interopRequireDefault(require("./api/WebhooksApi"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
