// @ziqni-tech/admin-api-client/dist/model/AccountMessageAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EventRefType = _interopRequireDefault(require("./EventRefType"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The AccountMessageAllOf model module.
 * @module model/AccountMessageAllOf
 * @version 0.0.26
 */
var AccountMessageAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AccountMessageAllOf</code>.
   * @alias module:model/AccountMessageAllOf
   * @param eventRefType {module:model/EventRefType} 
   * @param eventRefId {String} The reference ID of the event object
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param status {module:model/MessageStatus} 
   * @param userId {String} 
   */
  function AccountMessageAllOf(eventRefType, eventRefId, messageType, subject, body, status, userId) {
    _classCallCheck(this, AccountMessageAllOf);

    AccountMessageAllOf.initialize(this, eventRefType, eventRefId, messageType, subject, body, status, userId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AccountMessageAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, eventRefType, eventRefId, messageType, subject, body, status, userId) {
      obj['eventRefType'] = eventRefType;
      obj['eventRefId'] = eventRefId;
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['status'] = status;
      obj['userId'] = userId;
    }
    /**
     * Constructs a <code>AccountMessageAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AccountMessageAllOf} obj Optional instance to populate.
     * @return {module:model/AccountMessageAllOf} The populated <code>AccountMessageAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccountMessageAllOf();

        if (data.hasOwnProperty('expiry')) {
          obj['expiry'] = _ApiClient["default"].convertToType(data['expiry'], 'Date');
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _EventRefType["default"].constructFromObject(data['eventRefType']);
        }

        if (data.hasOwnProperty('eventRefId')) {
          obj['eventRefId'] = _ApiClient["default"].convertToType(data['eventRefId'], 'String');
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AccountMessageAllOf;
}();
/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */


AccountMessageAllOf.prototype['expiry'] = undefined;
/**
 * @member {module:model/EventRefType} eventRefType
 */

AccountMessageAllOf.prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

AccountMessageAllOf.prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

AccountMessageAllOf.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

AccountMessageAllOf.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

AccountMessageAllOf.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

AccountMessageAllOf.prototype['status'] = undefined;
/**
 * @member {String} userId
 */

AccountMessageAllOf.prototype['userId'] = undefined;
var _default = AccountMessageAllOf;
exports["default"] = _default;
