// @ziqni-tech/admin-api-client/dist/model/AccountMessage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccountMessageAllOf = _interopRequireDefault(require("./AccountMessageAllOf"));

var _EventRefType = _interopRequireDefault(require("./EventRefType"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The AccountMessage model module.
 * @module model/AccountMessage
 * @version 0.0.26
 */
var AccountMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AccountMessage</code>.
   * @alias module:model/AccountMessage
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/AccountMessageAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param eventRefType {module:model/EventRefType} 
   * @param eventRefId {String} The reference ID of the event object
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param status {module:model/MessageStatus} 
   * @param userId {String} 
   */
  function AccountMessage(id, spaceName, created, eventRefType, eventRefId, messageType, subject, body, status, userId) {
    _classCallCheck(this, AccountMessage);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _AccountMessageAllOf["default"].initialize(this, eventRefType, eventRefId, messageType, subject, body, status, userId);

    AccountMessage.initialize(this, id, spaceName, created, eventRefType, eventRefId, messageType, subject, body, status, userId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AccountMessage, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, eventRefType, eventRefId, messageType, subject, body, status, userId) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['eventRefType'] = eventRefType;
      obj['eventRefId'] = eventRefId;
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['status'] = status;
      obj['userId'] = userId;
    }
    /**
     * Constructs a <code>AccountMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AccountMessage} obj Optional instance to populate.
     * @return {module:model/AccountMessage} The populated <code>AccountMessage</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccountMessage();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _AccountMessageAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('expiry')) {
          obj['expiry'] = _ApiClient["default"].convertToType(data['expiry'], 'Date');
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _EventRefType["default"].constructFromObject(data['eventRefType']);
        }

        if (data.hasOwnProperty('eventRefId')) {
          obj['eventRefId'] = _ApiClient["default"].convertToType(data['eventRefId'], 'String');
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AccountMessage;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


AccountMessage.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

AccountMessage.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

AccountMessage.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

AccountMessage.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

AccountMessage.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

AccountMessage.prototype['metadata'] = undefined;
/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */

AccountMessage.prototype['expiry'] = undefined;
/**
 * @member {module:model/EventRefType} eventRefType
 */

AccountMessage.prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

AccountMessage.prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

AccountMessage.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

AccountMessage.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

AccountMessage.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

AccountMessage.prototype['status'] = undefined;
/**
 * @member {String} userId
 */

AccountMessage.prototype['userId'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement AccountMessageAllOf interface:

/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */

_AccountMessageAllOf["default"].prototype['expiry'] = undefined;
/**
 * @member {module:model/EventRefType} eventRefType
 */

_AccountMessageAllOf["default"].prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

_AccountMessageAllOf["default"].prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

_AccountMessageAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_AccountMessageAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_AccountMessageAllOf["default"].prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_AccountMessageAllOf["default"].prototype['status'] = undefined;
/**
 * @member {String} userId
 */

_AccountMessageAllOf["default"].prototype['userId'] = undefined;
var _default = AccountMessage;
exports["default"] = _default;
