// @ziqni-tech/admin-api-client/dist/model/AchievementAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AchievementStatus = _interopRequireDefault(require("./AchievementStatus"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The AchievementAllOf model module.
 * @module model/AchievementAllOf
 * @version 0.0.26
 */
var AchievementAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AchievementAllOf</code>.
   * @alias module:model/AchievementAllOf
   * @param name {String} A name for the Achievement. Can be translated
   * @param icon {String} An Icon id that has been pre uploaded to the system to display for Achievement
   * @param constraints {Array.<String>} Additional constraints, if set means true
   */
  function AchievementAllOf(name, icon, constraints) {
    _classCallCheck(this, AchievementAllOf);

    AchievementAllOf.initialize(this, name, icon, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AchievementAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, icon, constraints) {
      obj['name'] = name;
      obj['icon'] = icon;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>AchievementAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AchievementAllOf} obj Optional instance to populate.
     * @return {module:model/AchievementAllOf} The populated <code>AchievementAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AchievementAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('icon')) {
          obj['icon'] = _ApiClient["default"].convertToType(data['icon'], 'String');
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('maxNumberOfIssues')) {
          obj['maxNumberOfIssues'] = _ApiClient["default"].convertToType(data['maxNumberOfIssues'], 'Number');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _AchievementStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('statusCode')) {
          obj['statusCode'] = _ApiClient["default"].convertToType(data['statusCode'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('achievementDependencies')) {
          obj['achievementDependencies'] = _DependantOn["default"].constructFromObject(data['achievementDependencies']);
        }

        if (data.hasOwnProperty('memberTagsFilter')) {
          obj['memberTagsFilter'] = _DependantOn["default"].constructFromObject(data['memberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }
      }

      return obj;
    }
  }]);

  return AchievementAllOf;
}();
/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */


AchievementAllOf.prototype['name'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} description
 */

AchievementAllOf.prototype['description'] = undefined;
/**
 * Terms and conditions of an achievement. Can be translated
 * @member {String} termsAndConditions
 */

AchievementAllOf.prototype['termsAndConditions'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */

AchievementAllOf.prototype['icon'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

AchievementAllOf.prototype['scheduling'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

AchievementAllOf.prototype['maxNumberOfIssues'] = undefined;
/**
 * @member {module:model/AchievementStatus} status
 */

AchievementAllOf.prototype['status'] = undefined;
/**
 * The code of the contest
 * @member {Number} statusCode
 */

AchievementAllOf.prototype['statusCode'] = undefined;
/**
 * Additional constraints, if set means true
 * @member {Array.<String>} constraints
 */

AchievementAllOf.prototype['constraints'] = undefined;
/**
 * @member {module:model/DependantOn} achievementDependencies
 */

AchievementAllOf.prototype['achievementDependencies'] = undefined;
/**
 * @member {module:model/DependantOn} memberTagsFilter
 */

AchievementAllOf.prototype['memberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

AchievementAllOf.prototype['productTagsFilter'] = undefined;
var _default = AchievementAllOf;
exports["default"] = _default;
