// @ziqni-tech/admin-api-client/dist/model/Achievement.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AchievementAllOf = _interopRequireDefault(require("./AchievementAllOf"));

var _AchievementStatus = _interopRequireDefault(require("./AchievementStatus"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Achievement model module.
 * @module model/Achievement
 * @version 0.0.26
 */
var Achievement = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Achievement</code>.
   * @alias module:model/Achievement
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/AchievementAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} A name for the Achievement. Can be translated
   * @param icon {String} An Icon id that has been pre uploaded to the system to display for Achievement
   * @param constraints {Array.<String>} Additional constraints, if set means true
   */
  function Achievement(id, spaceName, created, name, icon, constraints) {
    _classCallCheck(this, Achievement);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _AchievementAllOf["default"].initialize(this, name, icon, constraints);

    Achievement.initialize(this, id, spaceName, created, name, icon, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Achievement, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, icon, constraints) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['icon'] = icon;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>Achievement</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Achievement} obj Optional instance to populate.
     * @return {module:model/Achievement} The populated <code>Achievement</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Achievement();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _AchievementAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('icon')) {
          obj['icon'] = _ApiClient["default"].convertToType(data['icon'], 'String');
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('maxNumberOfIssues')) {
          obj['maxNumberOfIssues'] = _ApiClient["default"].convertToType(data['maxNumberOfIssues'], 'Number');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _AchievementStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('statusCode')) {
          obj['statusCode'] = _ApiClient["default"].convertToType(data['statusCode'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('achievementDependencies')) {
          obj['achievementDependencies'] = _DependantOn["default"].constructFromObject(data['achievementDependencies']);
        }

        if (data.hasOwnProperty('memberTagsFilter')) {
          obj['memberTagsFilter'] = _DependantOn["default"].constructFromObject(data['memberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }
      }

      return obj;
    }
  }]);

  return Achievement;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Achievement.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Achievement.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Achievement.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Achievement.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

Achievement.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Achievement.prototype['metadata'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */

Achievement.prototype['name'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} description
 */

Achievement.prototype['description'] = undefined;
/**
 * Terms and conditions of an achievement. Can be translated
 * @member {String} termsAndConditions
 */

Achievement.prototype['termsAndConditions'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */

Achievement.prototype['icon'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

Achievement.prototype['scheduling'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

Achievement.prototype['maxNumberOfIssues'] = undefined;
/**
 * @member {module:model/AchievementStatus} status
 */

Achievement.prototype['status'] = undefined;
/**
 * The code of the contest
 * @member {Number} statusCode
 */

Achievement.prototype['statusCode'] = undefined;
/**
 * Additional constraints, if set means true
 * @member {Array.<String>} constraints
 */

Achievement.prototype['constraints'] = undefined;
/**
 * @member {module:model/DependantOn} achievementDependencies
 */

Achievement.prototype['achievementDependencies'] = undefined;
/**
 * @member {module:model/DependantOn} memberTagsFilter
 */

Achievement.prototype['memberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

Achievement.prototype['productTagsFilter'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement AchievementAllOf interface:

/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */

_AchievementAllOf["default"].prototype['name'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} description
 */

_AchievementAllOf["default"].prototype['description'] = undefined;
/**
 * Terms and conditions of an achievement. Can be translated
 * @member {String} termsAndConditions
 */

_AchievementAllOf["default"].prototype['termsAndConditions'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */

_AchievementAllOf["default"].prototype['icon'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

_AchievementAllOf["default"].prototype['scheduling'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

_AchievementAllOf["default"].prototype['maxNumberOfIssues'] = undefined;
/**
 * @member {module:model/AchievementStatus} status
 */

_AchievementAllOf["default"].prototype['status'] = undefined;
/**
 * The code of the contest
 * @member {Number} statusCode
 */

_AchievementAllOf["default"].prototype['statusCode'] = undefined;
/**
 * Additional constraints, if set means true
 * @member {Array.<String>} constraints
 */

_AchievementAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {module:model/DependantOn} achievementDependencies
 */

_AchievementAllOf["default"].prototype['achievementDependencies'] = undefined;
/**
 * @member {module:model/DependantOn} memberTagsFilter
 */

_AchievementAllOf["default"].prototype['memberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

_AchievementAllOf["default"].prototype['productTagsFilter'] = undefined;
var _default = Achievement;
exports["default"] = _default;
