// @ziqni-tech/admin-api-client/dist/model/AchievementReduced.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AchievementLiveStatus = _interopRequireDefault(require("./AchievementLiveStatus"));

var _AchievementReducedAllOf = _interopRequireDefault(require("./AchievementReducedAllOf"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The AchievementReduced model module.
 * @module model/AchievementReduced
 * @version 0.0.26
 */
var AchievementReduced = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AchievementReduced</code>.
   * @alias module:model/AchievementReduced
   * @implements module:model/ModelDefault
   * @implements module:model/AchievementReducedAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} A name for the Achievement. Can be translated
   * @param achievementLiveStatus {module:model/AchievementLiveStatus} 
   */
  function AchievementReduced(id, spaceName, created, name, achievementLiveStatus) {
    _classCallCheck(this, AchievementReduced);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _AchievementReducedAllOf["default"].initialize(this, name, achievementLiveStatus);

    AchievementReduced.initialize(this, id, spaceName, created, name, achievementLiveStatus);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AchievementReduced, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, achievementLiveStatus) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['achievementLiveStatus'] = achievementLiveStatus;
    }
    /**
     * Constructs a <code>AchievementReduced</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AchievementReduced} obj Optional instance to populate.
     * @return {module:model/AchievementReduced} The populated <code>AchievementReduced</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AchievementReduced();

        _ModelDefault["default"].constructFromObject(data, obj);

        _AchievementReducedAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('achievementLiveStatus')) {
          obj['achievementLiveStatus'] = _AchievementLiveStatus["default"].constructFromObject(data['achievementLiveStatus']);
        }

        if (data.hasOwnProperty('tagsId')) {
          obj['tagsId'] = _ApiClient["default"].convertToType(data['tagsId'], ['String']);
        }

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _ApiClient["default"].convertToType(data['dependantOn'], [_Dependancy["default"]]);
        }
      }

      return obj;
    }
  }]);

  return AchievementReduced;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


AchievementReduced.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

AchievementReduced.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

AchievementReduced.prototype['created'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */

AchievementReduced.prototype['name'] = undefined;
/**
 * @member {module:model/AchievementLiveStatus} achievementLiveStatus
 */

AchievementReduced.prototype['achievementLiveStatus'] = undefined;
/**
 * Categorisation of the achievements
 * @member {Array.<String>} tagsId
 */

AchievementReduced.prototype['tagsId'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

AchievementReduced.prototype['dependantOn'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement AchievementReducedAllOf interface:

/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */

_AchievementReducedAllOf["default"].prototype['name'] = undefined;
/**
 * @member {module:model/AchievementLiveStatus} achievementLiveStatus
 */

_AchievementReducedAllOf["default"].prototype['achievementLiveStatus'] = undefined;
/**
 * Categorisation of the achievements
 * @member {Array.<String>} tagsId
 */

_AchievementReducedAllOf["default"].prototype['tagsId'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

_AchievementReducedAllOf["default"].prototype['dependantOn'] = undefined;
var _default = AchievementReduced;
exports["default"] = _default;
