// @ziqni-tech/admin-api-client/dist/model/AchievementReducedAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AchievementLiveStatus = _interopRequireDefault(require("./AchievementLiveStatus"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The AchievementReducedAllOf model module.
 * @module model/AchievementReducedAllOf
 * @version 0.0.26
 */
var AchievementReducedAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AchievementReducedAllOf</code>.
   * @alias module:model/AchievementReducedAllOf
   * @param name {String} A name for the Achievement. Can be translated
   * @param achievementLiveStatus {module:model/AchievementLiveStatus} 
   */
  function AchievementReducedAllOf(name, achievementLiveStatus) {
    _classCallCheck(this, AchievementReducedAllOf);

    AchievementReducedAllOf.initialize(this, name, achievementLiveStatus);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AchievementReducedAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, achievementLiveStatus) {
      obj['name'] = name;
      obj['achievementLiveStatus'] = achievementLiveStatus;
    }
    /**
     * Constructs a <code>AchievementReducedAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AchievementReducedAllOf} obj Optional instance to populate.
     * @return {module:model/AchievementReducedAllOf} The populated <code>AchievementReducedAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AchievementReducedAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('achievementLiveStatus')) {
          obj['achievementLiveStatus'] = _AchievementLiveStatus["default"].constructFromObject(data['achievementLiveStatus']);
        }

        if (data.hasOwnProperty('tagsId')) {
          obj['tagsId'] = _ApiClient["default"].convertToType(data['tagsId'], ['String']);
        }

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _ApiClient["default"].convertToType(data['dependantOn'], [_Dependancy["default"]]);
        }
      }

      return obj;
    }
  }]);

  return AchievementReducedAllOf;
}();
/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */


AchievementReducedAllOf.prototype['name'] = undefined;
/**
 * @member {module:model/AchievementLiveStatus} achievementLiveStatus
 */

AchievementReducedAllOf.prototype['achievementLiveStatus'] = undefined;
/**
 * Categorisation of the achievements
 * @member {Array.<String>} tagsId
 */

AchievementReducedAllOf.prototype['tagsId'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

AchievementReducedAllOf.prototype['dependantOn'] = undefined;
var _default = AchievementReducedAllOf;
exports["default"] = _default;
