// @ziqni-tech/admin-api-client/dist/model/ActionType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActionTypeAllOf = _interopRequireDefault(require("./ActionTypeAllOf"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ActionType model module.
 * @module model/ActionType
 * @version 0.0.26
 */
var ActionType = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActionType</code>.
   * @alias module:model/ActionType
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/ActionTypeAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} The name of the Action Helper
   * @param key {String} A unique key that represents an action helper
   * @param unitOfMeasureType {module:model/UnitOfMeasureType} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function ActionType(id, spaceName, created, name, key, unitOfMeasureType, constraints) {
    _classCallCheck(this, ActionType);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _ActionTypeAllOf["default"].initialize(this, name, key, unitOfMeasureType, constraints);

    ActionType.initialize(this, id, spaceName, created, name, key, unitOfMeasureType, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ActionType, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, key, unitOfMeasureType, constraints) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['key'] = key;
      obj['unitOfMeasureType'] = unitOfMeasureType;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>ActionType</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActionType} obj Optional instance to populate.
     * @return {module:model/ActionType} The populated <code>ActionType</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActionType();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _ActionTypeAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return ActionType;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


ActionType.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

ActionType.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

ActionType.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

ActionType.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

ActionType.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

ActionType.prototype['metadata'] = undefined;
/**
 * The name of the Action Helper
 * @member {String} name
 */

ActionType.prototype['name'] = undefined;
/**
 * A unique key that represents an action helper
 * @member {String} key
 */

ActionType.prototype['key'] = undefined;
/**
 * The description of the rule action helper for your reference
 * @member {String} description
 */

ActionType.prototype['description'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

ActionType.prototype['unitOfMeasureType'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

ActionType.prototype['constraints'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement ActionTypeAllOf interface:

/**
 * The name of the Action Helper
 * @member {String} name
 */

_ActionTypeAllOf["default"].prototype['name'] = undefined;
/**
 * A unique key that represents an action helper
 * @member {String} key
 */

_ActionTypeAllOf["default"].prototype['key'] = undefined;
/**
 * The description of the rule action helper for your reference
 * @member {String} description
 */

_ActionTypeAllOf["default"].prototype['description'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

_ActionTypeAllOf["default"].prototype['unitOfMeasureType'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_ActionTypeAllOf["default"].prototype['constraints'] = undefined;
var _default = ActionType;
exports["default"] = _default;
