// @ziqni-tech/admin-api-client/dist/model/ApiKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ApiKeyAllOf = _interopRequireDefault(require("./ApiKeyAllOf"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _Role = _interopRequireDefault(require("./Role"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ApiKey model module.
 * @module model/ApiKey
 * @version 0.0.26
 */
var ApiKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ApiKey</code>.
   * @alias module:model/ApiKey
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/ApiKeyAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param key {String} An Api key hash
   * @param role {module:model/Role} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function ApiKey(id, spaceName, created, key, role, constraints) {
    _classCallCheck(this, ApiKey);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _ApiKeyAllOf["default"].initialize(this, key, role, constraints);

    ApiKey.initialize(this, id, spaceName, created, key, role, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ApiKey, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, key, role, constraints) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['key'] = key;
      obj['role'] = role;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>ApiKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ApiKey} obj Optional instance to populate.
     * @return {module:model/ApiKey} The populated <code>ApiKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ApiKey();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _ApiKeyAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('whiteListIPs')) {
          obj['whiteListIPs'] = _ApiClient["default"].convertToType(data['whiteListIPs'], ['String']);
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('role')) {
          obj['role'] = _Role["default"].constructFromObject(data['role']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return ApiKey;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


ApiKey.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

ApiKey.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

ApiKey.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

ApiKey.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

ApiKey.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

ApiKey.prototype['metadata'] = undefined;
/**
 * The description of an Api key
 * @member {String} description
 */

ApiKey.prototype['description'] = undefined;
/**
 * the IP's that need to be whitelisted with the Api key
 * @member {Array.<String>} whiteListIPs
 */

ApiKey.prototype['whiteListIPs'] = undefined;
/**
 * An Api key hash
 * @member {String} key
 */

ApiKey.prototype['key'] = undefined;
/**
 * @member {module:model/Role} role
 */

ApiKey.prototype['role'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

ApiKey.prototype['constraints'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement ApiKeyAllOf interface:

/**
 * The description of an Api key
 * @member {String} description
 */

_ApiKeyAllOf["default"].prototype['description'] = undefined;
/**
 * the IP's that need to be whitelisted with the Api key
 * @member {Array.<String>} whiteListIPs
 */

_ApiKeyAllOf["default"].prototype['whiteListIPs'] = undefined;
/**
 * An Api key hash
 * @member {String} key
 */

_ApiKeyAllOf["default"].prototype['key'] = undefined;
/**
 * @member {module:model/Role} role
 */

_ApiKeyAllOf["default"].prototype['role'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_ApiKeyAllOf["default"].prototype['constraints'] = undefined;
var _default = ApiKey;
exports["default"] = _default;
