// @ziqni-tech/admin-api-client/dist/model/ApiKeyAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Role = _interopRequireDefault(require("./Role"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ApiKeyAllOf model module.
 * @module model/ApiKeyAllOf
 * @version 0.0.26
 */
var ApiKeyAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ApiKeyAllOf</code>.
   * @alias module:model/ApiKeyAllOf
   * @param key {String} An Api key hash
   * @param role {module:model/Role} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function ApiKeyAllOf(key, role, constraints) {
    _classCallCheck(this, ApiKeyAllOf);

    ApiKeyAllOf.initialize(this, key, role, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ApiKeyAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, key, role, constraints) {
      obj['key'] = key;
      obj['role'] = role;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>ApiKeyAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ApiKeyAllOf} obj Optional instance to populate.
     * @return {module:model/ApiKeyAllOf} The populated <code>ApiKeyAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ApiKeyAllOf();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('whiteListIPs')) {
          obj['whiteListIPs'] = _ApiClient["default"].convertToType(data['whiteListIPs'], ['String']);
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('role')) {
          obj['role'] = _Role["default"].constructFromObject(data['role']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return ApiKeyAllOf;
}();
/**
 * The description of an Api key
 * @member {String} description
 */


ApiKeyAllOf.prototype['description'] = undefined;
/**
 * the IP's that need to be whitelisted with the Api key
 * @member {Array.<String>} whiteListIPs
 */

ApiKeyAllOf.prototype['whiteListIPs'] = undefined;
/**
 * An Api key hash
 * @member {String} key
 */

ApiKeyAllOf.prototype['key'] = undefined;
/**
 * @member {module:model/Role} role
 */

ApiKeyAllOf.prototype['role'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

ApiKeyAllOf.prototype['constraints'] = undefined;
var _default = ApiKeyAllOf;
exports["default"] = _default;
