// @ziqni-tech/admin-api-client/dist/model/AttachmentAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The AttachmentAllOf model module.
 * @module model/AttachmentAllOf
 * @version 0.0.5
 */
var AttachmentAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AttachmentAllOf</code>.
   * @alias module:model/AttachmentAllOf
   * @param attachmentType {String} A type of an attachement
   * @param md5Hash {String} Hash of the file contents to identify duplicates
   * @param name {String} Name of the original file uploaded
   * @param mimeType {String} Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
   * @param extension {String} Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
   * @param locationKey {String} The name of the attachment within the bucket
   * @param locationFolder {String} The folder name containing the attachment within the bucket
   * @param permission {Number} 
   */
  function AttachmentAllOf(attachmentType, md5Hash, name, mimeType, extension, locationKey, locationFolder, permission) {
    _classCallCheck(this, AttachmentAllOf);

    AttachmentAllOf.initialize(this, attachmentType, md5Hash, name, mimeType, extension, locationKey, locationFolder, permission);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AttachmentAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, attachmentType, md5Hash, name, mimeType, extension, locationKey, locationFolder, permission) {
      obj['attachmentType'] = attachmentType;
      obj['md5Hash'] = md5Hash;
      obj['name'] = name;
      obj['mimeType'] = mimeType;
      obj['extension'] = extension;
      obj['locationKey'] = locationKey;
      obj['locationFolder'] = locationFolder;
      obj['permission'] = permission;
    }
    /**
     * Constructs a <code>AttachmentAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AttachmentAllOf} obj Optional instance to populate.
     * @return {module:model/AttachmentAllOf} The populated <code>AttachmentAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AttachmentAllOf();

        if (data.hasOwnProperty('attachmentType')) {
          obj['attachmentType'] = _ApiClient["default"].convertToType(data['attachmentType'], 'String');
        }

        if (data.hasOwnProperty('md5Hash')) {
          obj['md5Hash'] = _ApiClient["default"].convertToType(data['md5Hash'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('mimeType')) {
          obj['mimeType'] = _ApiClient["default"].convertToType(data['mimeType'], 'String');
        }

        if (data.hasOwnProperty('extension')) {
          obj['extension'] = _ApiClient["default"].convertToType(data['extension'], 'String');
        }

        if (data.hasOwnProperty('locationKey')) {
          obj['locationKey'] = _ApiClient["default"].convertToType(data['locationKey'], 'String');
        }

        if (data.hasOwnProperty('locationFolder')) {
          obj['locationFolder'] = _ApiClient["default"].convertToType(data['locationFolder'], 'String');
        }

        if (data.hasOwnProperty('permission')) {
          obj['permission'] = _ApiClient["default"].convertToType(data['permission'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return AttachmentAllOf;
}();
/**
 * A type of an attachement
 * @member {String} attachmentType
 */


AttachmentAllOf.prototype['attachmentType'] = undefined;
/**
 * Hash of the file contents to identify duplicates
 * @member {String} md5Hash
 */

AttachmentAllOf.prototype['md5Hash'] = undefined;
/**
 * Name of the original file uploaded
 * @member {String} name
 */

AttachmentAllOf.prototype['name'] = undefined;
/**
 * Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
 * @member {String} mimeType
 */

AttachmentAllOf.prototype['mimeType'] = undefined;
/**
 * Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
 * @member {String} extension
 */

AttachmentAllOf.prototype['extension'] = undefined;
/**
 * The name of the attachment within the bucket
 * @member {String} locationKey
 */

AttachmentAllOf.prototype['locationKey'] = undefined;
/**
 * The folder name containing the attachment within the bucket
 * @member {String} locationFolder
 */

AttachmentAllOf.prototype['locationFolder'] = undefined;
/**
 * @member {Number} permission
 */

AttachmentAllOf.prototype['permission'] = undefined;
var _default = AttachmentAllOf;
exports["default"] = _default;
