// @ziqni-tech/admin-api-client/dist/model/Attachment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AttachmentAllOf = _interopRequireDefault(require("./AttachmentAllOf"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Attachment model module.
 * @module model/Attachment
 * @version 0.0.5
 */
var Attachment = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Attachment</code>.
   * @alias module:model/Attachment
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/AttachmentAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param attachmentType {String} A type of an attachement
   * @param md5Hash {String} Hash of the file contents to identify duplicates
   * @param name {String} Name of the original file uploaded
   * @param mimeType {String} Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
   * @param extension {String} Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
   * @param locationKey {String} The name of the attachment within the bucket
   * @param locationFolder {String} The folder name containing the attachment within the bucket
   * @param permission {Number} 
   */
  function Attachment(id, spaceName, created, attachmentType, md5Hash, name, mimeType, extension, locationKey, locationFolder, permission) {
    _classCallCheck(this, Attachment);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _AttachmentAllOf["default"].initialize(this, attachmentType, md5Hash, name, mimeType, extension, locationKey, locationFolder, permission);

    Attachment.initialize(this, id, spaceName, created, attachmentType, md5Hash, name, mimeType, extension, locationKey, locationFolder, permission);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Attachment, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, attachmentType, md5Hash, name, mimeType, extension, locationKey, locationFolder, permission) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['attachmentType'] = attachmentType;
      obj['md5Hash'] = md5Hash;
      obj['name'] = name;
      obj['mimeType'] = mimeType;
      obj['extension'] = extension;
      obj['locationKey'] = locationKey;
      obj['locationFolder'] = locationFolder;
      obj['permission'] = permission;
    }
    /**
     * Constructs a <code>Attachment</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Attachment} obj Optional instance to populate.
     * @return {module:model/Attachment} The populated <code>Attachment</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Attachment();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _AttachmentAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('attachmentType')) {
          obj['attachmentType'] = _ApiClient["default"].convertToType(data['attachmentType'], 'String');
        }

        if (data.hasOwnProperty('md5Hash')) {
          obj['md5Hash'] = _ApiClient["default"].convertToType(data['md5Hash'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('mimeType')) {
          obj['mimeType'] = _ApiClient["default"].convertToType(data['mimeType'], 'String');
        }

        if (data.hasOwnProperty('extension')) {
          obj['extension'] = _ApiClient["default"].convertToType(data['extension'], 'String');
        }

        if (data.hasOwnProperty('locationKey')) {
          obj['locationKey'] = _ApiClient["default"].convertToType(data['locationKey'], 'String');
        }

        if (data.hasOwnProperty('locationFolder')) {
          obj['locationFolder'] = _ApiClient["default"].convertToType(data['locationFolder'], 'String');
        }

        if (data.hasOwnProperty('permission')) {
          obj['permission'] = _ApiClient["default"].convertToType(data['permission'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return Attachment;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Attachment.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Attachment.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Attachment.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Attachment.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

Attachment.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Attachment.prototype['metadata'] = undefined;
/**
 * A type of an attachement
 * @member {String} attachmentType
 */

Attachment.prototype['attachmentType'] = undefined;
/**
 * Hash of the file contents to identify duplicates
 * @member {String} md5Hash
 */

Attachment.prototype['md5Hash'] = undefined;
/**
 * Name of the original file uploaded
 * @member {String} name
 */

Attachment.prototype['name'] = undefined;
/**
 * Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
 * @member {String} mimeType
 */

Attachment.prototype['mimeType'] = undefined;
/**
 * Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
 * @member {String} extension
 */

Attachment.prototype['extension'] = undefined;
/**
 * The name of the attachment within the bucket
 * @member {String} locationKey
 */

Attachment.prototype['locationKey'] = undefined;
/**
 * The folder name containing the attachment within the bucket
 * @member {String} locationFolder
 */

Attachment.prototype['locationFolder'] = undefined;
/**
 * @member {Number} permission
 */

Attachment.prototype['permission'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement AttachmentAllOf interface:

/**
 * A type of an attachement
 * @member {String} attachmentType
 */

_AttachmentAllOf["default"].prototype['attachmentType'] = undefined;
/**
 * Hash of the file contents to identify duplicates
 * @member {String} md5Hash
 */

_AttachmentAllOf["default"].prototype['md5Hash'] = undefined;
/**
 * Name of the original file uploaded
 * @member {String} name
 */

_AttachmentAllOf["default"].prototype['name'] = undefined;
/**
 * Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
 * @member {String} mimeType
 */

_AttachmentAllOf["default"].prototype['mimeType'] = undefined;
/**
 * Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
 * @member {String} extension
 */

_AttachmentAllOf["default"].prototype['extension'] = undefined;
/**
 * The name of the attachment within the bucket
 * @member {String} locationKey
 */

_AttachmentAllOf["default"].prototype['locationKey'] = undefined;
/**
 * The folder name containing the attachment within the bucket
 * @member {String} locationFolder
 */

_AttachmentAllOf["default"].prototype['locationFolder'] = undefined;
/**
 * @member {Number} permission
 */

_AttachmentAllOf["default"].prototype['permission'] = undefined;
var _default = Attachment;
exports["default"] = _default;
