// @ziqni-tech/admin-api-client/dist/model/AwardResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Award = _interopRequireDefault(require("./Award"));

var _Error = _interopRequireDefault(require("./Error"));

var _MetaExtended = _interopRequireDefault(require("./MetaExtended"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The AwardResponse model module.
 * @module model/AwardResponse
 * @version 0.0.26
 */
var AwardResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AwardResponse</code>.
   * @alias module:model/AwardResponse
   * @param meta {module:model/MetaExtended} 
   */
  function AwardResponse(meta) {
    _classCallCheck(this, AwardResponse);

    AwardResponse.initialize(this, meta);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AwardResponse, null, [{
    key: "initialize",
    value: function initialize(obj, meta) {
      obj['meta'] = meta;
    }
    /**
     * Constructs a <code>AwardResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AwardResponse} obj Optional instance to populate.
     * @return {module:model/AwardResponse} The populated <code>AwardResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AwardResponse();

        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _MetaExtended["default"].constructFromObject(data['meta']);
        }

        if (data.hasOwnProperty('results')) {
          obj['results'] = _ApiClient["default"].convertToType(data['results'], [_Award["default"]]);
        }

        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], [_Error["default"]]);
        }
      }

      return obj;
    }
  }]);

  return AwardResponse;
}();
/**
 * @member {module:model/MetaExtended} meta
 */


AwardResponse.prototype['meta'] = undefined;
/**
 * @member {Array.<module:model/Award>} results
 */

AwardResponse.prototype['results'] = undefined;
/**
 * @member {Array.<module:model/Error>} errors
 */

AwardResponse.prototype['errors'] = undefined;
var _default = AwardResponse;
exports["default"] = _default;
