// @ziqni-tech/admin-api-client/dist/model/ChangePasswordRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ChangePasswordRequest model module.
 * @module model/ChangePasswordRequest
 * @version 0.0.26
 */
var ChangePasswordRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ChangePasswordRequest</code>.
   * @alias module:model/ChangePasswordRequest
   * @param email {String} The account username
   * @param password {String} Current password
   * @param newPassword {String} New password
   * @param confirmPassword {String} Confirm password
   */
  function ChangePasswordRequest(email, password, newPassword, confirmPassword) {
    _classCallCheck(this, ChangePasswordRequest);

    ChangePasswordRequest.initialize(this, email, password, newPassword, confirmPassword);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ChangePasswordRequest, null, [{
    key: "initialize",
    value: function initialize(obj, email, password, newPassword, confirmPassword) {
      obj['email'] = email;
      obj['password'] = password;
      obj['newPassword'] = newPassword;
      obj['confirmPassword'] = confirmPassword;
    }
    /**
     * Constructs a <code>ChangePasswordRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ChangePasswordRequest} obj Optional instance to populate.
     * @return {module:model/ChangePasswordRequest} The populated <code>ChangePasswordRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ChangePasswordRequest();

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('newPassword')) {
          obj['newPassword'] = _ApiClient["default"].convertToType(data['newPassword'], 'String');
        }

        if (data.hasOwnProperty('confirmPassword')) {
          obj['confirmPassword'] = _ApiClient["default"].convertToType(data['confirmPassword'], 'String');
        }

        if (data.hasOwnProperty('twoFactorAccessCode')) {
          obj['twoFactorAccessCode'] = _ApiClient["default"].convertToType(data['twoFactorAccessCode'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ChangePasswordRequest;
}();
/**
 * The account username
 * @member {String} email
 */


ChangePasswordRequest.prototype['email'] = undefined;
/**
 * Current password
 * @member {String} password
 */

ChangePasswordRequest.prototype['password'] = undefined;
/**
 * New password
 * @member {String} newPassword
 */

ChangePasswordRequest.prototype['newPassword'] = undefined;
/**
 * Confirm password
 * @member {String} confirmPassword
 */

ChangePasswordRequest.prototype['confirmPassword'] = undefined;
/**
 * Two factor access code if 2FA enabled
 * @member {String} twoFactorAccessCode
 */

ChangePasswordRequest.prototype['twoFactorAccessCode'] = undefined;
var _default = ChangePasswordRequest;
exports["default"] = _default;
