// @ziqni-tech/admin-api-client/dist/model/Collaborator.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CollaboratorAllOf = _interopRequireDefault(require("./CollaboratorAllOf"));

var _CollaboratorRolesAvailable = _interopRequireDefault(require("./CollaboratorRolesAvailable"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Collaborator model module.
 * @module model/Collaborator
 * @version 0.0.26
 */
var Collaborator = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Collaborator</code>.
   * @alias module:model/Collaborator
   * @implements module:model/ModelDefault
   * @implements module:model/CollaboratorAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param email {String} The email of the user to log in
   */
  function Collaborator(id, spaceName, created, email) {
    _classCallCheck(this, Collaborator);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _CollaboratorAllOf["default"].initialize(this, email);

    Collaborator.initialize(this, id, spaceName, created, email);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Collaborator, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, email) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['email'] = email;
    }
    /**
     * Constructs a <code>Collaborator</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Collaborator} obj Optional instance to populate.
     * @return {module:model/Collaborator} The populated <code>Collaborator</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Collaborator();

        _ModelDefault["default"].constructFromObject(data, obj);

        _CollaboratorAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('roles')) {
          obj['roles'] = _ApiClient["default"].convertToType(data['roles'], ['String']);
        }

        if (data.hasOwnProperty('lastName')) {
          obj['lastName'] = _ApiClient["default"].convertToType(data['lastName'], 'String');
        }

        if (data.hasOwnProperty('firstName')) {
          obj['firstName'] = _ApiClient["default"].convertToType(data['firstName'], 'String');
        }

        if (data.hasOwnProperty('availableRoles')) {
          obj['availableRoles'] = _CollaboratorRolesAvailable["default"].constructFromObject(data['availableRoles']);
        }
      }

      return obj;
    }
  }]);

  return Collaborator;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Collaborator.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Collaborator.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Collaborator.prototype['created'] = undefined;
/**
 * The email of the user to log in
 * @member {String} email
 */

Collaborator.prototype['email'] = undefined;
/**
 * The access role of an individual
 * @member {Array.<String>} roles
 */

Collaborator.prototype['roles'] = undefined;
/**
 * The Surname of an individual
 * @member {String} lastName
 */

Collaborator.prototype['lastName'] = undefined;
/**
 * The Name of an individual
 * @member {String} firstName
 */

Collaborator.prototype['firstName'] = undefined;
/**
 * @member {module:model/CollaboratorRolesAvailable} availableRoles
 */

Collaborator.prototype['availableRoles'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement CollaboratorAllOf interface:

/**
 * The email of the user to log in
 * @member {String} email
 */

_CollaboratorAllOf["default"].prototype['email'] = undefined;
/**
 * The access role of an individual
 * @member {Array.<String>} roles
 */

_CollaboratorAllOf["default"].prototype['roles'] = undefined;
/**
 * The Surname of an individual
 * @member {String} lastName
 */

_CollaboratorAllOf["default"].prototype['lastName'] = undefined;
/**
 * The Name of an individual
 * @member {String} firstName
 */

_CollaboratorAllOf["default"].prototype['firstName'] = undefined;
/**
 * @member {module:model/CollaboratorRolesAvailable} availableRoles
 */

_CollaboratorAllOf["default"].prototype['availableRoles'] = undefined;
var _default = Collaborator;
exports["default"] = _default;
