// @ziqni-tech/admin-api-client/dist/model/CollaboratorAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CollaboratorRolesAvailable = _interopRequireDefault(require("./CollaboratorRolesAvailable"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CollaboratorAllOf model module.
 * @module model/CollaboratorAllOf
 * @version 0.0.26
 */
var CollaboratorAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CollaboratorAllOf</code>.
   * @alias module:model/CollaboratorAllOf
   * @param email {String} The email of the user to log in
   */
  function CollaboratorAllOf(email) {
    _classCallCheck(this, CollaboratorAllOf);

    CollaboratorAllOf.initialize(this, email);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CollaboratorAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, email) {
      obj['email'] = email;
    }
    /**
     * Constructs a <code>CollaboratorAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CollaboratorAllOf} obj Optional instance to populate.
     * @return {module:model/CollaboratorAllOf} The populated <code>CollaboratorAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CollaboratorAllOf();

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('roles')) {
          obj['roles'] = _ApiClient["default"].convertToType(data['roles'], ['String']);
        }

        if (data.hasOwnProperty('lastName')) {
          obj['lastName'] = _ApiClient["default"].convertToType(data['lastName'], 'String');
        }

        if (data.hasOwnProperty('firstName')) {
          obj['firstName'] = _ApiClient["default"].convertToType(data['firstName'], 'String');
        }

        if (data.hasOwnProperty('availableRoles')) {
          obj['availableRoles'] = _CollaboratorRolesAvailable["default"].constructFromObject(data['availableRoles']);
        }
      }

      return obj;
    }
  }]);

  return CollaboratorAllOf;
}();
/**
 * The email of the user to log in
 * @member {String} email
 */


CollaboratorAllOf.prototype['email'] = undefined;
/**
 * The access role of an individual
 * @member {Array.<String>} roles
 */

CollaboratorAllOf.prototype['roles'] = undefined;
/**
 * The Surname of an individual
 * @member {String} lastName
 */

CollaboratorAllOf.prototype['lastName'] = undefined;
/**
 * The Name of an individual
 * @member {String} firstName
 */

CollaboratorAllOf.prototype['firstName'] = undefined;
/**
 * @member {module:model/CollaboratorRolesAvailable} availableRoles
 */

CollaboratorAllOf.prototype['availableRoles'] = undefined;
var _default = CollaboratorAllOf;
exports["default"] = _default;
