// @ziqni-tech/admin-api-client/dist/model/ComputeEngine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ComputeEngineAllOf = _interopRequireDefault(require("./ComputeEngineAllOf"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ComputeEngine model module.
 * @module model/ComputeEngine
 * @version 0.0.26
 */
var ComputeEngine = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ComputeEngine</code>.
   * @alias module:model/ComputeEngine
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/ComputeEngineAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param description {String} 
   * @param name {String} 
   */
  function ComputeEngine(id, spaceName, created, description, name) {
    _classCallCheck(this, ComputeEngine);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _ComputeEngineAllOf["default"].initialize(this, description, name);

    ComputeEngine.initialize(this, id, spaceName, created, description, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ComputeEngine, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, description, name) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['description'] = description;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>ComputeEngine</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ComputeEngine} obj Optional instance to populate.
     * @return {module:model/ComputeEngine} The populated <code>ComputeEngine</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ComputeEngine();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _ComputeEngineAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('uptime')) {
          obj['uptime'] = _ApiClient["default"].convertToType(data['uptime'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('statistics')) {
          obj['statistics'] = _ApiClient["default"].convertToType(data['statistics'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('lastKnownStatusCode')) {
          obj['lastKnownStatusCode'] = _ApiClient["default"].convertToType(data['lastKnownStatusCode'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ComputeEngine;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


ComputeEngine.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

ComputeEngine.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

ComputeEngine.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

ComputeEngine.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

ComputeEngine.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

ComputeEngine.prototype['metadata'] = undefined;
/**
 * @member {String} description
 */

ComputeEngine.prototype['description'] = undefined;
/**
 * @member {Number} version
 */

ComputeEngine.prototype['version'] = undefined;
/**
 * @member {String} name
 */

ComputeEngine.prototype['name'] = undefined;
/**
 * Long
 * @member {Number} uptime
 */

ComputeEngine.prototype['uptime'] = undefined;
/**
 * @member {Array.<String>} constraints
 */

ComputeEngine.prototype['constraints'] = undefined;
/**
 * @member {Object.<String, String>} statistics
 */

ComputeEngine.prototype['statistics'] = undefined;
/**
 * @member {Number} lastKnownStatusCode
 */

ComputeEngine.prototype['lastKnownStatusCode'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement ComputeEngineAllOf interface:

/**
 * @member {String} description
 */

_ComputeEngineAllOf["default"].prototype['description'] = undefined;
/**
 * @member {Number} version
 */

_ComputeEngineAllOf["default"].prototype['version'] = undefined;
/**
 * @member {String} name
 */

_ComputeEngineAllOf["default"].prototype['name'] = undefined;
/**
 * Long
 * @member {Number} uptime
 */

_ComputeEngineAllOf["default"].prototype['uptime'] = undefined;
/**
 * @member {Array.<String>} constraints
 */

_ComputeEngineAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {Object.<String, String>} statistics
 */

_ComputeEngineAllOf["default"].prototype['statistics'] = undefined;
/**
 * @member {Number} lastKnownStatusCode
 */

_ComputeEngineAllOf["default"].prototype['lastKnownStatusCode'] = undefined;
var _default = ComputeEngine;
exports["default"] = _default;
