// @ziqni-tech/admin-api-client/dist/model/CompetitionReduced.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CompetitionReducedAllOf = _interopRequireDefault(require("./CompetitionReducedAllOf"));

var _CompetitionStatus = _interopRequireDefault(require("./CompetitionStatus"));

var _CompetitionType = _interopRequireDefault(require("./CompetitionType"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CompetitionReduced model module.
 * @module model/CompetitionReduced
 * @version 0.0.26
 */
var CompetitionReduced = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CompetitionReduced</code>.
   * @alias module:model/CompetitionReduced
   * @implements module:model/ModelDefault
   * @implements module:model/CompetitionReducedAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param competitionType {module:model/CompetitionType} 
   * @param numberOfRounds {Number} Number of rounds to be played in a competition
   * @param name {String} A name or a name of a competition. Can be translated
   * @param status {module:model/CompetitionStatus} 
   * @param statusCode {Number} The code of the competition
   * @param dependantOn {Array.<module:model/Dependancy>} 
   * @param scheduledStartDate {Date} ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
   * @param scheduledEndDate {Date} ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
   * @param constraints {Array.<String>} Additional constraints
   */
  function CompetitionReduced(id, spaceName, created, competitionType, numberOfRounds, name, status, statusCode, dependantOn, scheduledStartDate, scheduledEndDate, constraints) {
    _classCallCheck(this, CompetitionReduced);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _CompetitionReducedAllOf["default"].initialize(this, competitionType, numberOfRounds, name, status, statusCode, dependantOn, scheduledStartDate, scheduledEndDate, constraints);

    CompetitionReduced.initialize(this, id, spaceName, created, competitionType, numberOfRounds, name, status, statusCode, dependantOn, scheduledStartDate, scheduledEndDate, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CompetitionReduced, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, competitionType, numberOfRounds, name, status, statusCode, dependantOn, scheduledStartDate, scheduledEndDate, constraints) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['competitionType'] = competitionType;
      obj['numberOfRounds'] = numberOfRounds;
      obj['name'] = name;
      obj['status'] = status;
      obj['statusCode'] = statusCode;
      obj['dependantOn'] = dependantOn;
      obj['scheduledStartDate'] = scheduledStartDate;
      obj['scheduledEndDate'] = scheduledEndDate;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CompetitionReduced</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CompetitionReduced} obj Optional instance to populate.
     * @return {module:model/CompetitionReduced} The populated <code>CompetitionReduced</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CompetitionReduced();

        _ModelDefault["default"].constructFromObject(data, obj);

        _CompetitionReducedAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('competitionType')) {
          obj['competitionType'] = _CompetitionType["default"].constructFromObject(data['competitionType']);
        }

        if (data.hasOwnProperty('numberOfRounds')) {
          obj['numberOfRounds'] = _ApiClient["default"].convertToType(data['numberOfRounds'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _CompetitionStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('statusCode')) {
          obj['statusCode'] = _ApiClient["default"].convertToType(data['statusCode'], 'Number');
        }

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _ApiClient["default"].convertToType(data['dependantOn'], [_Dependancy["default"]]);
        }

        if (data.hasOwnProperty('entrantMemberType')) {
          obj['entrantMemberType'] = _ApiClient["default"].convertToType(data['entrantMemberType'], 'String');
        }

        if (data.hasOwnProperty('scheduledStartDate')) {
          obj['scheduledStartDate'] = _ApiClient["default"].convertToType(data['scheduledStartDate'], 'Date');
        }

        if (data.hasOwnProperty('scheduledEndDate')) {
          obj['scheduledEndDate'] = _ApiClient["default"].convertToType(data['scheduledEndDate'], 'Date');
        }

        if (data.hasOwnProperty('actualStartDate')) {
          obj['actualStartDate'] = _ApiClient["default"].convertToType(data['actualStartDate'], 'Date');
        }

        if (data.hasOwnProperty('actualEndDate')) {
          obj['actualEndDate'] = _ApiClient["default"].convertToType(data['actualEndDate'], 'Date');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CompetitionReduced;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


CompetitionReduced.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

CompetitionReduced.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

CompetitionReduced.prototype['created'] = undefined;
/**
 * @member {module:model/CompetitionType} competitionType
 */

CompetitionReduced.prototype['competitionType'] = undefined;
/**
 * Number of rounds to be played in a competition
 * @member {Number} numberOfRounds
 */

CompetitionReduced.prototype['numberOfRounds'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

CompetitionReduced.prototype['name'] = undefined;
/**
 * @member {module:model/CompetitionStatus} status
 */

CompetitionReduced.prototype['status'] = undefined;
/**
 * The code of the competition
 * @member {Number} statusCode
 */

CompetitionReduced.prototype['statusCode'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

CompetitionReduced.prototype['dependantOn'] = undefined;
/**
 * @member {String} entrantMemberType
 */

CompetitionReduced.prototype['entrantMemberType'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

CompetitionReduced.prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

CompetitionReduced.prototype['scheduledEndDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition started. All records are stored in UTC time zone
 * @member {Date} actualStartDate
 */

CompetitionReduced.prototype['actualStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition ended. All records are stored in UTC time zone
 * @member {Date} actualEndDate
 */

CompetitionReduced.prototype['actualEndDate'] = undefined;
/**
 * A list of id's used to tag
 * @member {Array.<String>} tags
 */

CompetitionReduced.prototype['tags'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

CompetitionReduced.prototype['constraints'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement CompetitionReducedAllOf interface:

/**
 * @member {module:model/CompetitionType} competitionType
 */

_CompetitionReducedAllOf["default"].prototype['competitionType'] = undefined;
/**
 * Number of rounds to be played in a competition
 * @member {Number} numberOfRounds
 */

_CompetitionReducedAllOf["default"].prototype['numberOfRounds'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

_CompetitionReducedAllOf["default"].prototype['name'] = undefined;
/**
 * @member {module:model/CompetitionStatus} status
 */

_CompetitionReducedAllOf["default"].prototype['status'] = undefined;
/**
 * The code of the competition
 * @member {Number} statusCode
 */

_CompetitionReducedAllOf["default"].prototype['statusCode'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

_CompetitionReducedAllOf["default"].prototype['dependantOn'] = undefined;
/**
 * @member {String} entrantMemberType
 */

_CompetitionReducedAllOf["default"].prototype['entrantMemberType'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

_CompetitionReducedAllOf["default"].prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

_CompetitionReducedAllOf["default"].prototype['scheduledEndDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition started. All records are stored in UTC time zone
 * @member {Date} actualStartDate
 */

_CompetitionReducedAllOf["default"].prototype['actualStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition ended. All records are stored in UTC time zone
 * @member {Date} actualEndDate
 */

_CompetitionReducedAllOf["default"].prototype['actualEndDate'] = undefined;
/**
 * A list of id's used to tag
 * @member {Array.<String>} tags
 */

_CompetitionReducedAllOf["default"].prototype['tags'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_CompetitionReducedAllOf["default"].prototype['constraints'] = undefined;
var _default = CompetitionReduced;
exports["default"] = _default;
