// @ziqni-tech/admin-api-client/dist/model/CompetitionAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CompetitionStatus = _interopRequireDefault(require("./CompetitionStatus"));

var _CompetitionType = _interopRequireDefault(require("./CompetitionType"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _ProductReduced = _interopRequireDefault(require("./ProductReduced"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CompetitionAllOf model module.
 * @module model/CompetitionAllOf
 * @version 0.0.26
 */
var CompetitionAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CompetitionAllOf</code>.
   * @alias module:model/CompetitionAllOf
   * @param competitionType {module:model/CompetitionType} 
   * @param numberOfRounds {Number} Number of rounds to be played in a competition
   * @param name {String} A name or a name of a competition. Can be translated
   * @param minNumberOfEntrants {Number} Maximum number of partiipants allowed in a competition
   * @param scheduledStartDate {Date} ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
   * @param scheduledEndDate {Date} ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
   * @param status {module:model/CompetitionStatus} 
   * @param constraints {Array.<String>} Additional constraints
   * @param products {Array.<module:model/ProductReduced>} 
   */
  function CompetitionAllOf(competitionType, numberOfRounds, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, status, constraints, products) {
    _classCallCheck(this, CompetitionAllOf);

    CompetitionAllOf.initialize(this, competitionType, numberOfRounds, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, status, constraints, products);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CompetitionAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, competitionType, numberOfRounds, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, status, constraints, products) {
      obj['competitionType'] = competitionType;
      obj['numberOfRounds'] = numberOfRounds;
      obj['name'] = name;
      obj['minNumberOfEntrants'] = minNumberOfEntrants;
      obj['scheduledStartDate'] = scheduledStartDate;
      obj['scheduledEndDate'] = scheduledEndDate;
      obj['status'] = status;
      obj['constraints'] = constraints;
      obj['products'] = products;
    }
    /**
     * Constructs a <code>CompetitionAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CompetitionAllOf} obj Optional instance to populate.
     * @return {module:model/CompetitionAllOf} The populated <code>CompetitionAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CompetitionAllOf();

        if (data.hasOwnProperty('competitionType')) {
          obj['competitionType'] = _CompetitionType["default"].constructFromObject(data['competitionType']);
        }

        if (data.hasOwnProperty('numberOfRounds')) {
          obj['numberOfRounds'] = _ApiClient["default"].convertToType(data['numberOfRounds'], 'Number');
        }

        if (data.hasOwnProperty('numberOfGroupStages')) {
          obj['numberOfGroupStages'] = _ApiClient["default"].convertToType(data['numberOfGroupStages'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('maxNumberOfEntrants')) {
          obj['maxNumberOfEntrants'] = _ApiClient["default"].convertToType(data['maxNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('minNumberOfEntrants')) {
          obj['minNumberOfEntrants'] = _ApiClient["default"].convertToType(data['minNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('entrantMemberType')) {
          obj['entrantMemberType'] = _ApiClient["default"].convertToType(data['entrantMemberType'], 'String');
        }

        if (data.hasOwnProperty('scheduledStartDate')) {
          obj['scheduledStartDate'] = _ApiClient["default"].convertToType(data['scheduledStartDate'], 'Date');
        }

        if (data.hasOwnProperty('scheduledEndDate')) {
          obj['scheduledEndDate'] = _ApiClient["default"].convertToType(data['scheduledEndDate'], 'Date');
        }

        if (data.hasOwnProperty('actualStartDate')) {
          obj['actualStartDate'] = _ApiClient["default"].convertToType(data['actualStartDate'], 'Date');
        }

        if (data.hasOwnProperty('actualEndDate')) {
          obj['actualEndDate'] = _ApiClient["default"].convertToType(data['actualEndDate'], 'Date');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _CompetitionStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('statusCode')) {
          obj['statusCode'] = _ApiClient["default"].convertToType(data['statusCode'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('products')) {
          obj['products'] = _ApiClient["default"].convertToType(data['products'], [_ProductReduced["default"]]);
        }

        if (data.hasOwnProperty('entrantMemberTagsFilter')) {
          obj['entrantMemberTagsFilter'] = _DependantOn["default"].constructFromObject(data['entrantMemberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }
      }

      return obj;
    }
  }]);

  return CompetitionAllOf;
}();
/**
 * @member {module:model/CompetitionType} competitionType
 */


CompetitionAllOf.prototype['competitionType'] = undefined;
/**
 * Number of rounds to be played in a competition
 * @member {Number} numberOfRounds
 */

CompetitionAllOf.prototype['numberOfRounds'] = undefined;
/**
 * Number of group stages
 * @member {Number} numberOfGroupStages
 */

CompetitionAllOf.prototype['numberOfGroupStages'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

CompetitionAllOf.prototype['name'] = undefined;
/**
 * Description of the competition. Can be translated
 * @member {String} description
 */

CompetitionAllOf.prototype['description'] = undefined;
/**
 * Terms and conditions of a competition. Can be translated
 * @member {String} termsAndConditions
 */

CompetitionAllOf.prototype['termsAndConditions'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} maxNumberOfEntrants
 */

CompetitionAllOf.prototype['maxNumberOfEntrants'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} minNumberOfEntrants
 */

CompetitionAllOf.prototype['minNumberOfEntrants'] = undefined;
/**
 * @member {String} entrantMemberType
 */

CompetitionAllOf.prototype['entrantMemberType'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

CompetitionAllOf.prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

CompetitionAllOf.prototype['scheduledEndDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition started. All records are stored in UTC time zone
 * @member {Date} actualStartDate
 */

CompetitionAllOf.prototype['actualStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition ended. All records are stored in UTC time zone
 * @member {Date} actualEndDate
 */

CompetitionAllOf.prototype['actualEndDate'] = undefined;
/**
 * @member {module:model/CompetitionStatus} status
 */

CompetitionAllOf.prototype['status'] = undefined;
/**
 * The code of the competition
 * @member {Number} statusCode
 */

CompetitionAllOf.prototype['statusCode'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

CompetitionAllOf.prototype['constraints'] = undefined;
/**
 * @member {Array.<module:model/ProductReduced>} products
 */

CompetitionAllOf.prototype['products'] = undefined;
/**
 * @member {module:model/DependantOn} entrantMemberTagsFilter
 */

CompetitionAllOf.prototype['entrantMemberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

CompetitionAllOf.prototype['productTagsFilter'] = undefined;
var _default = CompetitionAllOf;
exports["default"] = _default;
