// @ziqni-tech/admin-api-client/dist/model/Condition.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MatchCondition = _interopRequireDefault(require("./MatchCondition"));

var _Rule = _interopRequireDefault(require("./Rule"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Condition model module.
 * @module model/Condition
 * @version 0.0.26
 */
var Condition = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Condition</code>.
   * @alias module:model/Condition
   * @param matchCondition {module:model/MatchCondition} 
   * @param rules {Array.<module:model/Rule>} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function Condition(matchCondition, rules, constraints) {
    _classCallCheck(this, Condition);

    Condition.initialize(this, matchCondition, rules, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Condition, null, [{
    key: "initialize",
    value: function initialize(obj, matchCondition, rules, constraints) {
      obj['matchCondition'] = matchCondition;
      obj['rules'] = rules;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>Condition</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Condition} obj Optional instance to populate.
     * @return {module:model/Condition} The populated <code>Condition</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Condition();

        if (data.hasOwnProperty('matchCondition')) {
          obj['matchCondition'] = _MatchCondition["default"].constructFromObject(data['matchCondition']);
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _ApiClient["default"].convertToType(data['rules'], [_Rule["default"]]);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return Condition;
}();
/**
 * @member {module:model/MatchCondition} matchCondition
 */


Condition.prototype['matchCondition'] = undefined;
/**
 * @member {Array.<module:model/Rule>} rules
 */

Condition.prototype['rules'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

Condition.prototype['constraints'] = undefined;
var _default = Condition;
exports["default"] = _default;
