// @ziqni-tech/admin-api-client/dist/model/Contest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ContestAllOf = _interopRequireDefault(require("./ContestAllOf"));

var _ContestStatus = _interopRequireDefault(require("./ContestStatus"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _RoundType = _interopRequireDefault(require("./RoundType"));

var _Strategy = _interopRequireDefault(require("./Strategy"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Contest model module.
 * @module model/Contest
 * @version 0.0.26
 */
var Contest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Contest</code>.
   * @alias module:model/Contest
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/ContestAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param competitionId {String} A unique identifier of a Competition
   * @param row {Number} The row number for displaying the Contest in a table
   * @param name {String} A name for the Contest. Can be translated
   * @param round {Number} To what round does the contest belong
   * @param roundType {module:model/RoundType} 
   * @param entrantsFromContest {Array.<String>} A list of Ziqni contest identifiers to allow contests chaining. When a Progression competition is active and you have a 2 round competition the first round has to reference the second round then this parameter becomes mandatory instead of Optional.
   * @param minNumberOfEntrants {Number} Minimum number of entrants for the contest
   * @param scheduledStartDate {Date} ISO8601 timestamp for when a Contest should start. All records are stored in UTC time zone
   * @param scheduledEndDate {Date} ISO8601 timestamp for when a Contest should end. All records are stored in UTC time zone
   * @param status {module:model/ContestStatus} 
   * @param statusCode {Number} The code of the contest
   * @param constraints {Array.<String>} Additional constraints
   */
  function Contest(id, spaceName, created, competitionId, row, name, round, roundType, entrantsFromContest, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, status, statusCode, constraints) {
    _classCallCheck(this, Contest);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _ContestAllOf["default"].initialize(this, competitionId, row, name, round, roundType, entrantsFromContest, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, status, statusCode, constraints);

    Contest.initialize(this, id, spaceName, created, competitionId, row, name, round, roundType, entrantsFromContest, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, status, statusCode, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Contest, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, competitionId, row, name, round, roundType, entrantsFromContest, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, status, statusCode, constraints) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['competitionId'] = competitionId;
      obj['row'] = row;
      obj['name'] = name;
      obj['round'] = round;
      obj['roundType'] = roundType;
      obj['entrantsFromContest'] = entrantsFromContest;
      obj['minNumberOfEntrants'] = minNumberOfEntrants;
      obj['scheduledStartDate'] = scheduledStartDate;
      obj['scheduledEndDate'] = scheduledEndDate;
      obj['status'] = status;
      obj['statusCode'] = statusCode;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>Contest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Contest} obj Optional instance to populate.
     * @return {module:model/Contest} The populated <code>Contest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Contest();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _ContestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('competitionId')) {
          obj['competitionId'] = _ApiClient["default"].convertToType(data['competitionId'], 'String');
        }

        if (data.hasOwnProperty('row')) {
          obj['row'] = _ApiClient["default"].convertToType(data['row'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('round')) {
          obj['round'] = _ApiClient["default"].convertToType(data['round'], 'Number');
        }

        if (data.hasOwnProperty('roundType')) {
          obj['roundType'] = _RoundType["default"].constructFromObject(data['roundType']);
        }

        if (data.hasOwnProperty('groupStage')) {
          obj['groupStage'] = _ApiClient["default"].convertToType(data['groupStage'], 'Number');
        }

        if (data.hasOwnProperty('groupStageLabel')) {
          obj['groupStageLabel'] = _ApiClient["default"].convertToType(data['groupStageLabel'], 'String');
        }

        if (data.hasOwnProperty('entrantsFromContest')) {
          obj['entrantsFromContest'] = _ApiClient["default"].convertToType(data['entrantsFromContest'], ['String']);
        }

        if (data.hasOwnProperty('maxNumberOfEntrants')) {
          obj['maxNumberOfEntrants'] = _ApiClient["default"].convertToType(data['maxNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('minNumberOfEntrants')) {
          obj['minNumberOfEntrants'] = _ApiClient["default"].convertToType(data['minNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('scheduledStartDate')) {
          obj['scheduledStartDate'] = _ApiClient["default"].convertToType(data['scheduledStartDate'], 'Date');
        }

        if (data.hasOwnProperty('scheduledEndDate')) {
          obj['scheduledEndDate'] = _ApiClient["default"].convertToType(data['scheduledEndDate'], 'Date');
        }

        if (data.hasOwnProperty('actualStartDate')) {
          obj['actualStartDate'] = _ApiClient["default"].convertToType(data['actualStartDate'], 'Date');
        }

        if (data.hasOwnProperty('actualEndDate')) {
          obj['actualEndDate'] = _ApiClient["default"].convertToType(data['actualEndDate'], 'Date');
        }

        if (data.hasOwnProperty('strategies')) {
          obj['strategies'] = _Strategy["default"].constructFromObject(data['strategies']);
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ContestStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('statusCode')) {
          obj['statusCode'] = _ApiClient["default"].convertToType(data['statusCode'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return Contest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Contest.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Contest.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Contest.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Contest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

Contest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Contest.prototype['metadata'] = undefined;
/**
 * A unique identifier of a Competition
 * @member {String} competitionId
 */

Contest.prototype['competitionId'] = undefined;
/**
 * The row number for displaying the Contest in a table
 * @member {Number} row
 */

Contest.prototype['row'] = undefined;
/**
 * A name for the Contest. Can be translated
 * @member {String} name
 */

Contest.prototype['name'] = undefined;
/**
 * Description of the contest
 * @member {String} description
 */

Contest.prototype['description'] = undefined;
/**
 * Terms and conditions for participating in the Contest. Can be translated
 * @member {String} termsAndConditions
 */

Contest.prototype['termsAndConditions'] = undefined;
/**
 * To what round does the contest belong
 * @member {Number} round
 */

Contest.prototype['round'] = undefined;
/**
 * @member {module:model/RoundType} roundType
 */

Contest.prototype['roundType'] = undefined;
/**
 * Is used for more complex Competitions e.g. multi round competitions where round 1 group stage matched round 2
 * @member {Number} groupStage
 */

Contest.prototype['groupStage'] = undefined;
/**
 * The name of the group stages
 * @member {String} groupStageLabel
 */

Contest.prototype['groupStageLabel'] = undefined;
/**
 * A list of Ziqni contest identifiers to allow contests chaining. When a Progression competition is active and you have a 2 round competition the first round has to reference the second round then this parameter becomes mandatory instead of Optional.
 * @member {Array.<String>} entrantsFromContest
 */

Contest.prototype['entrantsFromContest'] = undefined;
/**
 * Maximum number of entrants for the contest
 * @member {Number} maxNumberOfEntrants
 */

Contest.prototype['maxNumberOfEntrants'] = undefined;
/**
 * Minimum number of entrants for the contest
 * @member {Number} minNumberOfEntrants
 */

Contest.prototype['minNumberOfEntrants'] = undefined;
/**
 * ISO8601 timestamp for when a Contest should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

Contest.prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Contest should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

Contest.prototype['scheduledEndDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition started. All records are stored in UTC time zone
 * @member {Date} actualStartDate
 */

Contest.prototype['actualStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition ended. All records are stored in UTC time zone
 * @member {Date} actualEndDate
 */

Contest.prototype['actualEndDate'] = undefined;
/**
 * @member {module:model/Strategy} strategies
 */

Contest.prototype['strategies'] = undefined;
/**
 * @member {module:model/ContestStatus} status
 */

Contest.prototype['status'] = undefined;
/**
 * The code of the contest
 * @member {Number} statusCode
 */

Contest.prototype['statusCode'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

Contest.prototype['constraints'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement ContestAllOf interface:

/**
 * A unique identifier of a Competition
 * @member {String} competitionId
 */

_ContestAllOf["default"].prototype['competitionId'] = undefined;
/**
 * The row number for displaying the Contest in a table
 * @member {Number} row
 */

_ContestAllOf["default"].prototype['row'] = undefined;
/**
 * A name for the Contest. Can be translated
 * @member {String} name
 */

_ContestAllOf["default"].prototype['name'] = undefined;
/**
 * Description of the contest
 * @member {String} description
 */

_ContestAllOf["default"].prototype['description'] = undefined;
/**
 * Terms and conditions for participating in the Contest. Can be translated
 * @member {String} termsAndConditions
 */

_ContestAllOf["default"].prototype['termsAndConditions'] = undefined;
/**
 * To what round does the contest belong
 * @member {Number} round
 */

_ContestAllOf["default"].prototype['round'] = undefined;
/**
 * @member {module:model/RoundType} roundType
 */

_ContestAllOf["default"].prototype['roundType'] = undefined;
/**
 * Is used for more complex Competitions e.g. multi round competitions where round 1 group stage matched round 2
 * @member {Number} groupStage
 */

_ContestAllOf["default"].prototype['groupStage'] = undefined;
/**
 * The name of the group stages
 * @member {String} groupStageLabel
 */

_ContestAllOf["default"].prototype['groupStageLabel'] = undefined;
/**
 * A list of Ziqni contest identifiers to allow contests chaining. When a Progression competition is active and you have a 2 round competition the first round has to reference the second round then this parameter becomes mandatory instead of Optional.
 * @member {Array.<String>} entrantsFromContest
 */

_ContestAllOf["default"].prototype['entrantsFromContest'] = undefined;
/**
 * Maximum number of entrants for the contest
 * @member {Number} maxNumberOfEntrants
 */

_ContestAllOf["default"].prototype['maxNumberOfEntrants'] = undefined;
/**
 * Minimum number of entrants for the contest
 * @member {Number} minNumberOfEntrants
 */

_ContestAllOf["default"].prototype['minNumberOfEntrants'] = undefined;
/**
 * ISO8601 timestamp for when a Contest should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

_ContestAllOf["default"].prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Contest should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

_ContestAllOf["default"].prototype['scheduledEndDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition started. All records are stored in UTC time zone
 * @member {Date} actualStartDate
 */

_ContestAllOf["default"].prototype['actualStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition ended. All records are stored in UTC time zone
 * @member {Date} actualEndDate
 */

_ContestAllOf["default"].prototype['actualEndDate'] = undefined;
/**
 * @member {module:model/Strategy} strategies
 */

_ContestAllOf["default"].prototype['strategies'] = undefined;
/**
 * @member {module:model/ContestStatus} status
 */

_ContestAllOf["default"].prototype['status'] = undefined;
/**
 * The code of the contest
 * @member {Number} statusCode
 */

_ContestAllOf["default"].prototype['statusCode'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_ContestAllOf["default"].prototype['constraints'] = undefined;
var _default = Contest;
exports["default"] = _default;
