// @ziqni-tech/admin-api-client/dist/model/Contact.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ContactAllOf = _interopRequireDefault(require("./ContactAllOf"));

var _Location = _interopRequireDefault(require("./Location"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Contact model module.
 * @module model/Contact
 * @version 0.0.5
 */
var Contact = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Contact</code>.
   * @alias module:model/Contact
   * @implements module:model/ModelDefault
   * @implements module:model/ContactAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param addressLine1 {String} The address
   * @param zipPostalCode {String} The Postal code of the address
   * @param country {String} The Country
   */
  function Contact(id, spaceName, created, addressLine1, zipPostalCode, country) {
    _classCallCheck(this, Contact);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _ContactAllOf["default"].initialize(this, addressLine1, zipPostalCode, country);

    Contact.initialize(this, id, spaceName, created, addressLine1, zipPostalCode, country);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Contact, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, addressLine1, zipPostalCode, country) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['addressLine1'] = addressLine1;
      obj['zipPostalCode'] = zipPostalCode;
      obj['country'] = country;
    }
    /**
     * Constructs a <code>Contact</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Contact} obj Optional instance to populate.
     * @return {module:model/Contact} The populated <code>Contact</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Contact();

        _ModelDefault["default"].constructFromObject(data, obj);

        _ContactAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('addressLine1')) {
          obj['addressLine1'] = _ApiClient["default"].convertToType(data['addressLine1'], 'String');
        }

        if (data.hasOwnProperty('addressLine2')) {
          obj['addressLine2'] = _ApiClient["default"].convertToType(data['addressLine2'], 'String');
        }

        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }

        if (data.hasOwnProperty('zipPostalCode')) {
          obj['zipPostalCode'] = _ApiClient["default"].convertToType(data['zipPostalCode'], 'String');
        }

        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }

        if (data.hasOwnProperty('mobileNumber')) {
          obj['mobileNumber'] = _ApiClient["default"].convertToType(data['mobileNumber'], 'String');
        }

        if (data.hasOwnProperty('phoneNumber')) {
          obj['phoneNumber'] = _ApiClient["default"].convertToType(data['phoneNumber'], 'String');
        }

        if (data.hasOwnProperty('geoLocation')) {
          obj['geoLocation'] = _Location["default"].constructFromObject(data['geoLocation']);
        }
      }

      return obj;
    }
  }]);

  return Contact;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Contact.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Contact.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Contact.prototype['created'] = undefined;
/**
 * The address
 * @member {String} addressLine1
 */

Contact.prototype['addressLine1'] = undefined;
/**
 * The secondary address
 * @member {String} addressLine2
 */

Contact.prototype['addressLine2'] = undefined;
/**
 * The city
 * @member {String} city
 */

Contact.prototype['city'] = undefined;
/**
 * The state of the city
 * @member {String} state
 */

Contact.prototype['state'] = undefined;
/**
 * The Postal code of the address
 * @member {String} zipPostalCode
 */

Contact.prototype['zipPostalCode'] = undefined;
/**
 * The Country
 * @member {String} country
 */

Contact.prototype['country'] = undefined;
/**
 * The mobile number
 * @member {String} mobileNumber
 */

Contact.prototype['mobileNumber'] = undefined;
/**
 * The phone number
 * @member {String} phoneNumber
 */

Contact.prototype['phoneNumber'] = undefined;
/**
 * @member {module:model/Location} geoLocation
 */

Contact.prototype['geoLocation'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement ContactAllOf interface:

/**
 * The address
 * @member {String} addressLine1
 */

_ContactAllOf["default"].prototype['addressLine1'] = undefined;
/**
 * The secondary address
 * @member {String} addressLine2
 */

_ContactAllOf["default"].prototype['addressLine2'] = undefined;
/**
 * The city
 * @member {String} city
 */

_ContactAllOf["default"].prototype['city'] = undefined;
/**
 * The state of the city
 * @member {String} state
 */

_ContactAllOf["default"].prototype['state'] = undefined;
/**
 * The Postal code of the address
 * @member {String} zipPostalCode
 */

_ContactAllOf["default"].prototype['zipPostalCode'] = undefined;
/**
 * The Country
 * @member {String} country
 */

_ContactAllOf["default"].prototype['country'] = undefined;
/**
 * The mobile number
 * @member {String} mobileNumber
 */

_ContactAllOf["default"].prototype['mobileNumber'] = undefined;
/**
 * The phone number
 * @member {String} phoneNumber
 */

_ContactAllOf["default"].prototype['phoneNumber'] = undefined;
/**
 * @member {module:model/Location} geoLocation
 */

_ContactAllOf["default"].prototype['geoLocation'] = undefined;
var _default = Contact;
exports["default"] = _default;
