// @ziqni-tech/admin-api-client/dist/model/ContactAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Location = _interopRequireDefault(require("./Location"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ContactAllOf model module.
 * @module model/ContactAllOf
 * @version 0.0.5
 */
var ContactAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ContactAllOf</code>.
   * @alias module:model/ContactAllOf
   * @param addressLine1 {String} The address
   * @param zipPostalCode {String} The Postal code of the address
   * @param country {String} The Country
   */
  function ContactAllOf(addressLine1, zipPostalCode, country) {
    _classCallCheck(this, ContactAllOf);

    ContactAllOf.initialize(this, addressLine1, zipPostalCode, country);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ContactAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, addressLine1, zipPostalCode, country) {
      obj['addressLine1'] = addressLine1;
      obj['zipPostalCode'] = zipPostalCode;
      obj['country'] = country;
    }
    /**
     * Constructs a <code>ContactAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ContactAllOf} obj Optional instance to populate.
     * @return {module:model/ContactAllOf} The populated <code>ContactAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ContactAllOf();

        if (data.hasOwnProperty('addressLine1')) {
          obj['addressLine1'] = _ApiClient["default"].convertToType(data['addressLine1'], 'String');
        }

        if (data.hasOwnProperty('addressLine2')) {
          obj['addressLine2'] = _ApiClient["default"].convertToType(data['addressLine2'], 'String');
        }

        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }

        if (data.hasOwnProperty('zipPostalCode')) {
          obj['zipPostalCode'] = _ApiClient["default"].convertToType(data['zipPostalCode'], 'String');
        }

        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }

        if (data.hasOwnProperty('mobileNumber')) {
          obj['mobileNumber'] = _ApiClient["default"].convertToType(data['mobileNumber'], 'String');
        }

        if (data.hasOwnProperty('phoneNumber')) {
          obj['phoneNumber'] = _ApiClient["default"].convertToType(data['phoneNumber'], 'String');
        }

        if (data.hasOwnProperty('geoLocation')) {
          obj['geoLocation'] = _Location["default"].constructFromObject(data['geoLocation']);
        }
      }

      return obj;
    }
  }]);

  return ContactAllOf;
}();
/**
 * The address
 * @member {String} addressLine1
 */


ContactAllOf.prototype['addressLine1'] = undefined;
/**
 * The secondary address
 * @member {String} addressLine2
 */

ContactAllOf.prototype['addressLine2'] = undefined;
/**
 * The city
 * @member {String} city
 */

ContactAllOf.prototype['city'] = undefined;
/**
 * The state of the city
 * @member {String} state
 */

ContactAllOf.prototype['state'] = undefined;
/**
 * The Postal code of the address
 * @member {String} zipPostalCode
 */

ContactAllOf.prototype['zipPostalCode'] = undefined;
/**
 * The Country
 * @member {String} country
 */

ContactAllOf.prototype['country'] = undefined;
/**
 * The mobile number
 * @member {String} mobileNumber
 */

ContactAllOf.prototype['mobileNumber'] = undefined;
/**
 * The phone number
 * @member {String} phoneNumber
 */

ContactAllOf.prototype['phoneNumber'] = undefined;
/**
 * @member {module:model/Location} geoLocation
 */

ContactAllOf.prototype['geoLocation'] = undefined;
var _default = ContactAllOf;
exports["default"] = _default;
