// @ziqni-tech/admin-api-client/dist/model/CreateAchievementRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateRewardRequest = _interopRequireDefault(require("./CreateRewardRequest"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _Rule = _interopRequireDefault(require("./Rule"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateAchievementRequestAllOf model module.
 * @module model/CreateAchievementRequestAllOf
 * @version 0.0.26
 */
var CreateAchievementRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAchievementRequestAllOf</code>.
   * @alias module:model/CreateAchievementRequestAllOf
   * @param name {String} A name for the Achievement. Can be translated
   * @param scheduling {module:model/Scheduling} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function CreateAchievementRequestAllOf(name, scheduling, constraints) {
    _classCallCheck(this, CreateAchievementRequestAllOf);

    CreateAchievementRequestAllOf.initialize(this, name, scheduling, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAchievementRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, scheduling, constraints) {
      obj['name'] = name;
      obj['scheduling'] = scheduling;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateAchievementRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAchievementRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateAchievementRequestAllOf} The populated <code>CreateAchievementRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAchievementRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('icon')) {
          obj['icon'] = _ApiClient["default"].convertToType(data['icon'], 'String');
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _ApiClient["default"].convertToType(data['rules'], [_Rule["default"]]);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('maxNumberOfIssues')) {
          obj['maxNumberOfIssues'] = _ApiClient["default"].convertToType(data['maxNumberOfIssues'], 'Number');
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('rewards')) {
          obj['rewards'] = _ApiClient["default"].convertToType(data['rewards'], [_CreateRewardRequest["default"]]);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('achievementDependencies')) {
          obj['achievementDependencies'] = _DependantOn["default"].constructFromObject(data['achievementDependencies']);
        }

        if (data.hasOwnProperty('memberTagsFilter')) {
          obj['memberTagsFilter'] = _DependantOn["default"].constructFromObject(data['memberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateAchievementRequestAllOf;
}();
/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */


CreateAchievementRequestAllOf.prototype['name'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} description
 */

CreateAchievementRequestAllOf.prototype['description'] = undefined;
/**
 * Terms and conditions of an achievement. Can be translated
 * @member {String} termsAndConditions
 */

CreateAchievementRequestAllOf.prototype['termsAndConditions'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */

CreateAchievementRequestAllOf.prototype['icon'] = undefined;
/**
 * @member {Array.<module:model/Rule>} rules
 */

CreateAchievementRequestAllOf.prototype['rules'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

CreateAchievementRequestAllOf.prototype['scheduling'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

CreateAchievementRequestAllOf.prototype['maxNumberOfIssues'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateAchievementRequestAllOf.prototype['translations'] = undefined;
/**
 * @member {Array.<module:model/CreateRewardRequest>} rewards
 */

CreateAchievementRequestAllOf.prototype['rewards'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

CreateAchievementRequestAllOf.prototype['constraints'] = undefined;
/**
 * @member {module:model/DependantOn} achievementDependencies
 */

CreateAchievementRequestAllOf.prototype['achievementDependencies'] = undefined;
/**
 * @member {module:model/DependantOn} memberTagsFilter
 */

CreateAchievementRequestAllOf.prototype['memberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

CreateAchievementRequestAllOf.prototype['productTagsFilter'] = undefined;
/**
 * @member {Array.<String>} addConstraints
 */

CreateAchievementRequestAllOf.prototype['addConstraints'] = undefined;
var _default = CreateAchievementRequestAllOf;
exports["default"] = _default;
