// @ziqni-tech/admin-api-client/dist/model/CreateSqsConnectionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _CreateSqsConnectionRequestAllOf = _interopRequireDefault(require("./CreateSqsConnectionRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateSqsConnectionRequest model module.
 * @module model/CreateSqsConnectionRequest
 * @version 0.0.26
 */
var CreateSqsConnectionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateSqsConnectionRequest</code>.
   * @alias module:model/CreateSqsConnectionRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateSqsConnectionRequestAllOf
   * @param name {String} The name of the consumer
   * @param uri {String} The SQS endpoint.
   * @param acessKey {String} The access key of the IAM user
   * @param secretKey {String} The secret key of IAM user
   * @param transformerId {String} The transformer to use, if empty the default system transformer will be used
   */
  function CreateSqsConnectionRequest(name, uri, acessKey, secretKey, transformerId) {
    _classCallCheck(this, CreateSqsConnectionRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateSqsConnectionRequestAllOf["default"].initialize(this, name, uri, acessKey, secretKey, transformerId);

    CreateSqsConnectionRequest.initialize(this, name, uri, acessKey, secretKey, transformerId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateSqsConnectionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, uri, acessKey, secretKey, transformerId) {
      obj['name'] = name;
      obj['uri'] = uri;
      obj['acessKey'] = acessKey;
      obj['secretKey'] = secretKey;
      obj['transformerId'] = transformerId;
    }
    /**
     * Constructs a <code>CreateSqsConnectionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateSqsConnectionRequest} obj Optional instance to populate.
     * @return {module:model/CreateSqsConnectionRequest} The populated <code>CreateSqsConnectionRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateSqsConnectionRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateSqsConnectionRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }

        if (data.hasOwnProperty('acessKey')) {
          obj['acessKey'] = _ApiClient["default"].convertToType(data['acessKey'], 'String');
        }

        if (data.hasOwnProperty('secretKey')) {
          obj['secretKey'] = _ApiClient["default"].convertToType(data['secretKey'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateSqsConnectionRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateSqsConnectionRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateSqsConnectionRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateSqsConnectionRequest.prototype['metadata'] = undefined;
/**
 * The name of the consumer
 * @member {String} name
 */

CreateSqsConnectionRequest.prototype['name'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

CreateSqsConnectionRequest.prototype['uri'] = undefined;
/**
 * The access key of the IAM user
 * @member {String} acessKey
 */

CreateSqsConnectionRequest.prototype['acessKey'] = undefined;
/**
 * The secret key of IAM user
 * @member {String} secretKey
 */

CreateSqsConnectionRequest.prototype['secretKey'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

CreateSqsConnectionRequest.prototype['description'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

CreateSqsConnectionRequest.prototype['transformerId'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateSqsConnectionRequestAllOf interface:

/**
 * The name of the consumer
 * @member {String} name
 */

_CreateSqsConnectionRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

_CreateSqsConnectionRequestAllOf["default"].prototype['uri'] = undefined;
/**
 * The access key of the IAM user
 * @member {String} acessKey
 */

_CreateSqsConnectionRequestAllOf["default"].prototype['acessKey'] = undefined;
/**
 * The secret key of IAM user
 * @member {String} secretKey
 */

_CreateSqsConnectionRequestAllOf["default"].prototype['secretKey'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

_CreateSqsConnectionRequestAllOf["default"].prototype['description'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

_CreateSqsConnectionRequestAllOf["default"].prototype['transformerId'] = undefined;
var _default = CreateSqsConnectionRequest;
exports["default"] = _default;
