// @ziqni-tech/admin-api-client/dist/model/CreateKafkaConnectionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateKafkaConnectionRequestAllOf = _interopRequireDefault(require("./CreateKafkaConnectionRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateKafkaConnectionRequest model module.
 * @module model/CreateKafkaConnectionRequest
 * @version 0.0.26
 */
var CreateKafkaConnectionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateKafkaConnectionRequest</code>.
   * @alias module:model/CreateKafkaConnectionRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateKafkaConnectionRequestAllOf
   * @param name {String} The name of the consumer
   * @param brokers {String} Kafka broker endpoints.
   * @param groupId {String} Group Id for connection
   * @param topic {String} Topic name
   * @param transformerId {String} The transformer to use, if empty the default system transformer will be used
   */
  function CreateKafkaConnectionRequest(name, brokers, groupId, topic, transformerId) {
    _classCallCheck(this, CreateKafkaConnectionRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateKafkaConnectionRequestAllOf["default"].initialize(this, name, brokers, groupId, topic, transformerId);

    CreateKafkaConnectionRequest.initialize(this, name, brokers, groupId, topic, transformerId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateKafkaConnectionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, brokers, groupId, topic, transformerId) {
      obj['name'] = name;
      obj['brokers'] = brokers;
      obj['groupId'] = groupId;
      obj['topic'] = topic;
      obj['transformerId'] = transformerId;
    }
    /**
     * Constructs a <code>CreateKafkaConnectionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateKafkaConnectionRequest} obj Optional instance to populate.
     * @return {module:model/CreateKafkaConnectionRequest} The populated <code>CreateKafkaConnectionRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateKafkaConnectionRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateKafkaConnectionRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('brokers')) {
          obj['brokers'] = _ApiClient["default"].convertToType(data['brokers'], 'String');
        }

        if (data.hasOwnProperty('groupId')) {
          obj['groupId'] = _ApiClient["default"].convertToType(data['groupId'], 'String');
        }

        if (data.hasOwnProperty('topic')) {
          obj['topic'] = _ApiClient["default"].convertToType(data['topic'], 'String');
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateKafkaConnectionRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateKafkaConnectionRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateKafkaConnectionRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateKafkaConnectionRequest.prototype['metadata'] = undefined;
/**
 * The name of the consumer
 * @member {String} name
 */

CreateKafkaConnectionRequest.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

CreateKafkaConnectionRequest.prototype['description'] = undefined;
/**
 * Kafka broker endpoints.
 * @member {String} brokers
 */

CreateKafkaConnectionRequest.prototype['brokers'] = undefined;
/**
 * Group Id for connection
 * @member {String} groupId
 */

CreateKafkaConnectionRequest.prototype['groupId'] = undefined;
/**
 * Topic name
 * @member {String} topic
 */

CreateKafkaConnectionRequest.prototype['topic'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

CreateKafkaConnectionRequest.prototype['transformerId'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateKafkaConnectionRequestAllOf interface:

/**
 * The name of the consumer
 * @member {String} name
 */

_CreateKafkaConnectionRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

_CreateKafkaConnectionRequestAllOf["default"].prototype['description'] = undefined;
/**
 * Kafka broker endpoints.
 * @member {String} brokers
 */

_CreateKafkaConnectionRequestAllOf["default"].prototype['brokers'] = undefined;
/**
 * Group Id for connection
 * @member {String} groupId
 */

_CreateKafkaConnectionRequestAllOf["default"].prototype['groupId'] = undefined;
/**
 * Topic name
 * @member {String} topic
 */

_CreateKafkaConnectionRequestAllOf["default"].prototype['topic'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

_CreateKafkaConnectionRequestAllOf["default"].prototype['transformerId'] = undefined;
var _default = CreateKafkaConnectionRequest;
exports["default"] = _default;
