// @ziqni-tech/admin-api-client/dist/model/CreateRepositoryRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _HostingOptions = _interopRequireDefault(require("./HostingOptions"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateRepositoryRequestAllOf model module.
 * @module model/CreateRepositoryRequestAllOf
 * @version 0.0.26
 */
var CreateRepositoryRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateRepositoryRequestAllOf</code>.
   * @alias module:model/CreateRepositoryRequestAllOf
   * @param name {String} The name of the repository. No special charaters or spaces permitted
   * @param constraints {Array.<String>} Additional constraints
   */
  function CreateRepositoryRequestAllOf(name, constraints) {
    _classCallCheck(this, CreateRepositoryRequestAllOf);

    CreateRepositoryRequestAllOf.initialize(this, name, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateRepositoryRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, constraints) {
      obj['name'] = name;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateRepositoryRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateRepositoryRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateRepositoryRequestAllOf} The populated <code>CreateRepositoryRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateRepositoryRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('hostingOptions')) {
          obj['hostingOptions'] = _HostingOptions["default"].constructFromObject(data['hostingOptions']);
        }
      }

      return obj;
    }
  }]);

  return CreateRepositoryRequestAllOf;
}();
/**
 * The name of the repository. No special charaters or spaces permitted
 * @member {String} name
 */


CreateRepositoryRequestAllOf.prototype['name'] = undefined;
/**
 * The description of the repository
 * @member {String} description
 */

CreateRepositoryRequestAllOf.prototype['description'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

CreateRepositoryRequestAllOf.prototype['constraints'] = undefined;
/**
 * @member {module:model/HostingOptions} hostingOptions
 */

CreateRepositoryRequestAllOf.prototype['hostingOptions'] = undefined;
var _default = CreateRepositoryRequestAllOf;
exports["default"] = _default;
