// @ziqni-tech/admin-api-client/dist/model/CreateFileObjectRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateFileObjectRequestAllOf model module.
 * @module model/CreateFileObjectRequestAllOf
 * @version 0.0.26
 */
var CreateFileObjectRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateFileObjectRequestAllOf</code>.
   * @alias module:model/CreateFileObjectRequestAllOf
   * @param repositoryId {String} The repository identifier this file belongs too
   * @param name {String} Name of the original file uploaded
   * @param mimeType {String} Mime type of the file. Example mime types - text/csv or application/vmd.ms-excel or application/x-directory for directories
   * @param parentFolderPath {String} The folder containing the file within the repository
   */
  function CreateFileObjectRequestAllOf(repositoryId, name, mimeType, parentFolderPath) {
    _classCallCheck(this, CreateFileObjectRequestAllOf);

    CreateFileObjectRequestAllOf.initialize(this, repositoryId, name, mimeType, parentFolderPath);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateFileObjectRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, repositoryId, name, mimeType, parentFolderPath) {
      obj['repositoryId'] = repositoryId;
      obj['name'] = name;
      obj['mimeType'] = mimeType;
      obj['parentFolderPath'] = parentFolderPath;
    }
    /**
     * Constructs a <code>CreateFileObjectRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateFileObjectRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateFileObjectRequestAllOf} The populated <code>CreateFileObjectRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateFileObjectRequestAllOf();

        if (data.hasOwnProperty('repositoryId')) {
          obj['repositoryId'] = _ApiClient["default"].convertToType(data['repositoryId'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('mimeType')) {
          obj['mimeType'] = _ApiClient["default"].convertToType(data['mimeType'], 'String');
        }

        if (data.hasOwnProperty('parentFolderPath')) {
          obj['parentFolderPath'] = _ApiClient["default"].convertToType(data['parentFolderPath'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateFileObjectRequestAllOf;
}();
/**
 * The repository identifier this file belongs too
 * @member {String} repositoryId
 */


CreateFileObjectRequestAllOf.prototype['repositoryId'] = undefined;
/**
 * Name of the original file uploaded
 * @member {String} name
 */

CreateFileObjectRequestAllOf.prototype['name'] = undefined;
/**
 * Mime type of the file. Example mime types - text/csv or application/vmd.ms-excel or application/x-directory for directories
 * @member {String} mimeType
 */

CreateFileObjectRequestAllOf.prototype['mimeType'] = undefined;
/**
 * The folder containing the file within the repository
 * @member {String} parentFolderPath
 */

CreateFileObjectRequestAllOf.prototype['parentFolderPath'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

CreateFileObjectRequestAllOf.prototype['constraints'] = undefined;
var _default = CreateFileObjectRequestAllOf;
exports["default"] = _default;
