// @ziqni-tech/admin-api-client/dist/model/CreateRepositoryRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _CreateRepositoryRequestAllOf = _interopRequireDefault(require("./CreateRepositoryRequestAllOf"));

var _HostingOptions = _interopRequireDefault(require("./HostingOptions"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateRepositoryRequest model module.
 * @module model/CreateRepositoryRequest
 * @version 0.0.26
 */
var CreateRepositoryRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateRepositoryRequest</code>.
   * @alias module:model/CreateRepositoryRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateRepositoryRequestAllOf
   * @param name {String} The name of the repository. No special charaters or spaces permitted
   * @param constraints {Array.<String>} Additional constraints
   */
  function CreateRepositoryRequest(name, constraints) {
    _classCallCheck(this, CreateRepositoryRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateRepositoryRequestAllOf["default"].initialize(this, name, constraints);

    CreateRepositoryRequest.initialize(this, name, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateRepositoryRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, constraints) {
      obj['name'] = name;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateRepositoryRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateRepositoryRequest} obj Optional instance to populate.
     * @return {module:model/CreateRepositoryRequest} The populated <code>CreateRepositoryRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateRepositoryRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateRepositoryRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('hostingOptions')) {
          obj['hostingOptions'] = _HostingOptions["default"].constructFromObject(data['hostingOptions']);
        }
      }

      return obj;
    }
  }]);

  return CreateRepositoryRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateRepositoryRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateRepositoryRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateRepositoryRequest.prototype['metadata'] = undefined;
/**
 * The name of the repository. No special charaters or spaces permitted
 * @member {String} name
 */

CreateRepositoryRequest.prototype['name'] = undefined;
/**
 * The description of the repository
 * @member {String} description
 */

CreateRepositoryRequest.prototype['description'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

CreateRepositoryRequest.prototype['constraints'] = undefined;
/**
 * @member {module:model/HostingOptions} hostingOptions
 */

CreateRepositoryRequest.prototype['hostingOptions'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateRepositoryRequestAllOf interface:

/**
 * The name of the repository. No special charaters or spaces permitted
 * @member {String} name
 */

_CreateRepositoryRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the repository
 * @member {String} description
 */

_CreateRepositoryRequestAllOf["default"].prototype['description'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_CreateRepositoryRequestAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {module:model/HostingOptions} hostingOptions
 */

_CreateRepositoryRequestAllOf["default"].prototype['hostingOptions'] = undefined;
var _default = CreateRepositoryRequest;
exports["default"] = _default;
