// @ziqni-tech/admin-api-client/dist/model/CreateActionTypeRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateActionTypeRequestAllOf = _interopRequireDefault(require("./CreateActionTypeRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateActionTypeRequest model module.
 * @module model/CreateActionTypeRequest
 * @version 0.0.26
 */
var CreateActionTypeRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateActionTypeRequest</code>.
   * @alias module:model/CreateActionTypeRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateActionTypeRequestAllOf
   * @param name {String} The name of the Action Helper
   * @param key {String} A unique key that represents an action helper
   * @param unitOfMeasureType {module:model/UnitOfMeasureType} 
   */
  function CreateActionTypeRequest(name, key, unitOfMeasureType) {
    _classCallCheck(this, CreateActionTypeRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateActionTypeRequestAllOf["default"].initialize(this, name, key, unitOfMeasureType);

    CreateActionTypeRequest.initialize(this, name, key, unitOfMeasureType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateActionTypeRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, key, unitOfMeasureType) {
      obj['name'] = name;
      obj['key'] = key;
      obj['unitOfMeasureType'] = unitOfMeasureType;
    }
    /**
     * Constructs a <code>CreateActionTypeRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateActionTypeRequest} obj Optional instance to populate.
     * @return {module:model/CreateActionTypeRequest} The populated <code>CreateActionTypeRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateActionTypeRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateActionTypeRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateActionTypeRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateActionTypeRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateActionTypeRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateActionTypeRequest.prototype['metadata'] = undefined;
/**
 * The name of the Action Helper
 * @member {String} name
 */

CreateActionTypeRequest.prototype['name'] = undefined;
/**
 * A unique key that represents an action helper
 * @member {String} key
 */

CreateActionTypeRequest.prototype['key'] = undefined;
/**
 * The description of the rule action helper for your reference
 * @member {String} description
 */

CreateActionTypeRequest.prototype['description'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

CreateActionTypeRequest.prototype['unitOfMeasureType'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

CreateActionTypeRequest.prototype['addConstraints'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateActionTypeRequestAllOf interface:

/**
 * The name of the Action Helper
 * @member {String} name
 */

_CreateActionTypeRequestAllOf["default"].prototype['name'] = undefined;
/**
 * A unique key that represents an action helper
 * @member {String} key
 */

_CreateActionTypeRequestAllOf["default"].prototype['key'] = undefined;
/**
 * The description of the rule action helper for your reference
 * @member {String} description
 */

_CreateActionTypeRequestAllOf["default"].prototype['description'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

_CreateActionTypeRequestAllOf["default"].prototype['unitOfMeasureType'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_CreateActionTypeRequestAllOf["default"].prototype['addConstraints'] = undefined;
var _default = CreateActionTypeRequest;
exports["default"] = _default;
