// @ziqni-tech/admin-api-client/dist/model/CreateAttachmentRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateAttachmentRequestAllOf = _interopRequireDefault(require("./CreateAttachmentRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateAttachmentRequest model module.
 * @module model/CreateAttachmentRequest
 * @version 0.0.5
 */
var CreateAttachmentRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAttachmentRequest</code>.
   * @alias module:model/CreateAttachmentRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateAttachmentRequestAllOf
   * @param name {String} Name of the original file uploaded
   * @param mimeType {String} Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
   * @param extension {String} Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
   */
  function CreateAttachmentRequest(name, mimeType, extension) {
    _classCallCheck(this, CreateAttachmentRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateAttachmentRequestAllOf["default"].initialize(this, name, mimeType, extension);

    CreateAttachmentRequest.initialize(this, name, mimeType, extension);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAttachmentRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, mimeType, extension) {
      obj['name'] = name;
      obj['mimeType'] = mimeType;
      obj['extension'] = extension;
    }
    /**
     * Constructs a <code>CreateAttachmentRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAttachmentRequest} obj Optional instance to populate.
     * @return {module:model/CreateAttachmentRequest} The populated <code>CreateAttachmentRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAttachmentRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateAttachmentRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('mimeType')) {
          obj['mimeType'] = _ApiClient["default"].convertToType(data['mimeType'], 'String');
        }

        if (data.hasOwnProperty('extension')) {
          obj['extension'] = _ApiClient["default"].convertToType(data['extension'], 'String');
        }

        if (data.hasOwnProperty('permission')) {
          obj['permission'] = _ApiClient["default"].convertToType(data['permission'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return CreateAttachmentRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateAttachmentRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateAttachmentRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateAttachmentRequest.prototype['metadata'] = undefined;
/**
 * Name of the original file uploaded
 * @member {String} name
 */

CreateAttachmentRequest.prototype['name'] = undefined;
/**
 * Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
 * @member {String} mimeType
 */

CreateAttachmentRequest.prototype['mimeType'] = undefined;
/**
 * Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
 * @member {String} extension
 */

CreateAttachmentRequest.prototype['extension'] = undefined;
/**
 * @member {Number} permission
 */

CreateAttachmentRequest.prototype['permission'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateAttachmentRequestAllOf interface:

/**
 * Name of the original file uploaded
 * @member {String} name
 */

_CreateAttachmentRequestAllOf["default"].prototype['name'] = undefined;
/**
 * Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
 * @member {String} mimeType
 */

_CreateAttachmentRequestAllOf["default"].prototype['mimeType'] = undefined;
/**
 * Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
 * @member {String} extension
 */

_CreateAttachmentRequestAllOf["default"].prototype['extension'] = undefined;
/**
 * @member {Number} permission
 */

_CreateAttachmentRequestAllOf["default"].prototype['permission'] = undefined;
var _default = CreateAttachmentRequest;
exports["default"] = _default;
