// @ziqni-tech/admin-api-client/dist/model/CreateEventRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateEventRequest model module.
 * @module model/CreateEventRequest
 * @version 0.0.26
 */
var CreateEventRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateEventRequest</code>.
   * @alias module:model/CreateEventRequest
   * @param memberRefId {String} The reference to this member in external system
   * @param action {String} The identifier that describes the meaning of this event
   * @param entityRefId {String} It is a reference to a game/product ID from external system
   * @param sourceValue {Number} The actual numerical value related to the event
   * @param transactionTimestamp {Date} ISO8601 timestamp for when this event happened. All records are stored in UTC time zone
   * @param eventRefId {String} 
   */
  function CreateEventRequest(memberRefId, action, entityRefId, sourceValue, transactionTimestamp, eventRefId) {
    _classCallCheck(this, CreateEventRequest);

    CreateEventRequest.initialize(this, memberRefId, action, entityRefId, sourceValue, transactionTimestamp, eventRefId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateEventRequest, null, [{
    key: "initialize",
    value: function initialize(obj, memberRefId, action, entityRefId, sourceValue, transactionTimestamp, eventRefId) {
      obj['memberRefId'] = memberRefId;
      obj['action'] = action;
      obj['entityRefId'] = entityRefId;
      obj['sourceValue'] = sourceValue;
      obj['transactionTimestamp'] = transactionTimestamp;
      obj['eventRefId'] = eventRefId;
    }
    /**
     * Constructs a <code>CreateEventRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateEventRequest} obj Optional instance to populate.
     * @return {module:model/CreateEventRequest} The populated <code>CreateEventRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateEventRequest();

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('action')) {
          obj['action'] = _ApiClient["default"].convertToType(data['action'], 'String');
        }

        if (data.hasOwnProperty('batchId')) {
          obj['batchId'] = _ApiClient["default"].convertToType(data['batchId'], 'String');
        }

        if (data.hasOwnProperty('entityRefId')) {
          obj['entityRefId'] = _ApiClient["default"].convertToType(data['entityRefId'], 'String');
        }

        if (data.hasOwnProperty('sourceValue')) {
          obj['sourceValue'] = _ApiClient["default"].convertToType(data['sourceValue'], 'Number');
        }

        if (data.hasOwnProperty('transactionTimestamp')) {
          obj['transactionTimestamp'] = _ApiClient["default"].convertToType(data['transactionTimestamp'], 'Date');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('eventRefId')) {
          obj['eventRefId'] = _ApiClient["default"].convertToType(data['eventRefId'], 'String');
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }
      }

      return obj;
    }
  }]);

  return CreateEventRequest;
}();
/**
 * The reference to this member in external system
 * @member {String} memberRefId
 */


CreateEventRequest.prototype['memberRefId'] = undefined;
/**
 * The identifier that describes the meaning of this event
 * @member {String} action
 */

CreateEventRequest.prototype['action'] = undefined;
/**
 * The batch identifier is used to group related event stream data together. This could be as simple as a UUID or your internal reference to that distinct player session event
 * @member {String} batchId
 */

CreateEventRequest.prototype['batchId'] = undefined;
/**
 * It is a reference to a game/product ID from external system
 * @member {String} entityRefId
 */

CreateEventRequest.prototype['entityRefId'] = undefined;
/**
 * The actual numerical value related to the event
 * @member {Number} sourceValue
 */

CreateEventRequest.prototype['sourceValue'] = undefined;
/**
 * ISO8601 timestamp for when this event happened. All records are stored in UTC time zone
 * @member {Date} transactionTimestamp
 */

CreateEventRequest.prototype['transactionTimestamp'] = undefined;
/**
 * A list of Strings of groups that the tag belongs to
 * @member {Array.<String>} tags
 */

CreateEventRequest.prototype['tags'] = undefined;
/**
 * @member {String} eventRefId
 */

CreateEventRequest.prototype['eventRefId'] = undefined;
/**
 * @member {String} memberId
 */

CreateEventRequest.prototype['memberId'] = undefined;
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

CreateEventRequest.prototype['customFields'] = undefined;
var _default = CreateEventRequest;
exports["default"] = _default;
