// @ziqni-tech/admin-api-client/dist/model/CreateAttachmentRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateAttachmentRequestAllOf model module.
 * @module model/CreateAttachmentRequestAllOf
 * @version 0.0.5
 */
var CreateAttachmentRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAttachmentRequestAllOf</code>.
   * @alias module:model/CreateAttachmentRequestAllOf
   * @param name {String} Name of the original file uploaded
   * @param mimeType {String} Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
   * @param extension {String} Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
   */
  function CreateAttachmentRequestAllOf(name, mimeType, extension) {
    _classCallCheck(this, CreateAttachmentRequestAllOf);

    CreateAttachmentRequestAllOf.initialize(this, name, mimeType, extension);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAttachmentRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, mimeType, extension) {
      obj['name'] = name;
      obj['mimeType'] = mimeType;
      obj['extension'] = extension;
    }
    /**
     * Constructs a <code>CreateAttachmentRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAttachmentRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateAttachmentRequestAllOf} The populated <code>CreateAttachmentRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAttachmentRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('mimeType')) {
          obj['mimeType'] = _ApiClient["default"].convertToType(data['mimeType'], 'String');
        }

        if (data.hasOwnProperty('extension')) {
          obj['extension'] = _ApiClient["default"].convertToType(data['extension'], 'String');
        }

        if (data.hasOwnProperty('permission')) {
          obj['permission'] = _ApiClient["default"].convertToType(data['permission'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return CreateAttachmentRequestAllOf;
}();
/**
 * Name of the original file uploaded
 * @member {String} name
 */


CreateAttachmentRequestAllOf.prototype['name'] = undefined;
/**
 * Mime type of the file. Valid mime types - text/csv or application/vmd.ms-excel
 * @member {String} mimeType
 */

CreateAttachmentRequestAllOf.prototype['mimeType'] = undefined;
/**
 * Extension of the file uploaded. Valid extensions supported - csv, xlsx, xls
 * @member {String} extension
 */

CreateAttachmentRequestAllOf.prototype['extension'] = undefined;
/**
 * @member {Number} permission
 */

CreateAttachmentRequestAllOf.prototype['permission'] = undefined;
var _default = CreateAttachmentRequestAllOf;
exports["default"] = _default;
