// @ziqni-tech/admin-api-client/dist/model/CreateActionTypeRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateActionTypeRequestAllOf model module.
 * @module model/CreateActionTypeRequestAllOf
 * @version 0.0.26
 */
var CreateActionTypeRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateActionTypeRequestAllOf</code>.
   * @alias module:model/CreateActionTypeRequestAllOf
   * @param name {String} The name of the Action Helper
   * @param key {String} A unique key that represents an action helper
   * @param unitOfMeasureType {module:model/UnitOfMeasureType} 
   */
  function CreateActionTypeRequestAllOf(name, key, unitOfMeasureType) {
    _classCallCheck(this, CreateActionTypeRequestAllOf);

    CreateActionTypeRequestAllOf.initialize(this, name, key, unitOfMeasureType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateActionTypeRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, key, unitOfMeasureType) {
      obj['name'] = name;
      obj['key'] = key;
      obj['unitOfMeasureType'] = unitOfMeasureType;
    }
    /**
     * Constructs a <code>CreateActionTypeRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateActionTypeRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateActionTypeRequestAllOf} The populated <code>CreateActionTypeRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateActionTypeRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateActionTypeRequestAllOf;
}();
/**
 * The name of the Action Helper
 * @member {String} name
 */


CreateActionTypeRequestAllOf.prototype['name'] = undefined;
/**
 * A unique key that represents an action helper
 * @member {String} key
 */

CreateActionTypeRequestAllOf.prototype['key'] = undefined;
/**
 * The description of the rule action helper for your reference
 * @member {String} description
 */

CreateActionTypeRequestAllOf.prototype['description'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

CreateActionTypeRequestAllOf.prototype['unitOfMeasureType'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

CreateActionTypeRequestAllOf.prototype['addConstraints'] = undefined;
var _default = CreateActionTypeRequestAllOf;
exports["default"] = _default;
