// @ziqni-tech/admin-api-client/dist/model/CreateMessageRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateMessageRequestAllOf = _interopRequireDefault(require("./CreateMessageRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Reward = _interopRequireDefault(require("./Reward"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateMessageRequest model module.
 * @module model/CreateMessageRequest
 * @version 0.0.5
 */
var CreateMessageRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateMessageRequest</code>.
   * @alias module:model/CreateMessageRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateMessageRequestAllOf
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param scheduling {module:model/Scheduling} 
   * @param constraints {Array.<String>} Additional constraints, if the value is present it means the
   */
  function CreateMessageRequest(messageType, subject, body, scheduling, constraints) {
    _classCallCheck(this, CreateMessageRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateMessageRequestAllOf["default"].initialize(this, messageType, subject, body, scheduling, constraints);

    CreateMessageRequest.initialize(this, messageType, subject, body, scheduling, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateMessageRequest, null, [{
    key: "initialize",
    value: function initialize(obj, messageType, subject, body, scheduling, constraints) {
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['scheduling'] = scheduling;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateMessageRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateMessageRequest} obj Optional instance to populate.
     * @return {module:model/CreateMessageRequest} The populated <code>CreateMessageRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateMessageRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateMessageRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _ApiClient["default"].convertToType(data['dependantOn'], [_Dependancy["default"]]);
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }

        if (data.hasOwnProperty('prize')) {
          obj['prize'] = _Reward["default"].constructFromObject(data['prize']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('eventReferenceId')) {
          obj['eventReferenceId'] = _ApiClient["default"].convertToType(data['eventReferenceId'], 'String');
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _ApiClient["default"].convertToType(data['eventRefType'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateMessageRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateMessageRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateMessageRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateMessageRequest.prototype['metadata'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

CreateMessageRequest.prototype['dependantOn'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

CreateMessageRequest.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

CreateMessageRequest.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

CreateMessageRequest.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

CreateMessageRequest.prototype['status'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

CreateMessageRequest.prototype['scheduling'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateMessageRequest.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

CreateMessageRequest.prototype['translatableFields'] = undefined;
/**
 * @member {module:model/Reward} prize
 */

CreateMessageRequest.prototype['prize'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

CreateMessageRequest.prototype['constraints'] = undefined;
/**
 * @member {String} eventReferenceId
 */

CreateMessageRequest.prototype['eventReferenceId'] = undefined;
/**
 * @member {String} eventRefType
 */

CreateMessageRequest.prototype['eventRefType'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateMessageRequestAllOf interface:

/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

_CreateMessageRequestAllOf["default"].prototype['dependantOn'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

_CreateMessageRequestAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_CreateMessageRequestAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_CreateMessageRequestAllOf["default"].prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_CreateMessageRequestAllOf["default"].prototype['status'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

_CreateMessageRequestAllOf["default"].prototype['scheduling'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_CreateMessageRequestAllOf["default"].prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

_CreateMessageRequestAllOf["default"].prototype['translatableFields'] = undefined;
/**
 * @member {module:model/Reward} prize
 */

_CreateMessageRequestAllOf["default"].prototype['prize'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

_CreateMessageRequestAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {String} eventReferenceId
 */

_CreateMessageRequestAllOf["default"].prototype['eventReferenceId'] = undefined;
/**
 * @member {String} eventRefType
 */

_CreateMessageRequestAllOf["default"].prototype['eventRefType'] = undefined;
var _default = CreateMessageRequest;
exports["default"] = _default;
