// @ziqni-tech/admin-api-client/dist/model/CreateCompetitionRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CompetitionType = _interopRequireDefault(require("./CompetitionType"));

var _CreateContestRequest = _interopRequireDefault(require("./CreateContestRequest"));

var _CreateRewardRequest = _interopRequireDefault(require("./CreateRewardRequest"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _Rule = _interopRequireDefault(require("./Rule"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateCompetitionRequestAllOf model module.
 * @module model/CreateCompetitionRequestAllOf
 * @version 0.0.26
 */
var CreateCompetitionRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateCompetitionRequestAllOf</code>.
   * @alias module:model/CreateCompetitionRequestAllOf
   * @param competitionType {module:model/CompetitionType} 
   * @param numberOfRounds {Number} Number of rounds to be played in a competition
   * @param name {String} A name or a name of a competition. Can be translated
   * @param minNumberOfEntrants {Number} Maximum number of partiipants allowed in a competition
   * @param scheduledStartDate {Date} ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
   * @param scheduledEndDate {Date} ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
   * @param constraints {Array.<String>} Additional constraints
   * @param productIds {Array.<String>} 
   */
  function CreateCompetitionRequestAllOf(competitionType, numberOfRounds, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, constraints, productIds) {
    _classCallCheck(this, CreateCompetitionRequestAllOf);

    CreateCompetitionRequestAllOf.initialize(this, competitionType, numberOfRounds, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, constraints, productIds);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateCompetitionRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, competitionType, numberOfRounds, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, constraints, productIds) {
      obj['competitionType'] = competitionType;
      obj['numberOfRounds'] = numberOfRounds;
      obj['name'] = name;
      obj['minNumberOfEntrants'] = minNumberOfEntrants;
      obj['scheduledStartDate'] = scheduledStartDate;
      obj['scheduledEndDate'] = scheduledEndDate;
      obj['constraints'] = constraints;
      obj['productIds'] = productIds;
    }
    /**
     * Constructs a <code>CreateCompetitionRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateCompetitionRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateCompetitionRequestAllOf} The populated <code>CreateCompetitionRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateCompetitionRequestAllOf();

        if (data.hasOwnProperty('competitionType')) {
          obj['competitionType'] = _CompetitionType["default"].constructFromObject(data['competitionType']);
        }

        if (data.hasOwnProperty('numberOfRounds')) {
          obj['numberOfRounds'] = _ApiClient["default"].convertToType(data['numberOfRounds'], 'Number');
        }

        if (data.hasOwnProperty('numberOfGroupStages')) {
          obj['numberOfGroupStages'] = _ApiClient["default"].convertToType(data['numberOfGroupStages'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('entrantMemberType')) {
          obj['entrantMemberType'] = _ApiClient["default"].convertToType(data['entrantMemberType'], 'String');
        }

        if (data.hasOwnProperty('maxNumberOfEntrants')) {
          obj['maxNumberOfEntrants'] = _ApiClient["default"].convertToType(data['maxNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('minNumberOfEntrants')) {
          obj['minNumberOfEntrants'] = _ApiClient["default"].convertToType(data['minNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _ApiClient["default"].convertToType(data['rules'], [_Rule["default"]]);
        }

        if (data.hasOwnProperty('scheduledStartDate')) {
          obj['scheduledStartDate'] = _ApiClient["default"].convertToType(data['scheduledStartDate'], 'Date');
        }

        if (data.hasOwnProperty('scheduledEndDate')) {
          obj['scheduledEndDate'] = _ApiClient["default"].convertToType(data['scheduledEndDate'], 'Date');
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('contests')) {
          obj['contests'] = _CreateContestRequest["default"].constructFromObject(data['contests']);
        }

        if (data.hasOwnProperty('rewards')) {
          obj['rewards'] = _ApiClient["default"].convertToType(data['rewards'], [_CreateRewardRequest["default"]]);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('productIds')) {
          obj['productIds'] = _ApiClient["default"].convertToType(data['productIds'], ['String']);
        }

        if (data.hasOwnProperty('entrantMemberTagsFilter')) {
          obj['entrantMemberTagsFilter'] = _DependantOn["default"].constructFromObject(data['entrantMemberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }
      }

      return obj;
    }
  }]);

  return CreateCompetitionRequestAllOf;
}();
/**
 * @member {module:model/CompetitionType} competitionType
 */


CreateCompetitionRequestAllOf.prototype['competitionType'] = undefined;
/**
 * Number of rounds to be played in a competition
 * @member {Number} numberOfRounds
 */

CreateCompetitionRequestAllOf.prototype['numberOfRounds'] = undefined;
/**
 * Number of group stages
 * @member {Number} numberOfGroupStages
 */

CreateCompetitionRequestAllOf.prototype['numberOfGroupStages'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

CreateCompetitionRequestAllOf.prototype['name'] = undefined;
/**
 * Description of the competition. Can be translated
 * @member {String} description
 */

CreateCompetitionRequestAllOf.prototype['description'] = undefined;
/**
 * Terms and conditions of a competition. Can be translated
 * @member {String} termsAndConditions
 */

CreateCompetitionRequestAllOf.prototype['termsAndConditions'] = undefined;
/**
 * @member {String} entrantMemberType
 */

CreateCompetitionRequestAllOf.prototype['entrantMemberType'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} maxNumberOfEntrants
 */

CreateCompetitionRequestAllOf.prototype['maxNumberOfEntrants'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} minNumberOfEntrants
 */

CreateCompetitionRequestAllOf.prototype['minNumberOfEntrants'] = undefined;
/**
 * @member {Array.<module:model/Rule>} rules
 */

CreateCompetitionRequestAllOf.prototype['rules'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

CreateCompetitionRequestAllOf.prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

CreateCompetitionRequestAllOf.prototype['scheduledEndDate'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateCompetitionRequestAllOf.prototype['translations'] = undefined;
/**
 * @member {module:model/CreateContestRequest} contests
 */

CreateCompetitionRequestAllOf.prototype['contests'] = undefined;
/**
 * @member {Array.<module:model/CreateRewardRequest>} rewards
 */

CreateCompetitionRequestAllOf.prototype['rewards'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

CreateCompetitionRequestAllOf.prototype['constraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

CreateCompetitionRequestAllOf.prototype['addConstraints'] = undefined;
/**
 * @member {Array.<String>} productIds
 */

CreateCompetitionRequestAllOf.prototype['productIds'] = undefined;
/**
 * @member {module:model/DependantOn} entrantMemberTagsFilter
 */

CreateCompetitionRequestAllOf.prototype['entrantMemberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

CreateCompetitionRequestAllOf.prototype['productTagsFilter'] = undefined;
var _default = CreateCompetitionRequestAllOf;
exports["default"] = _default;
