// @ziqni-tech/admin-api-client/dist/model/CreateUnitOfMeasureRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _CreateUnitOfMeasureRequestAllOf = _interopRequireDefault(require("./CreateUnitOfMeasureRequestAllOf"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateUnitOfMeasureRequest model module.
 * @module model/CreateUnitOfMeasureRequest
 * @version 0.0.26
 */
var CreateUnitOfMeasureRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateUnitOfMeasureRequest</code>.
   * @alias module:model/CreateUnitOfMeasureRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateUnitOfMeasureRequestAllOf
   * @param name {String} The name of a unit of measure
   * @param key {String} The reference to the unit of measure in your system
   * @param multiplier {Number} Is used to multiply the value from the standardised one that is being used
   * @param unitOfMeasureType {module:model/UnitOfMeasureType} 
   */
  function CreateUnitOfMeasureRequest(name, key, multiplier, unitOfMeasureType) {
    _classCallCheck(this, CreateUnitOfMeasureRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateUnitOfMeasureRequestAllOf["default"].initialize(this, name, key, multiplier, unitOfMeasureType);

    CreateUnitOfMeasureRequest.initialize(this, name, key, multiplier, unitOfMeasureType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateUnitOfMeasureRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, key, multiplier, unitOfMeasureType) {
      obj['name'] = name;
      obj['key'] = key;
      obj['multiplier'] = multiplier;
      obj['unitOfMeasureType'] = unitOfMeasureType;
    }
    /**
     * Constructs a <code>CreateUnitOfMeasureRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateUnitOfMeasureRequest} obj Optional instance to populate.
     * @return {module:model/CreateUnitOfMeasureRequest} The populated <code>CreateUnitOfMeasureRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateUnitOfMeasureRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateUnitOfMeasureRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('isoCode')) {
          obj['isoCode'] = _ApiClient["default"].convertToType(data['isoCode'], 'String');
        }

        if (data.hasOwnProperty('symbol')) {
          obj['symbol'] = _ApiClient["default"].convertToType(data['symbol'], 'String');
        }

        if (data.hasOwnProperty('multiplier')) {
          obj['multiplier'] = _ApiClient["default"].convertToType(data['multiplier'], 'Number');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }
      }

      return obj;
    }
  }]);

  return CreateUnitOfMeasureRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateUnitOfMeasureRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateUnitOfMeasureRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateUnitOfMeasureRequest.prototype['metadata'] = undefined;
/**
 * The name of a unit of measure
 * @member {String} name
 */

CreateUnitOfMeasureRequest.prototype['name'] = undefined;
/**
 * The reference to the unit of measure in your system
 * @member {String} key
 */

CreateUnitOfMeasureRequest.prototype['key'] = undefined;
/**
 * The description of a unit of measure
 * @member {String} description
 */

CreateUnitOfMeasureRequest.prototype['description'] = undefined;
/**
 * An alphabetical or numerical code to identify a unit of measure
 * @member {String} isoCode
 */

CreateUnitOfMeasureRequest.prototype['isoCode'] = undefined;
/**
 * The symbol of a unit of measure
 * @member {String} symbol
 */

CreateUnitOfMeasureRequest.prototype['symbol'] = undefined;
/**
 * Is used to multiply the value from the standardised one that is being used
 * @member {Number} multiplier
 */

CreateUnitOfMeasureRequest.prototype['multiplier'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

CreateUnitOfMeasureRequest.prototype['unitOfMeasureType'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateUnitOfMeasureRequestAllOf interface:

/**
 * The name of a unit of measure
 * @member {String} name
 */

_CreateUnitOfMeasureRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The reference to the unit of measure in your system
 * @member {String} key
 */

_CreateUnitOfMeasureRequestAllOf["default"].prototype['key'] = undefined;
/**
 * The description of a unit of measure
 * @member {String} description
 */

_CreateUnitOfMeasureRequestAllOf["default"].prototype['description'] = undefined;
/**
 * An alphabetical or numerical code to identify a unit of measure
 * @member {String} isoCode
 */

_CreateUnitOfMeasureRequestAllOf["default"].prototype['isoCode'] = undefined;
/**
 * The symbol of a unit of measure
 * @member {String} symbol
 */

_CreateUnitOfMeasureRequestAllOf["default"].prototype['symbol'] = undefined;
/**
 * Is used to multiply the value from the standardised one that is being used
 * @member {Number} multiplier
 */

_CreateUnitOfMeasureRequestAllOf["default"].prototype['multiplier'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

_CreateUnitOfMeasureRequestAllOf["default"].prototype['unitOfMeasureType'] = undefined;
var _default = CreateUnitOfMeasureRequest;
exports["default"] = _default;
