// @ziqni-tech/admin-api-client/dist/model/CreateComputeEngineRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateComputeEngineRequestAllOf = _interopRequireDefault(require("./CreateComputeEngineRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateComputeEngineRequest model module.
 * @module model/CreateComputeEngineRequest
 * @version 0.0.26
 */
var CreateComputeEngineRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateComputeEngineRequest</code>.
   * @alias module:model/CreateComputeEngineRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateComputeEngineRequestAllOf
   * @param constraints {Array.<String>} 
   */
  function CreateComputeEngineRequest(constraints) {
    _classCallCheck(this, CreateComputeEngineRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateComputeEngineRequestAllOf["default"].initialize(this, constraints);

    CreateComputeEngineRequest.initialize(this, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateComputeEngineRequest, null, [{
    key: "initialize",
    value: function initialize(obj, constraints) {
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateComputeEngineRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateComputeEngineRequest} obj Optional instance to populate.
     * @return {module:model/CreateComputeEngineRequest} The populated <code>CreateComputeEngineRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateComputeEngineRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateComputeEngineRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('uptime')) {
          obj['uptime'] = _ApiClient["default"].convertToType(data['uptime'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('statistics')) {
          obj['statistics'] = _ApiClient["default"].convertToType(data['statistics'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('lastKnownStatusCode')) {
          obj['lastKnownStatusCode'] = _ApiClient["default"].convertToType(data['lastKnownStatusCode'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return CreateComputeEngineRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateComputeEngineRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateComputeEngineRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateComputeEngineRequest.prototype['metadata'] = undefined;
/**
 * @member {String} description
 */

CreateComputeEngineRequest.prototype['description'] = undefined;
/**
 * @member {Number} version
 */

CreateComputeEngineRequest.prototype['version'] = undefined;
/**
 * @member {String} name
 */

CreateComputeEngineRequest.prototype['name'] = undefined;
/**
 * Long
 * @member {Number} uptime
 */

CreateComputeEngineRequest.prototype['uptime'] = undefined;
/**
 * @member {Array.<String>} constraints
 */

CreateComputeEngineRequest.prototype['constraints'] = undefined;
/**
 * @member {Object.<String, String>} statistics
 */

CreateComputeEngineRequest.prototype['statistics'] = undefined;
/**
 * @member {Number} lastKnownStatusCode
 */

CreateComputeEngineRequest.prototype['lastKnownStatusCode'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateComputeEngineRequestAllOf interface:

/**
 * @member {String} description
 */

_CreateComputeEngineRequestAllOf["default"].prototype['description'] = undefined;
/**
 * @member {Number} version
 */

_CreateComputeEngineRequestAllOf["default"].prototype['version'] = undefined;
/**
 * @member {String} name
 */

_CreateComputeEngineRequestAllOf["default"].prototype['name'] = undefined;
/**
 * Long
 * @member {Number} uptime
 */

_CreateComputeEngineRequestAllOf["default"].prototype['uptime'] = undefined;
/**
 * @member {Array.<String>} constraints
 */

_CreateComputeEngineRequestAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {Object.<String, String>} statistics
 */

_CreateComputeEngineRequestAllOf["default"].prototype['statistics'] = undefined;
/**
 * @member {Number} lastKnownStatusCode
 */

_CreateComputeEngineRequestAllOf["default"].prototype['lastKnownStatusCode'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateComputeEngineRequestAllOf["default"].prototype['metadata'] = undefined;
var _default = CreateComputeEngineRequest;
exports["default"] = _default;
