// @ziqni-tech/admin-api-client/dist/model/CreateFileObjectRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateFileObjectRequestAllOf = _interopRequireDefault(require("./CreateFileObjectRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateFileObjectRequest model module.
 * @module model/CreateFileObjectRequest
 * @version 0.0.26
 */
var CreateFileObjectRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateFileObjectRequest</code>.
   * @alias module:model/CreateFileObjectRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateFileObjectRequestAllOf
   * @param repositoryId {String} The repository identifier this file belongs too
   * @param name {String} Name of the original file uploaded
   * @param mimeType {String} Mime type of the file. Example mime types - text/csv or application/vmd.ms-excel or application/x-directory for directories
   * @param parentFolderPath {String} The folder containing the file within the repository
   */
  function CreateFileObjectRequest(repositoryId, name, mimeType, parentFolderPath) {
    _classCallCheck(this, CreateFileObjectRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateFileObjectRequestAllOf["default"].initialize(this, repositoryId, name, mimeType, parentFolderPath);

    CreateFileObjectRequest.initialize(this, repositoryId, name, mimeType, parentFolderPath);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateFileObjectRequest, null, [{
    key: "initialize",
    value: function initialize(obj, repositoryId, name, mimeType, parentFolderPath) {
      obj['repositoryId'] = repositoryId;
      obj['name'] = name;
      obj['mimeType'] = mimeType;
      obj['parentFolderPath'] = parentFolderPath;
    }
    /**
     * Constructs a <code>CreateFileObjectRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateFileObjectRequest} obj Optional instance to populate.
     * @return {module:model/CreateFileObjectRequest} The populated <code>CreateFileObjectRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateFileObjectRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateFileObjectRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('repositoryId')) {
          obj['repositoryId'] = _ApiClient["default"].convertToType(data['repositoryId'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('mimeType')) {
          obj['mimeType'] = _ApiClient["default"].convertToType(data['mimeType'], 'String');
        }

        if (data.hasOwnProperty('parentFolderPath')) {
          obj['parentFolderPath'] = _ApiClient["default"].convertToType(data['parentFolderPath'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateFileObjectRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateFileObjectRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateFileObjectRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateFileObjectRequest.prototype['metadata'] = undefined;
/**
 * The repository identifier this file belongs too
 * @member {String} repositoryId
 */

CreateFileObjectRequest.prototype['repositoryId'] = undefined;
/**
 * Name of the original file uploaded
 * @member {String} name
 */

CreateFileObjectRequest.prototype['name'] = undefined;
/**
 * Mime type of the file. Example mime types - text/csv or application/vmd.ms-excel or application/x-directory for directories
 * @member {String} mimeType
 */

CreateFileObjectRequest.prototype['mimeType'] = undefined;
/**
 * The folder containing the file within the repository
 * @member {String} parentFolderPath
 */

CreateFileObjectRequest.prototype['parentFolderPath'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

CreateFileObjectRequest.prototype['constraints'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateFileObjectRequestAllOf interface:

/**
 * The repository identifier this file belongs too
 * @member {String} repositoryId
 */

_CreateFileObjectRequestAllOf["default"].prototype['repositoryId'] = undefined;
/**
 * Name of the original file uploaded
 * @member {String} name
 */

_CreateFileObjectRequestAllOf["default"].prototype['name'] = undefined;
/**
 * Mime type of the file. Example mime types - text/csv or application/vmd.ms-excel or application/x-directory for directories
 * @member {String} mimeType
 */

_CreateFileObjectRequestAllOf["default"].prototype['mimeType'] = undefined;
/**
 * The folder containing the file within the repository
 * @member {String} parentFolderPath
 */

_CreateFileObjectRequestAllOf["default"].prototype['parentFolderPath'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

_CreateFileObjectRequestAllOf["default"].prototype['constraints'] = undefined;
var _default = CreateFileObjectRequest;
exports["default"] = _default;
