// @ziqni-tech/admin-api-client/dist/model/CreateWebhookRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _CreateWebhookRequestAllOf = _interopRequireDefault(require("./CreateWebhookRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateWebhookRequest model module.
 * @module model/CreateWebhookRequest
 * @version 0.0.26
 */
var CreateWebhookRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateWebhookRequest</code>.
   * @alias module:model/CreateWebhookRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateWebhookRequestAllOf
   * @param postToUrl {String} A URL to post the webhook to
   * @param triggers {Array.<String>} A list of event triggers
   * @param name {String} The name of a Webhook
   */
  function CreateWebhookRequest(postToUrl, triggers, name) {
    _classCallCheck(this, CreateWebhookRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateWebhookRequestAllOf["default"].initialize(this, postToUrl, triggers, name);

    CreateWebhookRequest.initialize(this, postToUrl, triggers, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateWebhookRequest, null, [{
    key: "initialize",
    value: function initialize(obj, postToUrl, triggers, name) {
      obj['postToUrl'] = postToUrl;
      obj['triggers'] = triggers;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateWebhookRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateWebhookRequest} obj Optional instance to populate.
     * @return {module:model/CreateWebhookRequest} The populated <code>CreateWebhookRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateWebhookRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateWebhookRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('postToUrl')) {
          obj['postToUrl'] = _ApiClient["default"].convertToType(data['postToUrl'], 'String');
        }

        if (data.hasOwnProperty('triggers')) {
          obj['triggers'] = _ApiClient["default"].convertToType(data['triggers'], ['String']);
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('headers')) {
          obj['headers'] = _ApiClient["default"].convertToType(data['headers'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return CreateWebhookRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateWebhookRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateWebhookRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateWebhookRequest.prototype['metadata'] = undefined;
/**
 * A URL to post the webhook to
 * @member {String} postToUrl
 */

CreateWebhookRequest.prototype['postToUrl'] = undefined;
/**
 * A list of event triggers
 * @member {Array.<String>} triggers
 */

CreateWebhookRequest.prototype['triggers'] = undefined;
/**
 * The description of a Webhook
 * @member {String} description
 */

CreateWebhookRequest.prototype['description'] = undefined;
/**
 * The name of a Webhook
 * @member {String} name
 */

CreateWebhookRequest.prototype['name'] = undefined;
/**
 * @member {Object.<String, String>} headers
 */

CreateWebhookRequest.prototype['headers'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateWebhookRequestAllOf interface:

/**
 * A URL to post the webhook to
 * @member {String} postToUrl
 */

_CreateWebhookRequestAllOf["default"].prototype['postToUrl'] = undefined;
/**
 * A list of event triggers
 * @member {Array.<String>} triggers
 */

_CreateWebhookRequestAllOf["default"].prototype['triggers'] = undefined;
/**
 * The description of a Webhook
 * @member {String} description
 */

_CreateWebhookRequestAllOf["default"].prototype['description'] = undefined;
/**
 * The name of a Webhook
 * @member {String} name
 */

_CreateWebhookRequestAllOf["default"].prototype['name'] = undefined;
/**
 * @member {Object.<String, String>} headers
 */

_CreateWebhookRequestAllOf["default"].prototype['headers'] = undefined;
var _default = CreateWebhookRequest;
exports["default"] = _default;
