// @ziqni-tech/admin-api-client/dist/model/CreateMessageRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Reward = _interopRequireDefault(require("./Reward"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateMessageRequestAllOf model module.
 * @module model/CreateMessageRequestAllOf
 * @version 0.0.5
 */
var CreateMessageRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateMessageRequestAllOf</code>.
   * @alias module:model/CreateMessageRequestAllOf
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param scheduling {module:model/Scheduling} 
   * @param constraints {Array.<String>} Additional constraints, if the value is present it means the
   */
  function CreateMessageRequestAllOf(messageType, subject, body, scheduling, constraints) {
    _classCallCheck(this, CreateMessageRequestAllOf);

    CreateMessageRequestAllOf.initialize(this, messageType, subject, body, scheduling, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateMessageRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, messageType, subject, body, scheduling, constraints) {
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['scheduling'] = scheduling;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateMessageRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateMessageRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateMessageRequestAllOf} The populated <code>CreateMessageRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateMessageRequestAllOf();

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _ApiClient["default"].convertToType(data['dependantOn'], [_Dependancy["default"]]);
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }

        if (data.hasOwnProperty('prize')) {
          obj['prize'] = _Reward["default"].constructFromObject(data['prize']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('eventReferenceId')) {
          obj['eventReferenceId'] = _ApiClient["default"].convertToType(data['eventReferenceId'], 'String');
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _ApiClient["default"].convertToType(data['eventRefType'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateMessageRequestAllOf;
}();
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */


CreateMessageRequestAllOf.prototype['dependantOn'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

CreateMessageRequestAllOf.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

CreateMessageRequestAllOf.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

CreateMessageRequestAllOf.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

CreateMessageRequestAllOf.prototype['status'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

CreateMessageRequestAllOf.prototype['scheduling'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateMessageRequestAllOf.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

CreateMessageRequestAllOf.prototype['translatableFields'] = undefined;
/**
 * @member {module:model/Reward} prize
 */

CreateMessageRequestAllOf.prototype['prize'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

CreateMessageRequestAllOf.prototype['constraints'] = undefined;
/**
 * @member {String} eventReferenceId
 */

CreateMessageRequestAllOf.prototype['eventReferenceId'] = undefined;
/**
 * @member {String} eventRefType
 */

CreateMessageRequestAllOf.prototype['eventRefType'] = undefined;
var _default = CreateMessageRequestAllOf;
exports["default"] = _default;
