// @ziqni-tech/admin-api-client/dist/model/CreateComputeEngineRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateComputeEngineRequestAllOf model module.
 * @module model/CreateComputeEngineRequestAllOf
 * @version 0.0.26
 */
var CreateComputeEngineRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateComputeEngineRequestAllOf</code>.
   * @alias module:model/CreateComputeEngineRequestAllOf
   * @param constraints {Array.<String>} 
   */
  function CreateComputeEngineRequestAllOf(constraints) {
    _classCallCheck(this, CreateComputeEngineRequestAllOf);

    CreateComputeEngineRequestAllOf.initialize(this, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateComputeEngineRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, constraints) {
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateComputeEngineRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateComputeEngineRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateComputeEngineRequestAllOf} The populated <code>CreateComputeEngineRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateComputeEngineRequestAllOf();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('uptime')) {
          obj['uptime'] = _ApiClient["default"].convertToType(data['uptime'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('statistics')) {
          obj['statistics'] = _ApiClient["default"].convertToType(data['statistics'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('lastKnownStatusCode')) {
          obj['lastKnownStatusCode'] = _ApiClient["default"].convertToType(data['lastKnownStatusCode'], 'Number');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return CreateComputeEngineRequestAllOf;
}();
/**
 * @member {String} description
 */


CreateComputeEngineRequestAllOf.prototype['description'] = undefined;
/**
 * @member {Number} version
 */

CreateComputeEngineRequestAllOf.prototype['version'] = undefined;
/**
 * @member {String} name
 */

CreateComputeEngineRequestAllOf.prototype['name'] = undefined;
/**
 * Long
 * @member {Number} uptime
 */

CreateComputeEngineRequestAllOf.prototype['uptime'] = undefined;
/**
 * @member {Array.<String>} constraints
 */

CreateComputeEngineRequestAllOf.prototype['constraints'] = undefined;
/**
 * @member {Object.<String, String>} statistics
 */

CreateComputeEngineRequestAllOf.prototype['statistics'] = undefined;
/**
 * @member {Number} lastKnownStatusCode
 */

CreateComputeEngineRequestAllOf.prototype['lastKnownStatusCode'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateComputeEngineRequestAllOf.prototype['metadata'] = undefined;
var _default = CreateComputeEngineRequestAllOf;
exports["default"] = _default;
