// @ziqni-tech/admin-api-client/dist/model/CreateTranslationRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _TranslationEntry = _interopRequireDefault(require("./TranslationEntry"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateTranslationRequest model module.
 * @module model/CreateTranslationRequest
 * @version 0.0.26
 */
var CreateTranslationRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateTranslationRequest</code>.
   * Translations to attach to the models.
   * @alias module:model/CreateTranslationRequest
   * @param entityId {String} 
   * @param entityType {String} 
   * @param translations {Array.<module:model/TranslationEntry>} 
   * @param languageId {String} 
   * @param languageKey {String} 
   */
  function CreateTranslationRequest(entityId, entityType, translations, languageId, languageKey) {
    _classCallCheck(this, CreateTranslationRequest);

    CreateTranslationRequest.initialize(this, entityId, entityType, translations, languageId, languageKey);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateTranslationRequest, null, [{
    key: "initialize",
    value: function initialize(obj, entityId, entityType, translations, languageId, languageKey) {
      obj['entityId'] = entityId;
      obj['entityType'] = entityType;
      obj['translations'] = translations;
      obj['languageId'] = languageId;
      obj['languageKey'] = languageKey;
    }
    /**
     * Constructs a <code>CreateTranslationRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateTranslationRequest} obj Optional instance to populate.
     * @return {module:model/CreateTranslationRequest} The populated <code>CreateTranslationRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateTranslationRequest();

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('entityType')) {
          obj['entityType'] = _ApiClient["default"].convertToType(data['entityType'], 'String');
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_TranslationEntry["default"]]);
        }

        if (data.hasOwnProperty('languageId')) {
          obj['languageId'] = _ApiClient["default"].convertToType(data['languageId'], 'String');
        }

        if (data.hasOwnProperty('languageKey')) {
          obj['languageKey'] = _ApiClient["default"].convertToType(data['languageKey'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateTranslationRequest;
}();
/**
 * @member {String} entityId
 */


CreateTranslationRequest.prototype['entityId'] = undefined;
/**
 * @member {String} entityType
 */

CreateTranslationRequest.prototype['entityType'] = undefined;
/**
 * @member {Array.<module:model/TranslationEntry>} translations
 */

CreateTranslationRequest.prototype['translations'] = undefined;
/**
 * @member {String} languageId
 */

CreateTranslationRequest.prototype['languageId'] = undefined;
/**
 * @member {String} languageKey
 */

CreateTranslationRequest.prototype['languageKey'] = undefined;
var _default = CreateTranslationRequest;
exports["default"] = _default;
