// @ziqni-tech/admin-api-client/dist/model/CreateWebhookRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateWebhookRequestAllOf model module.
 * @module model/CreateWebhookRequestAllOf
 * @version 0.0.26
 */
var CreateWebhookRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateWebhookRequestAllOf</code>.
   * @alias module:model/CreateWebhookRequestAllOf
   * @param postToUrl {String} A URL to post the webhook to
   * @param triggers {Array.<String>} A list of event triggers
   * @param name {String} The name of a Webhook
   */
  function CreateWebhookRequestAllOf(postToUrl, triggers, name) {
    _classCallCheck(this, CreateWebhookRequestAllOf);

    CreateWebhookRequestAllOf.initialize(this, postToUrl, triggers, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateWebhookRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, postToUrl, triggers, name) {
      obj['postToUrl'] = postToUrl;
      obj['triggers'] = triggers;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateWebhookRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateWebhookRequestAllOf} obj Optional instance to populate.
     * @return {module:model/CreateWebhookRequestAllOf} The populated <code>CreateWebhookRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateWebhookRequestAllOf();

        if (data.hasOwnProperty('postToUrl')) {
          obj['postToUrl'] = _ApiClient["default"].convertToType(data['postToUrl'], 'String');
        }

        if (data.hasOwnProperty('triggers')) {
          obj['triggers'] = _ApiClient["default"].convertToType(data['triggers'], ['String']);
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('headers')) {
          obj['headers'] = _ApiClient["default"].convertToType(data['headers'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return CreateWebhookRequestAllOf;
}();
/**
 * A URL to post the webhook to
 * @member {String} postToUrl
 */


CreateWebhookRequestAllOf.prototype['postToUrl'] = undefined;
/**
 * A list of event triggers
 * @member {Array.<String>} triggers
 */

CreateWebhookRequestAllOf.prototype['triggers'] = undefined;
/**
 * The description of a Webhook
 * @member {String} description
 */

CreateWebhookRequestAllOf.prototype['description'] = undefined;
/**
 * The name of a Webhook
 * @member {String} name
 */

CreateWebhookRequestAllOf.prototype['name'] = undefined;
/**
 * @member {Object.<String, String>} headers
 */

CreateWebhookRequestAllOf.prototype['headers'] = undefined;
var _default = CreateWebhookRequestAllOf;
exports["default"] = _default;
