// @ziqni-tech/admin-api-client/dist/model/CreateNotificationRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateNotificationRequestAllOf = _interopRequireDefault(require("./CreateNotificationRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateNotificationRequest model module.
 * @module model/CreateNotificationRequest
 * @version 0.0.26
 */
var CreateNotificationRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateNotificationRequest</code>.
   * @alias module:model/CreateNotificationRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateNotificationRequestAllOf
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param status {module:model/MessageStatus} 
   * @param memberId {Array.<String>} The reference ID of the event object
   * @param scheduling {module:model/Scheduling} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function CreateNotificationRequest(messageType, subject, body, status, memberId, scheduling, constraints) {
    _classCallCheck(this, CreateNotificationRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateNotificationRequestAllOf["default"].initialize(this, messageType, subject, body, status, memberId, scheduling, constraints);

    CreateNotificationRequest.initialize(this, messageType, subject, body, status, memberId, scheduling, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateNotificationRequest, null, [{
    key: "initialize",
    value: function initialize(obj, messageType, subject, body, status, memberId, scheduling, constraints) {
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['status'] = status;
      obj['memberId'] = memberId;
      obj['scheduling'] = scheduling;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>CreateNotificationRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateNotificationRequest} obj Optional instance to populate.
     * @return {module:model/CreateNotificationRequest} The populated <code>CreateNotificationRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateNotificationRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateNotificationRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('memberGroup')) {
          obj['memberGroup'] = _ApiClient["default"].convertToType(data['memberGroup'], ['String']);
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], ['String']);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateNotificationRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateNotificationRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateNotificationRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateNotificationRequest.prototype['metadata'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

CreateNotificationRequest.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

CreateNotificationRequest.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

CreateNotificationRequest.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

CreateNotificationRequest.prototype['status'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateNotificationRequest.prototype['translations'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberGroup
 */

CreateNotificationRequest.prototype['memberGroup'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

CreateNotificationRequest.prototype['memberId'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

CreateNotificationRequest.prototype['scheduling'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

CreateNotificationRequest.prototype['constraints'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateNotificationRequestAllOf interface:

/**
 * @member {module:model/MessageType} messageType
 */

_CreateNotificationRequestAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_CreateNotificationRequestAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_CreateNotificationRequestAllOf["default"].prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_CreateNotificationRequestAllOf["default"].prototype['status'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_CreateNotificationRequestAllOf["default"].prototype['translations'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberGroup
 */

_CreateNotificationRequestAllOf["default"].prototype['memberGroup'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

_CreateNotificationRequestAllOf["default"].prototype['memberId'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

_CreateNotificationRequestAllOf["default"].prototype['scheduling'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_CreateNotificationRequestAllOf["default"].prototype['constraints'] = undefined;
var _default = CreateNotificationRequest;
exports["default"] = _default;
