// @ziqni-tech/admin-api-client/dist/model/CreateApiKeyRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateApiKeyRequestAllOf = _interopRequireDefault(require("./CreateApiKeyRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _Role = _interopRequireDefault(require("./Role"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateApiKeyRequest model module.
 * @module model/CreateApiKeyRequest
 * @version 0.0.26
 */
var CreateApiKeyRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateApiKeyRequest</code>.
   * @alias module:model/CreateApiKeyRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateApiKeyRequestAllOf
   * @param role {module:model/Role} 
   */
  function CreateApiKeyRequest(role) {
    _classCallCheck(this, CreateApiKeyRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateApiKeyRequestAllOf["default"].initialize(this, role);

    CreateApiKeyRequest.initialize(this, role);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateApiKeyRequest, null, [{
    key: "initialize",
    value: function initialize(obj, role) {
      obj['role'] = role;
    }
    /**
     * Constructs a <code>CreateApiKeyRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateApiKeyRequest} obj Optional instance to populate.
     * @return {module:model/CreateApiKeyRequest} The populated <code>CreateApiKeyRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateApiKeyRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateApiKeyRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('whiteListIPs')) {
          obj['whiteListIPs'] = _ApiClient["default"].convertToType(data['whiteListIPs'], ['String']);
        }

        if (data.hasOwnProperty('role')) {
          obj['role'] = _Role["default"].constructFromObject(data['role']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateApiKeyRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateApiKeyRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateApiKeyRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateApiKeyRequest.prototype['metadata'] = undefined;
/**
 * The description of an Api key
 * @member {String} description
 */

CreateApiKeyRequest.prototype['description'] = undefined;
/**
 * the IP's that need to be whitelisted with the Api key
 * @member {Array.<String>} whiteListIPs
 */

CreateApiKeyRequest.prototype['whiteListIPs'] = undefined;
/**
 * @member {module:model/Role} role
 */

CreateApiKeyRequest.prototype['role'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

CreateApiKeyRequest.prototype['addConstraints'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateApiKeyRequestAllOf interface:

/**
 * The description of an Api key
 * @member {String} description
 */

_CreateApiKeyRequestAllOf["default"].prototype['description'] = undefined;
/**
 * the IP's that need to be whitelisted with the Api key
 * @member {Array.<String>} whiteListIPs
 */

_CreateApiKeyRequestAllOf["default"].prototype['whiteListIPs'] = undefined;
/**
 * @member {module:model/Role} role
 */

_CreateApiKeyRequestAllOf["default"].prototype['role'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_CreateApiKeyRequestAllOf["default"].prototype['addConstraints'] = undefined;
var _default = CreateApiKeyRequest;
exports["default"] = _default;
