// @ziqni-tech/admin-api-client/dist/model/CreateProductRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActionTypeAdjustmentFactor = _interopRequireDefault(require("./ActionTypeAdjustmentFactor"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _CreateProductRequestAllOf = _interopRequireDefault(require("./CreateProductRequestAllOf"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateProductRequest model module.
 * @module model/CreateProductRequest
 * @version 0.0.26
 */
var CreateProductRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateProductRequest</code>.
   * This is a request for Product creation
   * @alias module:model/CreateProductRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateProductRequestAllOf
   * @param name {String} The name of the product
   * @param adjustmentFactor {Number} The multiplier to apply to source values received for this product events
   * @param productRefId {String} The reference to this product in your system. The reference identifier can not be changed after the product has been created
   */
  function CreateProductRequest(name, adjustmentFactor, productRefId) {
    _classCallCheck(this, CreateProductRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateProductRequestAllOf["default"].initialize(this, name, adjustmentFactor, productRefId);

    CreateProductRequest.initialize(this, name, adjustmentFactor, productRefId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateProductRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, adjustmentFactor, productRefId) {
      obj['name'] = name;
      obj['adjustmentFactor'] = adjustmentFactor;
      obj['productRefId'] = productRefId;
    }
    /**
     * Constructs a <code>CreateProductRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateProductRequest} obj Optional instance to populate.
     * @return {module:model/CreateProductRequest} The populated <code>CreateProductRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateProductRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateProductRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('adjustmentFactor')) {
          obj['adjustmentFactor'] = _ApiClient["default"].convertToType(data['adjustmentFactor'], 'Number');
        }

        if (data.hasOwnProperty('productRefId')) {
          obj['productRefId'] = _ApiClient["default"].convertToType(data['productRefId'], 'String');
        }

        if (data.hasOwnProperty('actionTypeAdjustmentFactors')) {
          obj['actionTypeAdjustmentFactors'] = _ApiClient["default"].convertToType(data['actionTypeAdjustmentFactors'], [_ActionTypeAdjustmentFactor["default"]]);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }
      }

      return obj;
    }
  }]);

  return CreateProductRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateProductRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateProductRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateProductRequest.prototype['metadata'] = undefined;
/**
 * The name of the product
 * @member {String} name
 */

CreateProductRequest.prototype['name'] = undefined;
/**
 * The description of the product for your reference
 * @member {String} description
 */

CreateProductRequest.prototype['description'] = undefined;
/**
 * The multiplier to apply to source values received for this product events
 * @member {Number} adjustmentFactor
 */

CreateProductRequest.prototype['adjustmentFactor'] = undefined;
/**
 * The reference to this product in your system. The reference identifier can not be changed after the product has been created
 * @member {String} productRefId
 */

CreateProductRequest.prototype['productRefId'] = undefined;
/**
 * @member {Array.<module:model/ActionTypeAdjustmentFactor>} actionTypeAdjustmentFactors
 */

CreateProductRequest.prototype['actionTypeAdjustmentFactors'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateProductRequest.prototype['translations'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateProductRequestAllOf interface:

/**
 * The name of the product
 * @member {String} name
 */

_CreateProductRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the product for your reference
 * @member {String} description
 */

_CreateProductRequestAllOf["default"].prototype['description'] = undefined;
/**
 * The multiplier to apply to source values received for this product events
 * @member {Number} adjustmentFactor
 */

_CreateProductRequestAllOf["default"].prototype['adjustmentFactor'] = undefined;
/**
 * The reference to this product in your system. The reference identifier can not be changed after the product has been created
 * @member {String} productRefId
 */

_CreateProductRequestAllOf["default"].prototype['productRefId'] = undefined;
/**
 * @member {Array.<module:model/ActionTypeAdjustmentFactor>} actionTypeAdjustmentFactors
 */

_CreateProductRequestAllOf["default"].prototype['actionTypeAdjustmentFactors'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_CreateProductRequestAllOf["default"].prototype['translations'] = undefined;
var _default = CreateProductRequest;
exports["default"] = _default;
