// @ziqni-tech/admin-api-client/dist/model/CreateSimpleCompetitionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CompetitionType = _interopRequireDefault(require("./CompetitionType"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _CreateRewardRequest = _interopRequireDefault(require("./CreateRewardRequest"));

var _CreateSimpleCompetitionRequestAllOf = _interopRequireDefault(require("./CreateSimpleCompetitionRequestAllOf"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _Rule = _interopRequireDefault(require("./Rule"));

var _Strategy = _interopRequireDefault(require("./Strategy"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateSimpleCompetitionRequest model module.
 * @module model/CreateSimpleCompetitionRequest
 * @version 0.0.26
 */
var CreateSimpleCompetitionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateSimpleCompetitionRequest</code>.
   * @alias module:model/CreateSimpleCompetitionRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateSimpleCompetitionRequestAllOf
   * @param competitionType {module:model/CompetitionType} 
   * @param name {String} A name or a name of a competition. Can be translated
   * @param minNumberOfEntrants {Number} Maximum number of partiipants allowed in a competition
   * @param scheduledStartDate {Date} ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
   * @param scheduledEndDate {Date} ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
   * @param productIds {Array.<String>} 
   */
  function CreateSimpleCompetitionRequest(competitionType, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, productIds) {
    _classCallCheck(this, CreateSimpleCompetitionRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateSimpleCompetitionRequestAllOf["default"].initialize(this, competitionType, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, productIds);

    CreateSimpleCompetitionRequest.initialize(this, competitionType, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, productIds);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateSimpleCompetitionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, competitionType, name, minNumberOfEntrants, scheduledStartDate, scheduledEndDate, productIds) {
      obj['competitionType'] = competitionType;
      obj['name'] = name;
      obj['minNumberOfEntrants'] = minNumberOfEntrants;
      obj['scheduledStartDate'] = scheduledStartDate;
      obj['scheduledEndDate'] = scheduledEndDate;
      obj['productIds'] = productIds;
    }
    /**
     * Constructs a <code>CreateSimpleCompetitionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateSimpleCompetitionRequest} obj Optional instance to populate.
     * @return {module:model/CreateSimpleCompetitionRequest} The populated <code>CreateSimpleCompetitionRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateSimpleCompetitionRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateSimpleCompetitionRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('competitionType')) {
          obj['competitionType'] = _CompetitionType["default"].constructFromObject(data['competitionType']);
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('entrantMemberType')) {
          obj['entrantMemberType'] = _ApiClient["default"].convertToType(data['entrantMemberType'], 'String');
        }

        if (data.hasOwnProperty('maxNumberOfEntrants')) {
          obj['maxNumberOfEntrants'] = _ApiClient["default"].convertToType(data['maxNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('minNumberOfEntrants')) {
          obj['minNumberOfEntrants'] = _ApiClient["default"].convertToType(data['minNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _ApiClient["default"].convertToType(data['rules'], [_Rule["default"]]);
        }

        if (data.hasOwnProperty('scheduledStartDate')) {
          obj['scheduledStartDate'] = _ApiClient["default"].convertToType(data['scheduledStartDate'], 'Date');
        }

        if (data.hasOwnProperty('scheduledEndDate')) {
          obj['scheduledEndDate'] = _ApiClient["default"].convertToType(data['scheduledEndDate'], 'Date');
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('rewards')) {
          obj['rewards'] = _ApiClient["default"].convertToType(data['rewards'], [_CreateRewardRequest["default"]]);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('productIds')) {
          obj['productIds'] = _ApiClient["default"].convertToType(data['productIds'], ['String']);
        }

        if (data.hasOwnProperty('entrantMemberTagsFilter')) {
          obj['entrantMemberTagsFilter'] = _DependantOn["default"].constructFromObject(data['entrantMemberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }

        if (data.hasOwnProperty('strategies')) {
          obj['strategies'] = _Strategy["default"].constructFromObject(data['strategies']);
        }
      }

      return obj;
    }
  }]);

  return CreateSimpleCompetitionRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateSimpleCompetitionRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateSimpleCompetitionRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateSimpleCompetitionRequest.prototype['metadata'] = undefined;
/**
 * @member {module:model/CompetitionType} competitionType
 */

CreateSimpleCompetitionRequest.prototype['competitionType'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

CreateSimpleCompetitionRequest.prototype['name'] = undefined;
/**
 * Description of the competition. Can be translated
 * @member {String} description
 */

CreateSimpleCompetitionRequest.prototype['description'] = undefined;
/**
 * Terms and conditions of a competition. Can be translated
 * @member {String} termsAndConditions
 */

CreateSimpleCompetitionRequest.prototype['termsAndConditions'] = undefined;
/**
 * @member {String} entrantMemberType
 */

CreateSimpleCompetitionRequest.prototype['entrantMemberType'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} maxNumberOfEntrants
 */

CreateSimpleCompetitionRequest.prototype['maxNumberOfEntrants'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} minNumberOfEntrants
 */

CreateSimpleCompetitionRequest.prototype['minNumberOfEntrants'] = undefined;
/**
 * @member {Array.<module:model/Rule>} rules
 */

CreateSimpleCompetitionRequest.prototype['rules'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

CreateSimpleCompetitionRequest.prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

CreateSimpleCompetitionRequest.prototype['scheduledEndDate'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateSimpleCompetitionRequest.prototype['translations'] = undefined;
/**
 * @member {Array.<module:model/CreateRewardRequest>} rewards
 */

CreateSimpleCompetitionRequest.prototype['rewards'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

CreateSimpleCompetitionRequest.prototype['addConstraints'] = undefined;
/**
 * @member {Array.<String>} productIds
 */

CreateSimpleCompetitionRequest.prototype['productIds'] = undefined;
/**
 * @member {module:model/DependantOn} entrantMemberTagsFilter
 */

CreateSimpleCompetitionRequest.prototype['entrantMemberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

CreateSimpleCompetitionRequest.prototype['productTagsFilter'] = undefined;
/**
 * @member {module:model/Strategy} strategies
 */

CreateSimpleCompetitionRequest.prototype['strategies'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateSimpleCompetitionRequestAllOf interface:

/**
 * @member {module:model/CompetitionType} competitionType
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['competitionType'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['name'] = undefined;
/**
 * Description of the competition. Can be translated
 * @member {String} description
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['description'] = undefined;
/**
 * Terms and conditions of a competition. Can be translated
 * @member {String} termsAndConditions
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['termsAndConditions'] = undefined;
/**
 * @member {String} entrantMemberType
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['entrantMemberType'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} maxNumberOfEntrants
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['maxNumberOfEntrants'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} minNumberOfEntrants
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['minNumberOfEntrants'] = undefined;
/**
 * @member {Array.<module:model/Rule>} rules
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['rules'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['scheduledEndDate'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['translations'] = undefined;
/**
 * @member {Array.<module:model/CreateRewardRequest>} rewards
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['rewards'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['addConstraints'] = undefined;
/**
 * @member {Array.<String>} productIds
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['productIds'] = undefined;
/**
 * @member {module:model/DependantOn} entrantMemberTagsFilter
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['entrantMemberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['productTagsFilter'] = undefined;
/**
 * @member {module:model/Strategy} strategies
 */

_CreateSimpleCompetitionRequestAllOf["default"].prototype['strategies'] = undefined;
var _default = CreateSimpleCompetitionRequest;
exports["default"] = _default;
