// @ziqni-tech/admin-api-client/dist/model/CreateSpaceRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateSpaceRequest model module.
 * @module model/CreateSpaceRequest
 * @version 0.0.26
 */
var CreateSpaceRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateSpaceRequest</code>.
   * @alias module:model/CreateSpaceRequest
   * @param name {String} The space name in the Ziqni system
   * @param accountType {String} Type of account; personal or company or vendor
   */
  function CreateSpaceRequest(name, accountType) {
    _classCallCheck(this, CreateSpaceRequest);

    CreateSpaceRequest.initialize(this, name, accountType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateSpaceRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, accountType) {
      obj['name'] = name;
      obj['accountType'] = accountType;
    }
    /**
     * Constructs a <code>CreateSpaceRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateSpaceRequest} obj Optional instance to populate.
     * @return {module:model/CreateSpaceRequest} The populated <code>CreateSpaceRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateSpaceRequest();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('accountType')) {
          obj['accountType'] = _ApiClient["default"].convertToType(data['accountType'], 'String');
        }

        if (data.hasOwnProperty('masterSpace')) {
          obj['masterSpace'] = _ApiClient["default"].convertToType(data['masterSpace'], 'String');
        }

        if (data.hasOwnProperty('unitsOfMeasure')) {
          obj['unitsOfMeasure'] = _ApiClient["default"].convertToType(data['unitsOfMeasure'], ['String']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateSpaceRequest;
}();
/**
 * The space name in the Ziqni system
 * @member {String} name
 */


CreateSpaceRequest.prototype['name'] = undefined;
/**
 * Type of account; personal or company or vendor
 * @member {String} accountType
 */

CreateSpaceRequest.prototype['accountType'] = undefined;
/**
 * The master account for this account
 * @member {String} masterSpace
 */

CreateSpaceRequest.prototype['masterSpace'] = undefined;
/**
 * Foreign key to unit of measure id.
 * @member {Array.<String>} unitsOfMeasure
 */

CreateSpaceRequest.prototype['unitsOfMeasure'] = undefined;
/**
 * Additional constraints, if set means true
 * @member {Array.<String>} constraints
 */

CreateSpaceRequest.prototype['constraints'] = undefined;
var _default = CreateSpaceRequest;
exports["default"] = _default;
