// @ziqni-tech/admin-api-client/dist/model/CreateLanguageRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateLanguageRequestAllOf = _interopRequireDefault(require("./CreateLanguageRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateLanguageRequest model module.
 * @module model/CreateLanguageRequest
 * @version 0.0.26
 */
var CreateLanguageRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateLanguageRequest</code>.
   * @alias module:model/CreateLanguageRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateLanguageRequestAllOf
   * @param key {String} A reference to the pre created language keys
   */
  function CreateLanguageRequest(key) {
    _classCallCheck(this, CreateLanguageRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateLanguageRequestAllOf["default"].initialize(this, key);

    CreateLanguageRequest.initialize(this, key);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateLanguageRequest, null, [{
    key: "initialize",
    value: function initialize(obj, key) {
      obj['key'] = key;
    }
    /**
     * Constructs a <code>CreateLanguageRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateLanguageRequest} obj Optional instance to populate.
     * @return {module:model/CreateLanguageRequest} The populated <code>CreateLanguageRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateLanguageRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateLanguageRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateLanguageRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateLanguageRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateLanguageRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateLanguageRequest.prototype['metadata'] = undefined;
/**
 * A reference to the pre created language keys
 * @member {String} key
 */

CreateLanguageRequest.prototype['key'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateLanguageRequestAllOf interface:

/**
 * A reference to the pre created language keys
 * @member {String} key
 */

_CreateLanguageRequestAllOf["default"].prototype['key'] = undefined;
var _default = CreateLanguageRequest;
exports["default"] = _default;
