// @ziqni-tech/admin-api-client/dist/model/CreateMemberRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateMemberRequestAllOf = _interopRequireDefault(require("./CreateMemberRequestAllOf"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _MemberType = _interopRequireDefault(require("./MemberType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateMemberRequest model module.
 * @module model/CreateMemberRequest
 * @version 0.0.26
 */
var CreateMemberRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateMemberRequest</code>.
   * This is a request for Member creation
   * @alias module:model/CreateMemberRequest
   * @implements module:model/CreateMemberRequestAllOf
   * @implements module:model/CreateOptParamModels
   * @param name {String} The name of the member that is used on leader boards and public displays
   * @param memberRefId {String} The reference to this member in your system
   * @param memberType {module:model/MemberType} 
   */
  function CreateMemberRequest(name, memberRefId, memberType) {
    _classCallCheck(this, CreateMemberRequest);

    _CreateMemberRequestAllOf["default"].initialize(this, name, memberRefId, memberType);

    _CreateOptParamModels["default"].initialize(this);

    CreateMemberRequest.initialize(this, name, memberRefId, memberType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateMemberRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, memberRefId, memberType) {
      obj['name'] = name;
      obj['memberRefId'] = memberRefId;
      obj['memberType'] = memberType;
    }
    /**
     * Constructs a <code>CreateMemberRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateMemberRequest} obj Optional instance to populate.
     * @return {module:model/CreateMemberRequest} The populated <code>CreateMemberRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateMemberRequest();

        _CreateMemberRequestAllOf["default"].constructFromObject(data, obj);

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('memberType')) {
          obj['memberType'] = _MemberType["default"].constructFromObject(data['memberType']);
        }

        if (data.hasOwnProperty('teamMembers')) {
          obj['teamMembers'] = _ApiClient["default"].convertToType(data['teamMembers'], ['String']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('timeZoneOffset')) {
          obj['timeZoneOffset'] = _ApiClient["default"].convertToType(data['timeZoneOffset'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return CreateMemberRequest;
}();
/**
 * The name of the member that is used on leader boards and public displays
 * @member {String} name
 */


CreateMemberRequest.prototype['name'] = undefined;
/**
 * The reference to this member in your system
 * @member {String} memberRefId
 */

CreateMemberRequest.prototype['memberRefId'] = undefined;
/**
 * @member {module:model/MemberType} memberType
 */

CreateMemberRequest.prototype['memberType'] = undefined;
/**
 * A social group like Guilds
 * @member {Array.<String>} teamMembers
 */

CreateMemberRequest.prototype['teamMembers'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

CreateMemberRequest.prototype['addConstraints'] = undefined;
/**
 * To what time zone teh emmber belongs to
 * @member {String} timeZoneOffset
 */

CreateMemberRequest.prototype['timeZoneOffset'] = undefined;
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

CreateMemberRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateMemberRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateMemberRequest.prototype['metadata'] = undefined; // Implement CreateMemberRequestAllOf interface:

/**
 * The name of the member that is used on leader boards and public displays
 * @member {String} name
 */

_CreateMemberRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The reference to this member in your system
 * @member {String} memberRefId
 */

_CreateMemberRequestAllOf["default"].prototype['memberRefId'] = undefined;
/**
 * @member {module:model/MemberType} memberType
 */

_CreateMemberRequestAllOf["default"].prototype['memberType'] = undefined;
/**
 * A social group like Guilds
 * @member {Array.<String>} teamMembers
 */

_CreateMemberRequestAllOf["default"].prototype['teamMembers'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_CreateMemberRequestAllOf["default"].prototype['addConstraints'] = undefined;
/**
 * To what time zone teh emmber belongs to
 * @member {String} timeZoneOffset
 */

_CreateMemberRequestAllOf["default"].prototype['timeZoneOffset'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined;
var _default = CreateMemberRequest;
exports["default"] = _default;
