// @ziqni-tech/admin-api-client/dist/model/CreateRewardTypeRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CreateOptParamModels = _interopRequireDefault(require("./CreateOptParamModels"));

var _CreateRewardTypeRequestAllOf = _interopRequireDefault(require("./CreateRewardTypeRequestAllOf"));

var _Translation = _interopRequireDefault(require("./Translation"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CreateRewardTypeRequest model module.
 * @module model/CreateRewardTypeRequest
 * @version 0.0.26
 */
var CreateRewardTypeRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateRewardTypeRequest</code>.
   * This is a request for Reward Type creation
   * @alias module:model/CreateRewardTypeRequest
   * @implements module:model/CreateOptParamModels
   * @implements module:model/CreateRewardTypeRequestAllOf
   * @param name {String} The name of the Reward type
   * @param key {String} A unique key that represents the reward type
   * @param unitOfMeasureType {module:model/UnitOfMeasureType} 
   */
  function CreateRewardTypeRequest(name, key, unitOfMeasureType) {
    _classCallCheck(this, CreateRewardTypeRequest);

    _CreateOptParamModels["default"].initialize(this);

    _CreateRewardTypeRequestAllOf["default"].initialize(this, name, key, unitOfMeasureType);

    CreateRewardTypeRequest.initialize(this, name, key, unitOfMeasureType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateRewardTypeRequest, null, [{
    key: "initialize",
    value: function initialize(obj, name, key, unitOfMeasureType) {
      obj['name'] = name;
      obj['key'] = key;
      obj['unitOfMeasureType'] = unitOfMeasureType;
    }
    /**
     * Constructs a <code>CreateRewardTypeRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateRewardTypeRequest} obj Optional instance to populate.
     * @return {module:model/CreateRewardTypeRequest} The populated <code>CreateRewardTypeRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateRewardTypeRequest();

        _CreateOptParamModels["default"].constructFromObject(data, obj);

        _CreateRewardTypeRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return CreateRewardTypeRequest;
}();
/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */


CreateRewardTypeRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

CreateRewardTypeRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

CreateRewardTypeRequest.prototype['metadata'] = undefined;
/**
 * The name of the Reward type
 * @member {String} name
 */

CreateRewardTypeRequest.prototype['name'] = undefined;
/**
 * The description of a reward type
 * @member {String} description
 */

CreateRewardTypeRequest.prototype['description'] = undefined;
/**
 * A unique key that represents the reward type
 * @member {String} key
 */

CreateRewardTypeRequest.prototype['key'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

CreateRewardTypeRequest.prototype['unitOfMeasureType'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

CreateRewardTypeRequest.prototype['translations'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

CreateRewardTypeRequest.prototype['addConstraints'] = undefined; // Implement CreateOptParamModels interface:

/**
 * A list of custom field entries
 * @member {Object.<String, Object>} customFields
 */

_CreateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_CreateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_CreateOptParamModels["default"].prototype['metadata'] = undefined; // Implement CreateRewardTypeRequestAllOf interface:

/**
 * The name of the Reward type
 * @member {String} name
 */

_CreateRewardTypeRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of a reward type
 * @member {String} description
 */

_CreateRewardTypeRequestAllOf["default"].prototype['description'] = undefined;
/**
 * A unique key that represents the reward type
 * @member {String} key
 */

_CreateRewardTypeRequestAllOf["default"].prototype['key'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

_CreateRewardTypeRequestAllOf["default"].prototype['unitOfMeasureType'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_CreateRewardTypeRequestAllOf["default"].prototype['translations'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_CreateRewardTypeRequestAllOf["default"].prototype['addConstraints'] = undefined;
var _default = CreateRewardTypeRequest;
exports["default"] = _default;
