// @ziqni-tech/admin-api-client/dist/model/CustomFieldReduced.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CustomFieldReducedAllOf = _interopRequireDefault(require("./CustomFieldReducedAllOf"));

var _FieldType = _interopRequireDefault(require("./FieldType"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CustomFieldReduced model module.
 * @module model/CustomFieldReduced
 * @version 0.0.26
 */
var CustomFieldReduced = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CustomFieldReduced</code>.
   * This is a Custom field model with fewer parameters
   * @alias module:model/CustomFieldReduced
   * @implements module:model/ModelDefault
   * @implements module:model/CustomFieldReducedAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param key {String} The key of a Custom field
   * @param fieldType {module:model/FieldType} 
   */
  function CustomFieldReduced(id, spaceName, created, key, fieldType) {
    _classCallCheck(this, CustomFieldReduced);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _CustomFieldReducedAllOf["default"].initialize(this, key, fieldType);

    CustomFieldReduced.initialize(this, id, spaceName, created, key, fieldType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CustomFieldReduced, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, key, fieldType) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['key'] = key;
      obj['fieldType'] = fieldType;
    }
    /**
     * Constructs a <code>CustomFieldReduced</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CustomFieldReduced} obj Optional instance to populate.
     * @return {module:model/CustomFieldReduced} The populated <code>CustomFieldReduced</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CustomFieldReduced();

        _ModelDefault["default"].constructFromObject(data, obj);

        _CustomFieldReducedAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('fieldType')) {
          obj['fieldType'] = _FieldType["default"].constructFromObject(data['fieldType']);
        }
      }

      return obj;
    }
  }]);

  return CustomFieldReduced;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


CustomFieldReduced.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

CustomFieldReduced.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

CustomFieldReduced.prototype['created'] = undefined;
/**
 * The key of a Custom field
 * @member {String} key
 */

CustomFieldReduced.prototype['key'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

CustomFieldReduced.prototype['fieldType'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement CustomFieldReducedAllOf interface:

/**
 * The key of a Custom field
 * @member {String} key
 */

_CustomFieldReducedAllOf["default"].prototype['key'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

_CustomFieldReducedAllOf["default"].prototype['fieldType'] = undefined;
var _default = CustomFieldReduced;
exports["default"] = _default;
