// @ziqni-tech/admin-api-client/dist/model/CustomField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AppliesTo = _interopRequireDefault(require("./AppliesTo"));

var _CustomFieldAllOf = _interopRequireDefault(require("./CustomFieldAllOf"));

var _FieldType = _interopRequireDefault(require("./FieldType"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CustomField model module.
 * @module model/CustomField
 * @version 0.0.26
 */
var CustomField = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CustomField</code>.
   * Custom field model
   * @alias module:model/CustomField
   * @implements module:model/ModelDefault
   * @implements module:model/CustomFieldAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} The name of a Custom field
   * @param key {String} The key of a Custom field
   * @param fieldType {module:model/FieldType} 
   * @param appliesTo {module:model/AppliesTo} 
   */
  function CustomField(id, spaceName, created, name, key, fieldType, appliesTo) {
    _classCallCheck(this, CustomField);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _CustomFieldAllOf["default"].initialize(this, name, key, fieldType, appliesTo);

    CustomField.initialize(this, id, spaceName, created, name, key, fieldType, appliesTo);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CustomField, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, key, fieldType, appliesTo) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['key'] = key;
      obj['fieldType'] = fieldType;
      obj['appliesTo'] = appliesTo;
    }
    /**
     * Constructs a <code>CustomField</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CustomField} obj Optional instance to populate.
     * @return {module:model/CustomField} The populated <code>CustomField</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CustomField();

        _ModelDefault["default"].constructFromObject(data, obj);

        _CustomFieldAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('fieldType')) {
          obj['fieldType'] = _FieldType["default"].constructFromObject(data['fieldType']);
        }

        if (data.hasOwnProperty('appliesTo')) {
          obj['appliesTo'] = _AppliesTo["default"].constructFromObject(data['appliesTo']);
        }
      }

      return obj;
    }
  }]);

  return CustomField;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


CustomField.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

CustomField.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

CustomField.prototype['created'] = undefined;
/**
 * The name of a Custom field
 * @member {String} name
 */

CustomField.prototype['name'] = undefined;
/**
 * The key of a Custom field
 * @member {String} key
 */

CustomField.prototype['key'] = undefined;
/**
 * The description of a Custom field
 * @member {String} description
 */

CustomField.prototype['description'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

CustomField.prototype['fieldType'] = undefined;
/**
 * @member {module:model/AppliesTo} appliesTo
 */

CustomField.prototype['appliesTo'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement CustomFieldAllOf interface:

/**
 * The name of a Custom field
 * @member {String} name
 */

_CustomFieldAllOf["default"].prototype['name'] = undefined;
/**
 * The key of a Custom field
 * @member {String} key
 */

_CustomFieldAllOf["default"].prototype['key'] = undefined;
/**
 * The description of a Custom field
 * @member {String} description
 */

_CustomFieldAllOf["default"].prototype['description'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

_CustomFieldAllOf["default"].prototype['fieldType'] = undefined;
/**
 * @member {module:model/AppliesTo} appliesTo
 */

_CustomFieldAllOf["default"].prototype['appliesTo'] = undefined;
var _default = CustomField;
exports["default"] = _default;
