// @ziqni-tech/admin-api-client/dist/model/CustomFieldAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AppliesTo = _interopRequireDefault(require("./AppliesTo"));

var _FieldType = _interopRequireDefault(require("./FieldType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CustomFieldAllOf model module.
 * @module model/CustomFieldAllOf
 * @version 0.0.26
 */
var CustomFieldAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CustomFieldAllOf</code>.
   * @alias module:model/CustomFieldAllOf
   * @param name {String} The name of a Custom field
   * @param key {String} The key of a Custom field
   * @param fieldType {module:model/FieldType} 
   * @param appliesTo {module:model/AppliesTo} 
   */
  function CustomFieldAllOf(name, key, fieldType, appliesTo) {
    _classCallCheck(this, CustomFieldAllOf);

    CustomFieldAllOf.initialize(this, name, key, fieldType, appliesTo);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CustomFieldAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, key, fieldType, appliesTo) {
      obj['name'] = name;
      obj['key'] = key;
      obj['fieldType'] = fieldType;
      obj['appliesTo'] = appliesTo;
    }
    /**
     * Constructs a <code>CustomFieldAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CustomFieldAllOf} obj Optional instance to populate.
     * @return {module:model/CustomFieldAllOf} The populated <code>CustomFieldAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CustomFieldAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('fieldType')) {
          obj['fieldType'] = _FieldType["default"].constructFromObject(data['fieldType']);
        }

        if (data.hasOwnProperty('appliesTo')) {
          obj['appliesTo'] = _AppliesTo["default"].constructFromObject(data['appliesTo']);
        }
      }

      return obj;
    }
  }]);

  return CustomFieldAllOf;
}();
/**
 * The name of a Custom field
 * @member {String} name
 */


CustomFieldAllOf.prototype['name'] = undefined;
/**
 * The key of a Custom field
 * @member {String} key
 */

CustomFieldAllOf.prototype['key'] = undefined;
/**
 * The description of a Custom field
 * @member {String} description
 */

CustomFieldAllOf.prototype['description'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

CustomFieldAllOf.prototype['fieldType'] = undefined;
/**
 * @member {module:model/AppliesTo} appliesTo
 */

CustomFieldAllOf.prototype['appliesTo'] = undefined;
var _default = CustomFieldAllOf;
exports["default"] = _default;
