// @ziqni-tech/admin-api-client/dist/model/EntrantRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EntrantAction = _interopRequireDefault(require("./EntrantAction"));

var _EntrantStatus = _interopRequireDefault(require("./EntrantStatus"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The EntrantRequest model module.
 * @module model/EntrantRequest
 * @version 0.0.26
 */
var EntrantRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EntrantRequest</code>.
   * @alias module:model/EntrantRequest
   * @param memberRefId {String} Unique member reference identifier. A member has to exist in the Ziqni database
   * @param entityId {String} Unique competition identifier
   * @param entrantStatus {module:model/EntrantStatus} 
   * @param entrantAction {module:model/EntrantAction} 
   */
  function EntrantRequest(memberRefId, entityId, entrantStatus, entrantAction) {
    _classCallCheck(this, EntrantRequest);

    EntrantRequest.initialize(this, memberRefId, entityId, entrantStatus, entrantAction);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EntrantRequest, null, [{
    key: "initialize",
    value: function initialize(obj, memberRefId, entityId, entrantStatus, entrantAction) {
      obj['memberRefId'] = memberRefId;
      obj['entityId'] = entityId;
      obj['entrantStatus'] = entrantStatus;
      obj['entrantAction'] = entrantAction;
    }
    /**
     * Constructs a <code>EntrantRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EntrantRequest} obj Optional instance to populate.
     * @return {module:model/EntrantRequest} The populated <code>EntrantRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EntrantRequest();

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('entrantStatus')) {
          obj['entrantStatus'] = _EntrantStatus["default"].constructFromObject(data['entrantStatus']);
        }

        if (data.hasOwnProperty('entrantAction')) {
          obj['entrantAction'] = _EntrantAction["default"].constructFromObject(data['entrantAction']);
        }
      }

      return obj;
    }
  }]);

  return EntrantRequest;
}();
/**
 * Unique member identifier. A member has to exist in the Ziqni database
 * @member {String} memberId
 */


EntrantRequest.prototype['memberId'] = undefined;
/**
 * Unique member reference identifier. A member has to exist in the Ziqni database
 * @member {String} memberRefId
 */

EntrantRequest.prototype['memberRefId'] = undefined;
/**
 * Unique competition identifier
 * @member {String} entityId
 */

EntrantRequest.prototype['entityId'] = undefined;
/**
 * @member {module:model/EntrantStatus} entrantStatus
 */

EntrantRequest.prototype['entrantStatus'] = undefined;
/**
 * @member {module:model/EntrantAction} entrantAction
 */

EntrantRequest.prototype['entrantAction'] = undefined;
var _default = EntrantRequest;
exports["default"] = _default;
