// @ziqni-tech/admin-api-client/dist/model/Entrant.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EntityType = _interopRequireDefault(require("./EntityType"));

var _EntrantAction = _interopRequireDefault(require("./EntrantAction"));

var _EntrantStatus = _interopRequireDefault(require("./EntrantStatus"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Entrant model module.
 * @module model/Entrant
 * @version 0.0.26
 */
var Entrant = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Entrant</code>.
   * @alias module:model/Entrant
   * @param memberId {String} Unique member identifier. A member has to exist in the Ziqni database
   * @param memberRefId {String} Unique member reference identifier. A member has to exist in the Ziqni database
   * @param entrantStatus {module:model/EntrantStatus} 
   */
  function Entrant(memberId, memberRefId, entrantStatus) {
    _classCallCheck(this, Entrant);

    Entrant.initialize(this, memberId, memberRefId, entrantStatus);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Entrant, null, [{
    key: "initialize",
    value: function initialize(obj, memberId, memberRefId, entrantStatus) {
      obj['memberId'] = memberId;
      obj['memberRefId'] = memberRefId;
      obj['entrantStatus'] = entrantStatus;
    }
    /**
     * Constructs a <code>Entrant</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Entrant} obj Optional instance to populate.
     * @return {module:model/Entrant} The populated <code>Entrant</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Entrant();

        if (data.hasOwnProperty('entrantAction')) {
          obj['entrantAction'] = _EntrantAction["default"].constructFromObject(data['entrantAction']);
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('entrantStatus')) {
          obj['entrantStatus'] = _EntrantStatus["default"].constructFromObject(data['entrantStatus']);
        }

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('entityType')) {
          obj['entityType'] = _EntityType["default"].constructFromObject(data['entityType']);
        }
      }

      return obj;
    }
  }]);

  return Entrant;
}();
/**
 * @member {module:model/EntrantAction} entrantAction
 */


Entrant.prototype['entrantAction'] = undefined;
/**
 * Unique member identifier. A member has to exist in the Ziqni database
 * @member {String} memberId
 */

Entrant.prototype['memberId'] = undefined;
/**
 * Unique member reference identifier. A member has to exist in the Ziqni database
 * @member {String} memberRefId
 */

Entrant.prototype['memberRefId'] = undefined;
/**
 * @member {module:model/EntrantStatus} entrantStatus
 */

Entrant.prototype['entrantStatus'] = undefined;
/**
 * Unique identifier of the entity
 * @member {String} entityId
 */

Entrant.prototype['entityId'] = undefined;
/**
 * @member {module:model/EntityType} entityType
 */

Entrant.prototype['entityType'] = undefined;
var _default = Entrant;
exports["default"] = _default;
