// @ziqni-tech/admin-api-client/dist/model/ErrorDetail.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ErrorDetail model module.
 * @module model/ErrorDetail
 * @version 0.0.26
 */
var ErrorDetail = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ErrorDetail</code>.
   * @alias module:model/ErrorDetail
   * @param errorCode {Number} A unique system identifier that describes this specific error
   * @param message {String} The error description
   */
  function ErrorDetail(errorCode, message) {
    _classCallCheck(this, ErrorDetail);

    ErrorDetail.initialize(this, errorCode, message);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ErrorDetail, null, [{
    key: "initialize",
    value: function initialize(obj, errorCode, message) {
      obj['errorCode'] = errorCode;
      obj['message'] = message;
    }
    /**
     * Constructs a <code>ErrorDetail</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ErrorDetail} obj Optional instance to populate.
     * @return {module:model/ErrorDetail} The populated <code>ErrorDetail</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ErrorDetail();

        if (data.hasOwnProperty('modelId')) {
          obj['modelId'] = _ApiClient["default"].convertToType(data['modelId'], 'String');
        }

        if (data.hasOwnProperty('errorCode')) {
          obj['errorCode'] = _ApiClient["default"].convertToType(data['errorCode'], 'Number');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('errorAt')) {
          obj['errorAt'] = _ApiClient["default"].convertToType(data['errorAt'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ErrorDetail;
}();
/**
 * The identifier of the record this error relates to (This value only gets set if the error could relate to more than one object)
 * @member {String} modelId
 */


ErrorDetail.prototype['modelId'] = undefined;
/**
 * A unique system identifier that describes this specific error
 * @member {Number} errorCode
 */

ErrorDetail.prototype['errorCode'] = undefined;
/**
 * The error description
 * @member {String} message
 */

ErrorDetail.prototype['message'] = undefined;
/**
 * The object number that returned the error
 * @member {Number} errorAt
 */

ErrorDetail.prototype['errorAt'] = undefined;
var _default = ErrorDetail;
exports["default"] = _default;
