// @ziqni-tech/admin-api-client/dist/model/Event.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EventAllOf = _interopRequireDefault(require("./EventAllOf"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Event model module.
 * @module model/Event
 * @version 0.0.26
 */
var Event = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Event</code>.
   * @alias module:model/Event
   * @implements module:model/ModelDefault
   * @implements module:model/EventAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param action {String} The identifier that describes the meaning of this event
   * @param sourceValue {Number} The actual numerical value related to the event
   * @param points {Number} A value derived from the source value field. In the context of Product additionally an adjustment factor will be applied
   * @param transactionTimestamp {Date} ISO8601 timestamp for when this event happened. All records are stored in UTC time zone
   */
  function Event(id, spaceName, created, action, sourceValue, points, transactionTimestamp) {
    _classCallCheck(this, Event);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _EventAllOf["default"].initialize(this, action, sourceValue, points, transactionTimestamp);

    Event.initialize(this, id, spaceName, created, action, sourceValue, points, transactionTimestamp);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Event, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, action, sourceValue, points, transactionTimestamp) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['action'] = action;
      obj['sourceValue'] = sourceValue;
      obj['points'] = points;
      obj['transactionTimestamp'] = transactionTimestamp;
    }
    /**
     * Constructs a <code>Event</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Event} obj Optional instance to populate.
     * @return {module:model/Event} The populated <code>Event</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Event();

        _ModelDefault["default"].constructFromObject(data, obj);

        _EventAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('action')) {
          obj['action'] = _ApiClient["default"].convertToType(data['action'], 'String');
        }

        if (data.hasOwnProperty('batchId')) {
          obj['batchId'] = _ApiClient["default"].convertToType(data['batchId'], 'String');
        }

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('entityRefId')) {
          obj['entityRefId'] = _ApiClient["default"].convertToType(data['entityRefId'], 'String');
        }

        if (data.hasOwnProperty('sourceValue')) {
          obj['sourceValue'] = _ApiClient["default"].convertToType(data['sourceValue'], 'Number');
        }

        if (data.hasOwnProperty('points')) {
          obj['points'] = _ApiClient["default"].convertToType(data['points'], 'Number');
        }

        if (data.hasOwnProperty('transactionTimestamp')) {
          obj['transactionTimestamp'] = _ApiClient["default"].convertToType(data['transactionTimestamp'], 'Date');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('relatesTo')) {
          obj['relatesTo'] = _ApiClient["default"].convertToType(data['relatesTo'], ['String']);
        }

        if (data.hasOwnProperty('relatesToExternal')) {
          obj['relatesToExternal'] = _ApiClient["default"].convertToType(data['relatesToExternal'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return Event;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Event.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Event.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Event.prototype['created'] = undefined;
/**
 * Unique system identifier of a Member
 * @member {String} memberId
 */

Event.prototype['memberId'] = undefined;
/**
 * The reference to this member in external system
 * @member {String} memberRefId
 */

Event.prototype['memberRefId'] = undefined;
/**
 * The identifier that describes the meaning of this event
 * @member {String} action
 */

Event.prototype['action'] = undefined;
/**
 * The batch identifier is used to group related event stream data together. This could be as simple as a UUID or your internal reference to that distinct player session event
 * @member {String} batchId
 */

Event.prototype['batchId'] = undefined;
/**
 * It is a reference to a models ID for any of the following objects - Product, Achievement, Reward, Award
 * @member {String} entityId
 */

Event.prototype['entityId'] = undefined;
/**
 * It is a reference to a models ID for any of the following objects - Product, Achievement, Reward, Award
 * @member {String} entityRefId
 */

Event.prototype['entityRefId'] = undefined;
/**
 * The actual numerical value related to the event
 * @member {Number} sourceValue
 */

Event.prototype['sourceValue'] = undefined;
/**
 * A value derived from the source value field. In the context of Product additionally an adjustment factor will be applied
 * @member {Number} points
 */

Event.prototype['points'] = undefined;
/**
 * ISO8601 timestamp for when this event happened. All records are stored in UTC time zone
 * @member {Date} transactionTimestamp
 */

Event.prototype['transactionTimestamp'] = undefined;
/**
 * Event metadata map.
 * @member {Object.<String, Object>} metadata
 */

Event.prototype['metadata'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Event.prototype['customFields'] = undefined;
/**
 * A list of Strings of groups that the tag belongs to
 * @member {Array.<String>} tags
 */

Event.prototype['tags'] = undefined;
/**
 * @member {Array.<String>} relatesTo
 */

Event.prototype['relatesTo'] = undefined;
/**
 * @member {Array.<String>} relatesToExternal
 */

Event.prototype['relatesToExternal'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement EventAllOf interface:

/**
 * Unique system identifier of a Member
 * @member {String} memberId
 */

_EventAllOf["default"].prototype['memberId'] = undefined;
/**
 * The reference to this member in external system
 * @member {String} memberRefId
 */

_EventAllOf["default"].prototype['memberRefId'] = undefined;
/**
 * The identifier that describes the meaning of this event
 * @member {String} action
 */

_EventAllOf["default"].prototype['action'] = undefined;
/**
 * The batch identifier is used to group related event stream data together. This could be as simple as a UUID or your internal reference to that distinct player session event
 * @member {String} batchId
 */

_EventAllOf["default"].prototype['batchId'] = undefined;
/**
 * It is a reference to a models ID for any of the following objects - Product, Achievement, Reward, Award
 * @member {String} entityId
 */

_EventAllOf["default"].prototype['entityId'] = undefined;
/**
 * It is a reference to a models ID for any of the following objects - Product, Achievement, Reward, Award
 * @member {String} entityRefId
 */

_EventAllOf["default"].prototype['entityRefId'] = undefined;
/**
 * The actual numerical value related to the event
 * @member {Number} sourceValue
 */

_EventAllOf["default"].prototype['sourceValue'] = undefined;
/**
 * A value derived from the source value field. In the context of Product additionally an adjustment factor will be applied
 * @member {Number} points
 */

_EventAllOf["default"].prototype['points'] = undefined;
/**
 * ISO8601 timestamp for when this event happened. All records are stored in UTC time zone
 * @member {Date} transactionTimestamp
 */

_EventAllOf["default"].prototype['transactionTimestamp'] = undefined;
/**
 * Event metadata map.
 * @member {Object.<String, Object>} metadata
 */

_EventAllOf["default"].prototype['metadata'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

_EventAllOf["default"].prototype['customFields'] = undefined;
/**
 * A list of Strings of groups that the tag belongs to
 * @member {Array.<String>} tags
 */

_EventAllOf["default"].prototype['tags'] = undefined;
/**
 * @member {Array.<String>} relatesTo
 */

_EventAllOf["default"].prototype['relatesTo'] = undefined;
/**
 * @member {Array.<String>} relatesToExternal
 */

_EventAllOf["default"].prototype['relatesToExternal'] = undefined;
var _default = Event;
exports["default"] = _default;
