// @ziqni-tech/admin-api-client/dist/model/FactDefault.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Macro = _interopRequireDefault(require("./Macro"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The FactDefault model module.
 * @module model/FactDefault
 * @version 0.0.26
 */
var FactDefault = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>FactDefault</code>.
   * @alias module:model/FactDefault
   * @param operators {Array.<String>} The operators supported by the fact
   * @param constantType {String} The data type of the constant
   * @param constantExample {String} Example of the constant
   */
  function FactDefault(operators, constantType, constantExample) {
    _classCallCheck(this, FactDefault);

    FactDefault.initialize(this, operators, constantType, constantExample);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(FactDefault, null, [{
    key: "initialize",
    value: function initialize(obj, operators, constantType, constantExample) {
      obj['operators'] = operators;
      obj['constantType'] = constantType;
      obj['constantExample'] = constantExample;
    }
    /**
     * Constructs a <code>FactDefault</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/FactDefault} obj Optional instance to populate.
     * @return {module:model/FactDefault} The populated <code>FactDefault</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new FactDefault();

        if (data.hasOwnProperty('operators')) {
          obj['operators'] = _ApiClient["default"].convertToType(data['operators'], ['String']);
        }

        if (data.hasOwnProperty('constantType')) {
          obj['constantType'] = _ApiClient["default"].convertToType(data['constantType'], 'String');
        }

        if (data.hasOwnProperty('constantExample')) {
          obj['constantExample'] = _ApiClient["default"].convertToType(data['constantExample'], 'String');
        }

        if (data.hasOwnProperty('defaultVale')) {
          obj['defaultVale'] = _ApiClient["default"].convertToType(data['defaultVale'], 'String');
        }

        if (data.hasOwnProperty('macros')) {
          obj['macros'] = _ApiClient["default"].convertToType(data['macros'], [_Macro["default"]]);
        }
      }

      return obj;
    }
  }]);

  return FactDefault;
}();
/**
 * The operators supported by the fact
 * @member {Array.<String>} operators
 */


FactDefault.prototype['operators'] = undefined;
/**
 * The data type of the constant
 * @member {String} constantType
 */

FactDefault.prototype['constantType'] = undefined;
/**
 * Example of the constant
 * @member {String} constantExample
 */

FactDefault.prototype['constantExample'] = undefined;
/**
 * The default value of the constant for the fact
 * @member {String} defaultVale
 */

FactDefault.prototype['defaultVale'] = undefined;
/**
 * @member {Array.<module:model/Macro>} macros
 */

FactDefault.prototype['macros'] = undefined;
var _default = FactDefault;
exports["default"] = _default;
