// @ziqni-tech/admin-api-client/dist/model/KafkaConnection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _KafkaConnectionAllOf = _interopRequireDefault(require("./KafkaConnectionAllOf"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The KafkaConnection model module.
 * @module model/KafkaConnection
 * @version 0.0.26
 */
var KafkaConnection = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KafkaConnection</code>.
   * @alias module:model/KafkaConnection
   * @implements module:model/OptParamModels
   * @implements module:model/KafkaConnectionAllOf
   * @implements module:model/ModelDefault
   * @param name {String} The name of the consumer
   * @param brokers {String} Kafka broker endpoints.
   * @param groupId {String} Group Id for connection
   * @param topic {String} Topic name
   * @param lastKnownStatus {String} Last known status of the connection
   * @param lastKnownStatusCode {Number} Status code correspoding to the last known status
   * @param transformerId {String} The transformer to use, if empty the default system transformer will be used
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   */
  function KafkaConnection(name, brokers, groupId, topic, lastKnownStatus, lastKnownStatusCode, transformerId, id, spaceName, created) {
    _classCallCheck(this, KafkaConnection);

    _OptParamModels["default"].initialize(this);

    _KafkaConnectionAllOf["default"].initialize(this, name, brokers, groupId, topic, lastKnownStatus, lastKnownStatusCode, transformerId);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    KafkaConnection.initialize(this, name, brokers, groupId, topic, lastKnownStatus, lastKnownStatusCode, transformerId, id, spaceName, created);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KafkaConnection, null, [{
    key: "initialize",
    value: function initialize(obj, name, brokers, groupId, topic, lastKnownStatus, lastKnownStatusCode, transformerId, id, spaceName, created) {
      obj['name'] = name;
      obj['brokers'] = brokers;
      obj['groupId'] = groupId;
      obj['topic'] = topic;
      obj['lastKnownStatus'] = lastKnownStatus;
      obj['lastKnownStatusCode'] = lastKnownStatusCode;
      obj['transformerId'] = transformerId;
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
    }
    /**
     * Constructs a <code>KafkaConnection</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KafkaConnection} obj Optional instance to populate.
     * @return {module:model/KafkaConnection} The populated <code>KafkaConnection</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KafkaConnection();

        _OptParamModels["default"].constructFromObject(data, obj);

        _KafkaConnectionAllOf["default"].constructFromObject(data, obj);

        _ModelDefault["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('brokers')) {
          obj['brokers'] = _ApiClient["default"].convertToType(data['brokers'], 'String');
        }

        if (data.hasOwnProperty('groupId')) {
          obj['groupId'] = _ApiClient["default"].convertToType(data['groupId'], 'String');
        }

        if (data.hasOwnProperty('topic')) {
          obj['topic'] = _ApiClient["default"].convertToType(data['topic'], 'String');
        }

        if (data.hasOwnProperty('lastKnownStatus')) {
          obj['lastKnownStatus'] = _ApiClient["default"].convertToType(data['lastKnownStatus'], 'String');
        }

        if (data.hasOwnProperty('lastKnownStatusCode')) {
          obj['lastKnownStatusCode'] = _ApiClient["default"].convertToType(data['lastKnownStatusCode'], 'Number');
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }

        if (data.hasOwnProperty('connectionType')) {
          obj['connectionType'] = _ApiClient["default"].convertToType(data['connectionType'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return KafkaConnection;
}();
/**
 * @member {Object.<String, Object>} customFields
 */


KafkaConnection.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

KafkaConnection.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

KafkaConnection.prototype['metadata'] = undefined;
/**
 * The name of the consumer
 * @member {String} name
 */

KafkaConnection.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

KafkaConnection.prototype['description'] = undefined;
/**
 * Kafka broker endpoints.
 * @member {String} brokers
 */

KafkaConnection.prototype['brokers'] = undefined;
/**
 * Group Id for connection
 * @member {String} groupId
 */

KafkaConnection.prototype['groupId'] = undefined;
/**
 * Topic name
 * @member {String} topic
 */

KafkaConnection.prototype['topic'] = undefined;
/**
 * Last known status of the connection
 * @member {String} lastKnownStatus
 */

KafkaConnection.prototype['lastKnownStatus'] = undefined;
/**
 * Status code correspoding to the last known status
 * @member {Number} lastKnownStatusCode
 */

KafkaConnection.prototype['lastKnownStatusCode'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

KafkaConnection.prototype['transformerId'] = undefined;
/**
 * The type of connection
 * @member {String} connectionType
 */

KafkaConnection.prototype['connectionType'] = undefined;
/**
 * A unique system generated identifier
 * @member {String} id
 */

KafkaConnection.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

KafkaConnection.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

KafkaConnection.prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement KafkaConnectionAllOf interface:

/**
 * The name of the consumer
 * @member {String} name
 */

_KafkaConnectionAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

_KafkaConnectionAllOf["default"].prototype['description'] = undefined;
/**
 * Kafka broker endpoints.
 * @member {String} brokers
 */

_KafkaConnectionAllOf["default"].prototype['brokers'] = undefined;
/**
 * Group Id for connection
 * @member {String} groupId
 */

_KafkaConnectionAllOf["default"].prototype['groupId'] = undefined;
/**
 * Topic name
 * @member {String} topic
 */

_KafkaConnectionAllOf["default"].prototype['topic'] = undefined;
/**
 * Last known status of the connection
 * @member {String} lastKnownStatus
 */

_KafkaConnectionAllOf["default"].prototype['lastKnownStatus'] = undefined;
/**
 * Status code correspoding to the last known status
 * @member {Number} lastKnownStatusCode
 */

_KafkaConnectionAllOf["default"].prototype['lastKnownStatusCode'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

_KafkaConnectionAllOf["default"].prototype['transformerId'] = undefined;
/**
 * The type of connection
 * @member {String} connectionType
 */

_KafkaConnectionAllOf["default"].prototype['connectionType'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined;
var _default = KafkaConnection;
exports["default"] = _default;
