// @ziqni-tech/admin-api-client/dist/model/LeaderboardEntry.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _LeaderboardMember = _interopRequireDefault(require("./LeaderboardMember"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The LeaderboardEntry model module.
 * @module model/LeaderboardEntry
 * @version 0.0.26
 */
var LeaderboardEntry = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LeaderboardEntry</code>.
   * @alias module:model/LeaderboardEntry
   * @param rank {Number} The reank of the player
   */
  function LeaderboardEntry(rank) {
    _classCallCheck(this, LeaderboardEntry);

    LeaderboardEntry.initialize(this, rank);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LeaderboardEntry, null, [{
    key: "initialize",
    value: function initialize(obj, rank) {
      obj['rank'] = rank;
    }
    /**
     * Constructs a <code>LeaderboardEntry</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LeaderboardEntry} obj Optional instance to populate.
     * @return {module:model/LeaderboardEntry} The populated <code>LeaderboardEntry</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LeaderboardEntry();

        if (data.hasOwnProperty('rank')) {
          obj['rank'] = _ApiClient["default"].convertToType(data['rank'], 'Number');
        }

        if (data.hasOwnProperty('score')) {
          obj['score'] = _ApiClient["default"].convertToType(data['score'], 'Number');
        }

        if (data.hasOwnProperty('bestScores')) {
          obj['bestScores'] = _ApiClient["default"].convertToType(data['bestScores'], ['Number']);
        }

        if (data.hasOwnProperty('members')) {
          obj['members'] = _ApiClient["default"].convertToType(data['members'], [_LeaderboardMember["default"]]);
        }
      }

      return obj;
    }
  }]);

  return LeaderboardEntry;
}();
/**
 * The reank of the player
 * @member {Number} rank
 */


LeaderboardEntry.prototype['rank'] = undefined;
/**
 * The multiplier to apply to source values received for this product action helper events
 * @member {Number} score
 */

LeaderboardEntry.prototype['score'] = undefined;
/**
 * The multiplier to apply to source values received for this product action helper events
 * @member {Array.<Number>} bestScores
 */

LeaderboardEntry.prototype['bestScores'] = undefined;
/**
 * Member details
 * @member {Array.<module:model/LeaderboardMember>} members
 */

LeaderboardEntry.prototype['members'] = undefined;
var _default = LeaderboardEntry;
exports["default"] = _default;
