// @ziqni-tech/admin-api-client/dist/model/LeaderboardAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _LeaderboardEntry = _interopRequireDefault(require("./LeaderboardEntry"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The LeaderboardAllOf model module.
 * @module model/LeaderboardAllOf
 * @version 0.0.26
 */
var LeaderboardAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LeaderboardAllOf</code>.
   * @alias module:model/LeaderboardAllOf
   * @param constraints {Array.<String>} Additional constraints
   */
  function LeaderboardAllOf(constraints) {
    _classCallCheck(this, LeaderboardAllOf);

    LeaderboardAllOf.initialize(this, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LeaderboardAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, constraints) {
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>LeaderboardAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LeaderboardAllOf} obj Optional instance to populate.
     * @return {module:model/LeaderboardAllOf} The populated <code>LeaderboardAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LeaderboardAllOf();

        if (data.hasOwnProperty('leaderboardEntries')) {
          obj['leaderboardEntries'] = _ApiClient["default"].convertToType(data['leaderboardEntries'], [_LeaderboardEntry["default"]]);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('sequence')) {
          obj['sequence'] = _ApiClient["default"].convertToType(data['sequence'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return LeaderboardAllOf;
}();
/**
 * @member {Array.<module:model/LeaderboardEntry>} leaderboardEntries
 */


LeaderboardAllOf.prototype['leaderboardEntries'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

LeaderboardAllOf.prototype['constraints'] = undefined;
/**
 * how many changes were made to teh leaderboard
 * @member {Number} sequence
 */

LeaderboardAllOf.prototype['sequence'] = undefined;
var _default = LeaderboardAllOf;
exports["default"] = _default;
