// @ziqni-tech/admin-api-client/dist/model/LeaderboardResponseByContest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Leaderboard = _interopRequireDefault(require("./Leaderboard"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The LeaderboardResponseByContest model module.
 * @module model/LeaderboardResponseByContest
 * @version 0.0.26
 */
var LeaderboardResponseByContest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LeaderboardResponseByContest</code>.
   * @alias module:model/LeaderboardResponseByContest
   * @param contestId {String} The id of the contest
   * @param name {String} A name for the Contest. Can be translated
   * @param round {Number} To what round does the contest belong
   */
  function LeaderboardResponseByContest(contestId, name, round) {
    _classCallCheck(this, LeaderboardResponseByContest);

    LeaderboardResponseByContest.initialize(this, contestId, name, round);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LeaderboardResponseByContest, null, [{
    key: "initialize",
    value: function initialize(obj, contestId, name, round) {
      obj['contestId'] = contestId;
      obj['name'] = name;
      obj['round'] = round;
    }
    /**
     * Constructs a <code>LeaderboardResponseByContest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LeaderboardResponseByContest} obj Optional instance to populate.
     * @return {module:model/LeaderboardResponseByContest} The populated <code>LeaderboardResponseByContest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LeaderboardResponseByContest();

        if (data.hasOwnProperty('contestId')) {
          obj['contestId'] = _ApiClient["default"].convertToType(data['contestId'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('round')) {
          obj['round'] = _ApiClient["default"].convertToType(data['round'], 'Number');
        }

        if (data.hasOwnProperty('leaderboard')) {
          obj['leaderboard'] = _ApiClient["default"].convertToType(data['leaderboard'], [_Leaderboard["default"]]);
        }
      }

      return obj;
    }
  }]);

  return LeaderboardResponseByContest;
}();
/**
 * The id of the contest
 * @member {String} contestId
 */


LeaderboardResponseByContest.prototype['contestId'] = undefined;
/**
 * A name for the Contest. Can be translated
 * @member {String} name
 */

LeaderboardResponseByContest.prototype['name'] = undefined;
/**
 * To what round does the contest belong
 * @member {Number} round
 */

LeaderboardResponseByContest.prototype['round'] = undefined;
/**
 * @member {Array.<module:model/Leaderboard>} leaderboard
 */

LeaderboardResponseByContest.prototype['leaderboard'] = undefined;
var _default = LeaderboardResponseByContest;
exports["default"] = _default;
