// @ziqni-tech/admin-api-client/dist/model/Leaderboard.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _LeaderboardAllOf = _interopRequireDefault(require("./LeaderboardAllOf"));

var _LeaderboardEntry = _interopRequireDefault(require("./LeaderboardEntry"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Leaderboard model module.
 * @module model/Leaderboard
 * @version 0.0.26
 */
var Leaderboard = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Leaderboard</code>.
   * @alias module:model/Leaderboard
   * @implements module:model/ModelDefault
   * @implements module:model/LeaderboardAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param constraints {Array.<String>} Additional constraints
   */
  function Leaderboard(id, spaceName, created, constraints) {
    _classCallCheck(this, Leaderboard);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _LeaderboardAllOf["default"].initialize(this, constraints);

    Leaderboard.initialize(this, id, spaceName, created, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Leaderboard, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, constraints) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>Leaderboard</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Leaderboard} obj Optional instance to populate.
     * @return {module:model/Leaderboard} The populated <code>Leaderboard</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Leaderboard();

        _ModelDefault["default"].constructFromObject(data, obj);

        _LeaderboardAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('leaderboardEntries')) {
          obj['leaderboardEntries'] = _ApiClient["default"].convertToType(data['leaderboardEntries'], [_LeaderboardEntry["default"]]);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('sequence')) {
          obj['sequence'] = _ApiClient["default"].convertToType(data['sequence'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return Leaderboard;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Leaderboard.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Leaderboard.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Leaderboard.prototype['created'] = undefined;
/**
 * @member {Array.<module:model/LeaderboardEntry>} leaderboardEntries
 */

Leaderboard.prototype['leaderboardEntries'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

Leaderboard.prototype['constraints'] = undefined;
/**
 * how many changes were made to teh leaderboard
 * @member {Number} sequence
 */

Leaderboard.prototype['sequence'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement LeaderboardAllOf interface:

/**
 * @member {Array.<module:model/LeaderboardEntry>} leaderboardEntries
 */

_LeaderboardAllOf["default"].prototype['leaderboardEntries'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_LeaderboardAllOf["default"].prototype['constraints'] = undefined;
/**
 * how many changes were made to teh leaderboard
 * @member {Number} sequence
 */

_LeaderboardAllOf["default"].prototype['sequence'] = undefined;
var _default = Leaderboard;
exports["default"] = _default;
