// @ziqni-tech/admin-api-client/dist/model/LeaderboardMember.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The LeaderboardMember model module.
 * @module model/LeaderboardMember
 * @version 0.0.26
 */
var LeaderboardMember = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LeaderboardMember</code>.
   * @alias module:model/LeaderboardMember
   * @param name {String} The name of the player
   * @param memberId {String} Unique system identifier of a Member
   * @param memberRefId {String} The reference to this member in your system
   */
  function LeaderboardMember(name, memberId, memberRefId) {
    _classCallCheck(this, LeaderboardMember);

    LeaderboardMember.initialize(this, name, memberId, memberRefId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LeaderboardMember, null, [{
    key: "initialize",
    value: function initialize(obj, name, memberId, memberRefId) {
      obj['name'] = name;
      obj['memberId'] = memberId;
      obj['memberRefId'] = memberRefId;
    }
    /**
     * Constructs a <code>LeaderboardMember</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LeaderboardMember} obj Optional instance to populate.
     * @return {module:model/LeaderboardMember} The populated <code>LeaderboardMember</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LeaderboardMember();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('rankChangeFrom')) {
          obj['rankChangeFrom'] = _ApiClient["default"].convertToType(data['rankChangeFrom'], 'Number');
        }

        if (data.hasOwnProperty('rankChangeType')) {
          obj['rankChangeType'] = _ApiClient["default"].convertToType(data['rankChangeType'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return LeaderboardMember;
}();
/**
 * The name of the player
 * @member {String} name
 */


LeaderboardMember.prototype['name'] = undefined;
/**
 * Unique system identifier of a Member
 * @member {String} memberId
 */

LeaderboardMember.prototype['memberId'] = undefined;
/**
 * The reference to this member in your system
 * @member {String} memberRefId
 */

LeaderboardMember.prototype['memberRefId'] = undefined;
/**
 * Previous rank held, null if no changes are recorded
 * @member {Number} rankChangeFrom
 */

LeaderboardMember.prototype['rankChangeFrom'] = undefined;
/**
 * [-1] for rank-down, like 5 to 8. [0] for no change, [1] for rank-up like 9 to 3
 * @member {Number} rankChangeType
 */

LeaderboardMember.prototype['rankChangeType'] = undefined;
var _default = LeaderboardMember;
exports["default"] = _default;
