// @ziqni-tech/admin-api-client/dist/model/MemberMessage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EventRefType = _interopRequireDefault(require("./EventRefType"));

var _MemberMessageAllOf = _interopRequireDefault(require("./MemberMessageAllOf"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MemberMessage model module.
 * @module model/MemberMessage
 * @version 0.0.26
 */
var MemberMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemberMessage</code>.
   * This is the Member Message model
   * @alias module:model/MemberMessage
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/MemberMessageAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param eventRefType {module:model/EventRefType} 
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param prize {String} Unique system identifier of an Award
   * @param status {module:model/MessageStatus} 
   * @param expiry {Date} The time that the message will disappear after. ISO8601 timestamp
   * @param memberId {Array.<String>} The reference ID of the event object
   */
  function MemberMessage(id, spaceName, created, eventRefType, messageType, subject, body, prize, status, expiry, memberId) {
    _classCallCheck(this, MemberMessage);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _MemberMessageAllOf["default"].initialize(this, eventRefType, messageType, subject, body, prize, status, expiry, memberId);

    MemberMessage.initialize(this, id, spaceName, created, eventRefType, messageType, subject, body, prize, status, expiry, memberId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MemberMessage, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, eventRefType, messageType, subject, body, prize, status, expiry, memberId) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['eventRefType'] = eventRefType;
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['prize'] = prize;
      obj['status'] = status;
      obj['expiry'] = expiry;
      obj['memberId'] = memberId;
    }
    /**
     * Constructs a <code>MemberMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemberMessage} obj Optional instance to populate.
     * @return {module:model/MemberMessage} The populated <code>MemberMessage</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemberMessage();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _MemberMessageAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _EventRefType["default"].constructFromObject(data['eventRefType']);
        }

        if (data.hasOwnProperty('eventRefId')) {
          obj['eventRefId'] = _ApiClient["default"].convertToType(data['eventRefId'], 'String');
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('prize')) {
          obj['prize'] = _ApiClient["default"].convertToType(data['prize'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('expiry')) {
          obj['expiry'] = _ApiClient["default"].convertToType(data['expiry'], 'Date');
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], ['String']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return MemberMessage;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


MemberMessage.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

MemberMessage.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

MemberMessage.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

MemberMessage.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

MemberMessage.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

MemberMessage.prototype['metadata'] = undefined;
/**
 * @member {module:model/EventRefType} eventRefType
 */

MemberMessage.prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

MemberMessage.prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

MemberMessage.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

MemberMessage.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

MemberMessage.prototype['body'] = undefined;
/**
 * Unique system identifier of an Award
 * @member {String} prize
 */

MemberMessage.prototype['prize'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

MemberMessage.prototype['status'] = undefined;
/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */

MemberMessage.prototype['expiry'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

MemberMessage.prototype['memberId'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

MemberMessage.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

MemberMessage.prototype['translatableFields'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement MemberMessageAllOf interface:

/**
 * @member {module:model/EventRefType} eventRefType
 */

_MemberMessageAllOf["default"].prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

_MemberMessageAllOf["default"].prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

_MemberMessageAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_MemberMessageAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_MemberMessageAllOf["default"].prototype['body'] = undefined;
/**
 * Unique system identifier of an Award
 * @member {String} prize
 */

_MemberMessageAllOf["default"].prototype['prize'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_MemberMessageAllOf["default"].prototype['status'] = undefined;
/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */

_MemberMessageAllOf["default"].prototype['expiry'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

_MemberMessageAllOf["default"].prototype['memberId'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_MemberMessageAllOf["default"].prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

_MemberMessageAllOf["default"].prototype['translatableFields'] = undefined;
var _default = MemberMessage;
exports["default"] = _default;
