// @ziqni-tech/admin-api-client/dist/model/MemberMessageAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EventRefType = _interopRequireDefault(require("./EventRefType"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MemberMessageAllOf model module.
 * @module model/MemberMessageAllOf
 * @version 0.0.26
 */
var MemberMessageAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemberMessageAllOf</code>.
   * @alias module:model/MemberMessageAllOf
   * @param eventRefType {module:model/EventRefType} 
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param prize {String} Unique system identifier of an Award
   * @param status {module:model/MessageStatus} 
   * @param expiry {Date} The time that the message will disappear after. ISO8601 timestamp
   * @param memberId {Array.<String>} The reference ID of the event object
   */
  function MemberMessageAllOf(eventRefType, messageType, subject, body, prize, status, expiry, memberId) {
    _classCallCheck(this, MemberMessageAllOf);

    MemberMessageAllOf.initialize(this, eventRefType, messageType, subject, body, prize, status, expiry, memberId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MemberMessageAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, eventRefType, messageType, subject, body, prize, status, expiry, memberId) {
      obj['eventRefType'] = eventRefType;
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['prize'] = prize;
      obj['status'] = status;
      obj['expiry'] = expiry;
      obj['memberId'] = memberId;
    }
    /**
     * Constructs a <code>MemberMessageAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemberMessageAllOf} obj Optional instance to populate.
     * @return {module:model/MemberMessageAllOf} The populated <code>MemberMessageAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemberMessageAllOf();

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _EventRefType["default"].constructFromObject(data['eventRefType']);
        }

        if (data.hasOwnProperty('eventRefId')) {
          obj['eventRefId'] = _ApiClient["default"].convertToType(data['eventRefId'], 'String');
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('prize')) {
          obj['prize'] = _ApiClient["default"].convertToType(data['prize'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('expiry')) {
          obj['expiry'] = _ApiClient["default"].convertToType(data['expiry'], 'Date');
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], ['String']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return MemberMessageAllOf;
}();
/**
 * @member {module:model/EventRefType} eventRefType
 */


MemberMessageAllOf.prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

MemberMessageAllOf.prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

MemberMessageAllOf.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

MemberMessageAllOf.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

MemberMessageAllOf.prototype['body'] = undefined;
/**
 * Unique system identifier of an Award
 * @member {String} prize
 */

MemberMessageAllOf.prototype['prize'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

MemberMessageAllOf.prototype['status'] = undefined;
/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */

MemberMessageAllOf.prototype['expiry'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

MemberMessageAllOf.prototype['memberId'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

MemberMessageAllOf.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

MemberMessageAllOf.prototype['translatableFields'] = undefined;
var _default = MemberMessageAllOf;
exports["default"] = _default;
