// @ziqni-tech/admin-api-client/dist/model/MemberAchievementCount.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CountResponse = _interopRequireDefault(require("./CountResponse"));

var _MemberAchievementCountAllOf = _interopRequireDefault(require("./MemberAchievementCountAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MemberAchievementCount model module.
 * @module model/MemberAchievementCount
 * @version 0.0.26
 */
var MemberAchievementCount = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemberAchievementCount</code>.
   * This is the Member Achievement Count model
   * @alias module:model/MemberAchievementCount
   * @implements module:model/MemberAchievementCountAllOf
   * @implements module:model/CountResponse
   * @param name {String} A the name of the member
   * @param memberId {String} A unique identifier of a Member
   * @param memberRefId {String} The reference to this member in your system
   * @param latestIssuedTimestamp {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param entityId {String} A unique identifier of a Entity
   * @param countValue {Number} The value in how many times the achievemnt was issued
   */
  function MemberAchievementCount(name, memberId, memberRefId, latestIssuedTimestamp, entityId, countValue) {
    _classCallCheck(this, MemberAchievementCount);

    _MemberAchievementCountAllOf["default"].initialize(this, name, memberId, memberRefId, latestIssuedTimestamp);

    _CountResponse["default"].initialize(this, entityId, countValue);

    MemberAchievementCount.initialize(this, name, memberId, memberRefId, latestIssuedTimestamp, entityId, countValue);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MemberAchievementCount, null, [{
    key: "initialize",
    value: function initialize(obj, name, memberId, memberRefId, latestIssuedTimestamp, entityId, countValue) {
      obj['name'] = name;
      obj['memberId'] = memberId;
      obj['memberRefId'] = memberRefId;
      obj['latestIssuedTimestamp'] = latestIssuedTimestamp;
      obj['entityId'] = entityId;
      obj['countValue'] = countValue;
    }
    /**
     * Constructs a <code>MemberAchievementCount</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemberAchievementCount} obj Optional instance to populate.
     * @return {module:model/MemberAchievementCount} The populated <code>MemberAchievementCount</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemberAchievementCount();

        _MemberAchievementCountAllOf["default"].constructFromObject(data, obj);

        _CountResponse["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('latestIssuedTimestamp')) {
          obj['latestIssuedTimestamp'] = _ApiClient["default"].convertToType(data['latestIssuedTimestamp'], 'Date');
        }

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('countValue')) {
          obj['countValue'] = _ApiClient["default"].convertToType(data['countValue'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return MemberAchievementCount;
}();
/**
 * A the name of the member
 * @member {String} name
 */


MemberAchievementCount.prototype['name'] = undefined;
/**
 * A unique identifier of a Member
 * @member {String} memberId
 */

MemberAchievementCount.prototype['memberId'] = undefined;
/**
 * The reference to this member in your system
 * @member {String} memberRefId
 */

MemberAchievementCount.prototype['memberRefId'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} latestIssuedTimestamp
 */

MemberAchievementCount.prototype['latestIssuedTimestamp'] = undefined;
/**
 * A unique identifier of a Entity
 * @member {String} entityId
 */

MemberAchievementCount.prototype['entityId'] = undefined;
/**
 * The value in how many times the achievemnt was issued
 * @member {Number} countValue
 */

MemberAchievementCount.prototype['countValue'] = undefined; // Implement MemberAchievementCountAllOf interface:

/**
 * A the name of the member
 * @member {String} name
 */

_MemberAchievementCountAllOf["default"].prototype['name'] = undefined;
/**
 * A unique identifier of a Member
 * @member {String} memberId
 */

_MemberAchievementCountAllOf["default"].prototype['memberId'] = undefined;
/**
 * The reference to this member in your system
 * @member {String} memberRefId
 */

_MemberAchievementCountAllOf["default"].prototype['memberRefId'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} latestIssuedTimestamp
 */

_MemberAchievementCountAllOf["default"].prototype['latestIssuedTimestamp'] = undefined; // Implement CountResponse interface:

/**
 * A unique identifier of a Entity
 * @member {String} entityId
 */

_CountResponse["default"].prototype['entityId'] = undefined;
/**
 * The value in how many times the achievemnt was issued
 * @member {Number} countValue
 */

_CountResponse["default"].prototype['countValue'] = undefined;
var _default = MemberAchievementCount;
exports["default"] = _default;
