// @ziqni-tech/admin-api-client/dist/model/MemberAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MemberType = _interopRequireDefault(require("./MemberType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MemberAllOf model module.
 * @module model/MemberAllOf
 * @version 0.0.26
 */
var MemberAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemberAllOf</code>.
   * @alias module:model/MemberAllOf
   * @param name {String} The name of the member that is used on leader boards and public displays
   * @param memberRefId {String} The reference to this member in your system
   * @param memberType {module:model/MemberType} 
   */
  function MemberAllOf(name, memberRefId, memberType) {
    _classCallCheck(this, MemberAllOf);

    MemberAllOf.initialize(this, name, memberRefId, memberType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MemberAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, memberRefId, memberType) {
      obj['name'] = name;
      obj['memberRefId'] = memberRefId;
      obj['memberType'] = memberType;
    }
    /**
     * Constructs a <code>MemberAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemberAllOf} obj Optional instance to populate.
     * @return {module:model/MemberAllOf} The populated <code>MemberAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemberAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('memberType')) {
          obj['memberType'] = _MemberType["default"].constructFromObject(data['memberType']);
        }

        if (data.hasOwnProperty('teamMembers')) {
          obj['teamMembers'] = _ApiClient["default"].convertToType(data['teamMembers'], ['String']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('timeZoneOffset')) {
          obj['timeZoneOffset'] = _ApiClient["default"].convertToType(data['timeZoneOffset'], 'String');
        }
      }

      return obj;
    }
  }]);

  return MemberAllOf;
}();
/**
 * The name of the member that is used on leader boards and public displays
 * @member {String} name
 */


MemberAllOf.prototype['name'] = undefined;
/**
 * The reference to this member in your system
 * @member {String} memberRefId
 */

MemberAllOf.prototype['memberRefId'] = undefined;
/**
 * @member {module:model/MemberType} memberType
 */

MemberAllOf.prototype['memberType'] = undefined;
/**
 * A social group like Guilds
 * @member {Array.<String>} teamMembers
 */

MemberAllOf.prototype['teamMembers'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

MemberAllOf.prototype['constraints'] = undefined;
/**
 * To what time zone teh emmber belongs to
 * @member {String} timeZoneOffset
 */

MemberAllOf.prototype['timeZoneOffset'] = undefined;
var _default = MemberAllOf;
exports["default"] = _default;
