// @ziqni-tech/admin-api-client/dist/model/MemberAchievementCountAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MemberAchievementCountAllOf model module.
 * @module model/MemberAchievementCountAllOf
 * @version 0.0.26
 */
var MemberAchievementCountAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemberAchievementCountAllOf</code>.
   * @alias module:model/MemberAchievementCountAllOf
   * @param name {String} A the name of the member
   * @param memberId {String} A unique identifier of a Member
   * @param memberRefId {String} The reference to this member in your system
   * @param latestIssuedTimestamp {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   */
  function MemberAchievementCountAllOf(name, memberId, memberRefId, latestIssuedTimestamp) {
    _classCallCheck(this, MemberAchievementCountAllOf);

    MemberAchievementCountAllOf.initialize(this, name, memberId, memberRefId, latestIssuedTimestamp);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MemberAchievementCountAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, memberId, memberRefId, latestIssuedTimestamp) {
      obj['name'] = name;
      obj['memberId'] = memberId;
      obj['memberRefId'] = memberRefId;
      obj['latestIssuedTimestamp'] = latestIssuedTimestamp;
    }
    /**
     * Constructs a <code>MemberAchievementCountAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemberAchievementCountAllOf} obj Optional instance to populate.
     * @return {module:model/MemberAchievementCountAllOf} The populated <code>MemberAchievementCountAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemberAchievementCountAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], 'String');
        }

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('latestIssuedTimestamp')) {
          obj['latestIssuedTimestamp'] = _ApiClient["default"].convertToType(data['latestIssuedTimestamp'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return MemberAchievementCountAllOf;
}();
/**
 * A the name of the member
 * @member {String} name
 */


MemberAchievementCountAllOf.prototype['name'] = undefined;
/**
 * A unique identifier of a Member
 * @member {String} memberId
 */

MemberAchievementCountAllOf.prototype['memberId'] = undefined;
/**
 * The reference to this member in your system
 * @member {String} memberRefId
 */

MemberAchievementCountAllOf.prototype['memberRefId'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} latestIssuedTimestamp
 */

MemberAchievementCountAllOf.prototype['latestIssuedTimestamp'] = undefined;
var _default = MemberAchievementCountAllOf;
exports["default"] = _default;
