// @ziqni-tech/admin-api-client/dist/model/Message.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

var _MessageAllOf = _interopRequireDefault(require("./MessageAllOf"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _Reward = _interopRequireDefault(require("./Reward"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Message model module.
 * @module model/Message
 * @version 0.0.5
 */
var Message = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Message</code>.
   * @alias module:model/Message
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/MessageAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param status {module:model/MessageStatus} 
   * @param scheduling {module:model/Scheduling} 
   * @param constraints {Array.<String>} Additional constraints, if the value is present it means the
   */
  function Message(id, spaceName, created, messageType, subject, body, status, scheduling, constraints) {
    _classCallCheck(this, Message);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _MessageAllOf["default"].initialize(this, messageType, subject, body, status, scheduling, constraints);

    Message.initialize(this, id, spaceName, created, messageType, subject, body, status, scheduling, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Message, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, messageType, subject, body, status, scheduling, constraints) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['status'] = status;
      obj['scheduling'] = scheduling;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>Message</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Message} obj Optional instance to populate.
     * @return {module:model/Message} The populated <code>Message</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Message();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _MessageAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _ApiClient["default"].convertToType(data['dependantOn'], [_Dependancy["default"]]);
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }

        if (data.hasOwnProperty('prize')) {
          obj['prize'] = _Reward["default"].constructFromObject(data['prize']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('eventReferenceId')) {
          obj['eventReferenceId'] = _ApiClient["default"].convertToType(data['eventReferenceId'], 'String');
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _ApiClient["default"].convertToType(data['eventRefType'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Message;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Message.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Message.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Message.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Message.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

Message.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Message.prototype['metadata'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

Message.prototype['dependantOn'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

Message.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

Message.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

Message.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

Message.prototype['status'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

Message.prototype['scheduling'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

Message.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

Message.prototype['translatableFields'] = undefined;
/**
 * @member {module:model/Reward} prize
 */

Message.prototype['prize'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

Message.prototype['constraints'] = undefined;
/**
 * @member {String} eventReferenceId
 */

Message.prototype['eventReferenceId'] = undefined;
/**
 * @member {String} eventRefType
 */

Message.prototype['eventRefType'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement MessageAllOf interface:

/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

_MessageAllOf["default"].prototype['dependantOn'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

_MessageAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_MessageAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_MessageAllOf["default"].prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_MessageAllOf["default"].prototype['status'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

_MessageAllOf["default"].prototype['scheduling'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_MessageAllOf["default"].prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

_MessageAllOf["default"].prototype['translatableFields'] = undefined;
/**
 * @member {module:model/Reward} prize
 */

_MessageAllOf["default"].prototype['prize'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

_MessageAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {String} eventReferenceId
 */

_MessageAllOf["default"].prototype['eventReferenceId'] = undefined;
/**
 * @member {String} eventRefType
 */

_MessageAllOf["default"].prototype['eventRefType'] = undefined;
var _default = Message;
exports["default"] = _default;
