// @ziqni-tech/admin-api-client/dist/model/MetaInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MetaInfo model module.
 * @module model/MetaInfo
 * @version 0.0.26
 */
var MetaInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MetaInfo</code>.
   * @alias module:model/MetaInfo
   * @param objectType {String} 
   * @param resultCount {Number} The count of successful results
   * @param errorCount {Number} The count of errors
   */
  function MetaInfo(objectType, resultCount, errorCount) {
    _classCallCheck(this, MetaInfo);

    MetaInfo.initialize(this, objectType, resultCount, errorCount);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MetaInfo, null, [{
    key: "initialize",
    value: function initialize(obj, objectType, resultCount, errorCount) {
      obj['objectType'] = objectType;
      obj['resultCount'] = resultCount;
      obj['errorCount'] = errorCount;
    }
    /**
     * Constructs a <code>MetaInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MetaInfo} obj Optional instance to populate.
     * @return {module:model/MetaInfo} The populated <code>MetaInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MetaInfo();

        if (data.hasOwnProperty('objectType')) {
          obj['objectType'] = _ApiClient["default"].convertToType(data['objectType'], 'String');
        }

        if (data.hasOwnProperty('totalRecords')) {
          obj['totalRecords'] = _ApiClient["default"].convertToType(data['totalRecords'], 'Number');
        }

        if (data.hasOwnProperty('resultCount')) {
          obj['resultCount'] = _ApiClient["default"].convertToType(data['resultCount'], 'Number');
        }

        if (data.hasOwnProperty('errorCount')) {
          obj['errorCount'] = _ApiClient["default"].convertToType(data['errorCount'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return MetaInfo;
}();
/**
 * @member {String} objectType
 */


MetaInfo.prototype['objectType'] = undefined;
/**
 * The count of all results
 * @member {Number} totalRecords
 */

MetaInfo.prototype['totalRecords'] = undefined;
/**
 * The count of successful results
 * @member {Number} resultCount
 */

MetaInfo.prototype['resultCount'] = undefined;
/**
 * The count of errors
 * @member {Number} errorCount
 */

MetaInfo.prototype['errorCount'] = undefined;
var _default = MetaInfo;
exports["default"] = _default;
