// @ziqni-tech/admin-api-client/dist/model/ProductReducedAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ProductReducedAllOf model module.
 * @module model/ProductReducedAllOf
 * @version 0.0.26
 */
var ProductReducedAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProductReducedAllOf</code>.
   * @alias module:model/ProductReducedAllOf
   * @param productRefId {String} The reference to this product in your system. The reference identifier can not be changed after the product has been created
   */
  function ProductReducedAllOf(productRefId) {
    _classCallCheck(this, ProductReducedAllOf);

    ProductReducedAllOf.initialize(this, productRefId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProductReducedAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, productRefId) {
      obj['productRefId'] = productRefId;
    }
    /**
     * Constructs a <code>ProductReducedAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProductReducedAllOf} obj Optional instance to populate.
     * @return {module:model/ProductReducedAllOf} The populated <code>ProductReducedAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProductReducedAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('productRefId')) {
          obj['productRefId'] = _ApiClient["default"].convertToType(data['productRefId'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return ProductReducedAllOf;
}();
/**
 * The name of the product
 * @member {String} name
 */


ProductReducedAllOf.prototype['name'] = undefined;
/**
 * The reference to this product in your system. The reference identifier can not be changed after the product has been created
 * @member {String} productRefId
 */

ProductReducedAllOf.prototype['productRefId'] = undefined;
/**
 * A list of Strings used to tag products with taxonomy terms
 * @member {Array.<String>} tags
 */

ProductReducedAllOf.prototype['tags'] = undefined;
var _default = ProductReducedAllOf;
exports["default"] = _default;
