// @ziqni-tech/admin-api-client/dist/model/Product.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActionTypeAdjustmentFactor = _interopRequireDefault(require("./ActionTypeAdjustmentFactor"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _ProductAllOf = _interopRequireDefault(require("./ProductAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Product model module.
 * @module model/Product
 * @version 0.0.26
 */
var Product = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Product</code>.
   * This is the Product model
   * @alias module:model/Product
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/ProductAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} The name of the product
   * @param adjustmentFactor {Number} The multiplier to apply to source values received for this product events
   * @param productRefId {String} The reference to this product in your system. The reference identifier can not be changed after the product has been created
   */
  function Product(id, spaceName, created, name, adjustmentFactor, productRefId) {
    _classCallCheck(this, Product);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _ProductAllOf["default"].initialize(this, name, adjustmentFactor, productRefId);

    Product.initialize(this, id, spaceName, created, name, adjustmentFactor, productRefId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Product, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, adjustmentFactor, productRefId) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['adjustmentFactor'] = adjustmentFactor;
      obj['productRefId'] = productRefId;
    }
    /**
     * Constructs a <code>Product</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Product} obj Optional instance to populate.
     * @return {module:model/Product} The populated <code>Product</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Product();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _ProductAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('adjustmentFactor')) {
          obj['adjustmentFactor'] = _ApiClient["default"].convertToType(data['adjustmentFactor'], 'Number');
        }

        if (data.hasOwnProperty('productRefId')) {
          obj['productRefId'] = _ApiClient["default"].convertToType(data['productRefId'], 'String');
        }

        if (data.hasOwnProperty('actionTypeAdjustmentFactors')) {
          obj['actionTypeAdjustmentFactors'] = _ApiClient["default"].convertToType(data['actionTypeAdjustmentFactors'], [_ActionTypeAdjustmentFactor["default"]]);
        }
      }

      return obj;
    }
  }]);

  return Product;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Product.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Product.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Product.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Product.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

Product.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Product.prototype['metadata'] = undefined;
/**
 * The name of the product
 * @member {String} name
 */

Product.prototype['name'] = undefined;
/**
 * The description of the product for your reference
 * @member {String} description
 */

Product.prototype['description'] = undefined;
/**
 * The multiplier to apply to source values received for this product events
 * @member {Number} adjustmentFactor
 */

Product.prototype['adjustmentFactor'] = undefined;
/**
 * The reference to this product in your system. The reference identifier can not be changed after the product has been created
 * @member {String} productRefId
 */

Product.prototype['productRefId'] = undefined;
/**
 * @member {Array.<module:model/ActionTypeAdjustmentFactor>} actionTypeAdjustmentFactors
 */

Product.prototype['actionTypeAdjustmentFactors'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement ProductAllOf interface:

/**
 * The name of the product
 * @member {String} name
 */

_ProductAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the product for your reference
 * @member {String} description
 */

_ProductAllOf["default"].prototype['description'] = undefined;
/**
 * The multiplier to apply to source values received for this product events
 * @member {Number} adjustmentFactor
 */

_ProductAllOf["default"].prototype['adjustmentFactor'] = undefined;
/**
 * The reference to this product in your system. The reference identifier can not be changed after the product has been created
 * @member {String} productRefId
 */

_ProductAllOf["default"].prototype['productRefId'] = undefined;
/**
 * @member {Array.<module:model/ActionTypeAdjustmentFactor>} actionTypeAdjustmentFactors
 */

_ProductAllOf["default"].prototype['actionTypeAdjustmentFactors'] = undefined;
var _default = Product;
exports["default"] = _default;
