// @ziqni-tech/admin-api-client/dist/model/ProfileResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Error = _interopRequireDefault(require("./Error"));

var _MetaExtended = _interopRequireDefault(require("./MetaExtended"));

var _User = _interopRequireDefault(require("./User"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ProfileResponse model module.
 * @module model/ProfileResponse
 * @version 0.0.26
 */
var ProfileResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProfileResponse</code>.
   * @alias module:model/ProfileResponse
   * @param meta {module:model/MetaExtended} 
   */
  function ProfileResponse(meta) {
    _classCallCheck(this, ProfileResponse);

    ProfileResponse.initialize(this, meta);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProfileResponse, null, [{
    key: "initialize",
    value: function initialize(obj, meta) {
      obj['meta'] = meta;
    }
    /**
     * Constructs a <code>ProfileResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProfileResponse} obj Optional instance to populate.
     * @return {module:model/ProfileResponse} The populated <code>ProfileResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProfileResponse();

        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _MetaExtended["default"].constructFromObject(data['meta']);
        }

        if (data.hasOwnProperty('result')) {
          obj['result'] = _User["default"].constructFromObject(data['result']);
        }

        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], [_Error["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ProfileResponse;
}();
/**
 * @member {module:model/MetaExtended} meta
 */


ProfileResponse.prototype['meta'] = undefined;
/**
 * @member {module:model/User} result
 */

ProfileResponse.prototype['result'] = undefined;
/**
 * @member {Array.<module:model/Error>} errors
 */

ProfileResponse.prototype['errors'] = undefined;
var _default = ProfileResponse;
exports["default"] = _default;
