// @ziqni-tech/admin-api-client/dist/model/QueryRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _QueryMultiple = _interopRequireDefault(require("./QueryMultiple"));

var _QueryMultipleFields = _interopRequireDefault(require("./QueryMultipleFields"));

var _QuerySingle = _interopRequireDefault(require("./QuerySingle"));

var _QuerySortBy = _interopRequireDefault(require("./QuerySortBy"));

var _RangeQuery = _interopRequireDefault(require("./RangeQuery"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The QueryRequest model module.
 * @module model/QueryRequest
 * @version 0.0.26
 */
var QueryRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>QueryRequest</code>.
   * @alias module:model/QueryRequest
   */
  function QueryRequest() {
    _classCallCheck(this, QueryRequest);

    QueryRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(QueryRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>QueryRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/QueryRequest} obj Optional instance to populate.
     * @return {module:model/QueryRequest} The populated <code>QueryRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new QueryRequest();

        if (data.hasOwnProperty('must')) {
          obj['must'] = _ApiClient["default"].convertToType(data['must'], [_QueryMultiple["default"]]);
        }

        if (data.hasOwnProperty('mustNot')) {
          obj['mustNot'] = _ApiClient["default"].convertToType(data['mustNot'], [_QueryMultiple["default"]]);
        }

        if (data.hasOwnProperty('should')) {
          obj['should'] = _ApiClient["default"].convertToType(data['should'], [_QueryMultiple["default"]]);
        }

        if (data.hasOwnProperty('gte')) {
          obj['gte'] = _ApiClient["default"].convertToType(data['gte'], [_QuerySingle["default"]]);
        }

        if (data.hasOwnProperty('gt')) {
          obj['gt'] = _ApiClient["default"].convertToType(data['gt'], [_QuerySingle["default"]]);
        }

        if (data.hasOwnProperty('lte')) {
          obj['lte'] = _ApiClient["default"].convertToType(data['lte'], [_QuerySingle["default"]]);
        }

        if (data.hasOwnProperty('lt')) {
          obj['lt'] = _ApiClient["default"].convertToType(data['lt'], [_QuerySingle["default"]]);
        }

        if (data.hasOwnProperty('range')) {
          obj['range'] = _ApiClient["default"].convertToType(data['range'], [_RangeQuery["default"]]);
        }

        if (data.hasOwnProperty('sortBy')) {
          obj['sortBy'] = _ApiClient["default"].convertToType(data['sortBy'], [_QuerySortBy["default"]]);
        }

        if (data.hasOwnProperty('multiFields')) {
          obj['multiFields'] = _ApiClient["default"].convertToType(data['multiFields'], [_QueryMultipleFields["default"]]);
        }

        if (data.hasOwnProperty('includeFields')) {
          obj['includeFields'] = _ApiClient["default"].convertToType(data['includeFields'], ['String']);
        }

        if (data.hasOwnProperty('hasValue')) {
          obj['hasValue'] = _ApiClient["default"].convertToType(data['hasValue'], ['String']);
        }

        if (data.hasOwnProperty('hasNoValue')) {
          obj['hasNoValue'] = _ApiClient["default"].convertToType(data['hasNoValue'], ['String']);
        }

        if (data.hasOwnProperty('shouldMatch')) {
          obj['shouldMatch'] = _ApiClient["default"].convertToType(data['shouldMatch'], 'Number');
        }

        if (data.hasOwnProperty('skip')) {
          obj['skip'] = _ApiClient["default"].convertToType(data['skip'], 'Number');
        }

        if (data.hasOwnProperty('limit')) {
          obj['limit'] = _ApiClient["default"].convertToType(data['limit'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return QueryRequest;
}();
/**
 * @member {Array.<module:model/QueryMultiple>} must
 */


QueryRequest.prototype['must'] = undefined;
/**
 * @member {Array.<module:model/QueryMultiple>} mustNot
 */

QueryRequest.prototype['mustNot'] = undefined;
/**
 * @member {Array.<module:model/QueryMultiple>} should
 */

QueryRequest.prototype['should'] = undefined;
/**
 * @member {Array.<module:model/QuerySingle>} gte
 */

QueryRequest.prototype['gte'] = undefined;
/**
 * @member {Array.<module:model/QuerySingle>} gt
 */

QueryRequest.prototype['gt'] = undefined;
/**
 * @member {Array.<module:model/QuerySingle>} lte
 */

QueryRequest.prototype['lte'] = undefined;
/**
 * @member {Array.<module:model/QuerySingle>} lt
 */

QueryRequest.prototype['lt'] = undefined;
/**
 * @member {Array.<module:model/RangeQuery>} range
 */

QueryRequest.prototype['range'] = undefined;
/**
 * @member {Array.<module:model/QuerySortBy>} sortBy
 */

QueryRequest.prototype['sortBy'] = undefined;
/**
 * @member {Array.<module:model/QueryMultipleFields>} multiFields
 */

QueryRequest.prototype['multiFields'] = undefined;
/**
 * @member {Array.<String>} includeFields
 */

QueryRequest.prototype['includeFields'] = undefined;
/**
 * @member {Array.<String>} hasValue
 */

QueryRequest.prototype['hasValue'] = undefined;
/**
 * @member {Array.<String>} hasNoValue
 */

QueryRequest.prototype['hasNoValue'] = undefined;
/**
 * @member {Number} shouldMatch
 */

QueryRequest.prototype['shouldMatch'] = undefined;
/**
 * @member {Number} skip
 */

QueryRequest.prototype['skip'] = undefined;
/**
 * @member {Number} limit
 */

QueryRequest.prototype['limit'] = undefined;
/**
 * @member {Array.<String>} constraints
 */

QueryRequest.prototype['constraints'] = undefined;
var _default = QueryRequest;
exports["default"] = _default;
