// @ziqni-tech/admin-api-client/dist/model/RabbitMqConnectionAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RabbitMqConnectionAllOf model module.
 * @module model/RabbitMqConnectionAllOf
 * @version 0.0.26
 */
var RabbitMqConnectionAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RabbitMqConnectionAllOf</code>.
   * @alias module:model/RabbitMqConnectionAllOf
   * @param name {String} The name of the consumer
   * @param uri {String} The SQS endpoint.
   * @param virtualHost {String} The virtual host of the rabbitmq broker
   * @param port {Number} The port number on which consumer will connect on rabbitmq broker
   * @param userName {String} Consumer username for authentication
   * @param password {String} Consumer password for authentication
   * @param queueName {String} Name of the queue
   * @param lastKnownStatus {String} Last known status of the connection
   * @param lastKnownStatusCode {Number} Status code correspoding to the last known status
   * @param constraints {Array.<String>} Additional constraints
   * @param transformerId {String} The transformer to use, if empty the default system transformer will be used
   */
  function RabbitMqConnectionAllOf(name, uri, virtualHost, port, userName, password, queueName, lastKnownStatus, lastKnownStatusCode, constraints, transformerId) {
    _classCallCheck(this, RabbitMqConnectionAllOf);

    RabbitMqConnectionAllOf.initialize(this, name, uri, virtualHost, port, userName, password, queueName, lastKnownStatus, lastKnownStatusCode, constraints, transformerId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RabbitMqConnectionAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, uri, virtualHost, port, userName, password, queueName, lastKnownStatus, lastKnownStatusCode, constraints, transformerId) {
      obj['name'] = name;
      obj['uri'] = uri;
      obj['virtualHost'] = virtualHost;
      obj['port'] = port;
      obj['userName'] = userName;
      obj['password'] = password;
      obj['queueName'] = queueName;
      obj['lastKnownStatus'] = lastKnownStatus;
      obj['lastKnownStatusCode'] = lastKnownStatusCode;
      obj['constraints'] = constraints;
      obj['transformerId'] = transformerId;
    }
    /**
     * Constructs a <code>RabbitMqConnectionAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RabbitMqConnectionAllOf} obj Optional instance to populate.
     * @return {module:model/RabbitMqConnectionAllOf} The populated <code>RabbitMqConnectionAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RabbitMqConnectionAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }

        if (data.hasOwnProperty('virtualHost')) {
          obj['virtualHost'] = _ApiClient["default"].convertToType(data['virtualHost'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'Number');
        }

        if (data.hasOwnProperty('userName')) {
          obj['userName'] = _ApiClient["default"].convertToType(data['userName'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('queueName')) {
          obj['queueName'] = _ApiClient["default"].convertToType(data['queueName'], 'String');
        }

        if (data.hasOwnProperty('exchange')) {
          obj['exchange'] = _ApiClient["default"].convertToType(data['exchange'], 'String');
        }

        if (data.hasOwnProperty('routingKey')) {
          obj['routingKey'] = _ApiClient["default"].convertToType(data['routingKey'], 'String');
        }

        if (data.hasOwnProperty('lastKnownStatus')) {
          obj['lastKnownStatus'] = _ApiClient["default"].convertToType(data['lastKnownStatus'], 'String');
        }

        if (data.hasOwnProperty('lastKnownStatusCode')) {
          obj['lastKnownStatusCode'] = _ApiClient["default"].convertToType(data['lastKnownStatusCode'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }

        if (data.hasOwnProperty('connectionType')) {
          obj['connectionType'] = _ApiClient["default"].convertToType(data['connectionType'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RabbitMqConnectionAllOf;
}();
/**
 * The name of the consumer
 * @member {String} name
 */


RabbitMqConnectionAllOf.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

RabbitMqConnectionAllOf.prototype['description'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

RabbitMqConnectionAllOf.prototype['uri'] = undefined;
/**
 * The virtual host of the rabbitmq broker
 * @member {String} virtualHost
 */

RabbitMqConnectionAllOf.prototype['virtualHost'] = undefined;
/**
 * The port number on which consumer will connect on rabbitmq broker
 * @member {Number} port
 */

RabbitMqConnectionAllOf.prototype['port'] = undefined;
/**
 * Consumer username for authentication
 * @member {String} userName
 */

RabbitMqConnectionAllOf.prototype['userName'] = undefined;
/**
 * Consumer password for authentication
 * @member {String} password
 */

RabbitMqConnectionAllOf.prototype['password'] = undefined;
/**
 * Name of the queue
 * @member {String} queueName
 */

RabbitMqConnectionAllOf.prototype['queueName'] = undefined;
/**
 * Exchange to connect to
 * @member {String} exchange
 */

RabbitMqConnectionAllOf.prototype['exchange'] = undefined;
/**
 * Routing key to use
 * @member {String} routingKey
 */

RabbitMqConnectionAllOf.prototype['routingKey'] = undefined;
/**
 * Last known status of the connection
 * @member {String} lastKnownStatus
 */

RabbitMqConnectionAllOf.prototype['lastKnownStatus'] = undefined;
/**
 * Status code correspoding to the last known status
 * @member {Number} lastKnownStatusCode
 */

RabbitMqConnectionAllOf.prototype['lastKnownStatusCode'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

RabbitMqConnectionAllOf.prototype['constraints'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

RabbitMqConnectionAllOf.prototype['transformerId'] = undefined;
/**
 * The type of connection
 * @member {String} connectionType
 */

RabbitMqConnectionAllOf.prototype['connectionType'] = undefined;
var _default = RabbitMqConnectionAllOf;
exports["default"] = _default;
