// @ziqni-tech/admin-api-client/dist/model/Repository.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _HostingOptions = _interopRequireDefault(require("./HostingOptions"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _RepositoryAllOf = _interopRequireDefault(require("./RepositoryAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Repository model module.
 * @module model/Repository
 * @version 0.0.26
 */
var Repository = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Repository</code>.
   * @alias module:model/Repository
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/RepositoryAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} The name of the repository. This cannot contain spaces or specil characters.
   * @param constraints {Array.<String>} Additional constraints
   * @param hostingOptions {module:model/HostingOptions} 
   */
  function Repository(id, spaceName, created, name, constraints, hostingOptions) {
    _classCallCheck(this, Repository);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _RepositoryAllOf["default"].initialize(this, name, constraints, hostingOptions);

    Repository.initialize(this, id, spaceName, created, name, constraints, hostingOptions);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Repository, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, constraints, hostingOptions) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['constraints'] = constraints;
      obj['hostingOptions'] = hostingOptions;
    }
    /**
     * Constructs a <code>Repository</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Repository} obj Optional instance to populate.
     * @return {module:model/Repository} The populated <code>Repository</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Repository();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _RepositoryAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('hostingOptions')) {
          obj['hostingOptions'] = _HostingOptions["default"].constructFromObject(data['hostingOptions']);
        }
      }

      return obj;
    }
  }]);

  return Repository;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Repository.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Repository.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Repository.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Repository.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

Repository.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Repository.prototype['metadata'] = undefined;
/**
 * The name of the repository. This cannot contain spaces or specil characters.
 * @member {String} name
 */

Repository.prototype['name'] = undefined;
/**
 * The description of the promotion page
 * @member {String} description
 */

Repository.prototype['description'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

Repository.prototype['constraints'] = undefined;
/**
 * @member {module:model/HostingOptions} hostingOptions
 */

Repository.prototype['hostingOptions'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement RepositoryAllOf interface:

/**
 * The name of the repository. This cannot contain spaces or specil characters.
 * @member {String} name
 */

_RepositoryAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the promotion page
 * @member {String} description
 */

_RepositoryAllOf["default"].prototype['description'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_RepositoryAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {module:model/HostingOptions} hostingOptions
 */

_RepositoryAllOf["default"].prototype['hostingOptions'] = undefined;
var _default = Repository;
exports["default"] = _default;
