// @ziqni-tech/admin-api-client/dist/model/Result.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Relation = _interopRequireDefault(require("./Relation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Result model module.
 * @module model/Result
 * @version 0.0.26
 */
var Result = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Result</code>.
   * @alias module:model/Result
   * @param id {String} A unique system identifier of an updated record
   * @param result {String} The response of an action Possible values:   * created: Will be returned for a POST action to create entity   * updated: Will be returned for a PUT action to update entity   * deleted: Will be returned for a DELETE action on an entity 
   */
  function Result(id, result) {
    _classCallCheck(this, Result);

    Result.initialize(this, id, result);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Result, null, [{
    key: "initialize",
    value: function initialize(obj, id, result) {
      obj['id'] = id;
      obj['result'] = result;
    }
    /**
     * Constructs a <code>Result</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Result} obj Optional instance to populate.
     * @return {module:model/Result} The populated <code>Result</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Result();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('externalReference')) {
          obj['externalReference'] = _ApiClient["default"].convertToType(data['externalReference'], 'String');
        }

        if (data.hasOwnProperty('result')) {
          obj['result'] = _ApiClient["default"].convertToType(data['result'], 'String');
        }

        if (data.hasOwnProperty('relations')) {
          obj['relations'] = _ApiClient["default"].convertToType(data['relations'], [_Relation["default"]]);
        }
      }

      return obj;
    }
  }]);

  return Result;
}();
/**
 * A unique system identifier of an updated record
 * @member {String} id
 */


Result.prototype['id'] = undefined;
/**
 * A reference id from external system
 * @member {String} externalReference
 */

Result.prototype['externalReference'] = undefined;
/**
 * The response of an action Possible values:   * created: Will be returned for a POST action to create entity   * updated: Will be returned for a PUT action to update entity   * deleted: Will be returned for a DELETE action on an entity 
 * @member {String} result
 */

Result.prototype['result'] = undefined;
/**
 * @member {Array.<module:model/Relation>} relations
 */

Result.prototype['relations'] = undefined;
var _default = Result;
exports["default"] = _default;
