// @ziqni-tech/admin-api-client/dist/model/RewardTypeReduced.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _RewardTypeReducedAllOf = _interopRequireDefault(require("./RewardTypeReducedAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RewardTypeReduced model module.
 * @module model/RewardTypeReduced
 * @version 0.0.26
 */
var RewardTypeReduced = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RewardTypeReduced</code>.
   * @alias module:model/RewardTypeReduced
   * @implements module:model/ModelDefault
   * @implements module:model/RewardTypeReducedAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param key {String} A unique key that represents the reward type
   */
  function RewardTypeReduced(id, spaceName, created, key) {
    _classCallCheck(this, RewardTypeReduced);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _RewardTypeReducedAllOf["default"].initialize(this, key);

    RewardTypeReduced.initialize(this, id, spaceName, created, key);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RewardTypeReduced, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, key) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['key'] = key;
    }
    /**
     * Constructs a <code>RewardTypeReduced</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RewardTypeReduced} obj Optional instance to populate.
     * @return {module:model/RewardTypeReduced} The populated <code>RewardTypeReduced</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RewardTypeReduced();

        _ModelDefault["default"].constructFromObject(data, obj);

        _RewardTypeReducedAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RewardTypeReduced;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


RewardTypeReduced.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

RewardTypeReduced.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

RewardTypeReduced.prototype['created'] = undefined;
/**
 * A unique key that represents the reward type
 * @member {String} key
 */

RewardTypeReduced.prototype['key'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement RewardTypeReducedAllOf interface:

/**
 * A unique key that represents the reward type
 * @member {String} key
 */

_RewardTypeReducedAllOf["default"].prototype['key'] = undefined;
var _default = RewardTypeReduced;
exports["default"] = _default;
