// @ziqni-tech/admin-api-client/dist/model/RewardReduced.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _RewardReducedAllOf = _interopRequireDefault(require("./RewardReducedAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RewardReduced model module.
 * @module model/RewardReduced
 * @version 0.0.26
 */
var RewardReduced = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RewardReduced</code>.
   * @alias module:model/RewardReduced
   * @implements module:model/ModelDefault
   * @implements module:model/RewardReducedAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param rewardRank {String} If used in the context of contest this will associate with the rank of the leaderboard
   * @param name {String} The name of a reward
   * @param rewardValue {Number} Numerical value of the reward that will be issued based on the reward type
   * @param rewardType {String} Reward Type key
   * @param rewardTypeId {String} A unique id of the Reward Type
   */
  function RewardReduced(id, spaceName, created, rewardRank, name, rewardValue, rewardType, rewardTypeId) {
    _classCallCheck(this, RewardReduced);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _RewardReducedAllOf["default"].initialize(this, rewardRank, name, rewardValue, rewardType, rewardTypeId);

    RewardReduced.initialize(this, id, spaceName, created, rewardRank, name, rewardValue, rewardType, rewardTypeId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RewardReduced, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, rewardRank, name, rewardValue, rewardType, rewardTypeId) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['rewardRank'] = rewardRank;
      obj['name'] = name;
      obj['rewardValue'] = rewardValue;
      obj['rewardType'] = rewardType;
      obj['rewardTypeId'] = rewardTypeId;
    }
    /**
     * Constructs a <code>RewardReduced</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RewardReduced} obj Optional instance to populate.
     * @return {module:model/RewardReduced} The populated <code>RewardReduced</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RewardReduced();

        _ModelDefault["default"].constructFromObject(data, obj);

        _RewardReducedAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('rewardRank')) {
          obj['rewardRank'] = _ApiClient["default"].convertToType(data['rewardRank'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('rewardValue')) {
          obj['rewardValue'] = _ApiClient["default"].convertToType(data['rewardValue'], 'Number');
        }

        if (data.hasOwnProperty('rewardType')) {
          obj['rewardType'] = _ApiClient["default"].convertToType(data['rewardType'], 'String');
        }

        if (data.hasOwnProperty('rewardTypeId')) {
          obj['rewardTypeId'] = _ApiClient["default"].convertToType(data['rewardTypeId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RewardReduced;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


RewardReduced.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

RewardReduced.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

RewardReduced.prototype['created'] = undefined;
/**
 * If used in the context of contest this will associate with the rank of the leaderboard
 * @member {String} rewardRank
 */

RewardReduced.prototype['rewardRank'] = undefined;
/**
 * The name of a reward
 * @member {String} name
 */

RewardReduced.prototype['name'] = undefined;
/**
 * Numerical value of the reward that will be issued based on the reward type
 * @member {Number} rewardValue
 */

RewardReduced.prototype['rewardValue'] = undefined;
/**
 * Reward Type key
 * @member {String} rewardType
 */

RewardReduced.prototype['rewardType'] = undefined;
/**
 * A unique id of the Reward Type
 * @member {String} rewardTypeId
 */

RewardReduced.prototype['rewardTypeId'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement RewardReducedAllOf interface:

/**
 * If used in the context of contest this will associate with the rank of the leaderboard
 * @member {String} rewardRank
 */

_RewardReducedAllOf["default"].prototype['rewardRank'] = undefined;
/**
 * The name of a reward
 * @member {String} name
 */

_RewardReducedAllOf["default"].prototype['name'] = undefined;
/**
 * Numerical value of the reward that will be issued based on the reward type
 * @member {Number} rewardValue
 */

_RewardReducedAllOf["default"].prototype['rewardValue'] = undefined;
/**
 * Reward Type key
 * @member {String} rewardType
 */

_RewardReducedAllOf["default"].prototype['rewardType'] = undefined;
/**
 * A unique id of the Reward Type
 * @member {String} rewardTypeId
 */

_RewardReducedAllOf["default"].prototype['rewardTypeId'] = undefined;
var _default = RewardReduced;
exports["default"] = _default;
