// @ziqni-tech/admin-api-client/dist/model/RewardAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EntityType = _interopRequireDefault(require("./EntityType"));

var _RewardTypeReduced = _interopRequireDefault(require("./RewardTypeReduced"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RewardAllOf model module.
 * @module model/RewardAllOf
 * @version 0.0.26
 */
var RewardAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RewardAllOf</code>.
   * @alias module:model/RewardAllOf
   * @param rewardRank {String} If used in the context of contest this will associate with the rank of the leaderboard
   * @param rewardValue {Number} Numerical value of the reward that will be issued based on the reward type
   * @param rewardType {module:model/RewardTypeReduced} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function RewardAllOf(rewardRank, rewardValue, rewardType, constraints) {
    _classCallCheck(this, RewardAllOf);

    RewardAllOf.initialize(this, rewardRank, rewardValue, rewardType, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RewardAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, rewardRank, rewardValue, rewardType, constraints) {
      obj['rewardRank'] = rewardRank;
      obj['rewardValue'] = rewardValue;
      obj['rewardType'] = rewardType;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>RewardAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RewardAllOf} obj Optional instance to populate.
     * @return {module:model/RewardAllOf} The populated <code>RewardAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RewardAllOf();

        if (data.hasOwnProperty('entityType')) {
          obj['entityType'] = _EntityType["default"].constructFromObject(data['entityType']);
        }

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('rewardRank')) {
          obj['rewardRank'] = _ApiClient["default"].convertToType(data['rewardRank'], 'String');
        }

        if (data.hasOwnProperty('rewardValue')) {
          obj['rewardValue'] = _ApiClient["default"].convertToType(data['rewardValue'], 'Number');
        }

        if (data.hasOwnProperty('rewardType')) {
          obj['rewardType'] = _RewardTypeReduced["default"].constructFromObject(data['rewardType']);
        }

        if (data.hasOwnProperty('icon')) {
          obj['icon'] = _ApiClient["default"].convertToType(data['icon'], 'String');
        }

        if (data.hasOwnProperty('issueLimit')) {
          obj['issueLimit'] = _ApiClient["default"].convertToType(data['issueLimit'], 'Number');
        }

        if (data.hasOwnProperty('delay')) {
          obj['delay'] = _ApiClient["default"].convertToType(data['delay'], 'Number');
        }

        if (data.hasOwnProperty('pointInTime')) {
          obj['pointInTime'] = _ApiClient["default"].convertToType(data['pointInTime'], 'Date');
        }

        if (data.hasOwnProperty('period')) {
          obj['period'] = _ApiClient["default"].convertToType(data['period'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return RewardAllOf;
}();
/**
 * @member {module:model/EntityType} entityType
 */


RewardAllOf.prototype['entityType'] = undefined;
/**
 * A unique identifier of an achievement or contest. Dependant on entityType
 * @member {String} entityId
 */

RewardAllOf.prototype['entityId'] = undefined;
/**
 * The name of a reward
 * @member {String} name
 */

RewardAllOf.prototype['name'] = undefined;
/**
 * The description of a Reward
 * @member {String} description
 */

RewardAllOf.prototype['description'] = undefined;
/**
 * If used in the context of contest this will associate with the rank of the leaderboard
 * @member {String} rewardRank
 */

RewardAllOf.prototype['rewardRank'] = undefined;
/**
 * Numerical value of the reward that will be issued based on the reward type
 * @member {Number} rewardValue
 */

RewardAllOf.prototype['rewardValue'] = undefined;
/**
 * @member {module:model/RewardTypeReduced} rewardType
 */

RewardAllOf.prototype['rewardType'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for reward
 * @member {String} icon
 */

RewardAllOf.prototype['icon'] = undefined;
/**
 * how many of thsese rewards are
 * @member {Number} issueLimit
 */

RewardAllOf.prototype['issueLimit'] = undefined;
/**
 * Delay of issuing a reward in minutes
 * @member {Number} delay
 */

RewardAllOf.prototype['delay'] = undefined;
/**
 * ISO8601 timestamp for when a Reward is available until a specific point in time. All records are stored in UTC time zone
 * @member {Date} pointInTime
 */

RewardAllOf.prototype['pointInTime'] = undefined;
/**
 * Reward available for a period of time from issuing in minutes
 * @member {Number} period
 */

RewardAllOf.prototype['period'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

RewardAllOf.prototype['constraints'] = undefined;
var _default = RewardAllOf;
exports["default"] = _default;
