// @ziqni-tech/admin-api-client/dist/model/RewardReducedAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RewardReducedAllOf model module.
 * @module model/RewardReducedAllOf
 * @version 0.0.26
 */
var RewardReducedAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RewardReducedAllOf</code>.
   * @alias module:model/RewardReducedAllOf
   * @param rewardRank {String} If used in the context of contest this will associate with the rank of the leaderboard
   * @param name {String} The name of a reward
   * @param rewardValue {Number} Numerical value of the reward that will be issued based on the reward type
   * @param rewardType {String} Reward Type key
   * @param rewardTypeId {String} A unique id of the Reward Type
   */
  function RewardReducedAllOf(rewardRank, name, rewardValue, rewardType, rewardTypeId) {
    _classCallCheck(this, RewardReducedAllOf);

    RewardReducedAllOf.initialize(this, rewardRank, name, rewardValue, rewardType, rewardTypeId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RewardReducedAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, rewardRank, name, rewardValue, rewardType, rewardTypeId) {
      obj['rewardRank'] = rewardRank;
      obj['name'] = name;
      obj['rewardValue'] = rewardValue;
      obj['rewardType'] = rewardType;
      obj['rewardTypeId'] = rewardTypeId;
    }
    /**
     * Constructs a <code>RewardReducedAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RewardReducedAllOf} obj Optional instance to populate.
     * @return {module:model/RewardReducedAllOf} The populated <code>RewardReducedAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RewardReducedAllOf();

        if (data.hasOwnProperty('rewardRank')) {
          obj['rewardRank'] = _ApiClient["default"].convertToType(data['rewardRank'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('rewardValue')) {
          obj['rewardValue'] = _ApiClient["default"].convertToType(data['rewardValue'], 'Number');
        }

        if (data.hasOwnProperty('rewardType')) {
          obj['rewardType'] = _ApiClient["default"].convertToType(data['rewardType'], 'String');
        }

        if (data.hasOwnProperty('rewardTypeId')) {
          obj['rewardTypeId'] = _ApiClient["default"].convertToType(data['rewardTypeId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RewardReducedAllOf;
}();
/**
 * If used in the context of contest this will associate with the rank of the leaderboard
 * @member {String} rewardRank
 */


RewardReducedAllOf.prototype['rewardRank'] = undefined;
/**
 * The name of a reward
 * @member {String} name
 */

RewardReducedAllOf.prototype['name'] = undefined;
/**
 * Numerical value of the reward that will be issued based on the reward type
 * @member {Number} rewardValue
 */

RewardReducedAllOf.prototype['rewardValue'] = undefined;
/**
 * Reward Type key
 * @member {String} rewardType
 */

RewardReducedAllOf.prototype['rewardType'] = undefined;
/**
 * A unique id of the Reward Type
 * @member {String} rewardTypeId
 */

RewardReducedAllOf.prototype['rewardTypeId'] = undefined;
var _default = RewardReducedAllOf;
exports["default"] = _default;
