// @ziqni-tech/admin-api-client/dist/model/RewardTypeAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RewardTypeAllOf model module.
 * @module model/RewardTypeAllOf
 * @version 0.0.26
 */
var RewardTypeAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RewardTypeAllOf</code>.
   * @alias module:model/RewardTypeAllOf
   * @param name {String} The name of the Reward type
   * @param key {String} A unique key that represents the reward type
   * @param unitOfMeasureType {module:model/UnitOfMeasureType} 
   * @param constraints {Array.<String>} Additional constraints, if the value is present it means the
   */
  function RewardTypeAllOf(name, key, unitOfMeasureType, constraints) {
    _classCallCheck(this, RewardTypeAllOf);

    RewardTypeAllOf.initialize(this, name, key, unitOfMeasureType, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RewardTypeAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, key, unitOfMeasureType, constraints) {
      obj['name'] = name;
      obj['key'] = key;
      obj['unitOfMeasureType'] = unitOfMeasureType;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>RewardTypeAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RewardTypeAllOf} obj Optional instance to populate.
     * @return {module:model/RewardTypeAllOf} The populated <code>RewardTypeAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RewardTypeAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return RewardTypeAllOf;
}();
/**
 * The name of the Reward type
 * @member {String} name
 */


RewardTypeAllOf.prototype['name'] = undefined;
/**
 * The description of a reward type
 * @member {String} description
 */

RewardTypeAllOf.prototype['description'] = undefined;
/**
 * A unique key that represents the reward type
 * @member {String} key
 */

RewardTypeAllOf.prototype['key'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

RewardTypeAllOf.prototype['unitOfMeasureType'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

RewardTypeAllOf.prototype['constraints'] = undefined;
var _default = RewardTypeAllOf;
exports["default"] = _default;
