// @ziqni-tech/admin-api-client/dist/model/Rule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RuleExpression = _interopRequireDefault(require("./RuleExpression"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Rule model module.
 * @module model/Rule
 * @version 0.0.26
 */
var Rule = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Rule</code>.
   * @alias module:model/Rule
   * @param entityId {String} The id of the context object this rule refers to
   * @param action {String} The action this rule describes, like start.contest, or points.awarded
   * @param context {String} The context of this rule like achievement, competition, or contest 
   * @param rules {module:model/RuleExpression} 
   * @param id {String} Unique system identifier of a Main Rule
   * @param entityType {String} 
   */
  function Rule(entityId, action, context, rules, id, entityType) {
    _classCallCheck(this, Rule);

    Rule.initialize(this, entityId, action, context, rules, id, entityType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Rule, null, [{
    key: "initialize",
    value: function initialize(obj, entityId, action, context, rules, id, entityType) {
      obj['entityId'] = entityId;
      obj['action'] = action;
      obj['context'] = context;
      obj['rules'] = rules;
      obj['id'] = id;
      obj['entityType'] = entityType;
    }
    /**
     * Constructs a <code>Rule</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Rule} obj Optional instance to populate.
     * @return {module:model/Rule} The populated <code>Rule</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Rule();

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('action')) {
          obj['action'] = _ApiClient["default"].convertToType(data['action'], 'String');
        }

        if (data.hasOwnProperty('context')) {
          obj['context'] = _ApiClient["default"].convertToType(data['context'], 'String');
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _RuleExpression["default"].constructFromObject(data['rules']);
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('entityType')) {
          obj['entityType'] = _ApiClient["default"].convertToType(data['entityType'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Rule;
}();
/**
 * The id of the context object this rule refers to
 * @member {String} entityId
 */


Rule.prototype['entityId'] = undefined;
/**
 * The action this rule describes, like start.contest, or points.awarded
 * @member {String} action
 */

Rule.prototype['action'] = undefined;
/**
 * The context of this rule like achievement, competition, or contest 
 * @member {String} context
 */

Rule.prototype['context'] = undefined;
/**
 * @member {module:model/RuleExpression} rules
 */

Rule.prototype['rules'] = undefined;
/**
 * Unique system identifier of a Main Rule
 * @member {String} id
 */

Rule.prototype['id'] = undefined;
/**
 * @member {String} entityType
 */

Rule.prototype['entityType'] = undefined;
var _default = Rule;
exports["default"] = _default;
