// @ziqni-tech/admin-api-client/dist/model/RuleRoutine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RuleArgument = _interopRequireDefault(require("./RuleArgument"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RuleRoutine model module.
 * @module model/RuleRoutine
 * @version 0.0.26
 */
var RuleRoutine = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RuleRoutine</code>.
   * @alias module:model/RuleRoutine
   */
  function RuleRoutine() {
    _classCallCheck(this, RuleRoutine);

    RuleRoutine.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RuleRoutine, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RuleRoutine</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RuleRoutine} obj Optional instance to populate.
     * @return {module:model/RuleRoutine} The populated <code>RuleRoutine</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RuleRoutine();

        if (data.hasOwnProperty('routine')) {
          obj['routine'] = _ApiClient["default"].convertToType(data['routine'], 'String');
        }

        if (data.hasOwnProperty('arguments')) {
          obj['arguments'] = _ApiClient["default"].convertToType(data['arguments'], [_RuleArgument["default"]]);
        }

        if (data.hasOwnProperty('lineNumber')) {
          obj['lineNumber'] = _ApiClient["default"].convertToType(data['lineNumber'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return RuleRoutine;
}();
/**
 * A routine to execute if the conditional expressions in the matching group evaluted to true
 * @member {String} routine
 */


RuleRoutine.prototype['routine'] = undefined;
/**
 * The arguments of the routine.
 * @member {Array.<module:model/RuleArgument>} arguments
 */

RuleRoutine.prototype['arguments'] = undefined;
/**
 * @member {Number} lineNumber
 */

RuleRoutine.prototype['lineNumber'] = undefined;
var _default = RuleRoutine;
exports["default"] = _default;
