// @ziqni-tech/admin-api-client/dist/model/ScoringStrategy.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ScoringStrategy model module.
 * @module model/ScoringStrategy
 * @version 0.0.26
 */
var ScoringStrategy = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ScoringStrategy</code>.
   * @alias module:model/ScoringStrategy
   * @param limitUpdatesTo {Number} The scoring strategies types \"SumBest\" value
   * @param sumBestXOf {Number} The scoring strategies types \"LimitedTo\" value
   * @param lastUpdateTimeStamp {Number} The scoring strategies types \"FirstTo\" value
   * @param recordTimeWhenSumReaches {Number} Not supported at the moment
   */
  function ScoringStrategy(limitUpdatesTo, sumBestXOf, lastUpdateTimeStamp, recordTimeWhenSumReaches) {
    _classCallCheck(this, ScoringStrategy);

    ScoringStrategy.initialize(this, limitUpdatesTo, sumBestXOf, lastUpdateTimeStamp, recordTimeWhenSumReaches);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ScoringStrategy, null, [{
    key: "initialize",
    value: function initialize(obj, limitUpdatesTo, sumBestXOf, lastUpdateTimeStamp, recordTimeWhenSumReaches) {
      obj['limitUpdatesTo'] = limitUpdatesTo;
      obj['sumBestXOf'] = sumBestXOf;
      obj['lastUpdateTimeStamp'] = lastUpdateTimeStamp;
      obj['recordTimeWhenSumReaches'] = recordTimeWhenSumReaches;
    }
    /**
     * Constructs a <code>ScoringStrategy</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ScoringStrategy} obj Optional instance to populate.
     * @return {module:model/ScoringStrategy} The populated <code>ScoringStrategy</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ScoringStrategy();

        if (data.hasOwnProperty('limitUpdatesTo')) {
          obj['limitUpdatesTo'] = _ApiClient["default"].convertToType(data['limitUpdatesTo'], 'Number');
        }

        if (data.hasOwnProperty('sumBestXOf')) {
          obj['sumBestXOf'] = _ApiClient["default"].convertToType(data['sumBestXOf'], 'Number');
        }

        if (data.hasOwnProperty('lastUpdateTimeStamp')) {
          obj['lastUpdateTimeStamp'] = _ApiClient["default"].convertToType(data['lastUpdateTimeStamp'], 'Number');
        }

        if (data.hasOwnProperty('recordTimeWhenSumReaches')) {
          obj['recordTimeWhenSumReaches'] = _ApiClient["default"].convertToType(data['recordTimeWhenSumReaches'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ScoringStrategy;
}();
/**
 * The scoring strategies types \"SumBest\" value
 * @member {Number} limitUpdatesTo
 */


ScoringStrategy.prototype['limitUpdatesTo'] = undefined;
/**
 * The scoring strategies types \"LimitedTo\" value
 * @member {Number} sumBestXOf
 */

ScoringStrategy.prototype['sumBestXOf'] = undefined;
/**
 * The scoring strategies types \"FirstTo\" value
 * @member {Number} lastUpdateTimeStamp
 */

ScoringStrategy.prototype['lastUpdateTimeStamp'] = undefined;
/**
 * Not supported at the moment
 * @member {Number} recordTimeWhenSumReaches
 */

ScoringStrategy.prototype['recordTimeWhenSumReaches'] = undefined;
var _default = ScoringStrategy;
exports["default"] = _default;
