// @ziqni-tech/admin-api-client/dist/model/Space.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _SpaceAllOf = _interopRequireDefault(require("./SpaceAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Space model module.
 * @module model/Space
 * @version 0.0.26
 */
var Space = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Space</code>.
   * @alias module:model/Space
   * @implements module:model/ModelDefault
   * @implements module:model/SpaceAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param accountType {String} Type of account; personal or company or vendor
   */
  function Space(id, spaceName, created, accountType) {
    _classCallCheck(this, Space);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _SpaceAllOf["default"].initialize(this, accountType);

    Space.initialize(this, id, spaceName, created, accountType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Space, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, accountType) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['accountType'] = accountType;
    }
    /**
     * Constructs a <code>Space</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Space} obj Optional instance to populate.
     * @return {module:model/Space} The populated <code>Space</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Space();

        _ModelDefault["default"].constructFromObject(data, obj);

        _SpaceAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('accountType')) {
          obj['accountType'] = _ApiClient["default"].convertToType(data['accountType'], 'String');
        }

        if (data.hasOwnProperty('masterSpace')) {
          obj['masterSpace'] = _ApiClient["default"].convertToType(data['masterSpace'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Space;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Space.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Space.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Space.prototype['created'] = undefined;
/**
 * Type of account; personal or company or vendor
 * @member {String} accountType
 */

Space.prototype['accountType'] = undefined;
/**
 * The master account for this account
 * @member {String} masterSpace
 */

Space.prototype['masterSpace'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement SpaceAllOf interface:

/**
 * Type of account; personal or company or vendor
 * @member {String} accountType
 */

_SpaceAllOf["default"].prototype['accountType'] = undefined;
/**
 * The master account for this account
 * @member {String} masterSpace
 */

_SpaceAllOf["default"].prototype['masterSpace'] = undefined;
var _default = Space;
exports["default"] = _default;
