// @ziqni-tech/admin-api-client/dist/model/Strategy.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RankingStrategy = _interopRequireDefault(require("./RankingStrategy"));

var _ScoringStrategy = _interopRequireDefault(require("./ScoringStrategy"));

var _StrategyType = _interopRequireDefault(require("./StrategyType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Strategy model module.
 * @module model/Strategy
 * @version 0.0.26
 */
var Strategy = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Strategy</code>.
   * @alias module:model/Strategy
   * @param strategyType {module:model/StrategyType} 
   * @param rankingStrategy {module:model/RankingStrategy} 
   * @param scoringStrategy {module:model/ScoringStrategy} 
   */
  function Strategy(strategyType, rankingStrategy, scoringStrategy) {
    _classCallCheck(this, Strategy);

    Strategy.initialize(this, strategyType, rankingStrategy, scoringStrategy);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Strategy, null, [{
    key: "initialize",
    value: function initialize(obj, strategyType, rankingStrategy, scoringStrategy) {
      obj['strategyType'] = strategyType;
      obj['rankingStrategy'] = rankingStrategy;
      obj['scoringStrategy'] = scoringStrategy;
    }
    /**
     * Constructs a <code>Strategy</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Strategy} obj Optional instance to populate.
     * @return {module:model/Strategy} The populated <code>Strategy</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Strategy();

        if (data.hasOwnProperty('strategyType')) {
          obj['strategyType'] = _StrategyType["default"].constructFromObject(data['strategyType']);
        }

        if (data.hasOwnProperty('rankingStrategy')) {
          obj['rankingStrategy'] = _RankingStrategy["default"].constructFromObject(data['rankingStrategy']);
        }

        if (data.hasOwnProperty('scoringStrategy')) {
          obj['scoringStrategy'] = _ScoringStrategy["default"].constructFromObject(data['scoringStrategy']);
        }
      }

      return obj;
    }
  }]);

  return Strategy;
}();
/**
 * @member {module:model/StrategyType} strategyType
 */


Strategy.prototype['strategyType'] = undefined;
/**
 * @member {module:model/RankingStrategy} rankingStrategy
 */

Strategy.prototype['rankingStrategy'] = undefined;
/**
 * @member {module:model/ScoringStrategy} scoringStrategy
 */

Strategy.prototype['scoringStrategy'] = undefined;
var _default = Strategy;
exports["default"] = _default;
