// @ziqni-tech/admin-api-client/dist/model/Tag.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _TagAllOf = _interopRequireDefault(require("./TagAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Tag model module.
 * @module model/Tag
 * @version 0.0.26
 */
var Tag = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Tag</code>.
   * Tag model
   * @alias module:model/Tag
   * @implements module:model/ModelDefault
   * @implements module:model/TagAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} The name of the tag
   * @param key {String} A unique key that represents a tag
   */
  function Tag(id, spaceName, created, name, key) {
    _classCallCheck(this, Tag);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _TagAllOf["default"].initialize(this, name, key);

    Tag.initialize(this, id, spaceName, created, name, key);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Tag, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, key) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['key'] = key;
    }
    /**
     * Constructs a <code>Tag</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Tag} obj Optional instance to populate.
     * @return {module:model/Tag} The populated <code>Tag</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Tag();

        _ModelDefault["default"].constructFromObject(data, obj);

        _TagAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('entityTypes')) {
          obj['entityTypes'] = _ApiClient["default"].convertToType(data['entityTypes'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return Tag;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Tag.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Tag.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Tag.prototype['created'] = undefined;
/**
 * The name of the tag
 * @member {String} name
 */

Tag.prototype['name'] = undefined;
/**
 * A unique key that represents a tag
 * @member {String} key
 */

Tag.prototype['key'] = undefined;
/**
 * The description of the tag for your reference
 * @member {String} description
 */

Tag.prototype['description'] = undefined;
/**
 * The model name the tag refers to
 * @member {Array.<String>} entityTypes
 */

Tag.prototype['entityTypes'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Tag.prototype['metadata'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement TagAllOf interface:

/**
 * The name of the tag
 * @member {String} name
 */

_TagAllOf["default"].prototype['name'] = undefined;
/**
 * A unique key that represents a tag
 * @member {String} key
 */

_TagAllOf["default"].prototype['key'] = undefined;
/**
 * The description of the tag for your reference
 * @member {String} description
 */

_TagAllOf["default"].prototype['description'] = undefined;
/**
 * The model name the tag refers to
 * @member {Array.<String>} entityTypes
 */

_TagAllOf["default"].prototype['entityTypes'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_TagAllOf["default"].prototype['metadata'] = undefined;
var _default = Tag;
exports["default"] = _default;
