// @ziqni-tech/admin-api-client/dist/model/TagAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The TagAllOf model module.
 * @module model/TagAllOf
 * @version 0.0.26
 */
var TagAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TagAllOf</code>.
   * @alias module:model/TagAllOf
   * @param name {String} The name of the tag
   * @param key {String} A unique key that represents a tag
   */
  function TagAllOf(name, key) {
    _classCallCheck(this, TagAllOf);

    TagAllOf.initialize(this, name, key);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TagAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, name, key) {
      obj['name'] = name;
      obj['key'] = key;
    }
    /**
     * Constructs a <code>TagAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TagAllOf} obj Optional instance to populate.
     * @return {module:model/TagAllOf} The populated <code>TagAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TagAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('entityTypes')) {
          obj['entityTypes'] = _ApiClient["default"].convertToType(data['entityTypes'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return TagAllOf;
}();
/**
 * The name of the tag
 * @member {String} name
 */


TagAllOf.prototype['name'] = undefined;
/**
 * A unique key that represents a tag
 * @member {String} key
 */

TagAllOf.prototype['key'] = undefined;
/**
 * The description of the tag for your reference
 * @member {String} description
 */

TagAllOf.prototype['description'] = undefined;
/**
 * The model name the tag refers to
 * @member {Array.<String>} entityTypes
 */

TagAllOf.prototype['entityTypes'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

TagAllOf.prototype['metadata'] = undefined;
var _default = TagAllOf;
exports["default"] = _default;
