// @ziqni-tech/admin-api-client/dist/model/Transformer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _SourceLanguage = _interopRequireDefault(require("./SourceLanguage"));

var _TransformerAllOf = _interopRequireDefault(require("./TransformerAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Transformer model module.
 * @module model/Transformer
 * @version 0.0.26
 */
var Transformer = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Transformer</code>.
   * @alias module:model/Transformer
   * @implements module:model/ModelDefault
   * @implements module:model/TransformerAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} The name of the Transformer
   * @param source {String} Source code
   * @param sourceLanguage {module:model/SourceLanguage} 
   */
  function Transformer(id, spaceName, created, name, source, sourceLanguage) {
    _classCallCheck(this, Transformer);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _TransformerAllOf["default"].initialize(this, name, source, sourceLanguage);

    Transformer.initialize(this, id, spaceName, created, name, source, sourceLanguage);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Transformer, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, source, sourceLanguage) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['source'] = source;
      obj['sourceLanguage'] = sourceLanguage;
    }
    /**
     * Constructs a <code>Transformer</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Transformer} obj Optional instance to populate.
     * @return {module:model/Transformer} The populated <code>Transformer</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Transformer();

        _ModelDefault["default"].constructFromObject(data, obj);

        _TransformerAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('sourceLanguage')) {
          obj['sourceLanguage'] = _SourceLanguage["default"].constructFromObject(data['sourceLanguage']);
        }
      }

      return obj;
    }
  }]);

  return Transformer;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Transformer.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Transformer.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Transformer.prototype['created'] = undefined;
/**
 * The name of the Transformer
 * @member {String} name
 */

Transformer.prototype['name'] = undefined;
/**
 * Source code
 * @member {String} source
 */

Transformer.prototype['source'] = undefined;
/**
 * @member {module:model/SourceLanguage} sourceLanguage
 */

Transformer.prototype['sourceLanguage'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement TransformerAllOf interface:

/**
 * The name of the Transformer
 * @member {String} name
 */

_TransformerAllOf["default"].prototype['name'] = undefined;
/**
 * Source code
 * @member {String} source
 */

_TransformerAllOf["default"].prototype['source'] = undefined;
/**
 * @member {module:model/SourceLanguage} sourceLanguage
 */

_TransformerAllOf["default"].prototype['sourceLanguage'] = undefined;
var _default = Transformer;
exports["default"] = _default;
