// @ziqni-tech/admin-api-client/dist/model/UnitOfMeasure.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _UnitOfMeasureAllOf = _interopRequireDefault(require("./UnitOfMeasureAllOf"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UnitOfMeasure model module.
 * @module model/UnitOfMeasure
 * @version 0.0.26
 */
var UnitOfMeasure = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UnitOfMeasure</code>.
   * @alias module:model/UnitOfMeasure
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/UnitOfMeasureAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param name {String} The name of a unit of measure
   * @param key {String} The reference to the unit of measure in your system
   * @param multiplier {Number} Is used to multiply the value from the standardised one that is being used
   * @param unitOfMeasureType {module:model/UnitOfMeasureType} 
   */
  function UnitOfMeasure(id, spaceName, created, name, key, multiplier, unitOfMeasureType) {
    _classCallCheck(this, UnitOfMeasure);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _UnitOfMeasureAllOf["default"].initialize(this, name, key, multiplier, unitOfMeasureType);

    UnitOfMeasure.initialize(this, id, spaceName, created, name, key, multiplier, unitOfMeasureType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UnitOfMeasure, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, name, key, multiplier, unitOfMeasureType) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['name'] = name;
      obj['key'] = key;
      obj['multiplier'] = multiplier;
      obj['unitOfMeasureType'] = unitOfMeasureType;
    }
    /**
     * Constructs a <code>UnitOfMeasure</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UnitOfMeasure} obj Optional instance to populate.
     * @return {module:model/UnitOfMeasure} The populated <code>UnitOfMeasure</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UnitOfMeasure();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _UnitOfMeasureAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('isoCode')) {
          obj['isoCode'] = _ApiClient["default"].convertToType(data['isoCode'], 'String');
        }

        if (data.hasOwnProperty('symbol')) {
          obj['symbol'] = _ApiClient["default"].convertToType(data['symbol'], 'String');
        }

        if (data.hasOwnProperty('multiplier')) {
          obj['multiplier'] = _ApiClient["default"].convertToType(data['multiplier'], 'Number');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }
      }

      return obj;
    }
  }]);

  return UnitOfMeasure;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UnitOfMeasure.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

UnitOfMeasure.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

UnitOfMeasure.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

UnitOfMeasure.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UnitOfMeasure.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UnitOfMeasure.prototype['metadata'] = undefined;
/**
 * The name of a unit of measure
 * @member {String} name
 */

UnitOfMeasure.prototype['name'] = undefined;
/**
 * The reference to the unit of measure in your system
 * @member {String} key
 */

UnitOfMeasure.prototype['key'] = undefined;
/**
 * The description of a unit of measure
 * @member {String} description
 */

UnitOfMeasure.prototype['description'] = undefined;
/**
 * An alphabetical or numerical code to identify a unit of measure
 * @member {String} isoCode
 */

UnitOfMeasure.prototype['isoCode'] = undefined;
/**
 * The symbol of a unit of measure
 * @member {String} symbol
 */

UnitOfMeasure.prototype['symbol'] = undefined;
/**
 * Is used to multiply the value from the standardised one that is being used
 * @member {Number} multiplier
 */

UnitOfMeasure.prototype['multiplier'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

UnitOfMeasure.prototype['unitOfMeasureType'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement UnitOfMeasureAllOf interface:

/**
 * The name of a unit of measure
 * @member {String} name
 */

_UnitOfMeasureAllOf["default"].prototype['name'] = undefined;
/**
 * The reference to the unit of measure in your system
 * @member {String} key
 */

_UnitOfMeasureAllOf["default"].prototype['key'] = undefined;
/**
 * The description of a unit of measure
 * @member {String} description
 */

_UnitOfMeasureAllOf["default"].prototype['description'] = undefined;
/**
 * An alphabetical or numerical code to identify a unit of measure
 * @member {String} isoCode
 */

_UnitOfMeasureAllOf["default"].prototype['isoCode'] = undefined;
/**
 * The symbol of a unit of measure
 * @member {String} symbol
 */

_UnitOfMeasureAllOf["default"].prototype['symbol'] = undefined;
/**
 * Is used to multiply the value from the standardised one that is being used
 * @member {Number} multiplier
 */

_UnitOfMeasureAllOf["default"].prototype['multiplier'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

_UnitOfMeasureAllOf["default"].prototype['unitOfMeasureType'] = undefined;
var _default = UnitOfMeasure;
exports["default"] = _default;
