// @ziqni-tech/admin-api-client/dist/model/UpdateApiKeyRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Role = _interopRequireDefault(require("./Role"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateApiKeyRequestAllOf model module.
 * @module model/UpdateApiKeyRequestAllOf
 * @version 0.0.26
 */
var UpdateApiKeyRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateApiKeyRequestAllOf</code>.
   * @alias module:model/UpdateApiKeyRequestAllOf
   */
  function UpdateApiKeyRequestAllOf() {
    _classCallCheck(this, UpdateApiKeyRequestAllOf);

    UpdateApiKeyRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateApiKeyRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateApiKeyRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateApiKeyRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateApiKeyRequestAllOf} The populated <code>UpdateApiKeyRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateApiKeyRequestAllOf();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('whiteListIPs')) {
          obj['whiteListIPs'] = _ApiClient["default"].convertToType(data['whiteListIPs'], ['String']);
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('role')) {
          obj['role'] = _Role["default"].constructFromObject(data['role']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateApiKeyRequestAllOf;
}();
/**
 * The description of an Api key
 * @member {String} description
 */


UpdateApiKeyRequestAllOf.prototype['description'] = undefined;
/**
 * the IP's that need to be whitelisted with the Api key
 * @member {Array.<String>} whiteListIPs
 */

UpdateApiKeyRequestAllOf.prototype['whiteListIPs'] = undefined;
/**
 * An Api key hash
 * @member {String} key
 */

UpdateApiKeyRequestAllOf.prototype['key'] = undefined;
/**
 * @member {module:model/Role} role
 */

UpdateApiKeyRequestAllOf.prototype['role'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

UpdateApiKeyRequestAllOf.prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

UpdateApiKeyRequestAllOf.prototype['removeConstraints'] = undefined;
var _default = UpdateApiKeyRequestAllOf;
exports["default"] = _default;
