// @ziqni-tech/admin-api-client/dist/model/UpdateMemberRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MemberType = _interopRequireDefault(require("./MemberType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateMemberRequestAllOf model module.
 * @module model/UpdateMemberRequestAllOf
 * @version 0.0.26
 */
var UpdateMemberRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateMemberRequestAllOf</code>.
   * @alias module:model/UpdateMemberRequestAllOf
   */
  function UpdateMemberRequestAllOf() {
    _classCallCheck(this, UpdateMemberRequestAllOf);

    UpdateMemberRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateMemberRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateMemberRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateMemberRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateMemberRequestAllOf} The populated <code>UpdateMemberRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateMemberRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memberType')) {
          obj['memberType'] = _MemberType["default"].constructFromObject(data['memberType']);
        }

        if (data.hasOwnProperty('teamMembers')) {
          obj['teamMembers'] = _ApiClient["default"].convertToType(data['teamMembers'], ['String']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }

        if (data.hasOwnProperty('timeZoneOffset')) {
          obj['timeZoneOffset'] = _ApiClient["default"].convertToType(data['timeZoneOffset'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateMemberRequestAllOf;
}();
/**
 * The name of the member that is used on leader boards and public displays
 * @member {String} name
 */


UpdateMemberRequestAllOf.prototype['name'] = undefined;
/**
 * @member {module:model/MemberType} memberType
 */

UpdateMemberRequestAllOf.prototype['memberType'] = undefined;
/**
 * A social group like Guilds
 * @member {Array.<String>} teamMembers
 */

UpdateMemberRequestAllOf.prototype['teamMembers'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

UpdateMemberRequestAllOf.prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

UpdateMemberRequestAllOf.prototype['removeConstraints'] = undefined;
/**
 * To what time zone teh emmber belongs to
 * @member {String} timeZoneOffset
 */

UpdateMemberRequestAllOf.prototype['timeZoneOffset'] = undefined;
var _default = UpdateMemberRequestAllOf;
exports["default"] = _default;
