// @ziqni-tech/admin-api-client/dist/model/UpdateAchievementRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AchievementLiveStatus = _interopRequireDefault(require("./AchievementLiveStatus"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _UpdateAchievementRequestAllOf = _interopRequireDefault(require("./UpdateAchievementRequestAllOf"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateAchievementRequest model module.
 * @module model/UpdateAchievementRequest
 * @version 0.0.26
 */
var UpdateAchievementRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateAchievementRequest</code>.
   * @alias module:model/UpdateAchievementRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateAchievementRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateAchievementRequest(id) {
    _classCallCheck(this, UpdateAchievementRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateAchievementRequestAllOf["default"].initialize(this);

    UpdateAchievementRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateAchievementRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateAchievementRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateAchievementRequest} obj Optional instance to populate.
     * @return {module:model/UpdateAchievementRequest} The populated <code>UpdateAchievementRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateAchievementRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateAchievementRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('icon')) {
          obj['icon'] = _ApiClient["default"].convertToType(data['icon'], 'String');
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('achievementLiveStatus')) {
          obj['achievementLiveStatus'] = _AchievementLiveStatus["default"].constructFromObject(data['achievementLiveStatus']);
        }

        if (data.hasOwnProperty('maxNumberOfIssues')) {
          obj['maxNumberOfIssues'] = _ApiClient["default"].convertToType(data['maxNumberOfIssues'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('achievementDependencies')) {
          obj['achievementDependencies'] = _DependantOn["default"].constructFromObject(data['achievementDependencies']);
        }

        if (data.hasOwnProperty('memberTagsFilter')) {
          obj['memberTagsFilter'] = _DependantOn["default"].constructFromObject(data['memberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateAchievementRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateAchievementRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateAchievementRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateAchievementRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateAchievementRequest.prototype['metadata'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */

UpdateAchievementRequest.prototype['name'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} description
 */

UpdateAchievementRequest.prototype['description'] = undefined;
/**
 * Terms and conditions of an achievement. Can be translated
 * @member {String} termsAndConditions
 */

UpdateAchievementRequest.prototype['termsAndConditions'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */

UpdateAchievementRequest.prototype['icon'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

UpdateAchievementRequest.prototype['scheduling'] = undefined;
/**
 * @member {module:model/AchievementLiveStatus} achievementLiveStatus
 */

UpdateAchievementRequest.prototype['achievementLiveStatus'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

UpdateAchievementRequest.prototype['maxNumberOfIssues'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UpdateAchievementRequest.prototype['constraints'] = undefined;
/**
 * @member {module:model/DependantOn} achievementDependencies
 */

UpdateAchievementRequest.prototype['achievementDependencies'] = undefined;
/**
 * @member {module:model/DependantOn} memberTagsFilter
 */

UpdateAchievementRequest.prototype['memberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

UpdateAchievementRequest.prototype['productTagsFilter'] = undefined;
/**
 * @member {Array.<String>} addConstraints
 */

UpdateAchievementRequest.prototype['addConstraints'] = undefined;
/**
 * @member {Array.<String>} removeConstraints
 */

UpdateAchievementRequest.prototype['removeConstraints'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateAchievementRequestAllOf interface:

/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */

_UpdateAchievementRequestAllOf["default"].prototype['name'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} description
 */

_UpdateAchievementRequestAllOf["default"].prototype['description'] = undefined;
/**
 * Terms and conditions of an achievement. Can be translated
 * @member {String} termsAndConditions
 */

_UpdateAchievementRequestAllOf["default"].prototype['termsAndConditions'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */

_UpdateAchievementRequestAllOf["default"].prototype['icon'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

_UpdateAchievementRequestAllOf["default"].prototype['scheduling'] = undefined;
/**
 * @member {module:model/AchievementLiveStatus} achievementLiveStatus
 */

_UpdateAchievementRequestAllOf["default"].prototype['achievementLiveStatus'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

_UpdateAchievementRequestAllOf["default"].prototype['maxNumberOfIssues'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_UpdateAchievementRequestAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {module:model/DependantOn} achievementDependencies
 */

_UpdateAchievementRequestAllOf["default"].prototype['achievementDependencies'] = undefined;
/**
 * @member {module:model/DependantOn} memberTagsFilter
 */

_UpdateAchievementRequestAllOf["default"].prototype['memberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

_UpdateAchievementRequestAllOf["default"].prototype['productTagsFilter'] = undefined;
/**
 * @member {Array.<String>} addConstraints
 */

_UpdateAchievementRequestAllOf["default"].prototype['addConstraints'] = undefined;
/**
 * @member {Array.<String>} removeConstraints
 */

_UpdateAchievementRequestAllOf["default"].prototype['removeConstraints'] = undefined;
var _default = UpdateAchievementRequest;
exports["default"] = _default;
