// @ziqni-tech/admin-api-client/dist/model/UpdateUnitOfMeasureRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

var _UpdateUnitOfMeasureRequestAllOf = _interopRequireDefault(require("./UpdateUnitOfMeasureRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateUnitOfMeasureRequest model module.
 * @module model/UpdateUnitOfMeasureRequest
 * @version 0.0.26
 */
var UpdateUnitOfMeasureRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateUnitOfMeasureRequest</code>.
   * @alias module:model/UpdateUnitOfMeasureRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateUnitOfMeasureRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateUnitOfMeasureRequest(id) {
    _classCallCheck(this, UpdateUnitOfMeasureRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateUnitOfMeasureRequestAllOf["default"].initialize(this);

    UpdateUnitOfMeasureRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateUnitOfMeasureRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateUnitOfMeasureRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateUnitOfMeasureRequest} obj Optional instance to populate.
     * @return {module:model/UpdateUnitOfMeasureRequest} The populated <code>UpdateUnitOfMeasureRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateUnitOfMeasureRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateUnitOfMeasureRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('isoCode')) {
          obj['isoCode'] = _ApiClient["default"].convertToType(data['isoCode'], 'String');
        }

        if (data.hasOwnProperty('symbol')) {
          obj['symbol'] = _ApiClient["default"].convertToType(data['symbol'], 'String');
        }

        if (data.hasOwnProperty('multiplier')) {
          obj['multiplier'] = _ApiClient["default"].convertToType(data['multiplier'], 'Number');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }
      }

      return obj;
    }
  }]);

  return UpdateUnitOfMeasureRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateUnitOfMeasureRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateUnitOfMeasureRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateUnitOfMeasureRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateUnitOfMeasureRequest.prototype['metadata'] = undefined;
/**
 * The name of a unit of measure
 * @member {String} name
 */

UpdateUnitOfMeasureRequest.prototype['name'] = undefined;
/**
 * The description of a unit of measure
 * @member {String} description
 */

UpdateUnitOfMeasureRequest.prototype['description'] = undefined;
/**
 * An alphabetical or numerical code to identify a unit of measure
 * @member {String} isoCode
 */

UpdateUnitOfMeasureRequest.prototype['isoCode'] = undefined;
/**
 * The symbol of a unit of measure
 * @member {String} symbol
 */

UpdateUnitOfMeasureRequest.prototype['symbol'] = undefined;
/**
 * Is used to multiply the value from the standardised one that is being used
 * @member {Number} multiplier
 */

UpdateUnitOfMeasureRequest.prototype['multiplier'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

UpdateUnitOfMeasureRequest.prototype['unitOfMeasureType'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateUnitOfMeasureRequestAllOf interface:

/**
 * The name of a unit of measure
 * @member {String} name
 */

_UpdateUnitOfMeasureRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of a unit of measure
 * @member {String} description
 */

_UpdateUnitOfMeasureRequestAllOf["default"].prototype['description'] = undefined;
/**
 * An alphabetical or numerical code to identify a unit of measure
 * @member {String} isoCode
 */

_UpdateUnitOfMeasureRequestAllOf["default"].prototype['isoCode'] = undefined;
/**
 * The symbol of a unit of measure
 * @member {String} symbol
 */

_UpdateUnitOfMeasureRequestAllOf["default"].prototype['symbol'] = undefined;
/**
 * Is used to multiply the value from the standardised one that is being used
 * @member {Number} multiplier
 */

_UpdateUnitOfMeasureRequestAllOf["default"].prototype['multiplier'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

_UpdateUnitOfMeasureRequestAllOf["default"].prototype['unitOfMeasureType'] = undefined;
var _default = UpdateUnitOfMeasureRequest;
exports["default"] = _default;
