// @ziqni-tech/admin-api-client/dist/model/UpdateTransformerRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SourceLanguage = _interopRequireDefault(require("./SourceLanguage"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateTransformerRequestAllOf = _interopRequireDefault(require("./UpdateTransformerRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateTransformerRequest model module.
 * @module model/UpdateTransformerRequest
 * @version 0.0.26
 */
var UpdateTransformerRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTransformerRequest</code>.
   * @alias module:model/UpdateTransformerRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateTransformerRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateTransformerRequest(id) {
    _classCallCheck(this, UpdateTransformerRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateTransformerRequestAllOf["default"].initialize(this);

    UpdateTransformerRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateTransformerRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateTransformerRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTransformerRequest} obj Optional instance to populate.
     * @return {module:model/UpdateTransformerRequest} The populated <code>UpdateTransformerRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTransformerRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateTransformerRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('sourceLanguage')) {
          obj['sourceLanguage'] = _SourceLanguage["default"].constructFromObject(data['sourceLanguage']);
        }
      }

      return obj;
    }
  }]);

  return UpdateTransformerRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateTransformerRequest.prototype['id'] = undefined;
/**
 * The name of the Transformer
 * @member {String} name
 */

UpdateTransformerRequest.prototype['name'] = undefined;
/**
 * Source code
 * @member {String} source
 */

UpdateTransformerRequest.prototype['source'] = undefined;
/**
 * @member {module:model/SourceLanguage} sourceLanguage
 */

UpdateTransformerRequest.prototype['sourceLanguage'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateTransformerRequestAllOf interface:

/**
 * The name of the Transformer
 * @member {String} name
 */

_UpdateTransformerRequestAllOf["default"].prototype['name'] = undefined;
/**
 * Source code
 * @member {String} source
 */

_UpdateTransformerRequestAllOf["default"].prototype['source'] = undefined;
/**
 * @member {module:model/SourceLanguage} sourceLanguage
 */

_UpdateTransformerRequestAllOf["default"].prototype['sourceLanguage'] = undefined;
var _default = UpdateTransformerRequest;
exports["default"] = _default;
