// @ziqni-tech/admin-api-client/dist/model/UpdateCompetitionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CompetitionType = _interopRequireDefault(require("./CompetitionType"));

var _CreateContestForCompetitionRequest = _interopRequireDefault(require("./CreateContestForCompetitionRequest"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _UpdateCompetitionRequestAllOf = _interopRequireDefault(require("./UpdateCompetitionRequestAllOf"));

var _UpdateContestRequest = _interopRequireDefault(require("./UpdateContestRequest"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateCompetitionRequest model module.
 * @module model/UpdateCompetitionRequest
 * @version 0.0.26
 */
var UpdateCompetitionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateCompetitionRequest</code>.
   * @alias module:model/UpdateCompetitionRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateCompetitionRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateCompetitionRequest(id) {
    _classCallCheck(this, UpdateCompetitionRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateCompetitionRequestAllOf["default"].initialize(this);

    UpdateCompetitionRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateCompetitionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateCompetitionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateCompetitionRequest} obj Optional instance to populate.
     * @return {module:model/UpdateCompetitionRequest} The populated <code>UpdateCompetitionRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateCompetitionRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateCompetitionRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('competitionType')) {
          obj['competitionType'] = _CompetitionType["default"].constructFromObject(data['competitionType']);
        }

        if (data.hasOwnProperty('numberOfRounds')) {
          obj['numberOfRounds'] = _ApiClient["default"].convertToType(data['numberOfRounds'], 'Number');
        }

        if (data.hasOwnProperty('numberOfGroupStages')) {
          obj['numberOfGroupStages'] = _ApiClient["default"].convertToType(data['numberOfGroupStages'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('entrantMemberType')) {
          obj['entrantMemberType'] = _ApiClient["default"].convertToType(data['entrantMemberType'], 'String');
        }

        if (data.hasOwnProperty('maxNumberOfEntrants')) {
          obj['maxNumberOfEntrants'] = _ApiClient["default"].convertToType(data['maxNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('minNumberOfEntrants')) {
          obj['minNumberOfEntrants'] = _ApiClient["default"].convertToType(data['minNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('scheduledStartDate')) {
          obj['scheduledStartDate'] = _ApiClient["default"].convertToType(data['scheduledStartDate'], 'Date');
        }

        if (data.hasOwnProperty('scheduledEndDate')) {
          obj['scheduledEndDate'] = _ApiClient["default"].convertToType(data['scheduledEndDate'], 'Date');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }

        if (data.hasOwnProperty('productIds')) {
          obj['productIds'] = _ApiClient["default"].convertToType(data['productIds'], ['String']);
        }

        if (data.hasOwnProperty('createContests')) {
          obj['createContests'] = _CreateContestForCompetitionRequest["default"].constructFromObject(data['createContests']);
        }

        if (data.hasOwnProperty('updateContests')) {
          obj['updateContests'] = _UpdateContestRequest["default"].constructFromObject(data['updateContests']);
        }

        if (data.hasOwnProperty('entrantMemberTagsFilter')) {
          obj['entrantMemberTagsFilter'] = _DependantOn["default"].constructFromObject(data['entrantMemberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }
      }

      return obj;
    }
  }]);

  return UpdateCompetitionRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateCompetitionRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateCompetitionRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateCompetitionRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateCompetitionRequest.prototype['metadata'] = undefined;
/**
 * @member {module:model/CompetitionType} competitionType
 */

UpdateCompetitionRequest.prototype['competitionType'] = undefined;
/**
 * Number of rounds to be played in a competition
 * @member {Number} numberOfRounds
 */

UpdateCompetitionRequest.prototype['numberOfRounds'] = undefined;
/**
 * Number of group stages
 * @member {Number} numberOfGroupStages
 */

UpdateCompetitionRequest.prototype['numberOfGroupStages'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

UpdateCompetitionRequest.prototype['name'] = undefined;
/**
 * Description of the competition. Can be translated
 * @member {String} description
 */

UpdateCompetitionRequest.prototype['description'] = undefined;
/**
 * Terms and conditions of a competition. Can be translated
 * @member {String} termsAndConditions
 */

UpdateCompetitionRequest.prototype['termsAndConditions'] = undefined;
/**
 * @member {String} entrantMemberType
 */

UpdateCompetitionRequest.prototype['entrantMemberType'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} maxNumberOfEntrants
 */

UpdateCompetitionRequest.prototype['maxNumberOfEntrants'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} minNumberOfEntrants
 */

UpdateCompetitionRequest.prototype['minNumberOfEntrants'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

UpdateCompetitionRequest.prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

UpdateCompetitionRequest.prototype['scheduledEndDate'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UpdateCompetitionRequest.prototype['constraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

UpdateCompetitionRequest.prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

UpdateCompetitionRequest.prototype['removeConstraints'] = undefined;
/**
 * @member {Array.<String>} productIds
 */

UpdateCompetitionRequest.prototype['productIds'] = undefined;
/**
 * @member {module:model/CreateContestForCompetitionRequest} createContests
 */

UpdateCompetitionRequest.prototype['createContests'] = undefined;
/**
 * @member {module:model/UpdateContestRequest} updateContests
 */

UpdateCompetitionRequest.prototype['updateContests'] = undefined;
/**
 * @member {module:model/DependantOn} entrantMemberTagsFilter
 */

UpdateCompetitionRequest.prototype['entrantMemberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

UpdateCompetitionRequest.prototype['productTagsFilter'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateCompetitionRequestAllOf interface:

/**
 * @member {module:model/CompetitionType} competitionType
 */

_UpdateCompetitionRequestAllOf["default"].prototype['competitionType'] = undefined;
/**
 * Number of rounds to be played in a competition
 * @member {Number} numberOfRounds
 */

_UpdateCompetitionRequestAllOf["default"].prototype['numberOfRounds'] = undefined;
/**
 * Number of group stages
 * @member {Number} numberOfGroupStages
 */

_UpdateCompetitionRequestAllOf["default"].prototype['numberOfGroupStages'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

_UpdateCompetitionRequestAllOf["default"].prototype['name'] = undefined;
/**
 * Description of the competition. Can be translated
 * @member {String} description
 */

_UpdateCompetitionRequestAllOf["default"].prototype['description'] = undefined;
/**
 * Terms and conditions of a competition. Can be translated
 * @member {String} termsAndConditions
 */

_UpdateCompetitionRequestAllOf["default"].prototype['termsAndConditions'] = undefined;
/**
 * @member {String} entrantMemberType
 */

_UpdateCompetitionRequestAllOf["default"].prototype['entrantMemberType'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} maxNumberOfEntrants
 */

_UpdateCompetitionRequestAllOf["default"].prototype['maxNumberOfEntrants'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} minNumberOfEntrants
 */

_UpdateCompetitionRequestAllOf["default"].prototype['minNumberOfEntrants'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

_UpdateCompetitionRequestAllOf["default"].prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

_UpdateCompetitionRequestAllOf["default"].prototype['scheduledEndDate'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_UpdateCompetitionRequestAllOf["default"].prototype['constraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_UpdateCompetitionRequestAllOf["default"].prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

_UpdateCompetitionRequestAllOf["default"].prototype['removeConstraints'] = undefined;
/**
 * @member {Array.<String>} productIds
 */

_UpdateCompetitionRequestAllOf["default"].prototype['productIds'] = undefined;
/**
 * @member {module:model/CreateContestForCompetitionRequest} createContests
 */

_UpdateCompetitionRequestAllOf["default"].prototype['createContests'] = undefined;
/**
 * @member {module:model/UpdateContestRequest} updateContests
 */

_UpdateCompetitionRequestAllOf["default"].prototype['updateContests'] = undefined;
/**
 * @member {module:model/DependantOn} entrantMemberTagsFilter
 */

_UpdateCompetitionRequestAllOf["default"].prototype['entrantMemberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

_UpdateCompetitionRequestAllOf["default"].prototype['productTagsFilter'] = undefined;
var _default = UpdateCompetitionRequest;
exports["default"] = _default;
