// @ziqni-tech/admin-api-client/dist/model/UpdateKafkaConnectionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UpdateKafkaConnectionRequestAllOf = _interopRequireDefault(require("./UpdateKafkaConnectionRequestAllOf"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateKafkaConnectionRequest model module.
 * @module model/UpdateKafkaConnectionRequest
 * @version 0.0.26
 */
var UpdateKafkaConnectionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateKafkaConnectionRequest</code>.
   * @alias module:model/UpdateKafkaConnectionRequest
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateKafkaConnectionRequestAllOf
   * @implements module:model/UpdateModelDefault
   * @param id {String} A unique system generated identifier
   */
  function UpdateKafkaConnectionRequest(id) {
    _classCallCheck(this, UpdateKafkaConnectionRequest);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateKafkaConnectionRequestAllOf["default"].initialize(this);

    _UpdateModelDefault["default"].initialize(this, id);

    UpdateKafkaConnectionRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateKafkaConnectionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateKafkaConnectionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateKafkaConnectionRequest} obj Optional instance to populate.
     * @return {module:model/UpdateKafkaConnectionRequest} The populated <code>UpdateKafkaConnectionRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateKafkaConnectionRequest();

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateKafkaConnectionRequestAllOf["default"].constructFromObject(data, obj);

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('brokers')) {
          obj['brokers'] = _ApiClient["default"].convertToType(data['brokers'], 'String');
        }

        if (data.hasOwnProperty('groupId')) {
          obj['groupId'] = _ApiClient["default"].convertToType(data['groupId'], 'String');
        }

        if (data.hasOwnProperty('topic')) {
          obj['topic'] = _ApiClient["default"].convertToType(data['topic'], 'String');
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateKafkaConnectionRequest;
}();
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */


UpdateKafkaConnectionRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateKafkaConnectionRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateKafkaConnectionRequest.prototype['metadata'] = undefined;
/**
 * The name of the consumer
 * @member {String} name
 */

UpdateKafkaConnectionRequest.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

UpdateKafkaConnectionRequest.prototype['description'] = undefined;
/**
 * Kafka broker endpoints.
 * @member {String} brokers
 */

UpdateKafkaConnectionRequest.prototype['brokers'] = undefined;
/**
 * Group Id for connection
 * @member {String} groupId
 */

UpdateKafkaConnectionRequest.prototype['groupId'] = undefined;
/**
 * Topic name
 * @member {String} topic
 */

UpdateKafkaConnectionRequest.prototype['topic'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

UpdateKafkaConnectionRequest.prototype['transformerId'] = undefined;
/**
 * A unique system generated identifier
 * @member {String} id
 */

UpdateKafkaConnectionRequest.prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateKafkaConnectionRequestAllOf interface:

/**
 * The name of the consumer
 * @member {String} name
 */

_UpdateKafkaConnectionRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

_UpdateKafkaConnectionRequestAllOf["default"].prototype['description'] = undefined;
/**
 * Kafka broker endpoints.
 * @member {String} brokers
 */

_UpdateKafkaConnectionRequestAllOf["default"].prototype['brokers'] = undefined;
/**
 * Group Id for connection
 * @member {String} groupId
 */

_UpdateKafkaConnectionRequestAllOf["default"].prototype['groupId'] = undefined;
/**
 * Topic name
 * @member {String} topic
 */

_UpdateKafkaConnectionRequestAllOf["default"].prototype['topic'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

_UpdateKafkaConnectionRequestAllOf["default"].prototype['transformerId'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined;
var _default = UpdateKafkaConnectionRequest;
exports["default"] = _default;
