// @ziqni-tech/admin-api-client/dist/model/UpdateTransformerRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SourceLanguage = _interopRequireDefault(require("./SourceLanguage"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateTransformerRequestAllOf model module.
 * @module model/UpdateTransformerRequestAllOf
 * @version 0.0.26
 */
var UpdateTransformerRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTransformerRequestAllOf</code>.
   * @alias module:model/UpdateTransformerRequestAllOf
   */
  function UpdateTransformerRequestAllOf() {
    _classCallCheck(this, UpdateTransformerRequestAllOf);

    UpdateTransformerRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateTransformerRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateTransformerRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTransformerRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateTransformerRequestAllOf} The populated <code>UpdateTransformerRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTransformerRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('sourceLanguage')) {
          obj['sourceLanguage'] = _SourceLanguage["default"].constructFromObject(data['sourceLanguage']);
        }
      }

      return obj;
    }
  }]);

  return UpdateTransformerRequestAllOf;
}();
/**
 * The name of the Transformer
 * @member {String} name
 */


UpdateTransformerRequestAllOf.prototype['name'] = undefined;
/**
 * Source code
 * @member {String} source
 */

UpdateTransformerRequestAllOf.prototype['source'] = undefined;
/**
 * @member {module:model/SourceLanguage} sourceLanguage
 */

UpdateTransformerRequestAllOf.prototype['sourceLanguage'] = undefined;
var _default = UpdateTransformerRequestAllOf;
exports["default"] = _default;
