// @ziqni-tech/admin-api-client/dist/model/UpdateMemberRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MemberType = _interopRequireDefault(require("./MemberType"));

var _UpdateMemberRequestAllOf = _interopRequireDefault(require("./UpdateMemberRequestAllOf"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateMemberRequest model module.
 * @module model/UpdateMemberRequest
 * @version 0.0.26
 */
var UpdateMemberRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateMemberRequest</code>.
   * This is a request for Member update
   * @alias module:model/UpdateMemberRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateMemberRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateMemberRequest(id) {
    _classCallCheck(this, UpdateMemberRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateMemberRequestAllOf["default"].initialize(this);

    UpdateMemberRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateMemberRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateMemberRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateMemberRequest} obj Optional instance to populate.
     * @return {module:model/UpdateMemberRequest} The populated <code>UpdateMemberRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateMemberRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateMemberRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memberType')) {
          obj['memberType'] = _MemberType["default"].constructFromObject(data['memberType']);
        }

        if (data.hasOwnProperty('teamMembers')) {
          obj['teamMembers'] = _ApiClient["default"].convertToType(data['teamMembers'], ['String']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }

        if (data.hasOwnProperty('timeZoneOffset')) {
          obj['timeZoneOffset'] = _ApiClient["default"].convertToType(data['timeZoneOffset'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateMemberRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateMemberRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateMemberRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateMemberRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateMemberRequest.prototype['metadata'] = undefined;
/**
 * The name of the member that is used on leader boards and public displays
 * @member {String} name
 */

UpdateMemberRequest.prototype['name'] = undefined;
/**
 * @member {module:model/MemberType} memberType
 */

UpdateMemberRequest.prototype['memberType'] = undefined;
/**
 * A social group like Guilds
 * @member {Array.<String>} teamMembers
 */

UpdateMemberRequest.prototype['teamMembers'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

UpdateMemberRequest.prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

UpdateMemberRequest.prototype['removeConstraints'] = undefined;
/**
 * To what time zone teh emmber belongs to
 * @member {String} timeZoneOffset
 */

UpdateMemberRequest.prototype['timeZoneOffset'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateMemberRequestAllOf interface:

/**
 * The name of the member that is used on leader boards and public displays
 * @member {String} name
 */

_UpdateMemberRequestAllOf["default"].prototype['name'] = undefined;
/**
 * @member {module:model/MemberType} memberType
 */

_UpdateMemberRequestAllOf["default"].prototype['memberType'] = undefined;
/**
 * A social group like Guilds
 * @member {Array.<String>} teamMembers
 */

_UpdateMemberRequestAllOf["default"].prototype['teamMembers'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_UpdateMemberRequestAllOf["default"].prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

_UpdateMemberRequestAllOf["default"].prototype['removeConstraints'] = undefined;
/**
 * To what time zone teh emmber belongs to
 * @member {String} timeZoneOffset
 */

_UpdateMemberRequestAllOf["default"].prototype['timeZoneOffset'] = undefined;
var _default = UpdateMemberRequest;
exports["default"] = _default;
