// @ziqni-tech/admin-api-client/dist/model/UpdateTagRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateTagRequestAllOf model module.
 * @module model/UpdateTagRequestAllOf
 * @version 0.0.26
 */
var UpdateTagRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTagRequestAllOf</code>.
   * @alias module:model/UpdateTagRequestAllOf
   */
  function UpdateTagRequestAllOf() {
    _classCallCheck(this, UpdateTagRequestAllOf);

    UpdateTagRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateTagRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateTagRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTagRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateTagRequestAllOf} The populated <code>UpdateTagRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTagRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('entityTypes')) {
          obj['entityTypes'] = _ApiClient["default"].convertToType(data['entityTypes'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return UpdateTagRequestAllOf;
}();
/**
 * The name of the tag
 * @member {String} name
 */


UpdateTagRequestAllOf.prototype['name'] = undefined;
/**
 * The description of the tag for your reference
 * @member {String} description
 */

UpdateTagRequestAllOf.prototype['description'] = undefined;
/**
 * The model name the tag refers to
 * @member {Array.<String>} entityTypes
 */

UpdateTagRequestAllOf.prototype['entityTypes'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateTagRequestAllOf.prototype['metadata'] = undefined;
var _default = UpdateTagRequestAllOf;
exports["default"] = _default;
