// @ziqni-tech/admin-api-client/dist/model/UpdateAchievementRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AchievementLiveStatus = _interopRequireDefault(require("./AchievementLiveStatus"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateAchievementRequestAllOf model module.
 * @module model/UpdateAchievementRequestAllOf
 * @version 0.0.26
 */
var UpdateAchievementRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateAchievementRequestAllOf</code>.
   * @alias module:model/UpdateAchievementRequestAllOf
   */
  function UpdateAchievementRequestAllOf() {
    _classCallCheck(this, UpdateAchievementRequestAllOf);

    UpdateAchievementRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateAchievementRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateAchievementRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateAchievementRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateAchievementRequestAllOf} The populated <code>UpdateAchievementRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateAchievementRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('icon')) {
          obj['icon'] = _ApiClient["default"].convertToType(data['icon'], 'String');
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('achievementLiveStatus')) {
          obj['achievementLiveStatus'] = _AchievementLiveStatus["default"].constructFromObject(data['achievementLiveStatus']);
        }

        if (data.hasOwnProperty('maxNumberOfIssues')) {
          obj['maxNumberOfIssues'] = _ApiClient["default"].convertToType(data['maxNumberOfIssues'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('achievementDependencies')) {
          obj['achievementDependencies'] = _DependantOn["default"].constructFromObject(data['achievementDependencies']);
        }

        if (data.hasOwnProperty('memberTagsFilter')) {
          obj['memberTagsFilter'] = _DependantOn["default"].constructFromObject(data['memberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateAchievementRequestAllOf;
}();
/**
 * A name for the Achievement. Can be translated
 * @member {String} name
 */


UpdateAchievementRequestAllOf.prototype['name'] = undefined;
/**
 * A name for the Achievement. Can be translated
 * @member {String} description
 */

UpdateAchievementRequestAllOf.prototype['description'] = undefined;
/**
 * Terms and conditions of an achievement. Can be translated
 * @member {String} termsAndConditions
 */

UpdateAchievementRequestAllOf.prototype['termsAndConditions'] = undefined;
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */

UpdateAchievementRequestAllOf.prototype['icon'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

UpdateAchievementRequestAllOf.prototype['scheduling'] = undefined;
/**
 * @member {module:model/AchievementLiveStatus} achievementLiveStatus
 */

UpdateAchievementRequestAllOf.prototype['achievementLiveStatus'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

UpdateAchievementRequestAllOf.prototype['maxNumberOfIssues'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UpdateAchievementRequestAllOf.prototype['constraints'] = undefined;
/**
 * @member {module:model/DependantOn} achievementDependencies
 */

UpdateAchievementRequestAllOf.prototype['achievementDependencies'] = undefined;
/**
 * @member {module:model/DependantOn} memberTagsFilter
 */

UpdateAchievementRequestAllOf.prototype['memberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

UpdateAchievementRequestAllOf.prototype['productTagsFilter'] = undefined;
/**
 * @member {Array.<String>} addConstraints
 */

UpdateAchievementRequestAllOf.prototype['addConstraints'] = undefined;
/**
 * @member {Array.<String>} removeConstraints
 */

UpdateAchievementRequestAllOf.prototype['removeConstraints'] = undefined;
var _default = UpdateAchievementRequestAllOf;
exports["default"] = _default;
