// @ziqni-tech/admin-api-client/dist/model/UpdateRabbitMqConnectionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

var _UpdateRabbitMqConnectionRequestAllOf = _interopRequireDefault(require("./UpdateRabbitMqConnectionRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateRabbitMqConnectionRequest model module.
 * @module model/UpdateRabbitMqConnectionRequest
 * @version 0.0.26
 */
var UpdateRabbitMqConnectionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRabbitMqConnectionRequest</code>.
   * @alias module:model/UpdateRabbitMqConnectionRequest
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateRabbitMqConnectionRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateRabbitMqConnectionRequest(id) {
    _classCallCheck(this, UpdateRabbitMqConnectionRequest);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateRabbitMqConnectionRequestAllOf["default"].initialize(this);

    UpdateRabbitMqConnectionRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRabbitMqConnectionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateRabbitMqConnectionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRabbitMqConnectionRequest} obj Optional instance to populate.
     * @return {module:model/UpdateRabbitMqConnectionRequest} The populated <code>UpdateRabbitMqConnectionRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRabbitMqConnectionRequest();

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateRabbitMqConnectionRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }

        if (data.hasOwnProperty('virtualHost')) {
          obj['virtualHost'] = _ApiClient["default"].convertToType(data['virtualHost'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'Number');
        }

        if (data.hasOwnProperty('userName')) {
          obj['userName'] = _ApiClient["default"].convertToType(data['userName'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('queueName')) {
          obj['queueName'] = _ApiClient["default"].convertToType(data['queueName'], 'String');
        }

        if (data.hasOwnProperty('exchange')) {
          obj['exchange'] = _ApiClient["default"].convertToType(data['exchange'], 'String');
        }

        if (data.hasOwnProperty('routingKey')) {
          obj['routingKey'] = _ApiClient["default"].convertToType(data['routingKey'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateRabbitMqConnectionRequest;
}();
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */


UpdateRabbitMqConnectionRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateRabbitMqConnectionRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateRabbitMqConnectionRequest.prototype['metadata'] = undefined;
/**
 * A unique system generated identifier
 * @member {String} id
 */

UpdateRabbitMqConnectionRequest.prototype['id'] = undefined;
/**
 * The name of the consumer
 * @member {String} name
 */

UpdateRabbitMqConnectionRequest.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

UpdateRabbitMqConnectionRequest.prototype['description'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

UpdateRabbitMqConnectionRequest.prototype['uri'] = undefined;
/**
 * The virtual host of the rabbitmq broker
 * @member {String} virtualHost
 */

UpdateRabbitMqConnectionRequest.prototype['virtualHost'] = undefined;
/**
 * The port number on which consumer will connect on rabbitmq broker
 * @member {Number} port
 */

UpdateRabbitMqConnectionRequest.prototype['port'] = undefined;
/**
 * Consumer username for authentication
 * @member {String} userName
 */

UpdateRabbitMqConnectionRequest.prototype['userName'] = undefined;
/**
 * Consumer password for authentication
 * @member {String} password
 */

UpdateRabbitMqConnectionRequest.prototype['password'] = undefined;
/**
 * Name of the queue
 * @member {String} queueName
 */

UpdateRabbitMqConnectionRequest.prototype['queueName'] = undefined;
/**
 * Exchange to connect to
 * @member {String} exchange
 */

UpdateRabbitMqConnectionRequest.prototype['exchange'] = undefined;
/**
 * Routing key to use
 * @member {String} routingKey
 */

UpdateRabbitMqConnectionRequest.prototype['routingKey'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UpdateRabbitMqConnectionRequest.prototype['constraints'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

UpdateRabbitMqConnectionRequest.prototype['transformerId'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateRabbitMqConnectionRequestAllOf interface:

/**
 * The name of the consumer
 * @member {String} name
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['description'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['uri'] = undefined;
/**
 * The virtual host of the rabbitmq broker
 * @member {String} virtualHost
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['virtualHost'] = undefined;
/**
 * The port number on which consumer will connect on rabbitmq broker
 * @member {Number} port
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['port'] = undefined;
/**
 * Consumer username for authentication
 * @member {String} userName
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['userName'] = undefined;
/**
 * Consumer password for authentication
 * @member {String} password
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['password'] = undefined;
/**
 * Name of the queue
 * @member {String} queueName
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['queueName'] = undefined;
/**
 * Exchange to connect to
 * @member {String} exchange
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['exchange'] = undefined;
/**
 * Routing key to use
 * @member {String} routingKey
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['routingKey'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['constraints'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

_UpdateRabbitMqConnectionRequestAllOf["default"].prototype['transformerId'] = undefined;
var _default = UpdateRabbitMqConnectionRequest;
exports["default"] = _default;
