// @ziqni-tech/admin-api-client/dist/model/UpdateCustomFieldRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AppliesTo = _interopRequireDefault(require("./AppliesTo"));

var _FieldType = _interopRequireDefault(require("./FieldType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateCustomFieldRequestAllOf model module.
 * @module model/UpdateCustomFieldRequestAllOf
 * @version 0.0.26
 */
var UpdateCustomFieldRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateCustomFieldRequestAllOf</code>.
   * @alias module:model/UpdateCustomFieldRequestAllOf
   */
  function UpdateCustomFieldRequestAllOf() {
    _classCallCheck(this, UpdateCustomFieldRequestAllOf);

    UpdateCustomFieldRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateCustomFieldRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateCustomFieldRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateCustomFieldRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateCustomFieldRequestAllOf} The populated <code>UpdateCustomFieldRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateCustomFieldRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('fieldType')) {
          obj['fieldType'] = _FieldType["default"].constructFromObject(data['fieldType']);
        }

        if (data.hasOwnProperty('appliesTo')) {
          obj['appliesTo'] = _AppliesTo["default"].constructFromObject(data['appliesTo']);
        }
      }

      return obj;
    }
  }]);

  return UpdateCustomFieldRequestAllOf;
}();
/**
 * The name of a Custom field
 * @member {String} name
 */


UpdateCustomFieldRequestAllOf.prototype['name'] = undefined;
/**
 * The description of a Custom field
 * @member {String} description
 */

UpdateCustomFieldRequestAllOf.prototype['description'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

UpdateCustomFieldRequestAllOf.prototype['fieldType'] = undefined;
/**
 * @member {module:model/AppliesTo} appliesTo
 */

UpdateCustomFieldRequestAllOf.prototype['appliesTo'] = undefined;
var _default = UpdateCustomFieldRequestAllOf;
exports["default"] = _default;
