// @ziqni-tech/admin-api-client/dist/model/UpdateActionTypeRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateActionTypeRequestAllOf model module.
 * @module model/UpdateActionTypeRequestAllOf
 * @version 0.0.26
 */
var UpdateActionTypeRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateActionTypeRequestAllOf</code>.
   * @alias module:model/UpdateActionTypeRequestAllOf
   */
  function UpdateActionTypeRequestAllOf() {
    _classCallCheck(this, UpdateActionTypeRequestAllOf);

    UpdateActionTypeRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateActionTypeRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateActionTypeRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateActionTypeRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateActionTypeRequestAllOf} The populated <code>UpdateActionTypeRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateActionTypeRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateActionTypeRequestAllOf;
}();
/**
 * The name of the Action Helper
 * @member {String} name
 */


UpdateActionTypeRequestAllOf.prototype['name'] = undefined;
/**
 * The description of the rule action helper for your reference
 * @member {String} description
 */

UpdateActionTypeRequestAllOf.prototype['description'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

UpdateActionTypeRequestAllOf.prototype['unitOfMeasureType'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} addConstraints
 */

UpdateActionTypeRequestAllOf.prototype['addConstraints'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} removeConstraints
 */

UpdateActionTypeRequestAllOf.prototype['removeConstraints'] = undefined;
var _default = UpdateActionTypeRequestAllOf;
exports["default"] = _default;
