// @ziqni-tech/admin-api-client/dist/model/UpdateCompetitionRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CompetitionType = _interopRequireDefault(require("./CompetitionType"));

var _CreateContestForCompetitionRequest = _interopRequireDefault(require("./CreateContestForCompetitionRequest"));

var _DependantOn = _interopRequireDefault(require("./DependantOn"));

var _UpdateContestRequest = _interopRequireDefault(require("./UpdateContestRequest"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateCompetitionRequestAllOf model module.
 * @module model/UpdateCompetitionRequestAllOf
 * @version 0.0.26
 */
var UpdateCompetitionRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateCompetitionRequestAllOf</code>.
   * @alias module:model/UpdateCompetitionRequestAllOf
   */
  function UpdateCompetitionRequestAllOf() {
    _classCallCheck(this, UpdateCompetitionRequestAllOf);

    UpdateCompetitionRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateCompetitionRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateCompetitionRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateCompetitionRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateCompetitionRequestAllOf} The populated <code>UpdateCompetitionRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateCompetitionRequestAllOf();

        if (data.hasOwnProperty('competitionType')) {
          obj['competitionType'] = _CompetitionType["default"].constructFromObject(data['competitionType']);
        }

        if (data.hasOwnProperty('numberOfRounds')) {
          obj['numberOfRounds'] = _ApiClient["default"].convertToType(data['numberOfRounds'], 'Number');
        }

        if (data.hasOwnProperty('numberOfGroupStages')) {
          obj['numberOfGroupStages'] = _ApiClient["default"].convertToType(data['numberOfGroupStages'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('entrantMemberType')) {
          obj['entrantMemberType'] = _ApiClient["default"].convertToType(data['entrantMemberType'], 'String');
        }

        if (data.hasOwnProperty('maxNumberOfEntrants')) {
          obj['maxNumberOfEntrants'] = _ApiClient["default"].convertToType(data['maxNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('minNumberOfEntrants')) {
          obj['minNumberOfEntrants'] = _ApiClient["default"].convertToType(data['minNumberOfEntrants'], 'Number');
        }

        if (data.hasOwnProperty('scheduledStartDate')) {
          obj['scheduledStartDate'] = _ApiClient["default"].convertToType(data['scheduledStartDate'], 'Date');
        }

        if (data.hasOwnProperty('scheduledEndDate')) {
          obj['scheduledEndDate'] = _ApiClient["default"].convertToType(data['scheduledEndDate'], 'Date');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }

        if (data.hasOwnProperty('productIds')) {
          obj['productIds'] = _ApiClient["default"].convertToType(data['productIds'], ['String']);
        }

        if (data.hasOwnProperty('createContests')) {
          obj['createContests'] = _CreateContestForCompetitionRequest["default"].constructFromObject(data['createContests']);
        }

        if (data.hasOwnProperty('updateContests')) {
          obj['updateContests'] = _UpdateContestRequest["default"].constructFromObject(data['updateContests']);
        }

        if (data.hasOwnProperty('entrantMemberTagsFilter')) {
          obj['entrantMemberTagsFilter'] = _DependantOn["default"].constructFromObject(data['entrantMemberTagsFilter']);
        }

        if (data.hasOwnProperty('productTagsFilter')) {
          obj['productTagsFilter'] = _DependantOn["default"].constructFromObject(data['productTagsFilter']);
        }
      }

      return obj;
    }
  }]);

  return UpdateCompetitionRequestAllOf;
}();
/**
 * @member {module:model/CompetitionType} competitionType
 */


UpdateCompetitionRequestAllOf.prototype['competitionType'] = undefined;
/**
 * Number of rounds to be played in a competition
 * @member {Number} numberOfRounds
 */

UpdateCompetitionRequestAllOf.prototype['numberOfRounds'] = undefined;
/**
 * Number of group stages
 * @member {Number} numberOfGroupStages
 */

UpdateCompetitionRequestAllOf.prototype['numberOfGroupStages'] = undefined;
/**
 * A name or a name of a competition. Can be translated
 * @member {String} name
 */

UpdateCompetitionRequestAllOf.prototype['name'] = undefined;
/**
 * Description of the competition. Can be translated
 * @member {String} description
 */

UpdateCompetitionRequestAllOf.prototype['description'] = undefined;
/**
 * Terms and conditions of a competition. Can be translated
 * @member {String} termsAndConditions
 */

UpdateCompetitionRequestAllOf.prototype['termsAndConditions'] = undefined;
/**
 * @member {String} entrantMemberType
 */

UpdateCompetitionRequestAllOf.prototype['entrantMemberType'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} maxNumberOfEntrants
 */

UpdateCompetitionRequestAllOf.prototype['maxNumberOfEntrants'] = undefined;
/**
 * Maximum number of partiipants allowed in a competition
 * @member {Number} minNumberOfEntrants
 */

UpdateCompetitionRequestAllOf.prototype['minNumberOfEntrants'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should start. All records are stored in UTC time zone
 * @member {Date} scheduledStartDate
 */

UpdateCompetitionRequestAllOf.prototype['scheduledStartDate'] = undefined;
/**
 * ISO8601 timestamp for when a Competition should end. All records are stored in UTC time zone
 * @member {Date} scheduledEndDate
 */

UpdateCompetitionRequestAllOf.prototype['scheduledEndDate'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UpdateCompetitionRequestAllOf.prototype['constraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

UpdateCompetitionRequestAllOf.prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

UpdateCompetitionRequestAllOf.prototype['removeConstraints'] = undefined;
/**
 * @member {Array.<String>} productIds
 */

UpdateCompetitionRequestAllOf.prototype['productIds'] = undefined;
/**
 * @member {module:model/CreateContestForCompetitionRequest} createContests
 */

UpdateCompetitionRequestAllOf.prototype['createContests'] = undefined;
/**
 * @member {module:model/UpdateContestRequest} updateContests
 */

UpdateCompetitionRequestAllOf.prototype['updateContests'] = undefined;
/**
 * @member {module:model/DependantOn} entrantMemberTagsFilter
 */

UpdateCompetitionRequestAllOf.prototype['entrantMemberTagsFilter'] = undefined;
/**
 * @member {module:model/DependantOn} productTagsFilter
 */

UpdateCompetitionRequestAllOf.prototype['productTagsFilter'] = undefined;
var _default = UpdateCompetitionRequestAllOf;
exports["default"] = _default;
