// @ziqni-tech/admin-api-client/dist/model/UpdateCustomFieldRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AppliesTo = _interopRequireDefault(require("./AppliesTo"));

var _FieldType = _interopRequireDefault(require("./FieldType"));

var _UpdateCustomFieldRequestAllOf = _interopRequireDefault(require("./UpdateCustomFieldRequestAllOf"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateCustomFieldRequest model module.
 * @module model/UpdateCustomFieldRequest
 * @version 0.0.26
 */
var UpdateCustomFieldRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateCustomFieldRequest</code>.
   * Request to update custom fields
   * @alias module:model/UpdateCustomFieldRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateCustomFieldRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateCustomFieldRequest(id) {
    _classCallCheck(this, UpdateCustomFieldRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateCustomFieldRequestAllOf["default"].initialize(this);

    UpdateCustomFieldRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateCustomFieldRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateCustomFieldRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateCustomFieldRequest} obj Optional instance to populate.
     * @return {module:model/UpdateCustomFieldRequest} The populated <code>UpdateCustomFieldRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateCustomFieldRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateCustomFieldRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('fieldType')) {
          obj['fieldType'] = _FieldType["default"].constructFromObject(data['fieldType']);
        }

        if (data.hasOwnProperty('appliesTo')) {
          obj['appliesTo'] = _AppliesTo["default"].constructFromObject(data['appliesTo']);
        }
      }

      return obj;
    }
  }]);

  return UpdateCustomFieldRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateCustomFieldRequest.prototype['id'] = undefined;
/**
 * The name of a Custom field
 * @member {String} name
 */

UpdateCustomFieldRequest.prototype['name'] = undefined;
/**
 * The description of a Custom field
 * @member {String} description
 */

UpdateCustomFieldRequest.prototype['description'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

UpdateCustomFieldRequest.prototype['fieldType'] = undefined;
/**
 * @member {module:model/AppliesTo} appliesTo
 */

UpdateCustomFieldRequest.prototype['appliesTo'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateCustomFieldRequestAllOf interface:

/**
 * The name of a Custom field
 * @member {String} name
 */

_UpdateCustomFieldRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of a Custom field
 * @member {String} description
 */

_UpdateCustomFieldRequestAllOf["default"].prototype['description'] = undefined;
/**
 * @member {module:model/FieldType} fieldType
 */

_UpdateCustomFieldRequestAllOf["default"].prototype['fieldType'] = undefined;
/**
 * @member {module:model/AppliesTo} appliesTo
 */

_UpdateCustomFieldRequestAllOf["default"].prototype['appliesTo'] = undefined;
var _default = UpdateCustomFieldRequest;
exports["default"] = _default;
