// @ziqni-tech/admin-api-client/dist/model/UpdateSqsConnectionRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateSqsConnectionRequestAllOf model module.
 * @module model/UpdateSqsConnectionRequestAllOf
 * @version 0.0.26
 */
var UpdateSqsConnectionRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateSqsConnectionRequestAllOf</code>.
   * @alias module:model/UpdateSqsConnectionRequestAllOf
   */
  function UpdateSqsConnectionRequestAllOf() {
    _classCallCheck(this, UpdateSqsConnectionRequestAllOf);

    UpdateSqsConnectionRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateSqsConnectionRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateSqsConnectionRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateSqsConnectionRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateSqsConnectionRequestAllOf} The populated <code>UpdateSqsConnectionRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateSqsConnectionRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }

        if (data.hasOwnProperty('acessKey')) {
          obj['acessKey'] = _ApiClient["default"].convertToType(data['acessKey'], 'String');
        }

        if (data.hasOwnProperty('secretKey')) {
          obj['secretKey'] = _ApiClient["default"].convertToType(data['secretKey'], 'String');
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateSqsConnectionRequestAllOf;
}();
/**
 * The name of the consumer
 * @member {String} name
 */


UpdateSqsConnectionRequestAllOf.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

UpdateSqsConnectionRequestAllOf.prototype['description'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

UpdateSqsConnectionRequestAllOf.prototype['uri'] = undefined;
/**
 * The access key of the IAM user
 * @member {String} acessKey
 */

UpdateSqsConnectionRequestAllOf.prototype['acessKey'] = undefined;
/**
 * The secret key of IAM user
 * @member {String} secretKey
 */

UpdateSqsConnectionRequestAllOf.prototype['secretKey'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

UpdateSqsConnectionRequestAllOf.prototype['transformerId'] = undefined;
var _default = UpdateSqsConnectionRequestAllOf;
exports["default"] = _default;
