// @ziqni-tech/admin-api-client/dist/model/UpdateMessageRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Reward = _interopRequireDefault(require("./Reward"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

var _UpdateMessageRequestAllOf = _interopRequireDefault(require("./UpdateMessageRequestAllOf"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateMessageRequest model module.
 * @module model/UpdateMessageRequest
 * @version 0.0.5
 */
var UpdateMessageRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateMessageRequest</code>.
   * @alias module:model/UpdateMessageRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateMessageRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateMessageRequest(id) {
    _classCallCheck(this, UpdateMessageRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateMessageRequestAllOf["default"].initialize(this);

    UpdateMessageRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateMessageRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateMessageRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateMessageRequest} obj Optional instance to populate.
     * @return {module:model/UpdateMessageRequest} The populated <code>UpdateMessageRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateMessageRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateMessageRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _ApiClient["default"].convertToType(data['dependantOn'], [_Dependancy["default"]]);
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }

        if (data.hasOwnProperty('prize')) {
          obj['prize'] = _Reward["default"].constructFromObject(data['prize']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('eventReferenceId')) {
          obj['eventReferenceId'] = _ApiClient["default"].convertToType(data['eventReferenceId'], 'String');
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _ApiClient["default"].convertToType(data['eventRefType'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateMessageRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateMessageRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateMessageRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateMessageRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateMessageRequest.prototype['metadata'] = undefined;
/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

UpdateMessageRequest.prototype['dependantOn'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

UpdateMessageRequest.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

UpdateMessageRequest.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

UpdateMessageRequest.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

UpdateMessageRequest.prototype['status'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

UpdateMessageRequest.prototype['scheduling'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

UpdateMessageRequest.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

UpdateMessageRequest.prototype['translatableFields'] = undefined;
/**
 * @member {module:model/Reward} prize
 */

UpdateMessageRequest.prototype['prize'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

UpdateMessageRequest.prototype['constraints'] = undefined;
/**
 * @member {String} eventReferenceId
 */

UpdateMessageRequest.prototype['eventReferenceId'] = undefined;
/**
 * @member {String} eventRefType
 */

UpdateMessageRequest.prototype['eventRefType'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateMessageRequestAllOf interface:

/**
 * @member {Array.<module:model/Dependancy>} dependantOn
 */

_UpdateMessageRequestAllOf["default"].prototype['dependantOn'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

_UpdateMessageRequestAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_UpdateMessageRequestAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_UpdateMessageRequestAllOf["default"].prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_UpdateMessageRequestAllOf["default"].prototype['status'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

_UpdateMessageRequestAllOf["default"].prototype['scheduling'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_UpdateMessageRequestAllOf["default"].prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

_UpdateMessageRequestAllOf["default"].prototype['translatableFields'] = undefined;
/**
 * @member {module:model/Reward} prize
 */

_UpdateMessageRequestAllOf["default"].prototype['prize'] = undefined;
/**
 * Additional constraints, if the value is present it means the
 * @member {Array.<String>} constraints
 */

_UpdateMessageRequestAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {String} eventReferenceId
 */

_UpdateMessageRequestAllOf["default"].prototype['eventReferenceId'] = undefined;
/**
 * @member {String} eventRefType
 */

_UpdateMessageRequestAllOf["default"].prototype['eventRefType'] = undefined;
var _default = UpdateMessageRequest;
exports["default"] = _default;
