// @ziqni-tech/admin-api-client/dist/model/UpdateUnitOfMeasureRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UnitOfMeasureType = _interopRequireDefault(require("./UnitOfMeasureType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateUnitOfMeasureRequestAllOf model module.
 * @module model/UpdateUnitOfMeasureRequestAllOf
 * @version 0.0.26
 */
var UpdateUnitOfMeasureRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateUnitOfMeasureRequestAllOf</code>.
   * @alias module:model/UpdateUnitOfMeasureRequestAllOf
   */
  function UpdateUnitOfMeasureRequestAllOf() {
    _classCallCheck(this, UpdateUnitOfMeasureRequestAllOf);

    UpdateUnitOfMeasureRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateUnitOfMeasureRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateUnitOfMeasureRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateUnitOfMeasureRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateUnitOfMeasureRequestAllOf} The populated <code>UpdateUnitOfMeasureRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateUnitOfMeasureRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('isoCode')) {
          obj['isoCode'] = _ApiClient["default"].convertToType(data['isoCode'], 'String');
        }

        if (data.hasOwnProperty('symbol')) {
          obj['symbol'] = _ApiClient["default"].convertToType(data['symbol'], 'String');
        }

        if (data.hasOwnProperty('multiplier')) {
          obj['multiplier'] = _ApiClient["default"].convertToType(data['multiplier'], 'Number');
        }

        if (data.hasOwnProperty('unitOfMeasureType')) {
          obj['unitOfMeasureType'] = _UnitOfMeasureType["default"].constructFromObject(data['unitOfMeasureType']);
        }
      }

      return obj;
    }
  }]);

  return UpdateUnitOfMeasureRequestAllOf;
}();
/**
 * The name of a unit of measure
 * @member {String} name
 */


UpdateUnitOfMeasureRequestAllOf.prototype['name'] = undefined;
/**
 * The description of a unit of measure
 * @member {String} description
 */

UpdateUnitOfMeasureRequestAllOf.prototype['description'] = undefined;
/**
 * An alphabetical or numerical code to identify a unit of measure
 * @member {String} isoCode
 */

UpdateUnitOfMeasureRequestAllOf.prototype['isoCode'] = undefined;
/**
 * The symbol of a unit of measure
 * @member {String} symbol
 */

UpdateUnitOfMeasureRequestAllOf.prototype['symbol'] = undefined;
/**
 * Is used to multiply the value from the standardised one that is being used
 * @member {Number} multiplier
 */

UpdateUnitOfMeasureRequestAllOf.prototype['multiplier'] = undefined;
/**
 * @member {module:model/UnitOfMeasureType} unitOfMeasureType
 */

UpdateUnitOfMeasureRequestAllOf.prototype['unitOfMeasureType'] = undefined;
var _default = UpdateUnitOfMeasureRequestAllOf;
exports["default"] = _default;
