// @ziqni-tech/admin-api-client/dist/model/UpdateKafkaConnectionRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateKafkaConnectionRequestAllOf model module.
 * @module model/UpdateKafkaConnectionRequestAllOf
 * @version 0.0.26
 */
var UpdateKafkaConnectionRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateKafkaConnectionRequestAllOf</code>.
   * @alias module:model/UpdateKafkaConnectionRequestAllOf
   */
  function UpdateKafkaConnectionRequestAllOf() {
    _classCallCheck(this, UpdateKafkaConnectionRequestAllOf);

    UpdateKafkaConnectionRequestAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateKafkaConnectionRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateKafkaConnectionRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateKafkaConnectionRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateKafkaConnectionRequestAllOf} The populated <code>UpdateKafkaConnectionRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateKafkaConnectionRequestAllOf();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('brokers')) {
          obj['brokers'] = _ApiClient["default"].convertToType(data['brokers'], 'String');
        }

        if (data.hasOwnProperty('groupId')) {
          obj['groupId'] = _ApiClient["default"].convertToType(data['groupId'], 'String');
        }

        if (data.hasOwnProperty('topic')) {
          obj['topic'] = _ApiClient["default"].convertToType(data['topic'], 'String');
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateKafkaConnectionRequestAllOf;
}();
/**
 * The name of the consumer
 * @member {String} name
 */


UpdateKafkaConnectionRequestAllOf.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

UpdateKafkaConnectionRequestAllOf.prototype['description'] = undefined;
/**
 * Kafka broker endpoints.
 * @member {String} brokers
 */

UpdateKafkaConnectionRequestAllOf.prototype['brokers'] = undefined;
/**
 * Group Id for connection
 * @member {String} groupId
 */

UpdateKafkaConnectionRequestAllOf.prototype['groupId'] = undefined;
/**
 * Topic name
 * @member {String} topic
 */

UpdateKafkaConnectionRequestAllOf.prototype['topic'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

UpdateKafkaConnectionRequestAllOf.prototype['transformerId'] = undefined;
var _default = UpdateKafkaConnectionRequestAllOf;
exports["default"] = _default;
