// @ziqni-tech/admin-api-client/dist/model/UpdateComputeEngineRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UpdateComputeEngineRequestAllOf = _interopRequireDefault(require("./UpdateComputeEngineRequestAllOf"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateComputeEngineRequest model module.
 * @module model/UpdateComputeEngineRequest
 * @version 0.0.26
 */
var UpdateComputeEngineRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateComputeEngineRequest</code>.
   * @alias module:model/UpdateComputeEngineRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateComputeEngineRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateComputeEngineRequest(id) {
    _classCallCheck(this, UpdateComputeEngineRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateComputeEngineRequestAllOf["default"].initialize(this);

    UpdateComputeEngineRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateComputeEngineRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateComputeEngineRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateComputeEngineRequest} obj Optional instance to populate.
     * @return {module:model/UpdateComputeEngineRequest} The populated <code>UpdateComputeEngineRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateComputeEngineRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateComputeEngineRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('uptime')) {
          obj['uptime'] = _ApiClient["default"].convertToType(data['uptime'], 'Number');
        }

        if (data.hasOwnProperty('statistics')) {
          obj['statistics'] = _ApiClient["default"].convertToType(data['statistics'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('lastKnownStatusCode')) {
          obj['lastKnownStatusCode'] = _ApiClient["default"].convertToType(data['lastKnownStatusCode'], 'Number');
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateComputeEngineRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateComputeEngineRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateComputeEngineRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateComputeEngineRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateComputeEngineRequest.prototype['metadata'] = undefined;
/**
 * @member {String} description
 */

UpdateComputeEngineRequest.prototype['description'] = undefined;
/**
 * @member {Number} version
 */

UpdateComputeEngineRequest.prototype['version'] = undefined;
/**
 * @member {String} name
 */

UpdateComputeEngineRequest.prototype['name'] = undefined;
/**
 * Long
 * @member {Number} uptime
 */

UpdateComputeEngineRequest.prototype['uptime'] = undefined;
/**
 * @member {Object.<String, String>} statistics
 */

UpdateComputeEngineRequest.prototype['statistics'] = undefined;
/**
 * @member {Number} lastKnownStatusCode
 */

UpdateComputeEngineRequest.prototype['lastKnownStatusCode'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

UpdateComputeEngineRequest.prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

UpdateComputeEngineRequest.prototype['removeConstraints'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateComputeEngineRequestAllOf interface:

/**
 * @member {String} description
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['description'] = undefined;
/**
 * @member {Number} version
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['version'] = undefined;
/**
 * @member {String} name
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['name'] = undefined;
/**
 * Long
 * @member {Number} uptime
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['uptime'] = undefined;
/**
 * @member {Object.<String, String>} statistics
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['statistics'] = undefined;
/**
 * @member {Number} lastKnownStatusCode
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['lastKnownStatusCode'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['metadata'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

_UpdateComputeEngineRequestAllOf["default"].prototype['removeConstraints'] = undefined;
var _default = UpdateComputeEngineRequest;
exports["default"] = _default;
