// @ziqni-tech/admin-api-client/dist/model/UpdateNotificationRequestAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateNotificationRequestAllOf model module.
 * @module model/UpdateNotificationRequestAllOf
 * @version 0.0.5
 */
var UpdateNotificationRequestAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateNotificationRequestAllOf</code>.
   * @alias module:model/UpdateNotificationRequestAllOf
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param status {module:model/MessageStatus} 
   * @param memberId {Array.<String>} The reference ID of the event object
   * @param scheduling {module:model/Scheduling} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function UpdateNotificationRequestAllOf(messageType, subject, body, status, memberId, scheduling, constraints) {
    _classCallCheck(this, UpdateNotificationRequestAllOf);

    UpdateNotificationRequestAllOf.initialize(this, messageType, subject, body, status, memberId, scheduling, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateNotificationRequestAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, messageType, subject, body, status, memberId, scheduling, constraints) {
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['status'] = status;
      obj['memberId'] = memberId;
      obj['scheduling'] = scheduling;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>UpdateNotificationRequestAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateNotificationRequestAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateNotificationRequestAllOf} The populated <code>UpdateNotificationRequestAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateNotificationRequestAllOf();

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }

        if (data.hasOwnProperty('memberGroup')) {
          obj['memberGroup'] = _ApiClient["default"].convertToType(data['memberGroup'], ['String']);
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], ['String']);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateNotificationRequestAllOf;
}();
/**
 * @member {module:model/MessageType} messageType
 */


UpdateNotificationRequestAllOf.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

UpdateNotificationRequestAllOf.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

UpdateNotificationRequestAllOf.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

UpdateNotificationRequestAllOf.prototype['status'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

UpdateNotificationRequestAllOf.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

UpdateNotificationRequestAllOf.prototype['translatableFields'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberGroup
 */

UpdateNotificationRequestAllOf.prototype['memberGroup'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

UpdateNotificationRequestAllOf.prototype['memberId'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

UpdateNotificationRequestAllOf.prototype['scheduling'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UpdateNotificationRequestAllOf.prototype['constraints'] = undefined;
var _default = UpdateNotificationRequestAllOf;
exports["default"] = _default;
