// @ziqni-tech/admin-api-client/dist/model/UpdateTranslationRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _TranslationEntry = _interopRequireDefault(require("./TranslationEntry"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateTranslationRequestAllOf = _interopRequireDefault(require("./UpdateTranslationRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateTranslationRequest model module.
 * @module model/UpdateTranslationRequest
 * @version 0.0.26
 */
var UpdateTranslationRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTranslationRequest</code>.
   * @alias module:model/UpdateTranslationRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateTranslationRequestAllOf
   * @param id {String} A unique system generated identifier
   * @param translations {Array.<module:model/TranslationEntry>} 
   */
  function UpdateTranslationRequest(id, translations) {
    _classCallCheck(this, UpdateTranslationRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateTranslationRequestAllOf["default"].initialize(this, translations);

    UpdateTranslationRequest.initialize(this, id, translations);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateTranslationRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id, translations) {
      obj['id'] = id;
      obj['translations'] = translations;
    }
    /**
     * Constructs a <code>UpdateTranslationRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTranslationRequest} obj Optional instance to populate.
     * @return {module:model/UpdateTranslationRequest} The populated <code>UpdateTranslationRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTranslationRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateTranslationRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('entityType')) {
          obj['entityType'] = _ApiClient["default"].convertToType(data['entityType'], 'String');
        }

        if (data.hasOwnProperty('languageId')) {
          obj['languageId'] = _ApiClient["default"].convertToType(data['languageId'], 'String');
        }

        if (data.hasOwnProperty('languageKey')) {
          obj['languageKey'] = _ApiClient["default"].convertToType(data['languageKey'], 'String');
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_TranslationEntry["default"]]);
        }
      }

      return obj;
    }
  }]);

  return UpdateTranslationRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateTranslationRequest.prototype['id'] = undefined;
/**
 * @member {String} entityId
 */

UpdateTranslationRequest.prototype['entityId'] = undefined;
/**
 * @member {String} entityType
 */

UpdateTranslationRequest.prototype['entityType'] = undefined;
/**
 * @member {String} languageId
 */

UpdateTranslationRequest.prototype['languageId'] = undefined;
/**
 * @member {String} languageKey
 */

UpdateTranslationRequest.prototype['languageKey'] = undefined;
/**
 * @member {Array.<module:model/TranslationEntry>} translations
 */

UpdateTranslationRequest.prototype['translations'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateTranslationRequestAllOf interface:

/**
 * @member {String} entityId
 */

_UpdateTranslationRequestAllOf["default"].prototype['entityId'] = undefined;
/**
 * @member {String} entityType
 */

_UpdateTranslationRequestAllOf["default"].prototype['entityType'] = undefined;
/**
 * @member {String} languageId
 */

_UpdateTranslationRequestAllOf["default"].prototype['languageId'] = undefined;
/**
 * @member {String} languageKey
 */

_UpdateTranslationRequestAllOf["default"].prototype['languageKey'] = undefined;
/**
 * @member {Array.<module:model/TranslationEntry>} translations
 */

_UpdateTranslationRequestAllOf["default"].prototype['translations'] = undefined;
var _default = UpdateTranslationRequest;
exports["default"] = _default;
