// @ziqni-tech/admin-api-client/dist/model/UpdateSpaceRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateSpaceRequest model module.
 * @module model/UpdateSpaceRequest
 * @version 0.0.26
 */
var UpdateSpaceRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateSpaceRequest</code>.
   * @alias module:model/UpdateSpaceRequest
   */
  function UpdateSpaceRequest() {
    _classCallCheck(this, UpdateSpaceRequest);

    UpdateSpaceRequest.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateSpaceRequest, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateSpaceRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateSpaceRequest} obj Optional instance to populate.
     * @return {module:model/UpdateSpaceRequest} The populated <code>UpdateSpaceRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateSpaceRequest();

        if (data.hasOwnProperty('accountType')) {
          obj['accountType'] = _ApiClient["default"].convertToType(data['accountType'], 'String');
        }

        if (data.hasOwnProperty('masterSpace')) {
          obj['masterSpace'] = _ApiClient["default"].convertToType(data['masterSpace'], 'String');
        }

        if (data.hasOwnProperty('unitsOfMeasure')) {
          obj['unitsOfMeasure'] = _ApiClient["default"].convertToType(data['unitsOfMeasure'], ['String']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateSpaceRequest;
}();
/**
 * Type of account; personal or company or vendor
 * @member {String} accountType
 */


UpdateSpaceRequest.prototype['accountType'] = undefined;
/**
 * The master account for this account
 * @member {String} masterSpace
 */

UpdateSpaceRequest.prototype['masterSpace'] = undefined;
/**
 * Foreign key to unit of measure id.
 * @member {Array.<String>} unitsOfMeasure
 */

UpdateSpaceRequest.prototype['unitsOfMeasure'] = undefined;
/**
 * Additional constraints, if set means true
 * @member {Array.<String>} constraints
 */

UpdateSpaceRequest.prototype['constraints'] = undefined;
var _default = UpdateSpaceRequest;
exports["default"] = _default;
