// @ziqni-tech/admin-api-client/dist/model/UpdateRepositoryRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _HostingOptions = _interopRequireDefault(require("./HostingOptions"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

var _UpdateRepositoryRequestAllOf = _interopRequireDefault(require("./UpdateRepositoryRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateRepositoryRequest model module.
 * @module model/UpdateRepositoryRequest
 * @version 0.0.26
 */
var UpdateRepositoryRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRepositoryRequest</code>.
   * @alias module:model/UpdateRepositoryRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateRepositoryRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateRepositoryRequest(id) {
    _classCallCheck(this, UpdateRepositoryRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateRepositoryRequestAllOf["default"].initialize(this);

    UpdateRepositoryRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRepositoryRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateRepositoryRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRepositoryRequest} obj Optional instance to populate.
     * @return {module:model/UpdateRepositoryRequest} The populated <code>UpdateRepositoryRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRepositoryRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateRepositoryRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('addConstraints')) {
          obj['addConstraints'] = _ApiClient["default"].convertToType(data['addConstraints'], ['String']);
        }

        if (data.hasOwnProperty('removeConstraints')) {
          obj['removeConstraints'] = _ApiClient["default"].convertToType(data['removeConstraints'], ['String']);
        }

        if (data.hasOwnProperty('hostingOptions')) {
          obj['hostingOptions'] = _HostingOptions["default"].constructFromObject(data['hostingOptions']);
        }
      }

      return obj;
    }
  }]);

  return UpdateRepositoryRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateRepositoryRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateRepositoryRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateRepositoryRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateRepositoryRequest.prototype['metadata'] = undefined;
/**
 * The name of the repository. No special charaters or spaces permitted
 * @member {String} name
 */

UpdateRepositoryRequest.prototype['name'] = undefined;
/**
 * The description of the repository
 * @member {String} description
 */

UpdateRepositoryRequest.prototype['description'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

UpdateRepositoryRequest.prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

UpdateRepositoryRequest.prototype['removeConstraints'] = undefined;
/**
 * @member {module:model/HostingOptions} hostingOptions
 */

UpdateRepositoryRequest.prototype['hostingOptions'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateRepositoryRequestAllOf interface:

/**
 * The name of the repository. No special charaters or spaces permitted
 * @member {String} name
 */

_UpdateRepositoryRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the repository
 * @member {String} description
 */

_UpdateRepositoryRequestAllOf["default"].prototype['description'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} addConstraints
 */

_UpdateRepositoryRequestAllOf["default"].prototype['addConstraints'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} removeConstraints
 */

_UpdateRepositoryRequestAllOf["default"].prototype['removeConstraints'] = undefined;
/**
 * @member {module:model/HostingOptions} hostingOptions
 */

_UpdateRepositoryRequestAllOf["default"].prototype['hostingOptions'] = undefined;
var _default = UpdateRepositoryRequest;
exports["default"] = _default;
