// @ziqni-tech/admin-api-client/dist/model/UpdateNotificationRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

var _Translation = _interopRequireDefault(require("./Translation"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateNotificationRequestAllOf = _interopRequireDefault(require("./UpdateNotificationRequestAllOf"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateNotificationRequest model module.
 * @module model/UpdateNotificationRequest
 * @version 0.0.5
 */
var UpdateNotificationRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateNotificationRequest</code>.
   * @alias module:model/UpdateNotificationRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateNotificationRequestAllOf
   * @param id {String} A unique system generated identifier
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param status {module:model/MessageStatus} 
   * @param memberId {Array.<String>} The reference ID of the event object
   * @param scheduling {module:model/Scheduling} 
   * @param constraints {Array.<String>} Additional constraints
   */
  function UpdateNotificationRequest(id, messageType, subject, body, status, memberId, scheduling, constraints) {
    _classCallCheck(this, UpdateNotificationRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateNotificationRequestAllOf["default"].initialize(this, messageType, subject, body, status, memberId, scheduling, constraints);

    UpdateNotificationRequest.initialize(this, id, messageType, subject, body, status, memberId, scheduling, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateNotificationRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id, messageType, subject, body, status, memberId, scheduling, constraints) {
      obj['id'] = id;
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['status'] = status;
      obj['memberId'] = memberId;
      obj['scheduling'] = scheduling;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>UpdateNotificationRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateNotificationRequest} obj Optional instance to populate.
     * @return {module:model/UpdateNotificationRequest} The populated <code>UpdateNotificationRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateNotificationRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateNotificationRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }

        if (data.hasOwnProperty('memberGroup')) {
          obj['memberGroup'] = _ApiClient["default"].convertToType(data['memberGroup'], ['String']);
        }

        if (data.hasOwnProperty('memberId')) {
          obj['memberId'] = _ApiClient["default"].convertToType(data['memberId'], ['String']);
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateNotificationRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateNotificationRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateNotificationRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateNotificationRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateNotificationRequest.prototype['metadata'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

UpdateNotificationRequest.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

UpdateNotificationRequest.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

UpdateNotificationRequest.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

UpdateNotificationRequest.prototype['status'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

UpdateNotificationRequest.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

UpdateNotificationRequest.prototype['translatableFields'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberGroup
 */

UpdateNotificationRequest.prototype['memberGroup'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

UpdateNotificationRequest.prototype['memberId'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

UpdateNotificationRequest.prototype['scheduling'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UpdateNotificationRequest.prototype['constraints'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateNotificationRequestAllOf interface:

/**
 * @member {module:model/MessageType} messageType
 */

_UpdateNotificationRequestAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_UpdateNotificationRequestAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_UpdateNotificationRequestAllOf["default"].prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_UpdateNotificationRequestAllOf["default"].prototype['status'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_UpdateNotificationRequestAllOf["default"].prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

_UpdateNotificationRequestAllOf["default"].prototype['translatableFields'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberGroup
 */

_UpdateNotificationRequestAllOf["default"].prototype['memberGroup'] = undefined;
/**
 * The reference ID of the event object
 * @member {Array.<String>} memberId
 */

_UpdateNotificationRequestAllOf["default"].prototype['memberId'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

_UpdateNotificationRequestAllOf["default"].prototype['scheduling'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_UpdateNotificationRequestAllOf["default"].prototype['constraints'] = undefined;
var _default = UpdateNotificationRequest;
exports["default"] = _default;
