// @ziqni-tech/admin-api-client/dist/model/UpdateSqsConnectionRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

var _UpdateSqsConnectionRequestAllOf = _interopRequireDefault(require("./UpdateSqsConnectionRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateSqsConnectionRequest model module.
 * @module model/UpdateSqsConnectionRequest
 * @version 0.0.26
 */
var UpdateSqsConnectionRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateSqsConnectionRequest</code>.
   * @alias module:model/UpdateSqsConnectionRequest
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateSqsConnectionRequestAllOf
   * @implements module:model/UpdateModelDefault
   * @param id {String} A unique system generated identifier
   */
  function UpdateSqsConnectionRequest(id) {
    _classCallCheck(this, UpdateSqsConnectionRequest);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateSqsConnectionRequestAllOf["default"].initialize(this);

    _UpdateModelDefault["default"].initialize(this, id);

    UpdateSqsConnectionRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateSqsConnectionRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateSqsConnectionRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateSqsConnectionRequest} obj Optional instance to populate.
     * @return {module:model/UpdateSqsConnectionRequest} The populated <code>UpdateSqsConnectionRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateSqsConnectionRequest();

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateSqsConnectionRequestAllOf["default"].constructFromObject(data, obj);

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }

        if (data.hasOwnProperty('acessKey')) {
          obj['acessKey'] = _ApiClient["default"].convertToType(data['acessKey'], 'String');
        }

        if (data.hasOwnProperty('secretKey')) {
          obj['secretKey'] = _ApiClient["default"].convertToType(data['secretKey'], 'String');
        }

        if (data.hasOwnProperty('transformerId')) {
          obj['transformerId'] = _ApiClient["default"].convertToType(data['transformerId'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateSqsConnectionRequest;
}();
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */


UpdateSqsConnectionRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateSqsConnectionRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateSqsConnectionRequest.prototype['metadata'] = undefined;
/**
 * The name of the consumer
 * @member {String} name
 */

UpdateSqsConnectionRequest.prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

UpdateSqsConnectionRequest.prototype['description'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

UpdateSqsConnectionRequest.prototype['uri'] = undefined;
/**
 * The access key of the IAM user
 * @member {String} acessKey
 */

UpdateSqsConnectionRequest.prototype['acessKey'] = undefined;
/**
 * The secret key of IAM user
 * @member {String} secretKey
 */

UpdateSqsConnectionRequest.prototype['secretKey'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

UpdateSqsConnectionRequest.prototype['transformerId'] = undefined;
/**
 * A unique system generated identifier
 * @member {String} id
 */

UpdateSqsConnectionRequest.prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateSqsConnectionRequestAllOf interface:

/**
 * The name of the consumer
 * @member {String} name
 */

_UpdateSqsConnectionRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the consumer
 * @member {String} description
 */

_UpdateSqsConnectionRequestAllOf["default"].prototype['description'] = undefined;
/**
 * The SQS endpoint.
 * @member {String} uri
 */

_UpdateSqsConnectionRequestAllOf["default"].prototype['uri'] = undefined;
/**
 * The access key of the IAM user
 * @member {String} acessKey
 */

_UpdateSqsConnectionRequestAllOf["default"].prototype['acessKey'] = undefined;
/**
 * The secret key of IAM user
 * @member {String} secretKey
 */

_UpdateSqsConnectionRequestAllOf["default"].prototype['secretKey'] = undefined;
/**
 * The transformer to use, if empty the default system transformer will be used
 * @member {String} transformerId
 */

_UpdateSqsConnectionRequestAllOf["default"].prototype['transformerId'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined;
var _default = UpdateSqsConnectionRequest;
exports["default"] = _default;
