// @ziqni-tech/admin-api-client/dist/model/UpdateAccountMessageRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccountMessageAllOf = _interopRequireDefault(require("./AccountMessageAllOf"));

var _EventRefType = _interopRequireDefault(require("./EventRefType"));

var _MessageStatus = _interopRequireDefault(require("./MessageStatus"));

var _MessageType = _interopRequireDefault(require("./MessageType"));

var _Translation = _interopRequireDefault(require("./Translation"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateAccountMessageRequest model module.
 * @module model/UpdateAccountMessageRequest
 * @version 0.0.5
 */
var UpdateAccountMessageRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateAccountMessageRequest</code>.
   * @alias module:model/UpdateAccountMessageRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/AccountMessageAllOf
   * @param id {String} A unique system generated identifier
   * @param eventRefType {module:model/EventRefType} 
   * @param eventRefId {String} The reference ID of the event object
   * @param messageType {module:model/MessageType} 
   * @param subject {String} The title of the message
   * @param body {String} The context of the message
   * @param status {module:model/MessageStatus} 
   * @param userId {String} 
   */
  function UpdateAccountMessageRequest(id, eventRefType, eventRefId, messageType, subject, body, status, userId) {
    _classCallCheck(this, UpdateAccountMessageRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _AccountMessageAllOf["default"].initialize(this, eventRefType, eventRefId, messageType, subject, body, status, userId);

    UpdateAccountMessageRequest.initialize(this, id, eventRefType, eventRefId, messageType, subject, body, status, userId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateAccountMessageRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id, eventRefType, eventRefId, messageType, subject, body, status, userId) {
      obj['id'] = id;
      obj['eventRefType'] = eventRefType;
      obj['eventRefId'] = eventRefId;
      obj['messageType'] = messageType;
      obj['subject'] = subject;
      obj['body'] = body;
      obj['status'] = status;
      obj['userId'] = userId;
    }
    /**
     * Constructs a <code>UpdateAccountMessageRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateAccountMessageRequest} obj Optional instance to populate.
     * @return {module:model/UpdateAccountMessageRequest} The populated <code>UpdateAccountMessageRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateAccountMessageRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _AccountMessageAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('expiry')) {
          obj['expiry'] = _ApiClient["default"].convertToType(data['expiry'], 'Date');
        }

        if (data.hasOwnProperty('eventRefType')) {
          obj['eventRefType'] = _EventRefType["default"].constructFromObject(data['eventRefType']);
        }

        if (data.hasOwnProperty('eventRefId')) {
          obj['eventRefId'] = _ApiClient["default"].convertToType(data['eventRefId'], 'String');
        }

        if (data.hasOwnProperty('messageType')) {
          obj['messageType'] = _MessageType["default"].constructFromObject(data['messageType']);
        }

        if (data.hasOwnProperty('subject')) {
          obj['subject'] = _ApiClient["default"].convertToType(data['subject'], 'String');
        }

        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _MessageStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('translations')) {
          obj['translations'] = _ApiClient["default"].convertToType(data['translations'], [_Translation["default"]]);
        }

        if (data.hasOwnProperty('translatableFields')) {
          obj['translatableFields'] = _ApiClient["default"].convertToType(data['translatableFields'], ['String']);
        }

        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateAccountMessageRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateAccountMessageRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateAccountMessageRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateAccountMessageRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateAccountMessageRequest.prototype['metadata'] = undefined;
/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */

UpdateAccountMessageRequest.prototype['expiry'] = undefined;
/**
 * @member {module:model/EventRefType} eventRefType
 */

UpdateAccountMessageRequest.prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

UpdateAccountMessageRequest.prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

UpdateAccountMessageRequest.prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

UpdateAccountMessageRequest.prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

UpdateAccountMessageRequest.prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

UpdateAccountMessageRequest.prototype['status'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

UpdateAccountMessageRequest.prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

UpdateAccountMessageRequest.prototype['translatableFields'] = undefined;
/**
 * @member {String} userId
 */

UpdateAccountMessageRequest.prototype['userId'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement AccountMessageAllOf interface:

/**
 * The time that the message will disappear after. ISO8601 timestamp
 * @member {Date} expiry
 */

_AccountMessageAllOf["default"].prototype['expiry'] = undefined;
/**
 * @member {module:model/EventRefType} eventRefType
 */

_AccountMessageAllOf["default"].prototype['eventRefType'] = undefined;
/**
 * The reference ID of the event object
 * @member {String} eventRefId
 */

_AccountMessageAllOf["default"].prototype['eventRefId'] = undefined;
/**
 * @member {module:model/MessageType} messageType
 */

_AccountMessageAllOf["default"].prototype['messageType'] = undefined;
/**
 * The title of the message
 * @member {String} subject
 */

_AccountMessageAllOf["default"].prototype['subject'] = undefined;
/**
 * The context of the message
 * @member {String} body
 */

_AccountMessageAllOf["default"].prototype['body'] = undefined;
/**
 * @member {module:model/MessageStatus} status
 */

_AccountMessageAllOf["default"].prototype['status'] = undefined;
/**
 * @member {Array.<module:model/Translation>} translations
 */

_AccountMessageAllOf["default"].prototype['translations'] = undefined;
/**
 * Message translatable fields
 * @member {Array.<String>} translatableFields
 */

_AccountMessageAllOf["default"].prototype['translatableFields'] = undefined;
/**
 * @member {String} userId
 */

_AccountMessageAllOf["default"].prototype['userId'] = undefined;
var _default = UpdateAccountMessageRequest;
exports["default"] = _default;
