// @ziqni-tech/admin-api-client/dist/model/UpdateCollaboratorRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UpdateCollaboratorRequestAllOf = _interopRequireDefault(require("./UpdateCollaboratorRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateCollaboratorRequest model module.
 * @module model/UpdateCollaboratorRequest
 * @version 0.0.26
 */
var UpdateCollaboratorRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateCollaboratorRequest</code>.
   * @alias module:model/UpdateCollaboratorRequest
   * @implements module:model/UpdateCollaboratorRequestAllOf
   * @param email {String} 
   */
  function UpdateCollaboratorRequest(email) {
    _classCallCheck(this, UpdateCollaboratorRequest);

    _UpdateCollaboratorRequestAllOf["default"].initialize(this, email);

    UpdateCollaboratorRequest.initialize(this, email);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateCollaboratorRequest, null, [{
    key: "initialize",
    value: function initialize(obj, email) {
      obj['email'] = email;
    }
    /**
     * Constructs a <code>UpdateCollaboratorRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateCollaboratorRequest} obj Optional instance to populate.
     * @return {module:model/UpdateCollaboratorRequest} The populated <code>UpdateCollaboratorRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateCollaboratorRequest();

        _UpdateCollaboratorRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('addRoles')) {
          obj['addRoles'] = _ApiClient["default"].convertToType(data['addRoles'], ['String']);
        }

        if (data.hasOwnProperty('removeRoles')) {
          obj['removeRoles'] = _ApiClient["default"].convertToType(data['removeRoles'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UpdateCollaboratorRequest;
}();
/**
 * @member {String} email
 */


UpdateCollaboratorRequest.prototype['email'] = undefined;
/**
 * @member {Array.<String>} addRoles
 */

UpdateCollaboratorRequest.prototype['addRoles'] = undefined;
/**
 * @member {Array.<String>} removeRoles
 */

UpdateCollaboratorRequest.prototype['removeRoles'] = undefined; // Implement UpdateCollaboratorRequestAllOf interface:

/**
 * @member {String} email
 */

_UpdateCollaboratorRequestAllOf["default"].prototype['email'] = undefined;
/**
 * @member {Array.<String>} addRoles
 */

_UpdateCollaboratorRequestAllOf["default"].prototype['addRoles'] = undefined;
/**
 * @member {Array.<String>} removeRoles
 */

_UpdateCollaboratorRequestAllOf["default"].prototype['removeRoles'] = undefined;
var _default = UpdateCollaboratorRequest;
exports["default"] = _default;
