// @ziqni-tech/admin-api-client/dist/model/UpdateProductRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ActionTypeAdjustmentFactor = _interopRequireDefault(require("./ActionTypeAdjustmentFactor"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateOptParamModels = _interopRequireDefault(require("./UpdateOptParamModels"));

var _UpdateProductRequestAllOf = _interopRequireDefault(require("./UpdateProductRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateProductRequest model module.
 * @module model/UpdateProductRequest
 * @version 0.0.26
 */
var UpdateProductRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateProductRequest</code>.
   * This is a request for Product update
   * @alias module:model/UpdateProductRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateOptParamModels
   * @implements module:model/UpdateProductRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateProductRequest(id) {
    _classCallCheck(this, UpdateProductRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateOptParamModels["default"].initialize(this);

    _UpdateProductRequestAllOf["default"].initialize(this);

    UpdateProductRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateProductRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateProductRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateProductRequest} obj Optional instance to populate.
     * @return {module:model/UpdateProductRequest} The populated <code>UpdateProductRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateProductRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateOptParamModels["default"].constructFromObject(data, obj);

        _UpdateProductRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('adjustmentFactor')) {
          obj['adjustmentFactor'] = _ApiClient["default"].convertToType(data['adjustmentFactor'], 'Number');
        }

        if (data.hasOwnProperty('actionTypeAdjustmentFactors')) {
          obj['actionTypeAdjustmentFactors'] = _ApiClient["default"].convertToType(data['actionTypeAdjustmentFactors'], [_ActionTypeAdjustmentFactor["default"]]);
        }
      }

      return obj;
    }
  }]);

  return UpdateProductRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateProductRequest.prototype['id'] = undefined;
/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

UpdateProductRequest.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

UpdateProductRequest.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateProductRequest.prototype['metadata'] = undefined;
/**
 * The name of the product
 * @member {String} name
 */

UpdateProductRequest.prototype['name'] = undefined;
/**
 * The description of the product for your reference
 * @member {String} description
 */

UpdateProductRequest.prototype['description'] = undefined;
/**
 * The multiplier to apply to source values received for this product events
 * @member {Number} adjustmentFactor
 */

UpdateProductRequest.prototype['adjustmentFactor'] = undefined;
/**
 * @member {Array.<module:model/ActionTypeAdjustmentFactor>} actionTypeAdjustmentFactors
 */

UpdateProductRequest.prototype['actionTypeAdjustmentFactors'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateOptParamModels interface:

/**
 * A list of id's used to add cutom fields
 * @member {Object.<String, Object>} customFields
 */

_UpdateOptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_UpdateOptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateOptParamModels["default"].prototype['metadata'] = undefined; // Implement UpdateProductRequestAllOf interface:

/**
 * The name of the product
 * @member {String} name
 */

_UpdateProductRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the product for your reference
 * @member {String} description
 */

_UpdateProductRequestAllOf["default"].prototype['description'] = undefined;
/**
 * The multiplier to apply to source values received for this product events
 * @member {Number} adjustmentFactor
 */

_UpdateProductRequestAllOf["default"].prototype['adjustmentFactor'] = undefined;
/**
 * @member {Array.<module:model/ActionTypeAdjustmentFactor>} actionTypeAdjustmentFactors
 */

_UpdateProductRequestAllOf["default"].prototype['actionTypeAdjustmentFactors'] = undefined;
var _default = UpdateProductRequest;
exports["default"] = _default;
