// @ziqni-tech/admin-api-client/dist/model/UpdateTagsRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UpdateModelDefault = _interopRequireDefault(require("./UpdateModelDefault"));

var _UpdateTagsRequestAllOf = _interopRequireDefault(require("./UpdateTagsRequestAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UpdateTagsRequest model module.
 * @module model/UpdateTagsRequest
 * @version 0.0.5
 */
var UpdateTagsRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateTagsRequest</code>.
   * @alias module:model/UpdateTagsRequest
   * @implements module:model/UpdateModelDefault
   * @implements module:model/UpdateTagsRequestAllOf
   * @param id {String} A unique system generated identifier
   */
  function UpdateTagsRequest(id) {
    _classCallCheck(this, UpdateTagsRequest);

    _UpdateModelDefault["default"].initialize(this, id);

    _UpdateTagsRequestAllOf["default"].initialize(this);

    UpdateTagsRequest.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateTagsRequest, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>UpdateTagsRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateTagsRequest} obj Optional instance to populate.
     * @return {module:model/UpdateTagsRequest} The populated <code>UpdateTagsRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateTagsRequest();

        _UpdateModelDefault["default"].constructFromObject(data, obj);

        _UpdateTagsRequestAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('entityTypes')) {
          obj['entityTypes'] = _ApiClient["default"].convertToType(data['entityTypes'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }
      }

      return obj;
    }
  }]);

  return UpdateTagsRequest;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


UpdateTagsRequest.prototype['id'] = undefined;
/**
 * The name of the tag
 * @member {String} name
 */

UpdateTagsRequest.prototype['name'] = undefined;
/**
 * The description of the tag for your reference
 * @member {String} description
 */

UpdateTagsRequest.prototype['description'] = undefined;
/**
 * The model name the tag refers to
 * @member {Array.<String>} entityTypes
 */

UpdateTagsRequest.prototype['entityTypes'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

UpdateTagsRequest.prototype['metadata'] = undefined; // Implement UpdateModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_UpdateModelDefault["default"].prototype['id'] = undefined; // Implement UpdateTagsRequestAllOf interface:

/**
 * The name of the tag
 * @member {String} name
 */

_UpdateTagsRequestAllOf["default"].prototype['name'] = undefined;
/**
 * The description of the tag for your reference
 * @member {String} description
 */

_UpdateTagsRequestAllOf["default"].prototype['description'] = undefined;
/**
 * The model name the tag refers to
 * @member {Array.<String>} entityTypes
 */

_UpdateTagsRequestAllOf["default"].prototype['entityTypes'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_UpdateTagsRequestAllOf["default"].prototype['metadata'] = undefined;
var _default = UpdateTagsRequest;
exports["default"] = _default;
