// @ziqni-tech/admin-api-client/dist/model/UserAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The UserAllOf model module.
 * @module model/UserAllOf
 * @version 0.0.26
 */
var UserAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserAllOf</code>.
   * @alias module:model/UserAllOf
   * @param email {String} The email of the user to log in
   * @param constraints {Array.<String>} Additional constraints
   */
  function UserAllOf(email, constraints) {
    _classCallCheck(this, UserAllOf);

    UserAllOf.initialize(this, email, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UserAllOf, null, [{
    key: "initialize",
    value: function initialize(obj, email, constraints) {
      obj['email'] = email;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>UserAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserAllOf} obj Optional instance to populate.
     * @return {module:model/UserAllOf} The populated <code>UserAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserAllOf();

        if (data.hasOwnProperty('firstName')) {
          obj['firstName'] = _ApiClient["default"].convertToType(data['firstName'], 'String');
        }

        if (data.hasOwnProperty('lastName')) {
          obj['lastName'] = _ApiClient["default"].convertToType(data['lastName'], 'String');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('spaces')) {
          obj['spaces'] = _ApiClient["default"].convertToType(data['spaces'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return UserAllOf;
}();
/**
 * The Name of an individual
 * @member {String} firstName
 */


UserAllOf.prototype['firstName'] = undefined;
/**
 * The Surname of an individual
 * @member {String} lastName
 */

UserAllOf.prototype['lastName'] = undefined;
/**
 * The email of the user to log in
 * @member {String} email
 */

UserAllOf.prototype['email'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

UserAllOf.prototype['constraints'] = undefined;
/**
 * @member {Array.<String>} spaces
 */

UserAllOf.prototype['spaces'] = undefined;
var _default = UserAllOf;
exports["default"] = _default;
