// @ziqni-tech/admin-api-client/dist/model/User.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UserAllOf = _interopRequireDefault(require("./UserAllOf"));

var _UserModelDefault = _interopRequireDefault(require("./UserModelDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The User model module.
 * @module model/User
 * @version 0.0.26
 */
var User = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>User</code>.
   * @alias module:model/User
   * @implements module:model/UserModelDefault
   * @implements module:model/UserAllOf
   * @param objectType {String} 
   * @param email {String} The email of the user to log in
   * @param constraints {Array.<String>} Additional constraints
   */
  function User(objectType, email, constraints) {
    _classCallCheck(this, User);

    _UserModelDefault["default"].initialize(this, objectType);

    _UserAllOf["default"].initialize(this, email, constraints);

    User.initialize(this, objectType, email, constraints);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(User, null, [{
    key: "initialize",
    value: function initialize(obj, objectType, email, constraints) {
      obj['objectType'] = objectType;
      obj['email'] = email;
      obj['constraints'] = constraints;
    }
    /**
     * Constructs a <code>User</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/User} obj Optional instance to populate.
     * @return {module:model/User} The populated <code>User</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new User();

        _UserModelDefault["default"].constructFromObject(data, obj);

        _UserAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('objectType')) {
          obj['objectType'] = _ApiClient["default"].convertToType(data['objectType'], 'String');
        }

        if (data.hasOwnProperty('firstName')) {
          obj['firstName'] = _ApiClient["default"].convertToType(data['firstName'], 'String');
        }

        if (data.hasOwnProperty('lastName')) {
          obj['lastName'] = _ApiClient["default"].convertToType(data['lastName'], 'String');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('spaces')) {
          obj['spaces'] = _ApiClient["default"].convertToType(data['spaces'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return User;
}();
/**
 * @member {String} objectType
 */


User.prototype['objectType'] = undefined;
/**
 * The Name of an individual
 * @member {String} firstName
 */

User.prototype['firstName'] = undefined;
/**
 * The Surname of an individual
 * @member {String} lastName
 */

User.prototype['lastName'] = undefined;
/**
 * The email of the user to log in
 * @member {String} email
 */

User.prototype['email'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

User.prototype['constraints'] = undefined;
/**
 * @member {Array.<String>} spaces
 */

User.prototype['spaces'] = undefined; // Implement UserModelDefault interface:

/**
 * @member {String} objectType
 */

_UserModelDefault["default"].prototype['objectType'] = undefined; // Implement UserAllOf interface:

/**
 * The Name of an individual
 * @member {String} firstName
 */

_UserAllOf["default"].prototype['firstName'] = undefined;
/**
 * The Surname of an individual
 * @member {String} lastName
 */

_UserAllOf["default"].prototype['lastName'] = undefined;
/**
 * The email of the user to log in
 * @member {String} email
 */

_UserAllOf["default"].prototype['email'] = undefined;
/**
 * Additional constraints
 * @member {Array.<String>} constraints
 */

_UserAllOf["default"].prototype['constraints'] = undefined;
/**
 * @member {Array.<String>} spaces
 */

_UserAllOf["default"].prototype['spaces'] = undefined;
var _default = User;
exports["default"] = _default;
