// @ziqni-tech/admin-api-client/dist/model/ValueLookupResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ValueLookupResult model module.
 * @module model/ValueLookupResult
 * @version 0.0.26
 */
var ValueLookupResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ValueLookupResult</code>.
   * The result of a generic lookup query
   * @alias module:model/ValueLookupResult
   */
  function ValueLookupResult() {
    _classCallCheck(this, ValueLookupResult);

    ValueLookupResult.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ValueLookupResult, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ValueLookupResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ValueLookupResult} obj Optional instance to populate.
     * @return {module:model/ValueLookupResult} The populated <code>ValueLookupResult</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ValueLookupResult();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('referenceId')) {
          obj['referenceId'] = _ApiClient["default"].convertToType(data['referenceId'], 'String');
        }

        if (data.hasOwnProperty('relatesToId')) {
          obj['relatesToId'] = _ApiClient["default"].convertToType(data['relatesToId'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ValueLookupResult;
}();
/**
 * @member {String} id
 */


ValueLookupResult.prototype['id'] = undefined;
/**
 * @member {String} referenceId
 */

ValueLookupResult.prototype['referenceId'] = undefined;
/**
 * @member {String} relatesToId
 */

ValueLookupResult.prototype['relatesToId'] = undefined;
/**
 * @member {String} name
 */

ValueLookupResult.prototype['name'] = undefined;
/**
 * @member {String} key
 */

ValueLookupResult.prototype['key'] = undefined;
/**
 * @member {String} description
 */

ValueLookupResult.prototype['description'] = undefined;
var _default = ValueLookupResult;
exports["default"] = _default;
