// @ziqni-tech/admin-api-client/dist/model/Webhook.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ConnectionState = _interopRequireDefault(require("./ConnectionState"));

var _ModelDefault = _interopRequireDefault(require("./ModelDefault"));

var _OptParamModels = _interopRequireDefault(require("./OptParamModels"));

var _WebhookAllOf = _interopRequireDefault(require("./WebhookAllOf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The Webhook model module.
 * @module model/Webhook
 * @version 0.0.26
 */
var Webhook = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Webhook</code>.
   * @alias module:model/Webhook
   * @implements module:model/ModelDefault
   * @implements module:model/OptParamModels
   * @implements module:model/WebhookAllOf
   * @param id {String} A unique system generated identifier
   * @param spaceName {String} This is the space name which is linked to the account
   * @param created {Date} ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
   * @param postToUrl {String} A URL to post the webhook to
   * @param triggers {Array.<String>} A list of event triggers
   * @param name {String} The name of a Webhook
   * @param status {module:model/ConnectionState} 
   */
  function Webhook(id, spaceName, created, postToUrl, triggers, name, status) {
    _classCallCheck(this, Webhook);

    _ModelDefault["default"].initialize(this, id, spaceName, created);

    _OptParamModels["default"].initialize(this);

    _WebhookAllOf["default"].initialize(this, postToUrl, triggers, name, status);

    Webhook.initialize(this, id, spaceName, created, postToUrl, triggers, name, status);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Webhook, null, [{
    key: "initialize",
    value: function initialize(obj, id, spaceName, created, postToUrl, triggers, name, status) {
      obj['id'] = id;
      obj['spaceName'] = spaceName;
      obj['created'] = created;
      obj['postToUrl'] = postToUrl;
      obj['triggers'] = triggers;
      obj['name'] = name;
      obj['status'] = status;
    }
    /**
     * Constructs a <code>Webhook</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Webhook} obj Optional instance to populate.
     * @return {module:model/Webhook} The populated <code>Webhook</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Webhook();

        _ModelDefault["default"].constructFromObject(data, obj);

        _OptParamModels["default"].constructFromObject(data, obj);

        _WebhookAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('spaceName')) {
          obj['spaceName'] = _ApiClient["default"].convertToType(data['spaceName'], 'String');
        }

        if (data.hasOwnProperty('created')) {
          obj['created'] = _ApiClient["default"].convertToType(data['created'], 'Date');
        }

        if (data.hasOwnProperty('customFields')) {
          obj['customFields'] = _ApiClient["default"].convertToType(data['customFields'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('postToUrl')) {
          obj['postToUrl'] = _ApiClient["default"].convertToType(data['postToUrl'], 'String');
        }

        if (data.hasOwnProperty('triggers')) {
          obj['triggers'] = _ApiClient["default"].convertToType(data['triggers'], ['String']);
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('headers')) {
          obj['headers'] = _ApiClient["default"].convertToType(data['headers'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ConnectionState["default"].constructFromObject(data['status']);
        }
      }

      return obj;
    }
  }]);

  return Webhook;
}();
/**
 * A unique system generated identifier
 * @member {String} id
 */


Webhook.prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

Webhook.prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

Webhook.prototype['created'] = undefined;
/**
 * @member {Object.<String, Object>} customFields
 */

Webhook.prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

Webhook.prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

Webhook.prototype['metadata'] = undefined;
/**
 * A URL to post the webhook to
 * @member {String} postToUrl
 */

Webhook.prototype['postToUrl'] = undefined;
/**
 * A list of event triggers
 * @member {Array.<String>} triggers
 */

Webhook.prototype['triggers'] = undefined;
/**
 * The description of a Webhook
 * @member {String} description
 */

Webhook.prototype['description'] = undefined;
/**
 * The name of a Webhook
 * @member {String} name
 */

Webhook.prototype['name'] = undefined;
/**
 * @member {Object.<String, String>} headers
 */

Webhook.prototype['headers'] = undefined;
/**
 * @member {module:model/ConnectionState} status
 */

Webhook.prototype['status'] = undefined; // Implement ModelDefault interface:

/**
 * A unique system generated identifier
 * @member {String} id
 */

_ModelDefault["default"].prototype['id'] = undefined;
/**
 * This is the space name which is linked to the account
 * @member {String} spaceName
 */

_ModelDefault["default"].prototype['spaceName'] = undefined;
/**
 * ISO8601 timestamp for when a Model was created. All records are stored in UTC time zone
 * @member {Date} created
 */

_ModelDefault["default"].prototype['created'] = undefined; // Implement OptParamModels interface:

/**
 * @member {Object.<String, Object>} customFields
 */

_OptParamModels["default"].prototype['customFields'] = undefined;
/**
 * A list of id's used to tag models
 * @member {Array.<String>} tags
 */

_OptParamModels["default"].prototype['tags'] = undefined;
/**
 * @member {Object.<String, String>} metadata
 */

_OptParamModels["default"].prototype['metadata'] = undefined; // Implement WebhookAllOf interface:

/**
 * A URL to post the webhook to
 * @member {String} postToUrl
 */

_WebhookAllOf["default"].prototype['postToUrl'] = undefined;
/**
 * A list of event triggers
 * @member {Array.<String>} triggers
 */

_WebhookAllOf["default"].prototype['triggers'] = undefined;
/**
 * The description of a Webhook
 * @member {String} description
 */

_WebhookAllOf["default"].prototype['description'] = undefined;
/**
 * The name of a Webhook
 * @member {String} name
 */

_WebhookAllOf["default"].prototype['name'] = undefined;
/**
 * @member {Object.<String, String>} headers
 */

_WebhookAllOf["default"].prototype['headers'] = undefined;
/**
 * @member {module:model/ConnectionState} status
 */

_WebhookAllOf["default"].prototype['status'] = undefined;
var _default = Webhook;
exports["default"] = _default;
